// avatax/lib/models/AddressLocationInfo.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddressLocationInfo = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Represents an address to resolve.
 * @export
 * @class AddressLocationInfo
 */
var AddressLocationInfo = exports.AddressLocationInfo = /** @class */ (function () {
    function AddressLocationInfo() {
        /**
         * @type {string}
         * @memberof AddressLocationInfo
         */
        this.locationCode = undefined;
        /**
         * @type {string}
         * @memberof AddressLocationInfo
         */
        this.line1 = undefined;
        /**
         * @type {string}
         * @memberof AddressLocationInfo
         */
        this.line2 = undefined;
        /**
         * @type {string}
         * @memberof AddressLocationInfo
         */
        this.line3 = undefined;
        /**
         * @type {string}
         * @memberof AddressLocationInfo
         */
        this.city = undefined;
        /**
         * @type {string}
         * @memberof AddressLocationInfo
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof AddressLocationInfo
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof AddressLocationInfo
         */
        this.postalCode = undefined;
        /**
         * @type {number}
         * @memberof AddressLocationInfo
         */
        this.latitude = undefined;
        /**
         * @type {number}
         * @memberof AddressLocationInfo
         */
        this.longitude = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("locationCode", String, true),
        __metadata("design:type", String)
    ], AddressLocationInfo.prototype, "locationCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line1", String, true),
        __metadata("design:type", String)
    ], AddressLocationInfo.prototype, "line1", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line2", String, true),
        __metadata("design:type", String)
    ], AddressLocationInfo.prototype, "line2", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line3", String, true),
        __metadata("design:type", String)
    ], AddressLocationInfo.prototype, "line3", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("city", String, true),
        __metadata("design:type", String)
    ], AddressLocationInfo.prototype, "city", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], AddressLocationInfo.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], AddressLocationInfo.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("postalCode", String, true),
        __metadata("design:type", String)
    ], AddressLocationInfo.prototype, "postalCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("latitude", Number, true),
        __metadata("design:type", Number)
    ], AddressLocationInfo.prototype, "latitude", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("longitude", Number, true),
        __metadata("design:type", Number)
    ], AddressLocationInfo.prototype, "longitude", void 0);
    AddressLocationInfo = __decorate([
        (0, json2typescript_1.JsonObject)("AddressLocationInfo")
    ], AddressLocationInfo);
    return AddressLocationInfo;
}());
