// avatax/lib/models/AddressesModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddressesModel = void 0;
var AddressLocationInfo_1 = require("./AddressLocationInfo");
var json2typescript_1 = require("json2typescript");
/**
 * Information about all the addresses involved in this transaction.
            
For a physical in-person transaction at a retail point-of-sale location, please specify only one address using
the `singleLocation` field.
            
For a transaction that was shipped, delivered, or provided from an origin location such as a warehouse to
a destination location such as a customer, please specify the `shipFrom` and `shipTo` addresses.
            
In the United States, some jurisdictions recognize the address types `pointOfOrderOrigin` and `pointOfOrderAcceptance`.
These address types affect the sourcing models of some transactions.
            
If latitude and longitude information is provided for any of these addresses along with line, city, region, country and postal code information,
we will be using only latitude and longitude and will discard line, city, region, country and postal code information for the transaction.
Please ensure that you have the correct latitude/longitude information for the addresses prior to using the API.
If you provide either latitude or longitude information but not both, we will be using the line, city, region, country and postal code information for the addresses.
 * @export
 * @class AddressesModel
 */
var AddressesModel = exports.AddressesModel = /** @class */ (function () {
    function AddressesModel() {
        /**
         * @type {AddressLocationInfo}
         * @memberof AddressesModel
         */
        this.singleLocation = undefined;
        /**
         * @type {AddressLocationInfo}
         * @memberof AddressesModel
         */
        this.shipFrom = undefined;
        /**
         * @type {AddressLocationInfo}
         * @memberof AddressesModel
         */
        this.shipTo = undefined;
        /**
         * @type {AddressLocationInfo}
         * @memberof AddressesModel
         */
        this.pointOfOrderOrigin = undefined;
        /**
         * @type {AddressLocationInfo}
         * @memberof AddressesModel
         */
        this.pointOfOrderAcceptance = undefined;
        /**
         * @type {AddressLocationInfo}
         * @memberof AddressesModel
         */
        this.goodsPlaceOrServiceRendered = undefined;
        /**
         * @type {AddressLocationInfo}
         * @memberof AddressesModel
         */
        this.import = undefined;
        /**
         * @type {AddressLocationInfo}
         * @memberof AddressesModel
         */
        this.billTo = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("singleLocation", AddressLocationInfo_1.AddressLocationInfo, true),
        __metadata("design:type", AddressLocationInfo_1.AddressLocationInfo)
    ], AddressesModel.prototype, "singleLocation", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("shipFrom", AddressLocationInfo_1.AddressLocationInfo, true),
        __metadata("design:type", AddressLocationInfo_1.AddressLocationInfo)
    ], AddressesModel.prototype, "shipFrom", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("shipTo", AddressLocationInfo_1.AddressLocationInfo, true),
        __metadata("design:type", AddressLocationInfo_1.AddressLocationInfo)
    ], AddressesModel.prototype, "shipTo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("pointOfOrderOrigin", AddressLocationInfo_1.AddressLocationInfo, true),
        __metadata("design:type", AddressLocationInfo_1.AddressLocationInfo)
    ], AddressesModel.prototype, "pointOfOrderOrigin", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("pointOfOrderAcceptance", AddressLocationInfo_1.AddressLocationInfo, true),
        __metadata("design:type", AddressLocationInfo_1.AddressLocationInfo)
    ], AddressesModel.prototype, "pointOfOrderAcceptance", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("goodsPlaceOrServiceRendered", AddressLocationInfo_1.AddressLocationInfo, true),
        __metadata("design:type", AddressLocationInfo_1.AddressLocationInfo)
    ], AddressesModel.prototype, "goodsPlaceOrServiceRendered", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("import", AddressLocationInfo_1.AddressLocationInfo, true),
        __metadata("design:type", AddressLocationInfo_1.AddressLocationInfo)
    ], AddressesModel.prototype, "import", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("billTo", AddressLocationInfo_1.AddressLocationInfo, true),
        __metadata("design:type", AddressLocationInfo_1.AddressLocationInfo)
    ], AddressesModel.prototype, "billTo", void 0);
    AddressesModel = __decorate([
        (0, json2typescript_1.JsonObject)("AddressesModel")
    ], AddressesModel);
    return AddressesModel;
}());
