// avatax/lib/models/AdvancedRuleLookupFileModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdvancedRuleLookupFileModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Model representing a lookup file for a company
 * @export
 * @class AdvancedRuleLookupFileModel
 */
var AdvancedRuleLookupFileModel = exports.AdvancedRuleLookupFileModel = /** @class */ (function () {
    function AdvancedRuleLookupFileModel() {
        /**
         * @type {string}
         * @memberof AdvancedRuleLookupFileModel
         */
        this.lookupFileId = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleLookupFileModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleLookupFileModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleLookupFileModel
         */
        this.content = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleLookupFileModel
         */
        this.fileExtension = undefined;
        /**
         * @type {boolean}
         * @memberof AdvancedRuleLookupFileModel
         */
        this.isTest = undefined;
        /**
         * @type {boolean}
         * @memberof AdvancedRuleLookupFileModel
         */
        this.inUse = undefined;
        /**
         * @type {number}
         * @memberof AdvancedRuleLookupFileModel
         */
        this.version = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleLookupFileModel
         */
        this.createdDate = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleLookupFileModel
         */
        this.modifiedDate = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("lookupFileId", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleLookupFileModel.prototype, "lookupFileId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleLookupFileModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String),
        __metadata("design:type", String)
    ], AdvancedRuleLookupFileModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("content", String),
        __metadata("design:type", String)
    ], AdvancedRuleLookupFileModel.prototype, "content", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("fileExtension", String),
        __metadata("design:type", String)
    ], AdvancedRuleLookupFileModel.prototype, "fileExtension", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isTest", Boolean, true),
        __metadata("design:type", Boolean)
    ], AdvancedRuleLookupFileModel.prototype, "isTest", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("inUse", Boolean, true),
        __metadata("design:type", Boolean)
    ], AdvancedRuleLookupFileModel.prototype, "inUse", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("version", Number, true),
        __metadata("design:type", Number)
    ], AdvancedRuleLookupFileModel.prototype, "version", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleLookupFileModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleLookupFileModel.prototype, "modifiedDate", void 0);
    AdvancedRuleLookupFileModel = __decorate([
        (0, json2typescript_1.JsonObject)("AdvancedRuleLookupFileModel")
    ], AdvancedRuleLookupFileModel);
    return AdvancedRuleLookupFileModel;
}());
