// avatax/lib/models/AdvancedRuleModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdvancedRuleModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * A generic global advanced rule encapsulating a script
 * @export
 * @class AdvancedRuleModel
 */
var AdvancedRuleModel = exports.AdvancedRuleModel = /** @class */ (function () {
    function AdvancedRuleModel() {
        /**
         * @type {string}
         * @memberof AdvancedRuleModel
         */
        this.ruleId = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleModel
         */
        this.description = undefined;
        /**
         * @type {boolean}
         * @memberof AdvancedRuleModel
         */
        this.arEntitlementRequired = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleModel
         */
        this.executionPosition = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("ruleId", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleModel.prototype, "ruleId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String),
        __metadata("design:type", String)
    ], AdvancedRuleModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("arEntitlementRequired", Boolean, true),
        __metadata("design:type", Boolean)
    ], AdvancedRuleModel.prototype, "arEntitlementRequired", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("executionPosition", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleModel.prototype, "executionPosition", void 0);
    AdvancedRuleModel = __decorate([
        (0, json2typescript_1.JsonObject)("AdvancedRuleModel")
    ], AdvancedRuleModel);
    return AdvancedRuleModel;
}());
