// avatax/lib/models/AdvancedRuleFullDetailsModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdvancedRuleFullDetailsModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Model for Advanced Rules when full details are requested
 * @export
 * @class AdvancedRuleFullDetailsModel
 */
var AdvancedRuleFullDetailsModel = exports.AdvancedRuleFullDetailsModel = /** @class */ (function () {
    function AdvancedRuleFullDetailsModel() {
        /**
         * @type {string}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.script = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.customerDataValidatorScript = undefined;
        /**
         * @type {boolean}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.isApproved = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.createdBy = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.createdOn = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.modifiedBy = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.modifiedOn = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.approvedBy = undefined;
        /**
         * @type {boolean}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.isSystemRule = undefined;
        /**
         * @type {boolean}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.isVisibleInCUP = undefined;
        /**
         * @type {boolean}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.isTest = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.customerDataSchema = undefined;
        /**
         * @type {number}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.version = undefined;
        /**
         * @type {number[]}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.accountsVisibleFor = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.ruleId = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.description = undefined;
        /**
         * @type {boolean}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.arEntitlementRequired = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleFullDetailsModel
         */
        this.executionPosition = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("script", String),
        __metadata("design:type", String)
    ], AdvancedRuleFullDetailsModel.prototype, "script", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerDataValidatorScript", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleFullDetailsModel.prototype, "customerDataValidatorScript", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isApproved", Boolean, true),
        __metadata("design:type", Boolean)
    ], AdvancedRuleFullDetailsModel.prototype, "isApproved", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdBy", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleFullDetailsModel.prototype, "createdBy", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdOn", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleFullDetailsModel.prototype, "createdOn", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedBy", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleFullDetailsModel.prototype, "modifiedBy", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedOn", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleFullDetailsModel.prototype, "modifiedOn", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("approvedBy", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleFullDetailsModel.prototype, "approvedBy", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSystemRule", Boolean, true),
        __metadata("design:type", Boolean)
    ], AdvancedRuleFullDetailsModel.prototype, "isSystemRule", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isVisibleInCUP", Boolean, true),
        __metadata("design:type", Boolean)
    ], AdvancedRuleFullDetailsModel.prototype, "isVisibleInCUP", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isTest", Boolean, true),
        __metadata("design:type", Boolean)
    ], AdvancedRuleFullDetailsModel.prototype, "isTest", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerDataSchema", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleFullDetailsModel.prototype, "customerDataSchema", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("version", Number, true),
        __metadata("design:type", Number)
    ], AdvancedRuleFullDetailsModel.prototype, "version", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountsVisibleFor", [Number], true),
        __metadata("design:type", Array)
    ], AdvancedRuleFullDetailsModel.prototype, "accountsVisibleFor", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("ruleId", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleFullDetailsModel.prototype, "ruleId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String),
        __metadata("design:type", String)
    ], AdvancedRuleFullDetailsModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleFullDetailsModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("arEntitlementRequired", Boolean, true),
        __metadata("design:type", Boolean)
    ], AdvancedRuleFullDetailsModel.prototype, "arEntitlementRequired", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("executionPosition", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleFullDetailsModel.prototype, "executionPosition", void 0);
    AdvancedRuleFullDetailsModel = __decorate([
        (0, json2typescript_1.JsonObject)("AdvancedRuleFullDetailsModel")
    ], AdvancedRuleFullDetailsModel);
    return AdvancedRuleFullDetailsModel;
}());
