// avatax/lib/models/AuditEvent.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuditEvent = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 *
 * @export
 * @class AuditEvent
 */
var AuditEvent = exports.AuditEvent = /** @class */ (function () {
    function AuditEvent() {
        /**
         * @type {number}
         * @memberof AuditEvent
         */
        this.auditEventId = undefined;
        /**
         * @type {number}
         * @memberof AuditEvent
         */
        this.transactionId = undefined;
        /**
         * @type {string}
         * @memberof AuditEvent
         */
        this.correlationId = undefined;
        /**
         * @type {string}
         * @memberof AuditEvent
         */
        this.avalaraUid = undefined;
        /**
         * @type {number}
         * @memberof AuditEvent
         */
        this.auditEventLevelId = undefined;
        /**
         * @type {Date}
         * @memberof AuditEvent
         */
        this.eventTimestamp = undefined;
        /**
         * @type {string}
         * @memberof AuditEvent
         */
        this.source = undefined;
        /**
         * @type {string}
         * @memberof AuditEvent
         */
        this.summary = undefined;
        /**
         * @type {string}
         * @memberof AuditEvent
         */
        this.details = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("auditEventId", Number, true),
        __metadata("design:type", Number)
    ], AuditEvent.prototype, "auditEventId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("transactionId", Number, true),
        __metadata("design:type", Number)
    ], AuditEvent.prototype, "transactionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("correlationId", String, true),
        __metadata("design:type", String)
    ], AuditEvent.prototype, "correlationId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("avalaraUid", String, true),
        __metadata("design:type", String)
    ], AuditEvent.prototype, "avalaraUid", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("auditEventLevelId", Number, true),
        __metadata("design:type", Number)
    ], AuditEvent.prototype, "auditEventLevelId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("eventTimestamp", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AuditEvent.prototype, "eventTimestamp", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("source", String, true),
        __metadata("design:type", String)
    ], AuditEvent.prototype, "source", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("summary", String, true),
        __metadata("design:type", String)
    ], AuditEvent.prototype, "summary", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("details", String, true),
        __metadata("design:type", String)
    ], AuditEvent.prototype, "details", void 0);
    AuditEvent = __decorate([
        (0, json2typescript_1.JsonObject)("AuditEvent")
    ], AuditEvent);
    return AuditEvent;
}());
