// avatax/lib/models/AuditModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuditModel = void 0;
var AuditEvent_1 = require("./AuditEvent");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Provides detailed information about an API call.
            
The information on this record was captured by AvaTax when your API call was made.  If you are unsure why you
received an error, you can fetch these audit objects and examine the `RequestUrl`, `RequestBody`, and `ErrorMessage`
fields to determine root cause for the error.
 * @export
 * @class AuditModel
 */
var AuditModel = exports.AuditModel = /** @class */ (function () {
    function AuditModel() {
        /**
         * @type {number}
         * @memberof AuditModel
         */
        this.transactionId = undefined;
        /**
         * @type {number}
         * @memberof AuditModel
         */
        this.accountId = undefined;
        /**
         * @type {number}
         * @memberof AuditModel
         */
        this.userId = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.ipAddress = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.machineName = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.clientName = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.clientVersion = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.adapterName = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.adapterVersion = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.serverName = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.serverVersion = undefined;
        /**
         * @type {number}
         * @memberof AuditModel
         */
        this.referenceId = undefined;
        /**
         * @type {number}
         * @memberof AuditModel
         */
        this.severityLevelId = undefined;
        /**
         * @type {Date}
         * @memberof AuditModel
         */
        this.serverTimestamp = undefined;
        /**
         * @type {number}
         * @memberof AuditModel
         */
        this.serverDuration = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.serviceName = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.operation = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.referenceCode = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.errorMessage = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.auditMessage = undefined;
        /**
         * @type {number}
         * @memberof AuditModel
         */
        this.loadBalancerDuration = undefined;
        /**
         * @type {number}
         * @memberof AuditModel
         */
        this.recordCount = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.referenceAuthorization = undefined;
        /**
         * @type {boolean}
         * @memberof AuditModel
         */
        this.isQueued = undefined;
        /**
         * @type {number}
         * @memberof AuditModel
         */
        this.databaseCallCount = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.databaseCallDuration = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.remoteCallDuration = undefined;
        /**
         * @type {AuditEvent[]}
         * @memberof AuditModel
         */
        this.events = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.requestUrl = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.requestBody = undefined;
        /**
         * @type {number}
         * @memberof AuditModel
         */
        this.responseStatus = undefined;
        /**
         * @type {string}
         * @memberof AuditModel
         */
        this.responseBody = undefined;
        /**
         * @type {AuditModel[]}
         * @memberof AuditModel
         */
        this.remoteCalls = undefined;
    }
    AuditModel_1 = AuditModel;
    var AuditModel_1;
    __decorate([
        (0, json2typescript_1.JsonProperty)("transactionId", Number, true),
        __metadata("design:type", Number)
    ], AuditModel.prototype, "transactionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountId", Number, true),
        __metadata("design:type", Number)
    ], AuditModel.prototype, "accountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("userId", Number, true),
        __metadata("design:type", Number)
    ], AuditModel.prototype, "userId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("ipAddress", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "ipAddress", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("machineName", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "machineName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("clientName", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "clientName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("clientVersion", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "clientVersion", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("adapterName", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "adapterName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("adapterVersion", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "adapterVersion", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("serverName", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "serverName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("serverVersion", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "serverVersion", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("referenceId", Number, true),
        __metadata("design:type", Number)
    ], AuditModel.prototype, "referenceId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("severityLevelId", Number, true),
        __metadata("design:type", Number)
    ], AuditModel.prototype, "severityLevelId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("serverTimestamp", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AuditModel.prototype, "serverTimestamp", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("serverDuration", Number, true),
        __metadata("design:type", Number)
    ], AuditModel.prototype, "serverDuration", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("serviceName", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "serviceName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("operation", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "operation", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("referenceCode", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "referenceCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("errorMessage", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "errorMessage", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("auditMessage", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "auditMessage", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("loadBalancerDuration", Number, true),
        __metadata("design:type", Number)
    ], AuditModel.prototype, "loadBalancerDuration", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("recordCount", Number, true),
        __metadata("design:type", Number)
    ], AuditModel.prototype, "recordCount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("referenceAuthorization", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "referenceAuthorization", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isQueued", Boolean, true),
        __metadata("design:type", Boolean)
    ], AuditModel.prototype, "isQueued", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("databaseCallCount", Number, true),
        __metadata("design:type", Number)
    ], AuditModel.prototype, "databaseCallCount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("databaseCallDuration", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "databaseCallDuration", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("remoteCallDuration", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "remoteCallDuration", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("events", [AuditEvent_1.AuditEvent], true),
        __metadata("design:type", Array)
    ], AuditModel.prototype, "events", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("requestUrl", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "requestUrl", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("requestBody", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "requestBody", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("responseStatus", Number, true),
        __metadata("design:type", Number)
    ], AuditModel.prototype, "responseStatus", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("responseBody", String, true),
        __metadata("design:type", String)
    ], AuditModel.prototype, "responseBody", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("remoteCalls", [AuditModel_1], true),
        __metadata("design:type", Array)
    ], AuditModel.prototype, "remoteCalls", void 0);
    AuditModel = AuditModel_1 = __decorate([
        (0, json2typescript_1.JsonObject)("AuditModel")
    ], AuditModel);
    return AuditModel;
}());
