// avatax/lib/models/BatchModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchModel = void 0;
var Enums = require("../enums/index");
var BatchFileModel_1 = require("./BatchFileModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a batch of uploaded documents.
 * @export
 * @class BatchModel
 */
var BatchModel = exports.BatchModel = /** @class */ (function () {
    function BatchModel() {
        /**
         * @type {Enums.BatchType}
         * @memberof BatchModel
         */
        this.type = undefined;
        /**
         * @type {string}
         * @memberof BatchModel
         */
        this.batchAgent = undefined;
        /**
         * @type {string}
         * @memberof BatchModel
         */
        this.options = undefined;
        /**
         * @type {number}
         * @memberof BatchModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof BatchModel
         */
        this.name = undefined;
        /**
         * @type {number}
         * @memberof BatchModel
         */
        this.accountId = undefined;
        /**
         * @type {number}
         * @memberof BatchModel
         */
        this.companyId = undefined;
        /**
         * @type {Enums.BatchStatus}
         * @memberof BatchModel
         */
        this.status = undefined;
        /**
         * @type {Date}
         * @memberof BatchModel
         */
        this.startedDate = undefined;
        /**
         * @type {number}
         * @memberof BatchModel
         */
        this.recordCount = undefined;
        /**
         * @type {number}
         * @memberof BatchModel
         */
        this.currentRecord = undefined;
        /**
         * @type {Date}
         * @memberof BatchModel
         */
        this.completedDate = undefined;
        /**
         * @type {Date}
         * @memberof BatchModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof BatchModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof BatchModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof BatchModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {BatchFileModel[]}
         * @memberof BatchModel
         */
        this.files = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("type", Enums.BatchTypeConverter),
        __metadata("design:type", Number)
    ], BatchModel.prototype, "type", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("batchAgent", String, true),
        __metadata("design:type", String)
    ], BatchModel.prototype, "batchAgent", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("options", String, true),
        __metadata("design:type", String)
    ], BatchModel.prototype, "options", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], BatchModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String),
        __metadata("design:type", String)
    ], BatchModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountId", Number, true),
        __metadata("design:type", Number)
    ], BatchModel.prototype, "accountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], BatchModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("status", Enums.BatchStatusConverter, true),
        __metadata("design:type", Number)
    ], BatchModel.prototype, "status", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("startedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], BatchModel.prototype, "startedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("recordCount", Number, true),
        __metadata("design:type", Number)
    ], BatchModel.prototype, "recordCount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("currentRecord", Number, true),
        __metadata("design:type", Number)
    ], BatchModel.prototype, "currentRecord", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("completedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], BatchModel.prototype, "completedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], BatchModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], BatchModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], BatchModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], BatchModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("files", [BatchFileModel_1.BatchFileModel], true),
        __metadata("design:type", Array)
    ], BatchModel.prototype, "files", void 0);
    BatchModel = __decorate([
        (0, json2typescript_1.JsonObject)("BatchModel")
    ], BatchModel);
    return BatchModel;
}());
