// avatax/lib/models/BatchAdjustTransactionModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchAdjustTransactionModel = void 0;
var Enums = require("../enums/index");
var CreateTransactionModel_1 = require("./CreateTransactionModel");
var json2typescript_1 = require("json2typescript");
/**
 * Replace an existing transaction recorded in AvaTax with a new one.
 * @export
 * @class BatchAdjustTransactionModel
 */
var BatchAdjustTransactionModel = exports.BatchAdjustTransactionModel = /** @class */ (function () {
    function BatchAdjustTransactionModel() {
        /**
         * @type {string}
         * @memberof BatchAdjustTransactionModel
         */
        this.companyCode = undefined;
        /**
         * @type {string}
         * @memberof BatchAdjustTransactionModel
         */
        this.transactionCode = undefined;
        /**
         * @type {string}
         * @memberof BatchAdjustTransactionModel
         */
        this.documentType = undefined;
        /**
         * @type {Enums.AdjustmentReason}
         * @memberof BatchAdjustTransactionModel
         */
        this.adjustmentReason = undefined;
        /**
         * @type {string}
         * @memberof BatchAdjustTransactionModel
         */
        this.adjustmentDescription = undefined;
        /**
         * @type {CreateTransactionModel}
         * @memberof BatchAdjustTransactionModel
         */
        this.newTransaction = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyCode", String),
        __metadata("design:type", String)
    ], BatchAdjustTransactionModel.prototype, "companyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("transactionCode", String),
        __metadata("design:type", String)
    ], BatchAdjustTransactionModel.prototype, "transactionCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("documentType", String, true),
        __metadata("design:type", String)
    ], BatchAdjustTransactionModel.prototype, "documentType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("adjustmentReason", Enums.AdjustmentReasonConverter),
        __metadata("design:type", Number)
    ], BatchAdjustTransactionModel.prototype, "adjustmentReason", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("adjustmentDescription", String, true),
        __metadata("design:type", String)
    ], BatchAdjustTransactionModel.prototype, "adjustmentDescription", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("newTransaction", CreateTransactionModel_1.CreateTransactionModel),
        __metadata("design:type", CreateTransactionModel_1.CreateTransactionModel)
    ], BatchAdjustTransactionModel.prototype, "newTransaction", void 0);
    BatchAdjustTransactionModel = __decorate([
        (0, json2typescript_1.JsonObject)("BatchAdjustTransactionModel")
    ], BatchAdjustTransactionModel);
    return BatchAdjustTransactionModel;
}());
