// avatax/lib/models/ComplianceJurisdictionRateModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComplianceJurisdictionRateModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * A flattened model for jurisdictions and rates.
 * @export
 * @class ComplianceJurisdictionRateModel
 */
var ComplianceJurisdictionRateModel = exports.ComplianceJurisdictionRateModel = /** @class */ (function () {
    function ComplianceJurisdictionRateModel() {
        /**
         * @type {number}
         * @memberof ComplianceJurisdictionRateModel
         */
        this.jurisdictionId = undefined;
        /**
         * @type {string}
         * @memberof ComplianceJurisdictionRateModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof ComplianceJurisdictionRateModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof ComplianceJurisdictionRateModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof ComplianceJurisdictionRateModel
         */
        this.jurisdictionTypeId = undefined;
        /**
         * @type {number}
         * @memberof ComplianceJurisdictionRateModel
         */
        this.rate = undefined;
        /**
         * @type {string}
         * @memberof ComplianceJurisdictionRateModel
         */
        this.rateTypeId = undefined;
        /**
         * @type {string}
         * @memberof ComplianceJurisdictionRateModel
         */
        this.taxTypeId = undefined;
        /**
         * @type {Date}
         * @memberof ComplianceJurisdictionRateModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof ComplianceJurisdictionRateModel
         */
        this.endDate = undefined;
        /**
         * @type {string}
         * @memberof ComplianceJurisdictionRateModel
         */
        this.stateAssignedCode = undefined;
        /**
         * @type {number}
         * @memberof ComplianceJurisdictionRateModel
         */
        this.taxAuthorityId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionId", Number, true),
        __metadata("design:type", Number)
    ], ComplianceJurisdictionRateModel.prototype, "jurisdictionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], ComplianceJurisdictionRateModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], ComplianceJurisdictionRateModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String, true),
        __metadata("design:type", String)
    ], ComplianceJurisdictionRateModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionTypeId", String, true),
        __metadata("design:type", String)
    ], ComplianceJurisdictionRateModel.prototype, "jurisdictionTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rate", Number, true),
        __metadata("design:type", Number)
    ], ComplianceJurisdictionRateModel.prototype, "rate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rateTypeId", String, true),
        __metadata("design:type", String)
    ], ComplianceJurisdictionRateModel.prototype, "rateTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxTypeId", String, true),
        __metadata("design:type", String)
    ], ComplianceJurisdictionRateModel.prototype, "taxTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ComplianceJurisdictionRateModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ComplianceJurisdictionRateModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("stateAssignedCode", String, true),
        __metadata("design:type", String)
    ], ComplianceJurisdictionRateModel.prototype, "stateAssignedCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityId", Number, true),
        __metadata("design:type", Number)
    ], ComplianceJurisdictionRateModel.prototype, "taxAuthorityId", void 0);
    ComplianceJurisdictionRateModel = __decorate([
        (0, json2typescript_1.JsonObject)("ComplianceJurisdictionRateModel")
    ], ComplianceJurisdictionRateModel);
    return ComplianceJurisdictionRateModel;
}());
