// avatax/lib/models/ComplianceJurisdictionModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComplianceJurisdictionModel = void 0;
var ComplianceAggregatedTaxRateModel_1 = require("./ComplianceAggregatedTaxRateModel");
var json2typescript_1 = require("json2typescript");
/**
 * Model for distinct jurisdictions.
 * @export
 * @class ComplianceJurisdictionModel
 */
var ComplianceJurisdictionModel = exports.ComplianceJurisdictionModel = /** @class */ (function () {
    function ComplianceJurisdictionModel() {
        /**
         * @type {number}
         * @memberof ComplianceJurisdictionModel
         */
        this.taxRegionId = undefined;
        /**
         * @type {string}
         * @memberof ComplianceJurisdictionModel
         */
        this.stateAssignedCode = undefined;
        /**
         * @type {string}
         * @memberof ComplianceJurisdictionModel
         */
        this.jurisdictionTypeId = undefined;
        /**
         * @type {string}
         * @memberof ComplianceJurisdictionModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof ComplianceJurisdictionModel
         */
        this.county = undefined;
        /**
         * @type {string}
         * @memberof ComplianceJurisdictionModel
         */
        this.city = undefined;
        /**
         * @type {string}
         * @memberof ComplianceJurisdictionModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof ComplianceJurisdictionModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof ComplianceJurisdictionModel
         */
        this.taxRegionName = undefined;
        /**
         * @type {number}
         * @memberof ComplianceJurisdictionModel
         */
        this.taxAuthorityId = undefined;
        /**
         * @type {ComplianceAggregatedTaxRateModel[]}
         * @memberof ComplianceJurisdictionModel
         */
        this.rates = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxRegionId", Number, true),
        __metadata("design:type", Number)
    ], ComplianceJurisdictionModel.prototype, "taxRegionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("stateAssignedCode", String, true),
        __metadata("design:type", String)
    ], ComplianceJurisdictionModel.prototype, "stateAssignedCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionTypeId", String, true),
        __metadata("design:type", String)
    ], ComplianceJurisdictionModel.prototype, "jurisdictionTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String, true),
        __metadata("design:type", String)
    ], ComplianceJurisdictionModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("county", String, true),
        __metadata("design:type", String)
    ], ComplianceJurisdictionModel.prototype, "county", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("city", String, true),
        __metadata("design:type", String)
    ], ComplianceJurisdictionModel.prototype, "city", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], ComplianceJurisdictionModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], ComplianceJurisdictionModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxRegionName", String, true),
        __metadata("design:type", String)
    ], ComplianceJurisdictionModel.prototype, "taxRegionName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityId", Number, true),
        __metadata("design:type", Number)
    ], ComplianceJurisdictionModel.prototype, "taxAuthorityId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rates", [ComplianceAggregatedTaxRateModel_1.ComplianceAggregatedTaxRateModel], true),
        __metadata("design:type", Array)
    ], ComplianceJurisdictionModel.prototype, "rates", void 0);
    ComplianceJurisdictionModel = __decorate([
        (0, json2typescript_1.JsonObject)("ComplianceJurisdictionModel")
    ], ComplianceJurisdictionModel);
    return ComplianceJurisdictionModel;
}());
