// avatax/lib/models/CompanyModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CompanyModel = void 0;
var Enums = require("../enums/index");
var ContactModel_1 = require("./ContactModel");
var ItemModel_1 = require("./ItemModel");
var LocationModel_1 = require("./LocationModel");
var NexusModel_1 = require("./NexusModel");
var SettingModel_1 = require("./SettingModel");
var TaxCodeModel_1 = require("./TaxCodeModel");
var TaxRuleModel_1 = require("./TaxRuleModel");
var UPCModel_1 = require("./UPCModel");
var EcmsModel_1 = require("./EcmsModel");
var CompanyParameterDetailModel_1 = require("./CompanyParameterDetailModel");
var CustomerSupplierModel_1 = require("./CustomerSupplierModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * A company or business entity.
 * @export
 * @class CompanyModel
 */
var CompanyModel = exports.CompanyModel = /** @class */ (function () {
    function CompanyModel() {
        /**
         * @type {number}
         * @memberof CompanyModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof CompanyModel
         */
        this.accountId = undefined;
        /**
         * @type {number}
         * @memberof CompanyModel
         */
        this.parentCompanyId = undefined;
        /**
         * @type {string}
         * @memberof CompanyModel
         */
        this.sstPid = undefined;
        /**
         * @type {string}
         * @memberof CompanyModel
         */
        this.companyCode = undefined;
        /**
         * @type {string}
         * @memberof CompanyModel
         */
        this.name = undefined;
        /**
         * @type {boolean}
         * @memberof CompanyModel
         */
        this.isDefault = undefined;
        /**
         * @type {number}
         * @memberof CompanyModel
         */
        this.defaultLocationId = undefined;
        /**
         * @type {boolean}
         * @memberof CompanyModel
         */
        this.isActive = undefined;
        /**
         * @type {string}
         * @memberof CompanyModel
         */
        this.taxpayerIdNumber = undefined;
        /**
         * @type {boolean}
         * @memberof CompanyModel
         */
        this.isFein = undefined;
        /**
         * @type {boolean}
         * @memberof CompanyModel
         */
        this.hasProfile = undefined;
        /**
         * @type {boolean}
         * @memberof CompanyModel
         */
        this.isReportingEntity = undefined;
        /**
         * @type {Date}
         * @memberof CompanyModel
         */
        this.sstEffectiveDate = undefined;
        /**
         * @type {string}
         * @memberof CompanyModel
         */
        this.defaultCountry = undefined;
        /**
         * @type {string}
         * @memberof CompanyModel
         */
        this.baseCurrencyCode = undefined;
        /**
         * @type {Enums.RoundingLevelId}
         * @memberof CompanyModel
         */
        this.roundingLevelId = undefined;
        /**
         * @type {boolean}
         * @memberof CompanyModel
         */
        this.warningsEnabled = undefined;
        /**
         * @type {boolean}
         * @memberof CompanyModel
         */
        this.isTest = undefined;
        /**
         * @type {Enums.TaxDependencyLevelId}
         * @memberof CompanyModel
         */
        this.taxDependencyLevelId = undefined;
        /**
         * @type {boolean}
         * @memberof CompanyModel
         */
        this.inProgress = undefined;
        /**
         * @type {string}
         * @memberof CompanyModel
         */
        this.businessIdentificationNo = undefined;
        /**
         * @type {Date}
         * @memberof CompanyModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof CompanyModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof CompanyModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof CompanyModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {ContactModel[]}
         * @memberof CompanyModel
         */
        this.contacts = undefined;
        /**
         * @type {ItemModel[]}
         * @memberof CompanyModel
         */
        this.items = undefined;
        /**
         * @type {LocationModel[]}
         * @memberof CompanyModel
         */
        this.locations = undefined;
        /**
         * @type {NexusModel[]}
         * @memberof CompanyModel
         */
        this.nexus = undefined;
        /**
         * @type {SettingModel[]}
         * @memberof CompanyModel
         */
        this.settings = undefined;
        /**
         * @type {TaxCodeModel[]}
         * @memberof CompanyModel
         */
        this.taxCodes = undefined;
        /**
         * @type {TaxRuleModel[]}
         * @memberof CompanyModel
         */
        this.taxRules = undefined;
        /**
         * @type {UPCModel[]}
         * @memberof CompanyModel
         */
        this.upcs = undefined;
        /**
         * @type {CompanyModel[]}
         * @memberof CompanyModel
         */
        this.nonReportingChildCompanies = undefined;
        /**
         * @type {EcmsModel[]}
         * @memberof CompanyModel
         */
        this.exemptCerts = undefined;
        /**
         * @type {string}
         * @memberof CompanyModel
         */
        this.mossId = undefined;
        /**
         * @type {string}
         * @memberof CompanyModel
         */
        this.mossCountry = undefined;
        /**
         * @type {CompanyParameterDetailModel[]}
         * @memberof CompanyModel
         */
        this.parameters = undefined;
        /**
         * @type {CustomerSupplierModel[]}
         * @memberof CompanyModel
         */
        this.supplierandcustomers = undefined;
    }
    CompanyModel_1 = CompanyModel;
    var CompanyModel_1;
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number),
        __metadata("design:type", Number)
    ], CompanyModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountId", Number),
        __metadata("design:type", Number)
    ], CompanyModel.prototype, "accountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("parentCompanyId", Number, true),
        __metadata("design:type", Number)
    ], CompanyModel.prototype, "parentCompanyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("sstPid", String, true),
        __metadata("design:type", String)
    ], CompanyModel.prototype, "sstPid", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyCode", String, true),
        __metadata("design:type", String)
    ], CompanyModel.prototype, "companyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String),
        __metadata("design:type", String)
    ], CompanyModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isDefault", Boolean, true),
        __metadata("design:type", Boolean)
    ], CompanyModel.prototype, "isDefault", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("defaultLocationId", Number, true),
        __metadata("design:type", Number)
    ], CompanyModel.prototype, "defaultLocationId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isActive", Boolean, true),
        __metadata("design:type", Boolean)
    ], CompanyModel.prototype, "isActive", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxpayerIdNumber", String, true),
        __metadata("design:type", String)
    ], CompanyModel.prototype, "taxpayerIdNumber", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isFein", Boolean, true),
        __metadata("design:type", Boolean)
    ], CompanyModel.prototype, "isFein", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hasProfile", Boolean, true),
        __metadata("design:type", Boolean)
    ], CompanyModel.prototype, "hasProfile", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isReportingEntity", Boolean, true),
        __metadata("design:type", Boolean)
    ], CompanyModel.prototype, "isReportingEntity", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("sstEffectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CompanyModel.prototype, "sstEffectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("defaultCountry", String),
        __metadata("design:type", String)
    ], CompanyModel.prototype, "defaultCountry", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("baseCurrencyCode", String, true),
        __metadata("design:type", String)
    ], CompanyModel.prototype, "baseCurrencyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("roundingLevelId", Enums.RoundingLevelIdConverter, true),
        __metadata("design:type", Number)
    ], CompanyModel.prototype, "roundingLevelId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("warningsEnabled", Boolean, true),
        __metadata("design:type", Boolean)
    ], CompanyModel.prototype, "warningsEnabled", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isTest", Boolean, true),
        __metadata("design:type", Boolean)
    ], CompanyModel.prototype, "isTest", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxDependencyLevelId", Enums.TaxDependencyLevelIdConverter, true),
        __metadata("design:type", Number)
    ], CompanyModel.prototype, "taxDependencyLevelId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("inProgress", Boolean, true),
        __metadata("design:type", Boolean)
    ], CompanyModel.prototype, "inProgress", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("businessIdentificationNo", String, true),
        __metadata("design:type", String)
    ], CompanyModel.prototype, "businessIdentificationNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CompanyModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], CompanyModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CompanyModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], CompanyModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("contacts", [ContactModel_1.ContactModel], true),
        __metadata("design:type", Array)
    ], CompanyModel.prototype, "contacts", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("items", [ItemModel_1.ItemModel], true),
        __metadata("design:type", Array)
    ], CompanyModel.prototype, "items", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("locations", [LocationModel_1.LocationModel], true),
        __metadata("design:type", Array)
    ], CompanyModel.prototype, "locations", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nexus", [NexusModel_1.NexusModel], true),
        __metadata("design:type", Array)
    ], CompanyModel.prototype, "nexus", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("settings", [SettingModel_1.SettingModel], true),
        __metadata("design:type", Array)
    ], CompanyModel.prototype, "settings", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCodes", [TaxCodeModel_1.TaxCodeModel], true),
        __metadata("design:type", Array)
    ], CompanyModel.prototype, "taxCodes", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxRules", [TaxRuleModel_1.TaxRuleModel], true),
        __metadata("design:type", Array)
    ], CompanyModel.prototype, "taxRules", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("upcs", [UPCModel_1.UPCModel], true),
        __metadata("design:type", Array)
    ], CompanyModel.prototype, "upcs", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonReportingChildCompanies", [CompanyModel_1], true),
        __metadata("design:type", Array)
    ], CompanyModel.prototype, "nonReportingChildCompanies", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptCerts", [EcmsModel_1.EcmsModel], true),
        __metadata("design:type", Array)
    ], CompanyModel.prototype, "exemptCerts", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("mossId", String, true),
        __metadata("design:type", String)
    ], CompanyModel.prototype, "mossId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("mossCountry", String, true),
        __metadata("design:type", String)
    ], CompanyModel.prototype, "mossCountry", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("parameters", [CompanyParameterDetailModel_1.CompanyParameterDetailModel], true),
        __metadata("design:type", Array)
    ], CompanyModel.prototype, "parameters", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("supplierandcustomers", [CustomerSupplierModel_1.CustomerSupplierModel], true),
        __metadata("design:type", Array)
    ], CompanyModel.prototype, "supplierandcustomers", void 0);
    CompanyModel = CompanyModel_1 = __decorate([
        (0, json2typescript_1.JsonObject)("CompanyModel")
    ], CompanyModel);
    return CompanyModel;
}());
