// avatax/lib/models/ContactModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContactModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * A contact person for a company.
 * @export
 * @class ContactModel
 */
var ContactModel = exports.ContactModel = /** @class */ (function () {
    function ContactModel() {
        /**
         * @type {number}
         * @memberof ContactModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof ContactModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof ContactModel
         */
        this.contactCode = undefined;
        /**
         * @type {string}
         * @memberof ContactModel
         */
        this.firstName = undefined;
        /**
         * @type {string}
         * @memberof ContactModel
         */
        this.middleName = undefined;
        /**
         * @type {string}
         * @memberof ContactModel
         */
        this.lastName = undefined;
        /**
         * @type {string}
         * @memberof ContactModel
         */
        this.title = undefined;
        /**
         * @type {string}
         * @memberof ContactModel
         */
        this.line1 = undefined;
        /**
         * @type {string}
         * @memberof ContactModel
         */
        this.line2 = undefined;
        /**
         * @type {string}
         * @memberof ContactModel
         */
        this.line3 = undefined;
        /**
         * @type {string}
         * @memberof ContactModel
         */
        this.city = undefined;
        /**
         * @type {string}
         * @memberof ContactModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof ContactModel
         */
        this.postalCode = undefined;
        /**
         * @type {string}
         * @memberof ContactModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof ContactModel
         */
        this.email = undefined;
        /**
         * @type {string}
         * @memberof ContactModel
         */
        this.phone = undefined;
        /**
         * @type {string}
         * @memberof ContactModel
         */
        this.mobile = undefined;
        /**
         * @type {string}
         * @memberof ContactModel
         */
        this.fax = undefined;
        /**
         * @type {Date}
         * @memberof ContactModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof ContactModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof ContactModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof ContactModel
         */
        this.modifiedUserId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], ContactModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], ContactModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("contactCode", String),
        __metadata("design:type", String)
    ], ContactModel.prototype, "contactCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("firstName", String, true),
        __metadata("design:type", String)
    ], ContactModel.prototype, "firstName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("middleName", String, true),
        __metadata("design:type", String)
    ], ContactModel.prototype, "middleName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lastName", String, true),
        __metadata("design:type", String)
    ], ContactModel.prototype, "lastName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("title", String, true),
        __metadata("design:type", String)
    ], ContactModel.prototype, "title", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line1", String, true),
        __metadata("design:type", String)
    ], ContactModel.prototype, "line1", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line2", String, true),
        __metadata("design:type", String)
    ], ContactModel.prototype, "line2", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line3", String, true),
        __metadata("design:type", String)
    ], ContactModel.prototype, "line3", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("city", String, true),
        __metadata("design:type", String)
    ], ContactModel.prototype, "city", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], ContactModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("postalCode", String, true),
        __metadata("design:type", String)
    ], ContactModel.prototype, "postalCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], ContactModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("email", String, true),
        __metadata("design:type", String)
    ], ContactModel.prototype, "email", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("phone", String, true),
        __metadata("design:type", String)
    ], ContactModel.prototype, "phone", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("mobile", String, true),
        __metadata("design:type", String)
    ], ContactModel.prototype, "mobile", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("fax", String, true),
        __metadata("design:type", String)
    ], ContactModel.prototype, "fax", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ContactModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], ContactModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ContactModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], ContactModel.prototype, "modifiedUserId", void 0);
    ContactModel = __decorate([
        (0, json2typescript_1.JsonObject)("ContactModel")
    ], ContactModel);
    return ContactModel;
}());
