// avatax/lib/models/CoverLetterModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CoverLetterModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * The CoverLetter model represents a message sent along with an invitation to use CertExpress to
upload certificates.  An invitation allows customers to use CertExpress to upload their exemption
certificates directly; this cover letter explains why the invitation was sent.
 * @export
 * @class CoverLetterModel
 */
var CoverLetterModel = exports.CoverLetterModel = /** @class */ (function () {
    function CoverLetterModel() {
        /**
         * @type {number}
         * @memberof CoverLetterModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof CoverLetterModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof CoverLetterModel
         */
        this.title = undefined;
        /**
         * @type {string}
         * @memberof CoverLetterModel
         */
        this.subject = undefined;
        /**
         * @type {string}
         * @memberof CoverLetterModel
         */
        this.description = undefined;
        /**
         * @type {Date}
         * @memberof CoverLetterModel
         */
        this.createdDate = undefined;
        /**
         * @type {Date}
         * @memberof CoverLetterModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {boolean}
         * @memberof CoverLetterModel
         */
        this.active = undefined;
        /**
         * @type {number}
         * @memberof CoverLetterModel
         */
        this.pageCount = undefined;
        /**
         * @type {string}
         * @memberof CoverLetterModel
         */
        this.templateFilename = undefined;
        /**
         * @type {number}
         * @memberof CoverLetterModel
         */
        this.version = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], CoverLetterModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], CoverLetterModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("title", String, true),
        __metadata("design:type", String)
    ], CoverLetterModel.prototype, "title", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("subject", String, true),
        __metadata("design:type", String)
    ], CoverLetterModel.prototype, "subject", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], CoverLetterModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CoverLetterModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CoverLetterModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("active", Boolean, true),
        __metadata("design:type", Boolean)
    ], CoverLetterModel.prototype, "active", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("pageCount", Number, true),
        __metadata("design:type", Number)
    ], CoverLetterModel.prototype, "pageCount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("templateFilename", String, true),
        __metadata("design:type", String)
    ], CoverLetterModel.prototype, "templateFilename", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("version", Number, true),
        __metadata("design:type", Number)
    ], CoverLetterModel.prototype, "version", void 0);
    CoverLetterModel = __decorate([
        (0, json2typescript_1.JsonObject)("CoverLetterModel")
    ], CoverLetterModel);
    return CoverLetterModel;
}());
