// avatax/lib/models/CreateTransactionModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateTransactionModel = void 0;
var Enums = require("../enums/index");
var LineItemModel_1 = require("./LineItemModel");
var AddressesModel_1 = require("./AddressesModel");
var TransactionParameterModel_1 = require("./TransactionParameterModel");
var TransactionUserDefinedFieldModel_1 = require("./TransactionUserDefinedFieldModel");
var TaxOverrideModel_1 = require("./TaxOverrideModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Create a transaction
 * @export
 * @class CreateTransactionModel
 */
var CreateTransactionModel = exports.CreateTransactionModel = /** @class */ (function () {
    function CreateTransactionModel() {
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.code = undefined;
        /**
         * @type {LineItemModel[]}
         * @memberof CreateTransactionModel
         */
        this.lines = undefined;
        /**
         * @type {Enums.DocumentType}
         * @memberof CreateTransactionModel
         */
        this.type = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.companyCode = undefined;
        /**
         * @type {Date}
         * @memberof CreateTransactionModel
         */
        this.date = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.salespersonCode = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.customerCode = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.customerUsageType = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.entityUseCode = undefined;
        /**
         * @type {number}
         * @memberof CreateTransactionModel
         */
        this.discount = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.purchaseOrderNo = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.exemptionNo = undefined;
        /**
         * @type {AddressesModel}
         * @memberof CreateTransactionModel
         */
        this.addresses = undefined;
        /**
         * @type {TransactionParameterModel[]}
         * @memberof CreateTransactionModel
         */
        this.parameters = undefined;
        /**
         * @type {TransactionUserDefinedFieldModel[]}
         * @memberof CreateTransactionModel
         */
        this.userDefinedFields = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.referenceCode = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.reportingLocationCode = undefined;
        /**
         * @type {boolean}
         * @memberof CreateTransactionModel
         */
        this.commit = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.batchCode = undefined;
        /**
         * @type {TaxOverrideModel}
         * @memberof CreateTransactionModel
         */
        this.taxOverride = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.currencyCode = undefined;
        /**
         * @type {Enums.ServiceMode}
         * @memberof CreateTransactionModel
         */
        this.serviceMode = undefined;
        /**
         * @type {number}
         * @memberof CreateTransactionModel
         */
        this.exchangeRate = undefined;
        /**
         * @type {Date}
         * @memberof CreateTransactionModel
         */
        this.exchangeRateEffectiveDate = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.exchangeRateCurrencyCode = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.posLaneCode = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.businessIdentificationNo = undefined;
        /**
         * @type {boolean}
         * @memberof CreateTransactionModel
         */
        this.isSellerImporterOfRecord = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.description = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.email = undefined;
        /**
         * @type {Enums.TaxDebugLevel}
         * @memberof CreateTransactionModel
         */
        this.debugLevel = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionModel
         */
        this.customerSupplierName = undefined;
        /**
         * @type {number}
         * @memberof CreateTransactionModel
         */
        this.dataSourceId = undefined;
        /**
         * @type {Enums.DeliveryTerms}
         * @memberof CreateTransactionModel
         */
        this.deliveryTerms = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("code", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "code", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lines", [LineItemModel_1.LineItemModel]),
        __metadata("design:type", Array)
    ], CreateTransactionModel.prototype, "lines", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("type", Enums.DocumentTypeConverter, true),
        __metadata("design:type", Number)
    ], CreateTransactionModel.prototype, "type", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyCode", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "companyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("date", dateConverter_1.DateConverter),
        __metadata("design:type", Date)
    ], CreateTransactionModel.prototype, "date", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("salespersonCode", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "salespersonCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerCode", String),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "customerCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerUsageType", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "customerUsageType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("entityUseCode", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "entityUseCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("discount", Number, true),
        __metadata("design:type", Number)
    ], CreateTransactionModel.prototype, "discount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("purchaseOrderNo", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "purchaseOrderNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptionNo", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "exemptionNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("addresses", AddressesModel_1.AddressesModel, true),
        __metadata("design:type", AddressesModel_1.AddressesModel)
    ], CreateTransactionModel.prototype, "addresses", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("parameters", [TransactionParameterModel_1.TransactionParameterModel], true),
        __metadata("design:type", Array)
    ], CreateTransactionModel.prototype, "parameters", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("userDefinedFields", [TransactionUserDefinedFieldModel_1.TransactionUserDefinedFieldModel], true),
        __metadata("design:type", Array)
    ], CreateTransactionModel.prototype, "userDefinedFields", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("referenceCode", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "referenceCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportingLocationCode", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "reportingLocationCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("commit", Boolean, true),
        __metadata("design:type", Boolean)
    ], CreateTransactionModel.prototype, "commit", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("batchCode", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "batchCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxOverride", TaxOverrideModel_1.TaxOverrideModel, true),
        __metadata("design:type", TaxOverrideModel_1.TaxOverrideModel)
    ], CreateTransactionModel.prototype, "taxOverride", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("currencyCode", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "currencyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("serviceMode", Enums.ServiceModeConverter, true),
        __metadata("design:type", Number)
    ], CreateTransactionModel.prototype, "serviceMode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exchangeRate", Number, true),
        __metadata("design:type", Number)
    ], CreateTransactionModel.prototype, "exchangeRate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exchangeRateEffectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CreateTransactionModel.prototype, "exchangeRateEffectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exchangeRateCurrencyCode", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "exchangeRateCurrencyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("posLaneCode", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "posLaneCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("businessIdentificationNo", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "businessIdentificationNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSellerImporterOfRecord", Boolean, true),
        __metadata("design:type", Boolean)
    ], CreateTransactionModel.prototype, "isSellerImporterOfRecord", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("email", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "email", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("debugLevel", Enums.TaxDebugLevelConverter, true),
        __metadata("design:type", Number)
    ], CreateTransactionModel.prototype, "debugLevel", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerSupplierName", String, true),
        __metadata("design:type", String)
    ], CreateTransactionModel.prototype, "customerSupplierName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dataSourceId", Number, true),
        __metadata("design:type", Number)
    ], CreateTransactionModel.prototype, "dataSourceId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("deliveryTerms", Enums.DeliveryTermsConverter, true),
        __metadata("design:type", Number)
    ], CreateTransactionModel.prototype, "deliveryTerms", void 0);
    CreateTransactionModel = __decorate([
        (0, json2typescript_1.JsonObject)("CreateTransactionModel")
    ], CreateTransactionModel);
    return CreateTransactionModel;
}());
