// avatax/lib/models/CreateTransactionBatchResponseModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateTransactionBatchResponseModel = void 0;
var Enums = require("../enums/index");
var BatchFileModel_1 = require("./BatchFileModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a create transaction batch response model.
 * @export
 * @class CreateTransactionBatchResponseModel
 */
var CreateTransactionBatchResponseModel = exports.CreateTransactionBatchResponseModel = /** @class */ (function () {
    function CreateTransactionBatchResponseModel() {
        /**
         * @type {number}
         * @memberof CreateTransactionBatchResponseModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof CreateTransactionBatchResponseModel
         */
        this.name = undefined;
        /**
         * @type {number}
         * @memberof CreateTransactionBatchResponseModel
         */
        this.accountId = undefined;
        /**
         * @type {number}
         * @memberof CreateTransactionBatchResponseModel
         */
        this.companyId = undefined;
        /**
         * @type {Enums.BatchStatus}
         * @memberof CreateTransactionBatchResponseModel
         */
        this.status = undefined;
        /**
         * @type {Date}
         * @memberof CreateTransactionBatchResponseModel
         */
        this.startedDate = undefined;
        /**
         * @type {number}
         * @memberof CreateTransactionBatchResponseModel
         */
        this.recordCount = undefined;
        /**
         * @type {number}
         * @memberof CreateTransactionBatchResponseModel
         */
        this.currentRecord = undefined;
        /**
         * @type {Date}
         * @memberof CreateTransactionBatchResponseModel
         */
        this.completedDate = undefined;
        /**
         * @type {Date}
         * @memberof CreateTransactionBatchResponseModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof CreateTransactionBatchResponseModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof CreateTransactionBatchResponseModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof CreateTransactionBatchResponseModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {BatchFileModel[]}
         * @memberof CreateTransactionBatchResponseModel
         */
        this.files = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], CreateTransactionBatchResponseModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String),
        __metadata("design:type", String)
    ], CreateTransactionBatchResponseModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountId", Number, true),
        __metadata("design:type", Number)
    ], CreateTransactionBatchResponseModel.prototype, "accountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], CreateTransactionBatchResponseModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("status", Enums.BatchStatusConverter, true),
        __metadata("design:type", Number)
    ], CreateTransactionBatchResponseModel.prototype, "status", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("startedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CreateTransactionBatchResponseModel.prototype, "startedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("recordCount", Number, true),
        __metadata("design:type", Number)
    ], CreateTransactionBatchResponseModel.prototype, "recordCount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("currentRecord", Number, true),
        __metadata("design:type", Number)
    ], CreateTransactionBatchResponseModel.prototype, "currentRecord", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("completedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CreateTransactionBatchResponseModel.prototype, "completedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CreateTransactionBatchResponseModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], CreateTransactionBatchResponseModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CreateTransactionBatchResponseModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], CreateTransactionBatchResponseModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("files", [BatchFileModel_1.BatchFileModel], true),
        __metadata("design:type", Array)
    ], CreateTransactionBatchResponseModel.prototype, "files", void 0);
    CreateTransactionBatchResponseModel = __decorate([
        (0, json2typescript_1.JsonObject)("CreateTransactionBatchResponseModel")
    ], CreateTransactionBatchResponseModel);
    return CreateTransactionBatchResponseModel;
}());
