// avatax/lib/models/CustomerModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerModel = void 0;
var CertificateModel_1 = require("./CertificateModel");
var CustomFieldModel_1 = require("./CustomFieldModel");
var ExposureZoneModel_1 = require("./ExposureZoneModel");
var CustomerAttributeModel_1 = require("./CustomerAttributeModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a customer to whom you sell products and/or services.
 * @export
 * @class CustomerModel
 */
var CustomerModel = exports.CustomerModel = /** @class */ (function () {
    function CustomerModel() {
        /**
         * @type {number}
         * @memberof CustomerModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof CustomerModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof CustomerModel
         */
        this.customerCode = undefined;
        /**
         * @type {string}
         * @memberof CustomerModel
         */
        this.alternateId = undefined;
        /**
         * @type {string}
         * @memberof CustomerModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof CustomerModel
         */
        this.attnName = undefined;
        /**
         * @type {string}
         * @memberof CustomerModel
         */
        this.line1 = undefined;
        /**
         * @type {string}
         * @memberof CustomerModel
         */
        this.line2 = undefined;
        /**
         * @type {string}
         * @memberof CustomerModel
         */
        this.city = undefined;
        /**
         * @type {string}
         * @memberof CustomerModel
         */
        this.postalCode = undefined;
        /**
         * @type {string}
         * @memberof CustomerModel
         */
        this.phoneNumber = undefined;
        /**
         * @type {string}
         * @memberof CustomerModel
         */
        this.faxNumber = undefined;
        /**
         * @type {string}
         * @memberof CustomerModel
         */
        this.emailAddress = undefined;
        /**
         * @type {string}
         * @memberof CustomerModel
         */
        this.contactName = undefined;
        /**
         * @type {Date}
         * @memberof CustomerModel
         */
        this.lastTransaction = undefined;
        /**
         * @type {Date}
         * @memberof CustomerModel
         */
        this.createdDate = undefined;
        /**
         * @type {Date}
         * @memberof CustomerModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {string}
         * @memberof CustomerModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof CustomerModel
         */
        this.region = undefined;
        /**
         * @type {boolean}
         * @memberof CustomerModel
         */
        this.isBill = undefined;
        /**
         * @type {boolean}
         * @memberof CustomerModel
         */
        this.isShip = undefined;
        /**
         * @type {string}
         * @memberof CustomerModel
         */
        this.taxpayerIdNumber = undefined;
        /**
         * @type {CertificateModel[]}
         * @memberof CustomerModel
         */
        this.certificates = undefined;
        /**
         * @type {CustomFieldModel[]}
         * @memberof CustomerModel
         */
        this.customFields = undefined;
        /**
         * @type {ExposureZoneModel[]}
         * @memberof CustomerModel
         */
        this.exposureZones = undefined;
        /**
         * @type {CustomerModel[]}
         * @memberof CustomerModel
         */
        this.shipTos = undefined;
        /**
         * @type {CustomerAttributeModel[]}
         * @memberof CustomerModel
         */
        this.attributes = undefined;
    }
    CustomerModel_1 = CustomerModel;
    var CustomerModel_1;
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], CustomerModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number),
        __metadata("design:type", Number)
    ], CustomerModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerCode", String),
        __metadata("design:type", String)
    ], CustomerModel.prototype, "customerCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("alternateId", String, true),
        __metadata("design:type", String)
    ], CustomerModel.prototype, "alternateId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String),
        __metadata("design:type", String)
    ], CustomerModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("attnName", String, true),
        __metadata("design:type", String)
    ], CustomerModel.prototype, "attnName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line1", String),
        __metadata("design:type", String)
    ], CustomerModel.prototype, "line1", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line2", String, true),
        __metadata("design:type", String)
    ], CustomerModel.prototype, "line2", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("city", String),
        __metadata("design:type", String)
    ], CustomerModel.prototype, "city", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("postalCode", String),
        __metadata("design:type", String)
    ], CustomerModel.prototype, "postalCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("phoneNumber", String, true),
        __metadata("design:type", String)
    ], CustomerModel.prototype, "phoneNumber", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("faxNumber", String, true),
        __metadata("design:type", String)
    ], CustomerModel.prototype, "faxNumber", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("emailAddress", String, true),
        __metadata("design:type", String)
    ], CustomerModel.prototype, "emailAddress", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("contactName", String, true),
        __metadata("design:type", String)
    ], CustomerModel.prototype, "contactName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lastTransaction", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CustomerModel.prototype, "lastTransaction", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CustomerModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CustomerModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String),
        __metadata("design:type", String)
    ], CustomerModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], CustomerModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isBill", Boolean, true),
        __metadata("design:type", Boolean)
    ], CustomerModel.prototype, "isBill", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isShip", Boolean, true),
        __metadata("design:type", Boolean)
    ], CustomerModel.prototype, "isShip", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxpayerIdNumber", String, true),
        __metadata("design:type", String)
    ], CustomerModel.prototype, "taxpayerIdNumber", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("certificates", [CertificateModel_1.CertificateModel], true),
        __metadata("design:type", Array)
    ], CustomerModel.prototype, "certificates", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customFields", [CustomFieldModel_1.CustomFieldModel], true),
        __metadata("design:type", Array)
    ], CustomerModel.prototype, "customFields", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exposureZones", [ExposureZoneModel_1.ExposureZoneModel], true),
        __metadata("design:type", Array)
    ], CustomerModel.prototype, "exposureZones", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("shipTos", [CustomerModel_1], true),
        __metadata("design:type", Array)
    ], CustomerModel.prototype, "shipTos", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("attributes", [CustomerAttributeModel_1.CustomerAttributeModel], true),
        __metadata("design:type", Array)
    ], CustomerModel.prototype, "attributes", void 0);
    CustomerModel = CustomerModel_1 = __decorate([
        (0, json2typescript_1.JsonObject)("CustomerModel")
    ], CustomerModel);
    return CustomerModel;
}());
