// avatax/lib/models/CustomerSupplierCountryParamModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerSupplierCountryParamModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Represents a parameter associated with a company.
 * @export
 * @class CustomerSupplierCountryParamModel
 */
var CustomerSupplierCountryParamModel = exports.CustomerSupplierCountryParamModel = /** @class */ (function () {
    function CustomerSupplierCountryParamModel() {
        /**
         * @type {number}
         * @memberof CustomerSupplierCountryParamModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof CustomerSupplierCountryParamModel
         */
        this.companyId = undefined;
        /**
         * @type {number}
         * @memberof CustomerSupplierCountryParamModel
         */
        this.customerId = undefined;
        /**
         * @type {string}
         * @memberof CustomerSupplierCountryParamModel
         */
        this.customerCode = undefined;
        /**
         * @type {string}
         * @memberof CustomerSupplierCountryParamModel
         */
        this.country = undefined;
        /**
         * @type {boolean}
         * @memberof CustomerSupplierCountryParamModel
         */
        this.isEstablished = undefined;
        /**
         * @type {string}
         * @memberof CustomerSupplierCountryParamModel
         */
        this.businessIdentificationNo = undefined;
        /**
         * @type {boolean}
         * @memberof CustomerSupplierCountryParamModel
         */
        this.isRegisteredThroughFiscalRep = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], CustomerSupplierCountryParamModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], CustomerSupplierCountryParamModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerId", Number, true),
        __metadata("design:type", Number)
    ], CustomerSupplierCountryParamModel.prototype, "customerId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerCode", String, true),
        __metadata("design:type", String)
    ], CustomerSupplierCountryParamModel.prototype, "customerCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String),
        __metadata("design:type", String)
    ], CustomerSupplierCountryParamModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isEstablished", Boolean, true),
        __metadata("design:type", Boolean)
    ], CustomerSupplierCountryParamModel.prototype, "isEstablished", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("businessIdentificationNo", String, true),
        __metadata("design:type", String)
    ], CustomerSupplierCountryParamModel.prototype, "businessIdentificationNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isRegisteredThroughFiscalRep", Boolean, true),
        __metadata("design:type", Boolean)
    ], CustomerSupplierCountryParamModel.prototype, "isRegisteredThroughFiscalRep", void 0);
    CustomerSupplierCountryParamModel = __decorate([
        (0, json2typescript_1.JsonObject)("CustomerSupplierCountryParamModel")
    ], CustomerSupplierCountryParamModel);
    return CustomerSupplierCountryParamModel;
}());
