// avatax/lib/models/CycleAddOptionModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CycleAddOptionModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Model with options for adding a new filing calendar
 * @export
 * @class CycleAddOptionModel
 */
var CycleAddOptionModel = exports.CycleAddOptionModel = /** @class */ (function () {
    function CycleAddOptionModel() {
        /**
         * @type {boolean}
         * @memberof CycleAddOptionModel
         */
        this.available = undefined;
        /**
         * @type {Date}
         * @memberof CycleAddOptionModel
         */
        this.transactionalPeriodStart = undefined;
        /**
         * @type {Date}
         * @memberof CycleAddOptionModel
         */
        this.transactionalPeriodEnd = undefined;
        /**
         * @type {Date}
         * @memberof CycleAddOptionModel
         */
        this.filingDueDate = undefined;
        /**
         * @type {string}
         * @memberof CycleAddOptionModel
         */
        this.cycleName = undefined;
        /**
         * @type {string}
         * @memberof CycleAddOptionModel
         */
        this.frequencyName = undefined;
        /**
         * @type {string}
         * @memberof CycleAddOptionModel
         */
        this.filingFrequencyCode = undefined;
        /**
         * @type {Enums.FilingFrequencyId}
         * @memberof CycleAddOptionModel
         */
        this.filingFrequencyId = undefined;
        /**
         * @type {string}
         * @memberof CycleAddOptionModel
         */
        this.cycleUnavailableReason = undefined;
        /**
         * @type {string[]}
         * @memberof CycleAddOptionModel
         */
        this.availableLocationCodes = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("available", Boolean, true),
        __metadata("design:type", Boolean)
    ], CycleAddOptionModel.prototype, "available", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("transactionalPeriodStart", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CycleAddOptionModel.prototype, "transactionalPeriodStart", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("transactionalPeriodEnd", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CycleAddOptionModel.prototype, "transactionalPeriodEnd", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingDueDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CycleAddOptionModel.prototype, "filingDueDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("cycleName", String, true),
        __metadata("design:type", String)
    ], CycleAddOptionModel.prototype, "cycleName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("frequencyName", String, true),
        __metadata("design:type", String)
    ], CycleAddOptionModel.prototype, "frequencyName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingFrequencyCode", String, true),
        __metadata("design:type", String)
    ], CycleAddOptionModel.prototype, "filingFrequencyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingFrequencyId", Enums.FilingFrequencyIdConverter, true),
        __metadata("design:type", Number)
    ], CycleAddOptionModel.prototype, "filingFrequencyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("cycleUnavailableReason", String, true),
        __metadata("design:type", String)
    ], CycleAddOptionModel.prototype, "cycleUnavailableReason", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("availableLocationCodes", [String], true),
        __metadata("design:type", Array)
    ], CycleAddOptionModel.prototype, "availableLocationCodes", void 0);
    CycleAddOptionModel = __decorate([
        (0, json2typescript_1.JsonObject)("CycleAddOptionModel")
    ], CycleAddOptionModel);
    return CycleAddOptionModel;
}());
