// avatax/lib/models/CycleSafeFilingCalendarEditModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CycleSafeFilingCalendarEditModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Filing Calendar Edit
 * @export
 * @class CycleSafeFilingCalendarEditModel
 */
var CycleSafeFilingCalendarEditModel = exports.CycleSafeFilingCalendarEditModel = /** @class */ (function () {
    function CycleSafeFilingCalendarEditModel() {
        /**
         * @type {string}
         * @memberof CycleSafeFilingCalendarEditModel
         */
        this.fieldName = undefined;
        /**
         * @type {string}
         * @memberof CycleSafeFilingCalendarEditModel
         */
        this.destination = undefined;
        /**
         * @type {number}
         * @memberof CycleSafeFilingCalendarEditModel
         */
        this.questionId = undefined;
        /**
         * @type {string}
         * @memberof CycleSafeFilingCalendarEditModel
         */
        this.questionCode = undefined;
        /**
         * @type {object}
         * @memberof CycleSafeFilingCalendarEditModel
         */
        this.oldValue = undefined;
        /**
         * @type {object}
         * @memberof CycleSafeFilingCalendarEditModel
         */
        this.newValue = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("fieldName", String, true),
        __metadata("design:type", String)
    ], CycleSafeFilingCalendarEditModel.prototype, "fieldName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("destination", String, true),
        __metadata("design:type", String)
    ], CycleSafeFilingCalendarEditModel.prototype, "destination", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("questionId", Number, true),
        __metadata("design:type", Number)
    ], CycleSafeFilingCalendarEditModel.prototype, "questionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("questionCode", String, true),
        __metadata("design:type", String)
    ], CycleSafeFilingCalendarEditModel.prototype, "questionCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("oldValue", Object, true),
        __metadata("design:type", Object)
    ], CycleSafeFilingCalendarEditModel.prototype, "oldValue", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("newValue", Object, true),
        __metadata("design:type", Object)
    ], CycleSafeFilingCalendarEditModel.prototype, "newValue", void 0);
    CycleSafeFilingCalendarEditModel = __decorate([
        (0, json2typescript_1.JsonObject)("CycleSafeFilingCalendarEditModel")
    ], CycleSafeFilingCalendarEditModel);
    return CycleSafeFilingCalendarEditModel;
}());
