// avatax/lib/models/DeclareNexusByAddressModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeclareNexusByAddressModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Use this object to provide an address and date range where your company does business.
This address will be used to determine what jurisdictions you should declare nexus and
calculate tax.
 * @export
 * @class DeclareNexusByAddressModel
 */
var DeclareNexusByAddressModel = exports.DeclareNexusByAddressModel = /** @class */ (function () {
    function DeclareNexusByAddressModel() {
        /**
         * @type {Date}
         * @memberof DeclareNexusByAddressModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof DeclareNexusByAddressModel
         */
        this.endDate = undefined;
        /**
         * @type {string}
         * @memberof DeclareNexusByAddressModel
         */
        this.taxTypeGroup = undefined;
        /**
         * @type {string}
         * @memberof DeclareNexusByAddressModel
         */
        this.nexusTaxTypeGroup = undefined;
        /**
         * @type {string}
         * @memberof DeclareNexusByAddressModel
         */
        this.line1 = undefined;
        /**
         * @type {Enums.TextCase}
         * @memberof DeclareNexusByAddressModel
         */
        this.textCase = undefined;
        /**
         * @type {string}
         * @memberof DeclareNexusByAddressModel
         */
        this.line2 = undefined;
        /**
         * @type {string}
         * @memberof DeclareNexusByAddressModel
         */
        this.line3 = undefined;
        /**
         * @type {string}
         * @memberof DeclareNexusByAddressModel
         */
        this.city = undefined;
        /**
         * @type {string}
         * @memberof DeclareNexusByAddressModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof DeclareNexusByAddressModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof DeclareNexusByAddressModel
         */
        this.postalCode = undefined;
        /**
         * @type {number}
         * @memberof DeclareNexusByAddressModel
         */
        this.latitude = undefined;
        /**
         * @type {number}
         * @memberof DeclareNexusByAddressModel
         */
        this.longitude = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], DeclareNexusByAddressModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], DeclareNexusByAddressModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxTypeGroup", String, true),
        __metadata("design:type", String)
    ], DeclareNexusByAddressModel.prototype, "taxTypeGroup", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nexusTaxTypeGroup", String, true),
        __metadata("design:type", String)
    ], DeclareNexusByAddressModel.prototype, "nexusTaxTypeGroup", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line1", String, true),
        __metadata("design:type", String)
    ], DeclareNexusByAddressModel.prototype, "line1", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("textCase", Enums.TextCaseConverter, true),
        __metadata("design:type", Number)
    ], DeclareNexusByAddressModel.prototype, "textCase", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line2", String, true),
        __metadata("design:type", String)
    ], DeclareNexusByAddressModel.prototype, "line2", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line3", String, true),
        __metadata("design:type", String)
    ], DeclareNexusByAddressModel.prototype, "line3", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("city", String, true),
        __metadata("design:type", String)
    ], DeclareNexusByAddressModel.prototype, "city", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], DeclareNexusByAddressModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], DeclareNexusByAddressModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("postalCode", String, true),
        __metadata("design:type", String)
    ], DeclareNexusByAddressModel.prototype, "postalCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("latitude", Number, true),
        __metadata("design:type", Number)
    ], DeclareNexusByAddressModel.prototype, "latitude", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("longitude", Number, true),
        __metadata("design:type", Number)
    ], DeclareNexusByAddressModel.prototype, "longitude", void 0);
    DeclareNexusByAddressModel = __decorate([
        (0, json2typescript_1.JsonObject)("DeclareNexusByAddressModel")
    ], DeclareNexusByAddressModel);
    return DeclareNexusByAddressModel;
}());
