// avatax/lib/models/EcmsDetailModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EcmsDetailModel = void 0;
var EcmsDetailTaxCodeModel_1 = require("./EcmsDetailTaxCodeModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents an ECMS record, used internally by AvaTax to track information about exemptions.
 * @export
 * @class EcmsDetailModel
 */
var EcmsDetailModel = exports.EcmsDetailModel = /** @class */ (function () {
    function EcmsDetailModel() {
        /**
         * @type {number}
         * @memberof EcmsDetailModel
         */
        this.exemptCertDetailId = undefined;
        /**
         * @type {number}
         * @memberof EcmsDetailModel
         */
        this.exemptCertId = undefined;
        /**
         * @type {string}
         * @memberof EcmsDetailModel
         */
        this.stateFips = undefined;
        /**
         * @type {string}
         * @memberof EcmsDetailModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof EcmsDetailModel
         */
        this.idNo = undefined;
        /**
         * @type {string}
         * @memberof EcmsDetailModel
         */
        this.country = undefined;
        /**
         * @type {Date}
         * @memberof EcmsDetailModel
         */
        this.endDate = undefined;
        /**
         * @type {string}
         * @memberof EcmsDetailModel
         */
        this.idType = undefined;
        /**
         * @type {number}
         * @memberof EcmsDetailModel
         */
        this.isTaxCodeListExclusionList = undefined;
        /**
         * @type {EcmsDetailTaxCodeModel[]}
         * @memberof EcmsDetailModel
         */
        this.taxCodes = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptCertDetailId", Number),
        __metadata("design:type", Number)
    ], EcmsDetailModel.prototype, "exemptCertDetailId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptCertId", Number),
        __metadata("design:type", Number)
    ], EcmsDetailModel.prototype, "exemptCertId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("stateFips", String),
        __metadata("design:type", String)
    ], EcmsDetailModel.prototype, "stateFips", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String),
        __metadata("design:type", String)
    ], EcmsDetailModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("idNo", String, true),
        __metadata("design:type", String)
    ], EcmsDetailModel.prototype, "idNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String),
        __metadata("design:type", String)
    ], EcmsDetailModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], EcmsDetailModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("idType", String, true),
        __metadata("design:type", String)
    ], EcmsDetailModel.prototype, "idType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isTaxCodeListExclusionList", Number, true),
        __metadata("design:type", Number)
    ], EcmsDetailModel.prototype, "isTaxCodeListExclusionList", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCodes", [EcmsDetailTaxCodeModel_1.EcmsDetailTaxCodeModel], true),
        __metadata("design:type", Array)
    ], EcmsDetailModel.prototype, "taxCodes", void 0);
    EcmsDetailModel = __decorate([
        (0, json2typescript_1.JsonObject)("EcmsDetailModel")
    ], EcmsDetailModel);
    return EcmsDetailModel;
}());
