// avatax/lib/models/ExposureZoneModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExposureZoneModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Information about a physical area or zone in which a certificate can apply.
An exposure zone for an exemption certificate will generally be a tax authority such
as a state, country, or local government entity.
 * @export
 * @class ExposureZoneModel
 */
var ExposureZoneModel = exports.ExposureZoneModel = /** @class */ (function () {
    function ExposureZoneModel() {
        /**
         * @type {number}
         * @memberof ExposureZoneModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof ExposureZoneModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof ExposureZoneModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof ExposureZoneModel
         */
        this.tag = undefined;
        /**
         * @type {string}
         * @memberof ExposureZoneModel
         */
        this.description = undefined;
        /**
         * @type {Date}
         * @memberof ExposureZoneModel
         */
        this.created = undefined;
        /**
         * @type {Date}
         * @memberof ExposureZoneModel
         */
        this.modified = undefined;
        /**
         * @type {string}
         * @memberof ExposureZoneModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof ExposureZoneModel
         */
        this.country = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], ExposureZoneModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], ExposureZoneModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String, true),
        __metadata("design:type", String)
    ], ExposureZoneModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("tag", String, true),
        __metadata("design:type", String)
    ], ExposureZoneModel.prototype, "tag", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], ExposureZoneModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("created", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ExposureZoneModel.prototype, "created", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modified", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ExposureZoneModel.prototype, "modified", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], ExposureZoneModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], ExposureZoneModel.prototype, "country", void 0);
    ExposureZoneModel = __decorate([
        (0, json2typescript_1.JsonObject)("ExposureZoneModel")
    ], ExposureZoneModel);
    return ExposureZoneModel;
}());
