// avatax/lib/models/ExportDocumentLineModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExportDocumentLineModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * An input model for executing a report detailed to the document line level
 * @export
 * @class ExportDocumentLineModel
 */
var ExportDocumentLineModel = exports.ExportDocumentLineModel = /** @class */ (function () {
    function ExportDocumentLineModel() {
        /**
         * @type {Enums.ReportFormat}
         * @memberof ExportDocumentLineModel
         */
        this.format = undefined;
        /**
         * @type {Date}
         * @memberof ExportDocumentLineModel
         */
        this.startDate = undefined;
        /**
         * @type {Date}
         * @memberof ExportDocumentLineModel
         */
        this.endDate = undefined;
        /**
         * @type {string}
         * @memberof ExportDocumentLineModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof ExportDocumentLineModel
         */
        this.state = undefined;
        /**
         * @type {Enums.ReportDateFilter}
         * @memberof ExportDocumentLineModel
         */
        this.dateFilter = undefined;
        /**
         * @type {Enums.ReportDocType}
         * @memberof ExportDocumentLineModel
         */
        this.docType = undefined;
        /**
         * @type {string}
         * @memberof ExportDocumentLineModel
         */
        this.currencyCode = undefined;
        /**
         * @type {number}
         * @memberof ExportDocumentLineModel
         */
        this.numberOfPartitions = undefined;
        /**
         * @type {number}
         * @memberof ExportDocumentLineModel
         */
        this.partition = undefined;
        /**
         * @type {boolean}
         * @memberof ExportDocumentLineModel
         */
        this.isLocked = undefined;
        /**
         * @type {string}
         * @memberof ExportDocumentLineModel
         */
        this.merchantSellerIdentifier = undefined;
        /**
         * @type {Enums.DocumentStatus}
         * @memberof ExportDocumentLineModel
         */
        this.documentStatus = undefined;
        /**
         * @type {boolean}
         * @memberof ExportDocumentLineModel
         */
        this.isModifiedDateSameAsDocumentDate = undefined;
        /**
         * @type {string}
         * @memberof ExportDocumentLineModel
         */
        this.taxGroup = undefined;
        /**
         * @type {string}
         * @memberof ExportDocumentLineModel
         */
        this.taxName = undefined;
        /**
         * @type {string}
         * @memberof ExportDocumentLineModel
         */
        this.taxCode = undefined;
        /**
         * @type {string}
         * @memberof ExportDocumentLineModel
         */
        this.customerVendorCode = undefined;
        /**
         * @type {string}
         * @memberof ExportDocumentLineModel
         */
        this.taxSubType = undefined;
        /**
         * @type {Enums.ReportSource}
         * @memberof ExportDocumentLineModel
         */
        this.reportSource = undefined;
        /**
         * @type {Enums.Compression}
         * @memberof ExportDocumentLineModel
         */
        this.compression = undefined;
        /**
         * @type {boolean}
         * @memberof ExportDocumentLineModel
         */
        this.includeDocumentLineDetails = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("format", Enums.ReportFormatConverter, true),
        __metadata("design:type", Number)
    ], ExportDocumentLineModel.prototype, "format", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("startDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ExportDocumentLineModel.prototype, "startDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ExportDocumentLineModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], ExportDocumentLineModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("state", String, true),
        __metadata("design:type", String)
    ], ExportDocumentLineModel.prototype, "state", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dateFilter", Enums.ReportDateFilterConverter, true),
        __metadata("design:type", Number)
    ], ExportDocumentLineModel.prototype, "dateFilter", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("docType", Enums.ReportDocTypeConverter, true),
        __metadata("design:type", Number)
    ], ExportDocumentLineModel.prototype, "docType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("currencyCode", String, true),
        __metadata("design:type", String)
    ], ExportDocumentLineModel.prototype, "currencyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("numberOfPartitions", Number, true),
        __metadata("design:type", Number)
    ], ExportDocumentLineModel.prototype, "numberOfPartitions", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("partition", Number, true),
        __metadata("design:type", Number)
    ], ExportDocumentLineModel.prototype, "partition", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isLocked", Boolean, true),
        __metadata("design:type", Boolean)
    ], ExportDocumentLineModel.prototype, "isLocked", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("merchantSellerIdentifier", String, true),
        __metadata("design:type", String)
    ], ExportDocumentLineModel.prototype, "merchantSellerIdentifier", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("documentStatus", Enums.DocumentStatusConverter, true),
        __metadata("design:type", Number)
    ], ExportDocumentLineModel.prototype, "documentStatus", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isModifiedDateSameAsDocumentDate", Boolean, true),
        __metadata("design:type", Boolean)
    ], ExportDocumentLineModel.prototype, "isModifiedDateSameAsDocumentDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxGroup", String, true),
        __metadata("design:type", String)
    ], ExportDocumentLineModel.prototype, "taxGroup", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxName", String, true),
        __metadata("design:type", String)
    ], ExportDocumentLineModel.prototype, "taxName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCode", String, true),
        __metadata("design:type", String)
    ], ExportDocumentLineModel.prototype, "taxCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerVendorCode", String, true),
        __metadata("design:type", String)
    ], ExportDocumentLineModel.prototype, "customerVendorCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxSubType", String, true),
        __metadata("design:type", String)
    ], ExportDocumentLineModel.prototype, "taxSubType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportSource", Enums.ReportSourceConverter, true),
        __metadata("design:type", Number)
    ], ExportDocumentLineModel.prototype, "reportSource", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("compression", Enums.CompressionConverter, true),
        __metadata("design:type", Number)
    ], ExportDocumentLineModel.prototype, "compression", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("includeDocumentLineDetails", Boolean, true),
        __metadata("design:type", Boolean)
    ], ExportDocumentLineModel.prototype, "includeDocumentLineDetails", void 0);
    ExportDocumentLineModel = __decorate([
        (0, json2typescript_1.JsonObject)("ExportDocumentLineModel")
    ], ExportDocumentLineModel);
    return ExportDocumentLineModel;
}());
