// avatax/lib/models/FilingReturnModelBasic.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilingReturnModelBasic = void 0;
var Enums = require("../enums/index");
var FilingAttachmentModel_1 = require("./FilingAttachmentModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Filing Returns Model
 * @export
 * @class FilingReturnModelBasic
 */
var FilingReturnModelBasic = exports.FilingReturnModelBasic = /** @class */ (function () {
    function FilingReturnModelBasic() {
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.companyId = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.filingId = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.resourceFileId = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.filingRegionId = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.filingCalendarId = undefined;
        /**
         * @type {string}
         * @memberof FilingReturnModelBasic
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof FilingReturnModelBasic
         */
        this.region = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.endPeriodMonth = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.endPeriodYear = undefined;
        /**
         * @type {Enums.FilingStatusId}
         * @memberof FilingReturnModelBasic
         */
        this.status = undefined;
        /**
         * @type {Enums.FilingFrequencyId}
         * @memberof FilingReturnModelBasic
         */
        this.filingFrequency = undefined;
        /**
         * @type {Date}
         * @memberof FilingReturnModelBasic
         */
        this.filedDate = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.salesAmount = undefined;
        /**
         * @type {Enums.FilingTypeId}
         * @memberof FilingReturnModelBasic
         */
        this.filingType = undefined;
        /**
         * @type {string}
         * @memberof FilingReturnModelBasic
         */
        this.formName = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.remitAmount = undefined;
        /**
         * @type {string}
         * @memberof FilingReturnModelBasic
         */
        this.formCode = undefined;
        /**
         * @type {string}
         * @memberof FilingReturnModelBasic
         */
        this.description = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.taxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.taxAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.collectAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.taxDueAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.nonTaxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.nonTaxableDueAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.consumerUseTaxAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.consumerUseNonTaxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.consumerUseTaxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.excludedSalesAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.excludedNonTaxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.excludedTaxAmount = undefined;
        /**
         * @type {Enums.AccrualType}
         * @memberof FilingReturnModelBasic
         */
        this.accrualType = undefined;
        /**
         * @type {FilingAttachmentModel[]}
         * @memberof FilingReturnModelBasic
         */
        this.attachments = undefined;
        /**
         * @type {Date}
         * @memberof FilingReturnModelBasic
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModelBasic
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof FilingReturnModelBasic
         */
        this.modifiedDate = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingId", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "filingId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("resourceFileId", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "resourceFileId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingRegionId", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "filingRegionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingCalendarId", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "filingCalendarId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], FilingReturnModelBasic.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], FilingReturnModelBasic.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endPeriodMonth", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "endPeriodMonth", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endPeriodYear", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "endPeriodYear", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("status", Enums.FilingStatusIdConverter, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "status", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingFrequency", Enums.FilingFrequencyIdConverter, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "filingFrequency", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingReturnModelBasic.prototype, "filedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("salesAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "salesAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingType", Enums.FilingTypeIdConverter, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "filingType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("formName", String, true),
        __metadata("design:type", String)
    ], FilingReturnModelBasic.prototype, "formName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("remitAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "remitAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("formCode", String, true),
        __metadata("design:type", String)
    ], FilingReturnModelBasic.prototype, "formCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], FilingReturnModelBasic.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "taxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "taxAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("collectAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "collectAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxDueAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "taxDueAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "nonTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonTaxableDueAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "nonTaxableDueAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("consumerUseTaxAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "consumerUseTaxAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("consumerUseNonTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "consumerUseNonTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("consumerUseTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "consumerUseTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("excludedSalesAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "excludedSalesAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("excludedNonTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "excludedNonTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("excludedTaxAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "excludedTaxAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accrualType", Enums.AccrualTypeConverter, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "accrualType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("attachments", [FilingAttachmentModel_1.FilingAttachmentModel], true),
        __metadata("design:type", Array)
    ], FilingReturnModelBasic.prototype, "attachments", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingReturnModelBasic.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModelBasic.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingReturnModelBasic.prototype, "modifiedDate", void 0);
    FilingReturnModelBasic = __decorate([
        (0, json2typescript_1.JsonObject)("FilingReturnModelBasic")
    ], FilingReturnModelBasic);
    return FilingReturnModelBasic;
}());
