// avatax/lib/models/FilingReturnCreditModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilingReturnCreditModel = void 0;
var WorksheetDocument_1 = require("./WorksheetDocument");
var json2typescript_1 = require("json2typescript");
/**
 * An attachment associated with a filing return
 * @export
 * @class FilingReturnCreditModel
 */
var FilingReturnCreditModel = exports.FilingReturnCreditModel = /** @class */ (function () {
    function FilingReturnCreditModel() {
        /**
         * @type {number}
         * @memberof FilingReturnCreditModel
         */
        this.totalSales = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnCreditModel
         */
        this.totalExempt = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnCreditModel
         */
        this.totalTaxable = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnCreditModel
         */
        this.totalTax = undefined;
        /**
         * @type {WorksheetDocument[]}
         * @memberof FilingReturnCreditModel
         */
        this.transactionDetails = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalSales", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnCreditModel.prototype, "totalSales", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalExempt", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnCreditModel.prototype, "totalExempt", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalTaxable", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnCreditModel.prototype, "totalTaxable", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalTax", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnCreditModel.prototype, "totalTax", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("transactionDetails", [WorksheetDocument_1.WorksheetDocument], true),
        __metadata("design:type", Array)
    ], FilingReturnCreditModel.prototype, "transactionDetails", void 0);
    FilingReturnCreditModel = __decorate([
        (0, json2typescript_1.JsonObject)("FilingReturnCreditModel")
    ], FilingReturnCreditModel);
    return FilingReturnCreditModel;
}());
