// avatax/lib/models/FilingsTaxSummaryModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilingsTaxSummaryModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Represents a listing of all tax calculation data for filings and for accruing to future filings.
 * @export
 * @class FilingsTaxSummaryModel
 */
var FilingsTaxSummaryModel = exports.FilingsTaxSummaryModel = /** @class */ (function () {
    function FilingsTaxSummaryModel() {
        /**
         * @type {number}
         * @memberof FilingsTaxSummaryModel
         */
        this.salesAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxSummaryModel
         */
        this.taxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxSummaryModel
         */
        this.nonTaxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxSummaryModel
         */
        this.taxAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxSummaryModel
         */
        this.remittanceAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxSummaryModel
         */
        this.collectAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxSummaryModel
         */
        this.salesAccrualAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxSummaryModel
         */
        this.taxableAccrualAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxSummaryModel
         */
        this.nonTaxableAccrualAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxSummaryModel
         */
        this.taxAccrualAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxSummaryModel
         */
        this.reportableSalesAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxSummaryModel
         */
        this.reportableNonTaxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxSummaryModel
         */
        this.reportableTaxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxSummaryModel
         */
        this.reportableTaxAmount = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("salesAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxSummaryModel.prototype, "salesAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxSummaryModel.prototype, "taxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxSummaryModel.prototype, "nonTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxSummaryModel.prototype, "taxAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("remittanceAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxSummaryModel.prototype, "remittanceAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("collectAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxSummaryModel.prototype, "collectAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("salesAccrualAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxSummaryModel.prototype, "salesAccrualAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxableAccrualAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxSummaryModel.prototype, "taxableAccrualAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonTaxableAccrualAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxSummaryModel.prototype, "nonTaxableAccrualAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAccrualAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxSummaryModel.prototype, "taxAccrualAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportableSalesAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxSummaryModel.prototype, "reportableSalesAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportableNonTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxSummaryModel.prototype, "reportableNonTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportableTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxSummaryModel.prototype, "reportableTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportableTaxAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxSummaryModel.prototype, "reportableTaxAmount", void 0);
    FilingsTaxSummaryModel = __decorate([
        (0, json2typescript_1.JsonObject)("FilingsTaxSummaryModel")
    ], FilingsTaxSummaryModel);
    return FilingsTaxSummaryModel;
}());
