// avatax/lib/models/FilingAugmentationModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilingAugmentationModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * A model for return augmentations.
 * @export
 * @class FilingAugmentationModel
 */
var FilingAugmentationModel = exports.FilingAugmentationModel = /** @class */ (function () {
    function FilingAugmentationModel() {
        /**
         * @type {number}
         * @memberof FilingAugmentationModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof FilingAugmentationModel
         */
        this.filingId = undefined;
        /**
         * @type {number}
         * @memberof FilingAugmentationModel
         */
        this.fieldAmount = undefined;
        /**
         * @type {string}
         * @memberof FilingAugmentationModel
         */
        this.fieldName = undefined;
        /**
         * @type {Date}
         * @memberof FilingAugmentationModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof FilingAugmentationModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof FilingAugmentationModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof FilingAugmentationModel
         */
        this.modifiedUserId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], FilingAugmentationModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingId", Number, true),
        __metadata("design:type", Number)
    ], FilingAugmentationModel.prototype, "filingId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("fieldAmount", Number),
        __metadata("design:type", Number)
    ], FilingAugmentationModel.prototype, "fieldAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("fieldName", String),
        __metadata("design:type", String)
    ], FilingAugmentationModel.prototype, "fieldName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingAugmentationModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], FilingAugmentationModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingAugmentationModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], FilingAugmentationModel.prototype, "modifiedUserId", void 0);
    FilingAugmentationModel = __decorate([
        (0, json2typescript_1.JsonObject)("FilingAugmentationModel")
    ], FilingAugmentationModel);
    return FilingAugmentationModel;
}());
