// avatax/lib/models/FilingReturnModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilingReturnModel = void 0;
var Enums = require("../enums/index");
var FilingAdjustmentModel_1 = require("./FilingAdjustmentModel");
var FilingAugmentationModel_1 = require("./FilingAugmentationModel");
var FilingPaymentModel_1 = require("./FilingPaymentModel");
var FilingAttachmentModel_1 = require("./FilingAttachmentModel");
var FilingReturnCreditModel_1 = require("./FilingReturnCreditModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Filing Returns Model
 * @export
 * @class FilingReturnModel
 */
var FilingReturnModel = exports.FilingReturnModel = /** @class */ (function () {
    function FilingReturnModel() {
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.filingRegionId = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.filingCalendarId = undefined;
        /**
         * @type {string}
         * @memberof FilingReturnModel
         */
        this.registrationId = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.resourceFileId = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.taxAuthorityId = undefined;
        /**
         * @type {Enums.FilingStatusId}
         * @memberof FilingReturnModel
         */
        this.status = undefined;
        /**
         * @type {Enums.FilingFrequencyId}
         * @memberof FilingReturnModel
         */
        this.filingFrequency = undefined;
        /**
         * @type {Date}
         * @memberof FilingReturnModel
         */
        this.filedDate = undefined;
        /**
         * @type {Date}
         * @memberof FilingReturnModel
         */
        this.startPeriod = undefined;
        /**
         * @type {Date}
         * @memberof FilingReturnModel
         */
        this.endPeriod = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.salesAmount = undefined;
        /**
         * @type {Enums.FilingTypeId}
         * @memberof FilingReturnModel
         */
        this.filingType = undefined;
        /**
         * @type {string}
         * @memberof FilingReturnModel
         */
        this.formName = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.remitAmount = undefined;
        /**
         * @type {string}
         * @memberof FilingReturnModel
         */
        this.formCode = undefined;
        /**
         * @type {string}
         * @memberof FilingReturnModel
         */
        this.description = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.taxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.taxAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.collectAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.taxDueAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.nonTaxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.nonTaxableDueAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.consumerUseTaxAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.consumerUseTaxDueAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.consumerUseNonTaxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.consumerUseTaxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.totalAdjustments = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.excludedSalesAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.excludedNonTaxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.excludedTaxAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.carryOverSalesAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.carryOverNonTaxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.carryOverTaxAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.carryOverConsumerUseTaxAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.taxAccrualAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.salesAccrualAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.nonTaxableAccrualAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.taxableAccrualAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.salesTaxAccrualAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.sellersUseTaxAccrualAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.consumerUseTaxAccrualAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.consumerUseTaxableAccrualAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.consumerUseNonTaxableAccrualAmount = undefined;
        /**
         * @type {FilingAdjustmentModel[]}
         * @memberof FilingReturnModel
         */
        this.adjustments = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.totalAugmentations = undefined;
        /**
         * @type {FilingAugmentationModel[]}
         * @memberof FilingReturnModel
         */
        this.augmentations = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.totalPayments = undefined;
        /**
         * @type {FilingPaymentModel[]}
         * @memberof FilingReturnModel
         */
        this.payments = undefined;
        /**
         * @type {Enums.AccrualType}
         * @memberof FilingReturnModel
         */
        this.accrualType = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.month = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.year = undefined;
        /**
         * @type {string}
         * @memberof FilingReturnModel
         */
        this.type = undefined;
        /**
         * @type {FilingAttachmentModel[]}
         * @memberof FilingReturnModel
         */
        this.attachments = undefined;
        /**
         * @type {FilingReturnCreditModel}
         * @memberof FilingReturnModel
         */
        this.excludedCarryOverCredits = undefined;
        /**
         * @type {FilingReturnCreditModel}
         * @memberof FilingReturnModel
         */
        this.appliedCarryOverCredits = undefined;
        /**
         * @type {Date}
         * @memberof FilingReturnModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof FilingReturnModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof FilingReturnModel
         */
        this.modifiedDate = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingRegionId", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "filingRegionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingCalendarId", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "filingCalendarId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("registrationId", String, true),
        __metadata("design:type", String)
    ], FilingReturnModel.prototype, "registrationId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("resourceFileId", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "resourceFileId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityId", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "taxAuthorityId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("status", Enums.FilingStatusIdConverter, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "status", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingFrequency", Enums.FilingFrequencyIdConverter, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "filingFrequency", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingReturnModel.prototype, "filedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("startPeriod", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingReturnModel.prototype, "startPeriod", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endPeriod", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingReturnModel.prototype, "endPeriod", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("salesAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "salesAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingType", Enums.FilingTypeIdConverter, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "filingType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("formName", String, true),
        __metadata("design:type", String)
    ], FilingReturnModel.prototype, "formName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("remitAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "remitAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("formCode", String, true),
        __metadata("design:type", String)
    ], FilingReturnModel.prototype, "formCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], FilingReturnModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "taxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "taxAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("collectAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "collectAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxDueAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "taxDueAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "nonTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonTaxableDueAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "nonTaxableDueAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("consumerUseTaxAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "consumerUseTaxAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("consumerUseTaxDueAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "consumerUseTaxDueAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("consumerUseNonTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "consumerUseNonTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("consumerUseTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "consumerUseTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalAdjustments", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "totalAdjustments", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("excludedSalesAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "excludedSalesAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("excludedNonTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "excludedNonTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("excludedTaxAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "excludedTaxAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("carryOverSalesAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "carryOverSalesAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("carryOverNonTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "carryOverNonTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("carryOverTaxAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "carryOverTaxAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("carryOverConsumerUseTaxAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "carryOverConsumerUseTaxAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAccrualAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "taxAccrualAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("salesAccrualAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "salesAccrualAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonTaxableAccrualAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "nonTaxableAccrualAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxableAccrualAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "taxableAccrualAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("salesTaxAccrualAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "salesTaxAccrualAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("sellersUseTaxAccrualAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "sellersUseTaxAccrualAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("consumerUseTaxAccrualAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "consumerUseTaxAccrualAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("consumerUseTaxableAccrualAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "consumerUseTaxableAccrualAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("consumerUseNonTaxableAccrualAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "consumerUseNonTaxableAccrualAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("adjustments", [FilingAdjustmentModel_1.FilingAdjustmentModel], true),
        __metadata("design:type", Array)
    ], FilingReturnModel.prototype, "adjustments", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalAugmentations", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "totalAugmentations", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("augmentations", [FilingAugmentationModel_1.FilingAugmentationModel], true),
        __metadata("design:type", Array)
    ], FilingReturnModel.prototype, "augmentations", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalPayments", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "totalPayments", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("payments", [FilingPaymentModel_1.FilingPaymentModel], true),
        __metadata("design:type", Array)
    ], FilingReturnModel.prototype, "payments", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accrualType", Enums.AccrualTypeConverter, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "accrualType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("month", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "month", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("year", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "year", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("type", String, true),
        __metadata("design:type", String)
    ], FilingReturnModel.prototype, "type", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("attachments", [FilingAttachmentModel_1.FilingAttachmentModel], true),
        __metadata("design:type", Array)
    ], FilingReturnModel.prototype, "attachments", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("excludedCarryOverCredits", FilingReturnCreditModel_1.FilingReturnCreditModel, true),
        __metadata("design:type", FilingReturnCreditModel_1.FilingReturnCreditModel)
    ], FilingReturnModel.prototype, "excludedCarryOverCredits", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("appliedCarryOverCredits", FilingReturnCreditModel_1.FilingReturnCreditModel, true),
        __metadata("design:type", FilingReturnCreditModel_1.FilingReturnCreditModel)
    ], FilingReturnModel.prototype, "appliedCarryOverCredits", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingReturnModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], FilingReturnModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingReturnModel.prototype, "modifiedDate", void 0);
    FilingReturnModel = __decorate([
        (0, json2typescript_1.JsonObject)("FilingReturnModel")
    ], FilingReturnModel);
    return FilingReturnModel;
}());
