// avatax/lib/models/FilingRequestModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilingRequestModel = void 0;
var Enums = require("../enums/index");
var FilingRequestDataModel_1 = require("./FilingRequestDataModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a commitment to file a tax return on a recurring basis.
Only used if you subscribe to Avalara Returns.
 * @export
 * @class FilingRequestModel
 */
var FilingRequestModel = exports.FilingRequestModel = /** @class */ (function () {
    function FilingRequestModel() {
        /**
         * @type {number}
         * @memberof FilingRequestModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof FilingRequestModel
         */
        this.companyId = undefined;
        /**
         * @type {Enums.FilingRequestStatus}
         * @memberof FilingRequestModel
         */
        this.filingRequestStatusId = undefined;
        /**
         * @type {FilingRequestDataModel}
         * @memberof FilingRequestModel
         */
        this.data = undefined;
        /**
         * @type {Date}
         * @memberof FilingRequestModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof FilingRequestModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof FilingRequestModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof FilingRequestModel
         */
        this.modifiedUserId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], FilingRequestModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number),
        __metadata("design:type", Number)
    ], FilingRequestModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingRequestStatusId", Enums.FilingRequestStatusConverter, true),
        __metadata("design:type", Number)
    ], FilingRequestModel.prototype, "filingRequestStatusId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("data", FilingRequestDataModel_1.FilingRequestDataModel),
        __metadata("design:type", FilingRequestDataModel_1.FilingRequestDataModel)
    ], FilingRequestModel.prototype, "data", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingRequestModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], FilingRequestModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingRequestModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], FilingRequestModel.prototype, "modifiedUserId", void 0);
    FilingRequestModel = __decorate([
        (0, json2typescript_1.JsonObject)("FilingRequestModel")
    ], FilingRequestModel);
    return FilingRequestModel;
}());
