// avatax/lib/models/FormMasterModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormMasterModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents information about a tax form known to Avalara
 * @export
 * @class FormMasterModel
 */
var FormMasterModel = exports.FormMasterModel = /** @class */ (function () {
    function FormMasterModel() {
        /**
         * @type {number}
         * @memberof FormMasterModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof FormMasterModel
         */
        this.formTypeId = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.taxFormCode = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.legacyReturnName = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.taxFormName = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.description = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.isEffective = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.authorityName = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.shortCode = undefined;
        /**
         * @type {number}
         * @memberof FormMasterModel
         */
        this.dueDay = undefined;
        /**
         * @type {number}
         * @memberof FormMasterModel
         */
        this.delinquentDay = undefined;
        /**
         * @type {number}
         * @memberof FormMasterModel
         */
        this.fiscalYearStartMonth = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.hasMultiFrequencies = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.isPOARequired = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.isRegistrationRequired = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.hasMultiRegistrationMethods = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.hasSchedules = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.hasMultiFilingMethods = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.hasMultiPayMethods = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.isEFTRequired = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.isFilePayMethodLinked = undefined;
        /**
         * @type {number}
         * @memberof FormMasterModel
         */
        this.mailingReceivedRuleId = undefined;
        /**
         * @type {number}
         * @memberof FormMasterModel
         */
        this.proofOfMailingId = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.isNegAmountAllowed = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.allowNegativeOverallTax = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.isNettingRequired = undefined;
        /**
         * @type {number}
         * @memberof FormMasterModel
         */
        this.roundingMethodId = undefined;
        /**
         * @type {number}
         * @memberof FormMasterModel
         */
        this.vendorDiscountAnnualMax = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.versionsRequireAuthorityApproval = undefined;
        /**
         * @type {number}
         * @memberof FormMasterModel
         */
        this.outletReportingMethodId = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.hasReportingCodes = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.hasPrepayments = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.grossIncludesInterstateSales = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.grossIncludesTax = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.hasEfileFee = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.hasEpayFee = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.hasDependencies = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.requiredEfileTrigger = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.requiredEftTrigger = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.vendorDiscountEfile = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.vendorDiscountPaper = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.peerReviewed = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.peerReviewedId = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.peerReviewedDate = undefined;
        /**
         * @type {number}
         * @memberof FormMasterModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof FormMasterModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof FormMasterModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {Date}
         * @memberof FormMasterModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.dorAddressMailTo = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.dorAddress1 = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.dorAddress2 = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.dorAddressCity = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.dorAddressRegion = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.dorAddressPostalCode = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.dorAddressCountry = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.zeroAddressMailTo = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.zeroAddress1 = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.zeroAddress2 = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.zeroAddressCity = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.zeroAddressRegion = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.zeroAddressPostalCode = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.zeroAddressCountry = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.amendedAddressMailTo = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.amendedAddress1 = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.amendedAddress2 = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.amendedAddressCity = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.amendedAddressRegion = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.amendedAddressPostalCode = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.amendedAddressCountry = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.onlineBackFiling = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.onlineAmendedReturns = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.prepaymentFrequency = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.outletLocationIdentifiersRequired = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.listingSortOrder = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.dorWebsite = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.fileForAllOutlets = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.paperFormsDoNotHaveDiscounts = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.stackAggregation = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.roundingPrecision = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.inconsistencyTolerance = undefined;
        /**
         * @type {Date}
         * @memberof FormMasterModel
         */
        this.effDate = undefined;
        /**
         * @type {Date}
         * @memberof FormMasterModel
         */
        this.endDate = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.visibleToCustomers = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.requiresOutletSetup = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.achCreditAllowed = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.reportLevel = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.postOfficeValidated = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.stackAggregationOption = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.sstBehavior = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.nonSstBehavior = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.dorPhoneNumber = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.averageCheckClearDays = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.filterZeroRatedLineDetails = undefined;
        /**
         * @type {boolean}
         * @memberof FormMasterModel
         */
        this.allowsBulkFilingAccounts = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.bulkAccountInstructionLink = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.registrationIdFormat = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.thresholdTrigger = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.transactionSortingOption = undefined;
        /**
         * @type {number}
         * @memberof FormMasterModel
         */
        this.contentReviewFrequencyId = undefined;
        /**
         * @type {string}
         * @memberof FormMasterModel
         */
        this.aliasForFormMasterId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], FormMasterModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("formTypeId", Number, true),
        __metadata("design:type", Number)
    ], FormMasterModel.prototype, "formTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxFormCode", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "taxFormCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("legacyReturnName", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "legacyReturnName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxFormName", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "taxFormName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isEffective", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "isEffective", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("authorityName", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "authorityName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("shortCode", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "shortCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dueDay", Number, true),
        __metadata("design:type", Number)
    ], FormMasterModel.prototype, "dueDay", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("delinquentDay", Number, true),
        __metadata("design:type", Number)
    ], FormMasterModel.prototype, "delinquentDay", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("fiscalYearStartMonth", Number, true),
        __metadata("design:type", Number)
    ], FormMasterModel.prototype, "fiscalYearStartMonth", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hasMultiFrequencies", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "hasMultiFrequencies", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isPOARequired", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "isPOARequired", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isRegistrationRequired", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "isRegistrationRequired", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hasMultiRegistrationMethods", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "hasMultiRegistrationMethods", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hasSchedules", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "hasSchedules", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hasMultiFilingMethods", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "hasMultiFilingMethods", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hasMultiPayMethods", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "hasMultiPayMethods", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isEFTRequired", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "isEFTRequired", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isFilePayMethodLinked", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "isFilePayMethodLinked", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("mailingReceivedRuleId", Number, true),
        __metadata("design:type", Number)
    ], FormMasterModel.prototype, "mailingReceivedRuleId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("proofOfMailingId", Number, true),
        __metadata("design:type", Number)
    ], FormMasterModel.prototype, "proofOfMailingId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isNegAmountAllowed", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "isNegAmountAllowed", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("allowNegativeOverallTax", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "allowNegativeOverallTax", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isNettingRequired", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "isNettingRequired", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("roundingMethodId", Number, true),
        __metadata("design:type", Number)
    ], FormMasterModel.prototype, "roundingMethodId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("vendorDiscountAnnualMax", Number, true),
        __metadata("design:type", Number)
    ], FormMasterModel.prototype, "vendorDiscountAnnualMax", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("versionsRequireAuthorityApproval", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "versionsRequireAuthorityApproval", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("outletReportingMethodId", Number, true),
        __metadata("design:type", Number)
    ], FormMasterModel.prototype, "outletReportingMethodId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hasReportingCodes", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "hasReportingCodes", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hasPrepayments", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "hasPrepayments", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("grossIncludesInterstateSales", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "grossIncludesInterstateSales", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("grossIncludesTax", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "grossIncludesTax", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hasEfileFee", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "hasEfileFee", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hasEpayFee", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "hasEpayFee", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hasDependencies", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "hasDependencies", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("requiredEfileTrigger", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "requiredEfileTrigger", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("requiredEftTrigger", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "requiredEftTrigger", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("vendorDiscountEfile", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "vendorDiscountEfile", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("vendorDiscountPaper", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "vendorDiscountPaper", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("peerReviewed", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "peerReviewed", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("peerReviewedId", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "peerReviewedId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("peerReviewedDate", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "peerReviewedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], FormMasterModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FormMasterModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], FormMasterModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FormMasterModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dorAddressMailTo", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "dorAddressMailTo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dorAddress1", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "dorAddress1", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dorAddress2", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "dorAddress2", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dorAddressCity", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "dorAddressCity", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dorAddressRegion", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "dorAddressRegion", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dorAddressPostalCode", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "dorAddressPostalCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dorAddressCountry", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "dorAddressCountry", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("zeroAddressMailTo", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "zeroAddressMailTo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("zeroAddress1", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "zeroAddress1", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("zeroAddress2", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "zeroAddress2", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("zeroAddressCity", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "zeroAddressCity", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("zeroAddressRegion", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "zeroAddressRegion", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("zeroAddressPostalCode", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "zeroAddressPostalCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("zeroAddressCountry", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "zeroAddressCountry", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("amendedAddressMailTo", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "amendedAddressMailTo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("amendedAddress1", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "amendedAddress1", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("amendedAddress2", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "amendedAddress2", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("amendedAddressCity", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "amendedAddressCity", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("amendedAddressRegion", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "amendedAddressRegion", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("amendedAddressPostalCode", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "amendedAddressPostalCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("amendedAddressCountry", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "amendedAddressCountry", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("onlineBackFiling", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "onlineBackFiling", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("onlineAmendedReturns", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "onlineAmendedReturns", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("prepaymentFrequency", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "prepaymentFrequency", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("outletLocationIdentifiersRequired", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "outletLocationIdentifiersRequired", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("listingSortOrder", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "listingSortOrder", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dorWebsite", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "dorWebsite", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("fileForAllOutlets", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "fileForAllOutlets", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("paperFormsDoNotHaveDiscounts", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "paperFormsDoNotHaveDiscounts", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("stackAggregation", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "stackAggregation", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("roundingPrecision", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "roundingPrecision", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("inconsistencyTolerance", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "inconsistencyTolerance", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FormMasterModel.prototype, "effDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FormMasterModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("visibleToCustomers", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "visibleToCustomers", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("requiresOutletSetup", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "requiresOutletSetup", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("achCreditAllowed", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "achCreditAllowed", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportLevel", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "reportLevel", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("postOfficeValidated", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "postOfficeValidated", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("stackAggregationOption", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "stackAggregationOption", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("sstBehavior", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "sstBehavior", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonSstBehavior", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "nonSstBehavior", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dorPhoneNumber", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "dorPhoneNumber", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("averageCheckClearDays", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "averageCheckClearDays", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filterZeroRatedLineDetails", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "filterZeroRatedLineDetails", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("allowsBulkFilingAccounts", Boolean, true),
        __metadata("design:type", Boolean)
    ], FormMasterModel.prototype, "allowsBulkFilingAccounts", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("bulkAccountInstructionLink", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "bulkAccountInstructionLink", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("registrationIdFormat", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "registrationIdFormat", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("thresholdTrigger", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "thresholdTrigger", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("transactionSortingOption", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "transactionSortingOption", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("contentReviewFrequencyId", Number, true),
        __metadata("design:type", Number)
    ], FormMasterModel.prototype, "contentReviewFrequencyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("aliasForFormMasterId", String, true),
        __metadata("design:type", String)
    ], FormMasterModel.prototype, "aliasForFormMasterId", void 0);
    FormMasterModel = __decorate([
        (0, json2typescript_1.JsonObject)("FormMasterModel")
    ], FormMasterModel);
    return FormMasterModel;
}());
