// avatax/lib/models/HsCodeModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HsCodeModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Describes an element in the harmonized tariff system.
            
According to the [United States International Trade Commission](https://www.usitc.gov), the harmonized tariff schedule is defined as follows:
            
The HTS is a U.S. nomenclature system used to classify traded goods based on their material composition, product name, and/or intended
function. The HTS is designed so that each article falls into only one category. It is divided into chapters, each of which has a 2-digit
number. Each product category within the various chapters is designated by 4, 6, 8, or 10 digits. The 4-digit categories are called
"headings." The 6-, 8- and 10-digit classifications are called "subheadings."
            
Within AvaTax, the `HsCodeModel` object can refer to sections, chapters, headings, subheadings, or articles.  Each object represents one
classification.  Many of these objects have child objects underneath them; these child objects are more specific than their parent objects.
 * @export
 * @class HsCodeModel
 */
var HsCodeModel = exports.HsCodeModel = /** @class */ (function () {
    function HsCodeModel() {
        /**
         * @type {string}
         * @memberof HsCodeModel
         */
        this.hsCode = undefined;
        /**
         * @type {number}
         * @memberof HsCodeModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof HsCodeModel
         */
        this.parentHsCodeId = undefined;
        /**
         * @type {string}
         * @memberof HsCodeModel
         */
        this.description = undefined;
        /**
         * @type {string}
         * @memberof HsCodeModel
         */
        this.system = undefined;
        /**
         * @type {string}
         * @memberof HsCodeModel
         */
        this.destinationCountry = undefined;
        /**
         * @type {Date}
         * @memberof HsCodeModel
         */
        this.effDate = undefined;
        /**
         * @type {Date}
         * @memberof HsCodeModel
         */
        this.endDate = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("hsCode", String, true),
        __metadata("design:type", String)
    ], HsCodeModel.prototype, "hsCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number),
        __metadata("design:type", Number)
    ], HsCodeModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("parentHsCodeId", Number, true),
        __metadata("design:type", Number)
    ], HsCodeModel.prototype, "parentHsCodeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String),
        __metadata("design:type", String)
    ], HsCodeModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("system", String, true),
        __metadata("design:type", String)
    ], HsCodeModel.prototype, "system", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("destinationCountry", String, true),
        __metadata("design:type", String)
    ], HsCodeModel.prototype, "destinationCountry", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], HsCodeModel.prototype, "effDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], HsCodeModel.prototype, "endDate", void 0);
    HsCodeModel = __decorate([
        (0, json2typescript_1.JsonObject)("HsCodeModel")
    ], HsCodeModel);
    return HsCodeModel;
}());
