// avatax/lib/models/ItemPremiumClassificationOutputModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemPremiumClassificationOutputModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a Premium Classification output model associated with an item's SystemCode..
 * @export
 * @class ItemPremiumClassificationOutputModel
 */
var ItemPremiumClassificationOutputModel = exports.ItemPremiumClassificationOutputModel = /** @class */ (function () {
    function ItemPremiumClassificationOutputModel() {
        /**
         * @type {string}
         * @memberof ItemPremiumClassificationOutputModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof ItemPremiumClassificationOutputModel
         */
        this.itemCode = undefined;
        /**
         * @type {number}
         * @memberof ItemPremiumClassificationOutputModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof ItemPremiumClassificationOutputModel
         */
        this.hsCode = undefined;
        /**
         * @type {string}
         * @memberof ItemPremiumClassificationOutputModel
         */
        this.systemCode = undefined;
        /**
         * @type {string}
         * @memberof ItemPremiumClassificationOutputModel
         */
        this.justification = undefined;
        /**
         * @type {Date}
         * @memberof ItemPremiumClassificationOutputModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof ItemPremiumClassificationOutputModel
         */
        this.createdUserId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", String, true),
        __metadata("design:type", String)
    ], ItemPremiumClassificationOutputModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("itemCode", String, true),
        __metadata("design:type", String)
    ], ItemPremiumClassificationOutputModel.prototype, "itemCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], ItemPremiumClassificationOutputModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hsCode", String, true),
        __metadata("design:type", String)
    ], ItemPremiumClassificationOutputModel.prototype, "hsCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("systemCode", String, true),
        __metadata("design:type", String)
    ], ItemPremiumClassificationOutputModel.prototype, "systemCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("justification", String, true),
        __metadata("design:type", String)
    ], ItemPremiumClassificationOutputModel.prototype, "justification", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ItemPremiumClassificationOutputModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], ItemPremiumClassificationOutputModel.prototype, "createdUserId", void 0);
    ItemPremiumClassificationOutputModel = __decorate([
        (0, json2typescript_1.JsonObject)("ItemPremiumClassificationOutputModel")
    ], ItemPremiumClassificationOutputModel);
    return ItemPremiumClassificationOutputModel;
}());
