// avatax/lib/models/ItemClassificationOutputModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemClassificationOutputModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Product classification output model.
 * @export
 * @class ItemClassificationOutputModel
 */
var ItemClassificationOutputModel = exports.ItemClassificationOutputModel = /** @class */ (function () {
    function ItemClassificationOutputModel() {
        /**
         * @type {number}
         * @memberof ItemClassificationOutputModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof ItemClassificationOutputModel
         */
        this.itemId = undefined;
        /**
         * @type {number}
         * @memberof ItemClassificationOutputModel
         */
        this.systemId = undefined;
        /**
         * @type {string}
         * @memberof ItemClassificationOutputModel
         */
        this.productCode = undefined;
        /**
         * @type {string}
         * @memberof ItemClassificationOutputModel
         */
        this.systemCode = undefined;
        /**
         * @type {boolean}
         * @memberof ItemClassificationOutputModel
         */
        this.isPremium = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], ItemClassificationOutputModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("itemId", Number, true),
        __metadata("design:type", Number)
    ], ItemClassificationOutputModel.prototype, "itemId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("systemId", Number, true),
        __metadata("design:type", Number)
    ], ItemClassificationOutputModel.prototype, "systemId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("productCode", String, true),
        __metadata("design:type", String)
    ], ItemClassificationOutputModel.prototype, "productCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("systemCode", String, true),
        __metadata("design:type", String)
    ], ItemClassificationOutputModel.prototype, "systemCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isPremium", Boolean, true),
        __metadata("design:type", Boolean)
    ], ItemClassificationOutputModel.prototype, "isPremium", void 0);
    ItemClassificationOutputModel = __decorate([
        (0, json2typescript_1.JsonObject)("ItemClassificationOutputModel")
    ], ItemClassificationOutputModel);
    return ItemClassificationOutputModel;
}());
