// avatax/lib/models/ItemModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemModel = void 0;
var ClassificationModel_1 = require("./ClassificationModel");
var ItemParameterModel_1 = require("./ItemParameterModel");
var ItemTagDetailInputModel_1 = require("./ItemTagDetailInputModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents an item in your company's product catalog.
 * @export
 * @class ItemModel
 */
var ItemModel = exports.ItemModel = /** @class */ (function () {
    function ItemModel() {
        /**
         * @type {number}
         * @memberof ItemModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof ItemModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof ItemModel
         */
        this.itemCode = undefined;
        /**
         * @type {number}
         * @memberof ItemModel
         */
        this.taxCodeId = undefined;
        /**
         * @type {string}
         * @memberof ItemModel
         */
        this.taxCode = undefined;
        /**
         * @type {string}
         * @memberof ItemModel
         */
        this.description = undefined;
        /**
         * @type {string}
         * @memberof ItemModel
         */
        this.itemGroup = undefined;
        /**
         * @type {string}
         * @memberof ItemModel
         */
        this.category = undefined;
        /**
         * @type {Date}
         * @memberof ItemModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof ItemModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof ItemModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof ItemModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {string}
         * @memberof ItemModel
         */
        this.source = undefined;
        /**
         * @type {string}
         * @memberof ItemModel
         */
        this.sourceEntityId = undefined;
        /**
         * @type {string}
         * @memberof ItemModel
         */
        this.upc = undefined;
        /**
         * @type {ClassificationModel[]}
         * @memberof ItemModel
         */
        this.classifications = undefined;
        /**
         * @type {ItemParameterModel[]}
         * @memberof ItemModel
         */
        this.parameters = undefined;
        /**
         * @type {ItemTagDetailInputModel[]}
         * @memberof ItemModel
         */
        this.tags = undefined;
        /**
         * @type {object}
         * @memberof ItemModel
         */
        this.properties = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number),
        __metadata("design:type", Number)
    ], ItemModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], ItemModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("itemCode", String),
        __metadata("design:type", String)
    ], ItemModel.prototype, "itemCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCodeId", Number, true),
        __metadata("design:type", Number)
    ], ItemModel.prototype, "taxCodeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCode", String, true),
        __metadata("design:type", String)
    ], ItemModel.prototype, "taxCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String),
        __metadata("design:type", String)
    ], ItemModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("itemGroup", String, true),
        __metadata("design:type", String)
    ], ItemModel.prototype, "itemGroup", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("category", String, true),
        __metadata("design:type", String)
    ], ItemModel.prototype, "category", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ItemModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], ItemModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ItemModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], ItemModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("source", String, true),
        __metadata("design:type", String)
    ], ItemModel.prototype, "source", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("sourceEntityId", String, true),
        __metadata("design:type", String)
    ], ItemModel.prototype, "sourceEntityId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("upc", String, true),
        __metadata("design:type", String)
    ], ItemModel.prototype, "upc", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("classifications", [ClassificationModel_1.ClassificationModel], true),
        __metadata("design:type", Array)
    ], ItemModel.prototype, "classifications", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("parameters", [ItemParameterModel_1.ItemParameterModel], true),
        __metadata("design:type", Array)
    ], ItemModel.prototype, "parameters", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("tags", [ItemTagDetailInputModel_1.ItemTagDetailInputModel], true),
        __metadata("design:type", Array)
    ], ItemModel.prototype, "tags", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("properties", Object, true),
        __metadata("design:type", Object)
    ], ItemModel.prototype, "properties", void 0);
    ItemModel = __decorate([
        (0, json2typescript_1.JsonObject)("ItemModel")
    ], ItemModel);
    return ItemModel;
}());
