// avatax/lib/models/ItemCatalogueInputModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemCatalogueInputModel = void 0;
var ClassificationModel_1 = require("./ClassificationModel");
var ItemParameterModel_1 = require("./ItemParameterModel");
var json2typescript_1 = require("json2typescript");
/**
 * Represents an item in your company's product catalog.
 * @export
 * @class ItemCatalogueInputModel
 */
var ItemCatalogueInputModel = exports.ItemCatalogueInputModel = /** @class */ (function () {
    function ItemCatalogueInputModel() {
        /**
         * @type {number}
         * @memberof ItemCatalogueInputModel
         */
        this.itemId = undefined;
        /**
         * @type {string}
         * @memberof ItemCatalogueInputModel
         */
        this.itemCode = undefined;
        /**
         * @type {string}
         * @memberof ItemCatalogueInputModel
         */
        this.description = undefined;
        /**
         * @type {string}
         * @memberof ItemCatalogueInputModel
         */
        this.summary = undefined;
        /**
         * @type {string}
         * @memberof ItemCatalogueInputModel
         */
        this.taxCode = undefined;
        /**
         * @type {string}
         * @memberof ItemCatalogueInputModel
         */
        this.upc = undefined;
        /**
         * @type {string}
         * @memberof ItemCatalogueInputModel
         */
        this.itemGroup = undefined;
        /**
         * @type {string}
         * @memberof ItemCatalogueInputModel
         */
        this.category = undefined;
        /**
         * @type {string}
         * @memberof ItemCatalogueInputModel
         */
        this.source = undefined;
        /**
         * @type {string}
         * @memberof ItemCatalogueInputModel
         */
        this.sourceEntityId = undefined;
        /**
         * @type {object}
         * @memberof ItemCatalogueInputModel
         */
        this.properties = undefined;
        /**
         * @type {ClassificationModel[]}
         * @memberof ItemCatalogueInputModel
         */
        this.classifications = undefined;
        /**
         * @type {ItemParameterModel[]}
         * @memberof ItemCatalogueInputModel
         */
        this.parameters = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("itemId", Number, true),
        __metadata("design:type", Number)
    ], ItemCatalogueInputModel.prototype, "itemId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("itemCode", String),
        __metadata("design:type", String)
    ], ItemCatalogueInputModel.prototype, "itemCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String),
        __metadata("design:type", String)
    ], ItemCatalogueInputModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("summary", String, true),
        __metadata("design:type", String)
    ], ItemCatalogueInputModel.prototype, "summary", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCode", String, true),
        __metadata("design:type", String)
    ], ItemCatalogueInputModel.prototype, "taxCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("upc", String, true),
        __metadata("design:type", String)
    ], ItemCatalogueInputModel.prototype, "upc", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("itemGroup", String, true),
        __metadata("design:type", String)
    ], ItemCatalogueInputModel.prototype, "itemGroup", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("category", String, true),
        __metadata("design:type", String)
    ], ItemCatalogueInputModel.prototype, "category", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("source", String),
        __metadata("design:type", String)
    ], ItemCatalogueInputModel.prototype, "source", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("sourceEntityId", String, true),
        __metadata("design:type", String)
    ], ItemCatalogueInputModel.prototype, "sourceEntityId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("properties", Object, true),
        __metadata("design:type", Object)
    ], ItemCatalogueInputModel.prototype, "properties", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("classifications", [ClassificationModel_1.ClassificationModel], true),
        __metadata("design:type", Array)
    ], ItemCatalogueInputModel.prototype, "classifications", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("parameters", [ItemParameterModel_1.ItemParameterModel], true),
        __metadata("design:type", Array)
    ], ItemCatalogueInputModel.prototype, "parameters", void 0);
    ItemCatalogueInputModel = __decorate([
        (0, json2typescript_1.JsonObject)("ItemCatalogueInputModel")
    ], ItemCatalogueInputModel);
    return ItemCatalogueInputModel;
}());
