// avatax/lib/models/JurisdictionOverrideModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.JurisdictionOverrideModel = void 0;
var Enums = require("../enums/index");
var JurisdictionModel_1 = require("./JurisdictionModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents an override of tax jurisdictions for a specific address.
            
During the time period represented by EffDate through EndDate, all tax decisions for addresses matching
this override object will be assigned to the list of jurisdictions designated in this object.
 * @export
 * @class JurisdictionOverrideModel
 */
var JurisdictionOverrideModel = exports.JurisdictionOverrideModel = /** @class */ (function () {
    function JurisdictionOverrideModel() {
        /**
         * @type {number}
         * @memberof JurisdictionOverrideModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof JurisdictionOverrideModel
         */
        this.accountId = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionOverrideModel
         */
        this.description = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionOverrideModel
         */
        this.line1 = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionOverrideModel
         */
        this.city = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionOverrideModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionOverrideModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionOverrideModel
         */
        this.postalCode = undefined;
        /**
         * @type {Date}
         * @memberof JurisdictionOverrideModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof JurisdictionOverrideModel
         */
        this.endDate = undefined;
        /**
         * @type {Date}
         * @memberof JurisdictionOverrideModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof JurisdictionOverrideModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof JurisdictionOverrideModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof JurisdictionOverrideModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {JurisdictionModel[]}
         * @memberof JurisdictionOverrideModel
         */
        this.jurisdictions = undefined;
        /**
         * @type {number}
         * @memberof JurisdictionOverrideModel
         */
        this.taxRegionId = undefined;
        /**
         * @type {Enums.BoundaryLevel}
         * @memberof JurisdictionOverrideModel
         */
        this.boundaryLevel = undefined;
        /**
         * @type {boolean}
         * @memberof JurisdictionOverrideModel
         */
        this.isDefault = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], JurisdictionOverrideModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountId", Number, true),
        __metadata("design:type", Number)
    ], JurisdictionOverrideModel.prototype, "accountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String),
        __metadata("design:type", String)
    ], JurisdictionOverrideModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line1", String, true),
        __metadata("design:type", String)
    ], JurisdictionOverrideModel.prototype, "line1", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("city", String, true),
        __metadata("design:type", String)
    ], JurisdictionOverrideModel.prototype, "city", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String),
        __metadata("design:type", String)
    ], JurisdictionOverrideModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], JurisdictionOverrideModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("postalCode", String),
        __metadata("design:type", String)
    ], JurisdictionOverrideModel.prototype, "postalCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], JurisdictionOverrideModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], JurisdictionOverrideModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], JurisdictionOverrideModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], JurisdictionOverrideModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], JurisdictionOverrideModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], JurisdictionOverrideModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictions", [JurisdictionModel_1.JurisdictionModel]),
        __metadata("design:type", Array)
    ], JurisdictionOverrideModel.prototype, "jurisdictions", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxRegionId", Number),
        __metadata("design:type", Number)
    ], JurisdictionOverrideModel.prototype, "taxRegionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("boundaryLevel", Enums.BoundaryLevelConverter, true),
        __metadata("design:type", Number)
    ], JurisdictionOverrideModel.prototype, "boundaryLevel", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isDefault", Boolean, true),
        __metadata("design:type", Boolean)
    ], JurisdictionOverrideModel.prototype, "isDefault", void 0);
    JurisdictionOverrideModel = __decorate([
        (0, json2typescript_1.JsonObject)("JurisdictionOverrideModel")
    ], JurisdictionOverrideModel);
    return JurisdictionOverrideModel;
}());
