// avatax/lib/models/JurisdictionRateTypeTaxTypeMappingModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.JurisdictionRateTypeTaxTypeMappingModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a Jurisdiction with applicable TaxType, TaxSubType and RateType.
 * @export
 * @class JurisdictionRateTypeTaxTypeMappingModel
 */
var JurisdictionRateTypeTaxTypeMappingModel = exports.JurisdictionRateTypeTaxTypeMappingModel = /** @class */ (function () {
    function JurisdictionRateTypeTaxTypeMappingModel() {
        /**
         * @type {number}
         * @memberof JurisdictionRateTypeTaxTypeMappingModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionRateTypeTaxTypeMappingModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionRateTypeTaxTypeMappingModel
         */
        this.state = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionRateTypeTaxTypeMappingModel
         */
        this.jurisdictionType = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionRateTypeTaxTypeMappingModel
         */
        this.jurisdictionCode = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionRateTypeTaxTypeMappingModel
         */
        this.longName = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionRateTypeTaxTypeMappingModel
         */
        this.taxTypeId = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionRateTypeTaxTypeMappingModel
         */
        this.taxSubTypeId = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionRateTypeTaxTypeMappingModel
         */
        this.taxTypeGroupId = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionRateTypeTaxTypeMappingModel
         */
        this.rateTypeId = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionRateTypeTaxTypeMappingModel
         */
        this.stateFips = undefined;
        /**
         * @type {Date}
         * @memberof JurisdictionRateTypeTaxTypeMappingModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof JurisdictionRateTypeTaxTypeMappingModel
         */
        this.endDate = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], JurisdictionRateTypeTaxTypeMappingModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], JurisdictionRateTypeTaxTypeMappingModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("state", String, true),
        __metadata("design:type", String)
    ], JurisdictionRateTypeTaxTypeMappingModel.prototype, "state", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionType", String, true),
        __metadata("design:type", String)
    ], JurisdictionRateTypeTaxTypeMappingModel.prototype, "jurisdictionType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionCode", String, true),
        __metadata("design:type", String)
    ], JurisdictionRateTypeTaxTypeMappingModel.prototype, "jurisdictionCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("longName", String, true),
        __metadata("design:type", String)
    ], JurisdictionRateTypeTaxTypeMappingModel.prototype, "longName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxTypeId", String, true),
        __metadata("design:type", String)
    ], JurisdictionRateTypeTaxTypeMappingModel.prototype, "taxTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxSubTypeId", String, true),
        __metadata("design:type", String)
    ], JurisdictionRateTypeTaxTypeMappingModel.prototype, "taxSubTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxTypeGroupId", String, true),
        __metadata("design:type", String)
    ], JurisdictionRateTypeTaxTypeMappingModel.prototype, "taxTypeGroupId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rateTypeId", String, true),
        __metadata("design:type", String)
    ], JurisdictionRateTypeTaxTypeMappingModel.prototype, "rateTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("stateFips", String, true),
        __metadata("design:type", String)
    ], JurisdictionRateTypeTaxTypeMappingModel.prototype, "stateFips", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], JurisdictionRateTypeTaxTypeMappingModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], JurisdictionRateTypeTaxTypeMappingModel.prototype, "endDate", void 0);
    JurisdictionRateTypeTaxTypeMappingModel = __decorate([
        (0, json2typescript_1.JsonObject)("JurisdictionRateTypeTaxTypeMappingModel")
    ], JurisdictionRateTypeTaxTypeMappingModel);
    return JurisdictionRateTypeTaxTypeMappingModel;
}());
