// avatax/lib/models/MultiTaxFilingRegionModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultiTaxFilingRegionModel = void 0;
var Enums = require("../enums/index");
var FilingsTaxSummaryModel_1 = require("./FilingsTaxSummaryModel");
var FilingsTaxDetailsModel_1 = require("./FilingsTaxDetailsModel");
var FilingsCheckupSuggestedFormModel_1 = require("./FilingsCheckupSuggestedFormModel");
var MultiTaxFilingReturnModel_1 = require("./MultiTaxFilingReturnModel");
var json2typescript_1 = require("json2typescript");
/**
 * Regions
 * @export
 * @class MultiTaxFilingRegionModel
 */
var MultiTaxFilingRegionModel = exports.MultiTaxFilingRegionModel = /** @class */ (function () {
    function MultiTaxFilingRegionModel() {
        /**
         * @type {string}
         * @memberof MultiTaxFilingRegionModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof MultiTaxFilingRegionModel
         */
        this.region = undefined;
        /**
         * @type {boolean}
         * @memberof MultiTaxFilingRegionModel
         */
        this.hasNexus = undefined;
        /**
         * @type {Enums.FilingStatusId}
         * @memberof MultiTaxFilingRegionModel
         */
        this.status = undefined;
        /**
         * @type {FilingsTaxSummaryModel}
         * @memberof MultiTaxFilingRegionModel
         */
        this.regionTaxSummary = undefined;
        /**
         * @type {FilingsTaxDetailsModel[]}
         * @memberof MultiTaxFilingRegionModel
         */
        this.regionTaxDetails = undefined;
        /**
         * @type {FilingsCheckupSuggestedFormModel[]}
         * @memberof MultiTaxFilingRegionModel
         */
        this.suggestReturns = undefined;
        /**
         * @type {MultiTaxFilingReturnModel[]}
         * @memberof MultiTaxFilingRegionModel
         */
        this.returns = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], MultiTaxFilingRegionModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], MultiTaxFilingRegionModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hasNexus", Boolean, true),
        __metadata("design:type", Boolean)
    ], MultiTaxFilingRegionModel.prototype, "hasNexus", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("status", Enums.FilingStatusIdConverter, true),
        __metadata("design:type", Number)
    ], MultiTaxFilingRegionModel.prototype, "status", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("regionTaxSummary", FilingsTaxSummaryModel_1.FilingsTaxSummaryModel, true),
        __metadata("design:type", FilingsTaxSummaryModel_1.FilingsTaxSummaryModel)
    ], MultiTaxFilingRegionModel.prototype, "regionTaxSummary", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("regionTaxDetails", [FilingsTaxDetailsModel_1.FilingsTaxDetailsModel], true),
        __metadata("design:type", Array)
    ], MultiTaxFilingRegionModel.prototype, "regionTaxDetails", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("suggestReturns", [FilingsCheckupSuggestedFormModel_1.FilingsCheckupSuggestedFormModel], true),
        __metadata("design:type", Array)
    ], MultiTaxFilingRegionModel.prototype, "suggestReturns", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("returns", [MultiTaxFilingReturnModel_1.MultiTaxFilingReturnModel], true),
        __metadata("design:type", Array)
    ], MultiTaxFilingRegionModel.prototype, "returns", void 0);
    MultiTaxFilingRegionModel = __decorate([
        (0, json2typescript_1.JsonObject)("MultiTaxFilingRegionModel")
    ], MultiTaxFilingRegionModel);
    return MultiTaxFilingRegionModel;
}());
