// avatax/lib/models/MultiTaxFilingModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultiTaxFilingModel = void 0;
var Enums = require("../enums/index");
var FilingsTaxSummaryModel_1 = require("./FilingsTaxSummaryModel");
var FilingsTaxDetailsModel_1 = require("./FilingsTaxDetailsModel");
var MultiTaxFilingRegionModel_1 = require("./MultiTaxFilingRegionModel");
var json2typescript_1 = require("json2typescript");
/**
 * Represents a listing of all tax calculation data for filings and for accruing to future filings.
 * @export
 * @class MultiTaxFilingModel
 */
var MultiTaxFilingModel = exports.MultiTaxFilingModel = /** @class */ (function () {
    function MultiTaxFilingModel() {
        /**
         * @type {number}
         * @memberof MultiTaxFilingModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof MultiTaxFilingModel
         */
        this.companyId = undefined;
        /**
         * @type {number}
         * @memberof MultiTaxFilingModel
         */
        this.month = undefined;
        /**
         * @type {number}
         * @memberof MultiTaxFilingModel
         */
        this.year = undefined;
        /**
         * @type {Enums.WorksheetTypeId}
         * @memberof MultiTaxFilingModel
         */
        this.type = undefined;
        /**
         * @type {FilingsTaxSummaryModel}
         * @memberof MultiTaxFilingModel
         */
        this.taxSummary = undefined;
        /**
         * @type {FilingsTaxDetailsModel[]}
         * @memberof MultiTaxFilingModel
         */
        this.taxDetails = undefined;
        /**
         * @type {MultiTaxFilingRegionModel[]}
         * @memberof MultiTaxFilingModel
         */
        this.filingRegions = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], MultiTaxFilingModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], MultiTaxFilingModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("month", Number, true),
        __metadata("design:type", Number)
    ], MultiTaxFilingModel.prototype, "month", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("year", Number, true),
        __metadata("design:type", Number)
    ], MultiTaxFilingModel.prototype, "year", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("type", Enums.WorksheetTypeIdConverter, true),
        __metadata("design:type", Number)
    ], MultiTaxFilingModel.prototype, "type", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxSummary", FilingsTaxSummaryModel_1.FilingsTaxSummaryModel, true),
        __metadata("design:type", FilingsTaxSummaryModel_1.FilingsTaxSummaryModel)
    ], MultiTaxFilingModel.prototype, "taxSummary", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxDetails", [FilingsTaxDetailsModel_1.FilingsTaxDetailsModel], true),
        __metadata("design:type", Array)
    ], MultiTaxFilingModel.prototype, "taxDetails", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingRegions", [MultiTaxFilingRegionModel_1.MultiTaxFilingRegionModel], true),
        __metadata("design:type", Array)
    ], MultiTaxFilingModel.prototype, "filingRegions", void 0);
    MultiTaxFilingModel = __decorate([
        (0, json2typescript_1.JsonObject)("MultiTaxFilingModel")
    ], MultiTaxFilingModel);
    return MultiTaxFilingModel;
}());
