// avatax/lib/models/NewAccountRequestModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NewAccountRequestModel = void 0;
var Enums = require("../enums/index");
var CompanyAddress_1 = require("./CompanyAddress");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a request for a new account with Avalara for a new subscriber.
Contains information about the account requested and the rate plan selected.
 * @export
 * @class NewAccountRequestModel
 */
var NewAccountRequestModel = exports.NewAccountRequestModel = /** @class */ (function () {
    function NewAccountRequestModel() {
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.offer = undefined;
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.connectorId = undefined;
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.campaign = undefined;
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.leadSource = undefined;
        /**
         * @type {Date}
         * @memberof NewAccountRequestModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof NewAccountRequestModel
         */
        this.endDate = undefined;
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.accountName = undefined;
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.website = undefined;
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.paymentMethodId = undefined;
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.firstName = undefined;
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.lastName = undefined;
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.title = undefined;
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.phoneNumber = undefined;
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.email = undefined;
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.username = undefined;
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.userPassword = undefined;
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.welcomeEmail = undefined;
        /**
         * @type {CompanyAddress}
         * @memberof NewAccountRequestModel
         */
        this.companyAddress = undefined;
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.companyCode = undefined;
        /**
         * @type {string[]}
         * @memberof NewAccountRequestModel
         */
        this.properties = undefined;
        /**
         * @type {boolean}
         * @memberof NewAccountRequestModel
         */
        this.acceptAvalaraTermsAndConditions = undefined;
        /**
         * @type {boolean}
         * @memberof NewAccountRequestModel
         */
        this.haveReadAvalaraTermsAndConditions = undefined;
        /**
         * @type {object}
         * @memberof NewAccountRequestModel
         */
        this.marketingContext = undefined;
        /**
         * @type {Enums.AccountTypeId}
         * @memberof NewAccountRequestModel
         */
        this.accountType = undefined;
        /**
         * @type {string}
         * @memberof NewAccountRequestModel
         */
        this.taxPayerIdNumber = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("offer", String),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "offer", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("connectorId", String, true),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "connectorId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("campaign", String, true),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "campaign", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("leadSource", String, true),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "leadSource", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NewAccountRequestModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NewAccountRequestModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountName", String),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "accountName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("website", String, true),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "website", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("paymentMethodId", String, true),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "paymentMethodId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("firstName", String),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "firstName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lastName", String),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "lastName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("title", String, true),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "title", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("phoneNumber", String, true),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "phoneNumber", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("email", String),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "email", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("username", String, true),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "username", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("userPassword", String, true),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "userPassword", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("welcomeEmail", String, true),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "welcomeEmail", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyAddress", CompanyAddress_1.CompanyAddress),
        __metadata("design:type", CompanyAddress_1.CompanyAddress)
    ], NewAccountRequestModel.prototype, "companyAddress", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyCode", String, true),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "companyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("properties", [String], true),
        __metadata("design:type", Array)
    ], NewAccountRequestModel.prototype, "properties", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("acceptAvalaraTermsAndConditions", Boolean, true),
        __metadata("design:type", Boolean)
    ], NewAccountRequestModel.prototype, "acceptAvalaraTermsAndConditions", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("haveReadAvalaraTermsAndConditions", Boolean, true),
        __metadata("design:type", Boolean)
    ], NewAccountRequestModel.prototype, "haveReadAvalaraTermsAndConditions", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("marketingContext", Object, true),
        __metadata("design:type", Object)
    ], NewAccountRequestModel.prototype, "marketingContext", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountType", Enums.AccountTypeIdConverter, true),
        __metadata("design:type", Number)
    ], NewAccountRequestModel.prototype, "accountType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxPayerIdNumber", String, true),
        __metadata("design:type", String)
    ], NewAccountRequestModel.prototype, "taxPayerIdNumber", void 0);
    NewAccountRequestModel = __decorate([
        (0, json2typescript_1.JsonObject)("NewAccountRequestModel")
    ], NewAccountRequestModel);
    return NewAccountRequestModel;
}());
