// avatax/lib/models/NotificationModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a single notification.
            
A notification is a message from Avalara that may have relevance to your business.  You may want
to regularly review notifications and then dismiss them when you are certain that you have addressed
any relevant concerns raised by this notification.
            
An example of a notification would be a message about new software, or a change to AvaTax that may
affect you, or a potential issue with your company's tax profile.
 * @export
 * @class NotificationModel
 */
var NotificationModel = exports.NotificationModel = /** @class */ (function () {
    function NotificationModel() {
        /**
         * @type {number}
         * @memberof NotificationModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof NotificationModel
         */
        this.accountId = undefined;
        /**
         * @type {number}
         * @memberof NotificationModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof NotificationModel
         */
        this.referenceObject = undefined;
        /**
         * @type {number}
         * @memberof NotificationModel
         */
        this.referenceId = undefined;
        /**
         * @type {Enums.NotificationSeverityLevel}
         * @memberof NotificationModel
         */
        this.severityLevelId = undefined;
        /**
         * @type {string}
         * @memberof NotificationModel
         */
        this.category = undefined;
        /**
         * @type {string}
         * @memberof NotificationModel
         */
        this.topic = undefined;
        /**
         * @type {string}
         * @memberof NotificationModel
         */
        this.message = undefined;
        /**
         * @type {boolean}
         * @memberof NotificationModel
         */
        this.needsAction = undefined;
        /**
         * @type {string}
         * @memberof NotificationModel
         */
        this.actionName = undefined;
        /**
         * @type {string}
         * @memberof NotificationModel
         */
        this.actionLink = undefined;
        /**
         * @type {Date}
         * @memberof NotificationModel
         */
        this.actionDueDate = undefined;
        /**
         * @type {boolean}
         * @memberof NotificationModel
         */
        this.dismissed = undefined;
        /**
         * @type {number}
         * @memberof NotificationModel
         */
        this.dismissedByUserId = undefined;
        /**
         * @type {Date}
         * @memberof NotificationModel
         */
        this.dismissedDate = undefined;
        /**
         * @type {Date}
         * @memberof NotificationModel
         */
        this.expireDate = undefined;
        /**
         * @type {number}
         * @memberof NotificationModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof NotificationModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof NotificationModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {Date}
         * @memberof NotificationModel
         */
        this.modifiedDate = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], NotificationModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountId", Number, true),
        __metadata("design:type", Number)
    ], NotificationModel.prototype, "accountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], NotificationModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("referenceObject", String, true),
        __metadata("design:type", String)
    ], NotificationModel.prototype, "referenceObject", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("referenceId", Number, true),
        __metadata("design:type", Number)
    ], NotificationModel.prototype, "referenceId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("severityLevelId", Enums.NotificationSeverityLevelConverter),
        __metadata("design:type", Number)
    ], NotificationModel.prototype, "severityLevelId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("category", String, true),
        __metadata("design:type", String)
    ], NotificationModel.prototype, "category", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("topic", String, true),
        __metadata("design:type", String)
    ], NotificationModel.prototype, "topic", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("message", String),
        __metadata("design:type", String)
    ], NotificationModel.prototype, "message", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("needsAction", Boolean, true),
        __metadata("design:type", Boolean)
    ], NotificationModel.prototype, "needsAction", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("actionName", String, true),
        __metadata("design:type", String)
    ], NotificationModel.prototype, "actionName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("actionLink", String, true),
        __metadata("design:type", String)
    ], NotificationModel.prototype, "actionLink", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("actionDueDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NotificationModel.prototype, "actionDueDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dismissed", Boolean, true),
        __metadata("design:type", Boolean)
    ], NotificationModel.prototype, "dismissed", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dismissedByUserId", Number, true),
        __metadata("design:type", Number)
    ], NotificationModel.prototype, "dismissedByUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dismissedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NotificationModel.prototype, "dismissedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("expireDate", dateConverter_1.DateConverter),
        __metadata("design:type", Date)
    ], NotificationModel.prototype, "expireDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], NotificationModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NotificationModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], NotificationModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NotificationModel.prototype, "modifiedDate", void 0);
    NotificationModel = __decorate([
        (0, json2typescript_1.JsonObject)("NotificationModel")
    ], NotificationModel);
    return NotificationModel;
}());
