// avatax/lib/models/NoticeFinanceModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NoticeFinanceModel = void 0;
var ResourceFileUploadRequestModel_1 = require("./ResourceFileUploadRequestModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents estimated financial results from responding to a tax notice.
 * @export
 * @class NoticeFinanceModel
 */
var NoticeFinanceModel = exports.NoticeFinanceModel = /** @class */ (function () {
    function NoticeFinanceModel() {
        /**
         * @type {number}
         * @memberof NoticeFinanceModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof NoticeFinanceModel
         */
        this.noticeId = undefined;
        /**
         * @type {Date}
         * @memberof NoticeFinanceModel
         */
        this.noticeDate = undefined;
        /**
         * @type {Date}
         * @memberof NoticeFinanceModel
         */
        this.dueDate = undefined;
        /**
         * @type {string}
         * @memberof NoticeFinanceModel
         */
        this.noticeNumber = undefined;
        /**
         * @type {number}
         * @memberof NoticeFinanceModel
         */
        this.taxDue = undefined;
        /**
         * @type {number}
         * @memberof NoticeFinanceModel
         */
        this.penalty = undefined;
        /**
         * @type {number}
         * @memberof NoticeFinanceModel
         */
        this.interest = undefined;
        /**
         * @type {number}
         * @memberof NoticeFinanceModel
         */
        this.credits = undefined;
        /**
         * @type {number}
         * @memberof NoticeFinanceModel
         */
        this.taxAbated = undefined;
        /**
         * @type {number}
         * @memberof NoticeFinanceModel
         */
        this.customerPenalty = undefined;
        /**
         * @type {number}
         * @memberof NoticeFinanceModel
         */
        this.customerInterest = undefined;
        /**
         * @type {number}
         * @memberof NoticeFinanceModel
         */
        this.cspFeeRefund = undefined;
        /**
         * @type {string}
         * @memberof NoticeFinanceModel
         */
        this.fileName = undefined;
        /**
         * @type {string}
         * @memberof NoticeFinanceModel
         */
        this.paymentMethod = undefined;
        /**
         * @type {number}
         * @memberof NoticeFinanceModel
         */
        this.resourceFileId = undefined;
        /**
         * @type {Date}
         * @memberof NoticeFinanceModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof NoticeFinanceModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof NoticeFinanceModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof NoticeFinanceModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {ResourceFileUploadRequestModel}
         * @memberof NoticeFinanceModel
         */
        this.attachmentUploadRequest = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], NoticeFinanceModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("noticeId", Number, true),
        __metadata("design:type", Number)
    ], NoticeFinanceModel.prototype, "noticeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("noticeDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NoticeFinanceModel.prototype, "noticeDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dueDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NoticeFinanceModel.prototype, "dueDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("noticeNumber", String, true),
        __metadata("design:type", String)
    ], NoticeFinanceModel.prototype, "noticeNumber", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxDue", Number, true),
        __metadata("design:type", Number)
    ], NoticeFinanceModel.prototype, "taxDue", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("penalty", Number, true),
        __metadata("design:type", Number)
    ], NoticeFinanceModel.prototype, "penalty", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("interest", Number, true),
        __metadata("design:type", Number)
    ], NoticeFinanceModel.prototype, "interest", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("credits", Number, true),
        __metadata("design:type", Number)
    ], NoticeFinanceModel.prototype, "credits", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAbated", Number, true),
        __metadata("design:type", Number)
    ], NoticeFinanceModel.prototype, "taxAbated", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerPenalty", Number, true),
        __metadata("design:type", Number)
    ], NoticeFinanceModel.prototype, "customerPenalty", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerInterest", Number, true),
        __metadata("design:type", Number)
    ], NoticeFinanceModel.prototype, "customerInterest", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("cspFeeRefund", Number, true),
        __metadata("design:type", Number)
    ], NoticeFinanceModel.prototype, "cspFeeRefund", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("fileName", String, true),
        __metadata("design:type", String)
    ], NoticeFinanceModel.prototype, "fileName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("paymentMethod", String, true),
        __metadata("design:type", String)
    ], NoticeFinanceModel.prototype, "paymentMethod", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("resourceFileId", Number, true),
        __metadata("design:type", Number)
    ], NoticeFinanceModel.prototype, "resourceFileId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NoticeFinanceModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], NoticeFinanceModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NoticeFinanceModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], NoticeFinanceModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("attachmentUploadRequest", ResourceFileUploadRequestModel_1.ResourceFileUploadRequestModel, true),
        __metadata("design:type", ResourceFileUploadRequestModel_1.ResourceFileUploadRequestModel)
    ], NoticeFinanceModel.prototype, "attachmentUploadRequest", void 0);
    NoticeFinanceModel = __decorate([
        (0, json2typescript_1.JsonObject)("NoticeFinanceModel")
    ], NoticeFinanceModel);
    return NoticeFinanceModel;
}());
