// avatax/lib/models/NoticeCommentModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NoticeCommentModel = void 0;
var Enums = require("../enums/index");
var ResourceFileUploadRequestModel_1 = require("./ResourceFileUploadRequestModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents communication between Avalara and the company regarding the processing of a tax notice.
 * @export
 * @class NoticeCommentModel
 */
var NoticeCommentModel = exports.NoticeCommentModel = /** @class */ (function () {
    function NoticeCommentModel() {
        /**
         * @type {number}
         * @memberof NoticeCommentModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof NoticeCommentModel
         */
        this.noticeId = undefined;
        /**
         * @type {Date}
         * @memberof NoticeCommentModel
         */
        this.date = undefined;
        /**
         * @type {string}
         * @memberof NoticeCommentModel
         */
        this.comment = undefined;
        /**
         * @type {number}
         * @memberof NoticeCommentModel
         */
        this.commentUserId = undefined;
        /**
         * @type {string}
         * @memberof NoticeCommentModel
         */
        this.commentUserName = undefined;
        /**
         * @type {number}
         * @memberof NoticeCommentModel
         */
        this.commentTypeId = undefined;
        /**
         * @type {Enums.CommentType}
         * @memberof NoticeCommentModel
         */
        this.commentType = undefined;
        /**
         * @type {string}
         * @memberof NoticeCommentModel
         */
        this.commentLink = undefined;
        /**
         * @type {string}
         * @memberof NoticeCommentModel
         */
        this.taxNoticeFileName = undefined;
        /**
         * @type {number}
         * @memberof NoticeCommentModel
         */
        this.resourceFileId = undefined;
        /**
         * @type {Date}
         * @memberof NoticeCommentModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof NoticeCommentModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {ResourceFileUploadRequestModel}
         * @memberof NoticeCommentModel
         */
        this.attachmentUploadRequest = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], NoticeCommentModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("noticeId", Number),
        __metadata("design:type", Number)
    ], NoticeCommentModel.prototype, "noticeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("date", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NoticeCommentModel.prototype, "date", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("comment", String, true),
        __metadata("design:type", String)
    ], NoticeCommentModel.prototype, "comment", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("commentUserId", Number, true),
        __metadata("design:type", Number)
    ], NoticeCommentModel.prototype, "commentUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("commentUserName", String, true),
        __metadata("design:type", String)
    ], NoticeCommentModel.prototype, "commentUserName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("commentTypeId", Number, true),
        __metadata("design:type", Number)
    ], NoticeCommentModel.prototype, "commentTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("commentType", Enums.CommentTypeConverter),
        __metadata("design:type", Number)
    ], NoticeCommentModel.prototype, "commentType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("commentLink", String, true),
        __metadata("design:type", String)
    ], NoticeCommentModel.prototype, "commentLink", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxNoticeFileName", String, true),
        __metadata("design:type", String)
    ], NoticeCommentModel.prototype, "taxNoticeFileName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("resourceFileId", Number, true),
        __metadata("design:type", Number)
    ], NoticeCommentModel.prototype, "resourceFileId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NoticeCommentModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], NoticeCommentModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("attachmentUploadRequest", ResourceFileUploadRequestModel_1.ResourceFileUploadRequestModel, true),
        __metadata("design:type", ResourceFileUploadRequestModel_1.ResourceFileUploadRequestModel)
    ], NoticeCommentModel.prototype, "attachmentUploadRequest", void 0);
    NoticeCommentModel = __decorate([
        (0, json2typescript_1.JsonObject)("NoticeCommentModel")
    ], NoticeCommentModel);
    return NoticeCommentModel;
}());
