// avatax/lib/models/ParameterModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParameterModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * An extra property that can change the behavior of tax transactions.
 * @export
 * @class ParameterModel
 */
var ParameterModel = exports.ParameterModel = /** @class */ (function () {
    function ParameterModel() {
        /**
         * @type {number}
         * @memberof ParameterModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof ParameterModel
         */
        this.category = undefined;
        /**
         * @type {string}
         * @memberof ParameterModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof ParameterModel
         */
        this.dataType = undefined;
        /**
         * @type {string}
         * @memberof ParameterModel
         */
        this.helpText = undefined;
        /**
         * @type {string[]}
         * @memberof ParameterModel
         */
        this.serviceTypes = undefined;
        /**
         * @type {string}
         * @memberof ParameterModel
         */
        this.prompt = undefined;
        /**
         * @type {string}
         * @memberof ParameterModel
         */
        this.regularExpression = undefined;
        /**
         * @type {string}
         * @memberof ParameterModel
         */
        this.label = undefined;
        /**
         * @type {string}
         * @memberof ParameterModel
         */
        this.helpUrl = undefined;
        /**
         * @type {string}
         * @memberof ParameterModel
         */
        this.attributeType = undefined;
        /**
         * @type {string[]}
         * @memberof ParameterModel
         */
        this.values = undefined;
        /**
         * @type {string}
         * @memberof ParameterModel
         */
        this.measurementType = undefined;
        /**
         * @type {boolean}
         * @memberof ParameterModel
         */
        this.isNeededForCalculation = undefined;
        /**
         * @type {boolean}
         * @memberof ParameterModel
         */
        this.isNeededForReturns = undefined;
        /**
         * @type {boolean}
         * @memberof ParameterModel
         */
        this.isNeededForClassification = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], ParameterModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("category", String, true),
        __metadata("design:type", String)
    ], ParameterModel.prototype, "category", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String, true),
        __metadata("design:type", String)
    ], ParameterModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dataType", String, true),
        __metadata("design:type", String)
    ], ParameterModel.prototype, "dataType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("helpText", String, true),
        __metadata("design:type", String)
    ], ParameterModel.prototype, "helpText", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("serviceTypes", [String], true),
        __metadata("design:type", Array)
    ], ParameterModel.prototype, "serviceTypes", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("prompt", String, true),
        __metadata("design:type", String)
    ], ParameterModel.prototype, "prompt", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("regularExpression", String, true),
        __metadata("design:type", String)
    ], ParameterModel.prototype, "regularExpression", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("label", String, true),
        __metadata("design:type", String)
    ], ParameterModel.prototype, "label", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("helpUrl", String, true),
        __metadata("design:type", String)
    ], ParameterModel.prototype, "helpUrl", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("attributeType", String, true),
        __metadata("design:type", String)
    ], ParameterModel.prototype, "attributeType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("values", [String], true),
        __metadata("design:type", Array)
    ], ParameterModel.prototype, "values", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("measurementType", String, true),
        __metadata("design:type", String)
    ], ParameterModel.prototype, "measurementType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isNeededForCalculation", Boolean, true),
        __metadata("design:type", Boolean)
    ], ParameterModel.prototype, "isNeededForCalculation", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isNeededForReturns", Boolean, true),
        __metadata("design:type", Boolean)
    ], ParameterModel.prototype, "isNeededForReturns", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isNeededForClassification", Boolean, true),
        __metadata("design:type", Boolean)
    ], ParameterModel.prototype, "isNeededForClassification", void 0);
    ParameterModel = __decorate([
        (0, json2typescript_1.JsonObject)("ParameterModel")
    ], ParameterModel);
    return ParameterModel;
}());
