// avatax/lib/models/PowerOfAttorneyCheckModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PowerOfAttorneyCheckModel = void 0;
var ResourceFileDownloadResult_1 = require("./ResourceFileDownloadResult");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Response when checking if a company has a POA on file with Avalara
 * @export
 * @class PowerOfAttorneyCheckModel
 */
var PowerOfAttorneyCheckModel = exports.PowerOfAttorneyCheckModel = /** @class */ (function () {
    function PowerOfAttorneyCheckModel() {
        /**
         * @type {number}
         * @memberof PowerOfAttorneyCheckModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof PowerOfAttorneyCheckModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof PowerOfAttorneyCheckModel
         */
        this.region = undefined;
        /**
         * @type {boolean}
         * @memberof PowerOfAttorneyCheckModel
         */
        this.activePoa = undefined;
        /**
         * @type {Date}
         * @memberof PowerOfAttorneyCheckModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof PowerOfAttorneyCheckModel
         */
        this.expirationDate = undefined;
        /**
         * @type {ResourceFileDownloadResult}
         * @memberof PowerOfAttorneyCheckModel
         */
        this.availablePoa = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], PowerOfAttorneyCheckModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], PowerOfAttorneyCheckModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], PowerOfAttorneyCheckModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("activePoa", Boolean, true),
        __metadata("design:type", Boolean)
    ], PowerOfAttorneyCheckModel.prototype, "activePoa", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], PowerOfAttorneyCheckModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("expirationDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], PowerOfAttorneyCheckModel.prototype, "expirationDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("availablePoa", ResourceFileDownloadResult_1.ResourceFileDownloadResult, true),
        __metadata("design:type", ResourceFileDownloadResult_1.ResourceFileDownloadResult)
    ], PowerOfAttorneyCheckModel.prototype, "availablePoa", void 0);
    PowerOfAttorneyCheckModel = __decorate([
        (0, json2typescript_1.JsonObject)("PowerOfAttorneyCheckModel")
    ], PowerOfAttorneyCheckModel);
    return PowerOfAttorneyCheckModel;
}());
