// avatax/lib/models/ProductClassificationSystemModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductClassificationSystemModel = void 0;
var ProductSystemCountryModel_1 = require("./ProductSystemCountryModel");
var json2typescript_1 = require("json2typescript");
/**
 * Represents a product classification system.
 * @export
 * @class ProductClassificationSystemModel
 */
var ProductClassificationSystemModel = exports.ProductClassificationSystemModel = /** @class */ (function () {
    function ProductClassificationSystemModel() {
        /**
         * @type {number}
         * @memberof ProductClassificationSystemModel
         */
        this.systemId = undefined;
        /**
         * @type {string}
         * @memberof ProductClassificationSystemModel
         */
        this.systemCode = undefined;
        /**
         * @type {string}
         * @memberof ProductClassificationSystemModel
         */
        this.description = undefined;
        /**
         * @type {string}
         * @memberof ProductClassificationSystemModel
         */
        this.customsValue = undefined;
        /**
         * @type {ProductSystemCountryModel[]}
         * @memberof ProductClassificationSystemModel
         */
        this.countries = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("systemId", Number, true),
        __metadata("design:type", Number)
    ], ProductClassificationSystemModel.prototype, "systemId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("systemCode", String, true),
        __metadata("design:type", String)
    ], ProductClassificationSystemModel.prototype, "systemCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], ProductClassificationSystemModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customsValue", String, true),
        __metadata("design:type", String)
    ], ProductClassificationSystemModel.prototype, "customsValue", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("countries", [ProductSystemCountryModel_1.ProductSystemCountryModel], true),
        __metadata("design:type", Array)
    ], ProductClassificationSystemModel.prototype, "countries", void 0);
    ProductClassificationSystemModel = __decorate([
        (0, json2typescript_1.JsonObject)("ProductClassificationSystemModel")
    ], ProductClassificationSystemModel);
    return ProductClassificationSystemModel;
}());
