// avatax/lib/models/RefundTransactionModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RefundTransactionModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Refund a committed transaction
 * @export
 * @class RefundTransactionModel
 */
var RefundTransactionModel = exports.RefundTransactionModel = /** @class */ (function () {
    function RefundTransactionModel() {
        /**
         * @type {string}
         * @memberof RefundTransactionModel
         */
        this.refundTransactionCode = undefined;
        /**
         * @type {Date}
         * @memberof RefundTransactionModel
         */
        this.refundDate = undefined;
        /**
         * @type {Enums.RefundType}
         * @memberof RefundTransactionModel
         */
        this.refundType = undefined;
        /**
         * @type {number}
         * @memberof RefundTransactionModel
         */
        this.refundPercentage = undefined;
        /**
         * @type {string[]}
         * @memberof RefundTransactionModel
         */
        this.refundLines = undefined;
        /**
         * @type {string}
         * @memberof RefundTransactionModel
         */
        this.referenceCode = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("refundTransactionCode", String, true),
        __metadata("design:type", String)
    ], RefundTransactionModel.prototype, "refundTransactionCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("refundDate", dateConverter_1.DateConverter),
        __metadata("design:type", Date)
    ], RefundTransactionModel.prototype, "refundDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("refundType", Enums.RefundTypeConverter, true),
        __metadata("design:type", Number)
    ], RefundTransactionModel.prototype, "refundType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("refundPercentage", Number, true),
        __metadata("design:type", Number)
    ], RefundTransactionModel.prototype, "refundPercentage", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("refundLines", [String], true),
        __metadata("design:type", Array)
    ], RefundTransactionModel.prototype, "refundLines", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("referenceCode", String, true),
        __metadata("design:type", String)
    ], RefundTransactionModel.prototype, "referenceCode", void 0);
    RefundTransactionModel = __decorate([
        (0, json2typescript_1.JsonObject)("RefundTransactionModel")
    ], RefundTransactionModel);
    return RefundTransactionModel;
}());
