// avatax/lib/models/ReportModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportModel = void 0;
var ReportParametersModel_1 = require("./ReportParametersModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * A model for displaying report task metadata
 * @export
 * @class ReportModel
 */
var ReportModel = exports.ReportModel = /** @class */ (function () {
    function ReportModel() {
        /**
         * @type {number}
         * @memberof ReportModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof ReportModel
         */
        this.accountId = undefined;
        /**
         * @type {number}
         * @memberof ReportModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof ReportModel
         */
        this.reportType = undefined;
        /**
         * @type {ReportParametersModel}
         * @memberof ReportModel
         */
        this.parameters = undefined;
        /**
         * @type {string}
         * @memberof ReportModel
         */
        this.status = undefined;
        /**
         * @type {number}
         * @memberof ReportModel
         */
        this.size = undefined;
        /**
         * @type {string}
         * @memberof ReportModel
         */
        this.format = undefined;
        /**
         * @type {string}
         * @memberof ReportModel
         */
        this.file = undefined;
        /**
         * @type {Date}
         * @memberof ReportModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof ReportModel
         */
        this.createdUserId = undefined;
        /**
         * @type {string}
         * @memberof ReportModel
         */
        this.createdUser = undefined;
        /**
         * @type {Date}
         * @memberof ReportModel
         */
        this.completedDate = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], ReportModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountId", Number, true),
        __metadata("design:type", Number)
    ], ReportModel.prototype, "accountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], ReportModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportType", String, true),
        __metadata("design:type", String)
    ], ReportModel.prototype, "reportType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("parameters", ReportParametersModel_1.ReportParametersModel, true),
        __metadata("design:type", ReportParametersModel_1.ReportParametersModel)
    ], ReportModel.prototype, "parameters", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("status", String, true),
        __metadata("design:type", String)
    ], ReportModel.prototype, "status", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("size", Number, true),
        __metadata("design:type", Number)
    ], ReportModel.prototype, "size", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("format", String, true),
        __metadata("design:type", String)
    ], ReportModel.prototype, "format", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("file", String, true),
        __metadata("design:type", String)
    ], ReportModel.prototype, "file", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ReportModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], ReportModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUser", String, true),
        __metadata("design:type", String)
    ], ReportModel.prototype, "createdUser", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("completedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ReportModel.prototype, "completedDate", void 0);
    ReportModel = __decorate([
        (0, json2typescript_1.JsonObject)("ReportModel")
    ], ReportModel);
    return ReportModel;
}());
