// avatax/lib/models/ResourceFileUploadRequestModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceFileUploadRequestModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * A request to upload a file to Resource Files
 * @export
 * @class ResourceFileUploadRequestModel
 */
var ResourceFileUploadRequestModel = exports.ResourceFileUploadRequestModel = /** @class */ (function () {
    function ResourceFileUploadRequestModel() {
        /**
         * @type {string}
         * @memberof ResourceFileUploadRequestModel
         */
        this.content = undefined;
        /**
         * @type {string}
         * @memberof ResourceFileUploadRequestModel
         */
        this.username = undefined;
        /**
         * @type {number}
         * @memberof ResourceFileUploadRequestModel
         */
        this.accountId = undefined;
        /**
         * @type {number}
         * @memberof ResourceFileUploadRequestModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof ResourceFileUploadRequestModel
         */
        this.name = undefined;
        /**
         * @type {number}
         * @memberof ResourceFileUploadRequestModel
         */
        this.resourceFileTypeId = undefined;
        /**
         * @type {number}
         * @memberof ResourceFileUploadRequestModel
         */
        this.length = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("content", String),
        __metadata("design:type", String)
    ], ResourceFileUploadRequestModel.prototype, "content", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("username", String, true),
        __metadata("design:type", String)
    ], ResourceFileUploadRequestModel.prototype, "username", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountId", Number, true),
        __metadata("design:type", Number)
    ], ResourceFileUploadRequestModel.prototype, "accountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], ResourceFileUploadRequestModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String, true),
        __metadata("design:type", String)
    ], ResourceFileUploadRequestModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("resourceFileTypeId", Number, true),
        __metadata("design:type", Number)
    ], ResourceFileUploadRequestModel.prototype, "resourceFileTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("length", Number, true),
        __metadata("design:type", Number)
    ], ResourceFileUploadRequestModel.prototype, "length", void 0);
    ResourceFileUploadRequestModel = __decorate([
        (0, json2typescript_1.JsonObject)("ResourceFileUploadRequestModel")
    ], ResourceFileUploadRequestModel);
    return ResourceFileUploadRequestModel;
}());
