// avatax/lib/models/TaxCodeModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxCodeModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a tax code that can be applied to items on a transaction.
A tax code can have specific rules for specific jurisdictions that change the tax calculation behavior.
 * @export
 * @class TaxCodeModel
 */
var TaxCodeModel = exports.TaxCodeModel = /** @class */ (function () {
    function TaxCodeModel() {
        /**
         * @type {number}
         * @memberof TaxCodeModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof TaxCodeModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof TaxCodeModel
         */
        this.taxCode = undefined;
        /**
         * @type {string}
         * @memberof TaxCodeModel
         */
        this.taxCodeTypeId = undefined;
        /**
         * @type {string}
         * @memberof TaxCodeModel
         */
        this.description = undefined;
        /**
         * @type {string}
         * @memberof TaxCodeModel
         */
        this.parentTaxCode = undefined;
        /**
         * @type {boolean}
         * @memberof TaxCodeModel
         */
        this.isPhysical = undefined;
        /**
         * @type {number}
         * @memberof TaxCodeModel
         */
        this.goodsServiceCode = undefined;
        /**
         * @type {string}
         * @memberof TaxCodeModel
         */
        this.entityUseCode = undefined;
        /**
         * @type {boolean}
         * @memberof TaxCodeModel
         */
        this.isActive = undefined;
        /**
         * @type {boolean}
         * @memberof TaxCodeModel
         */
        this.isSSTCertified = undefined;
        /**
         * @type {Date}
         * @memberof TaxCodeModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof TaxCodeModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof TaxCodeModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof TaxCodeModel
         */
        this.modifiedUserId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], TaxCodeModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], TaxCodeModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCode", String),
        __metadata("design:type", String)
    ], TaxCodeModel.prototype, "taxCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCodeTypeId", String),
        __metadata("design:type", String)
    ], TaxCodeModel.prototype, "taxCodeTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], TaxCodeModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("parentTaxCode", String, true),
        __metadata("design:type", String)
    ], TaxCodeModel.prototype, "parentTaxCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isPhysical", Boolean, true),
        __metadata("design:type", Boolean)
    ], TaxCodeModel.prototype, "isPhysical", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("goodsServiceCode", Number, true),
        __metadata("design:type", Number)
    ], TaxCodeModel.prototype, "goodsServiceCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("entityUseCode", String, true),
        __metadata("design:type", String)
    ], TaxCodeModel.prototype, "entityUseCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isActive", Boolean, true),
        __metadata("design:type", Boolean)
    ], TaxCodeModel.prototype, "isActive", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSSTCertified", Boolean, true),
        __metadata("design:type", Boolean)
    ], TaxCodeModel.prototype, "isSSTCertified", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TaxCodeModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], TaxCodeModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TaxCodeModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], TaxCodeModel.prototype, "modifiedUserId", void 0);
    TaxCodeModel = __decorate([
        (0, json2typescript_1.JsonObject)("TaxCodeModel")
    ], TaxCodeModel);
    return TaxCodeModel;
}());
