// avatax/lib/models/TaxOverrideModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxOverrideModel = void 0;
var Enums = require("../enums/index");
var TransactionLineTaxAmountByTaxTypeModel_1 = require("./TransactionLineTaxAmountByTaxTypeModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a tax override for a transaction
 * @export
 * @class TaxOverrideModel
 */
var TaxOverrideModel = exports.TaxOverrideModel = /** @class */ (function () {
    function TaxOverrideModel() {
        /**
         * @type {Enums.TaxOverrideType}
         * @memberof TaxOverrideModel
         */
        this.type = undefined;
        /**
         * @type {number}
         * @memberof TaxOverrideModel
         */
        this.taxAmount = undefined;
        /**
         * @type {Date}
         * @memberof TaxOverrideModel
         */
        this.taxDate = undefined;
        /**
         * @type {string}
         * @memberof TaxOverrideModel
         */
        this.reason = undefined;
        /**
         * @type {TransactionLineTaxAmountByTaxTypeModel[]}
         * @memberof TaxOverrideModel
         */
        this.taxAmountByTaxTypes = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("type", Enums.TaxOverrideTypeConverter, true),
        __metadata("design:type", Number)
    ], TaxOverrideModel.prototype, "type", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAmount", Number, true),
        __metadata("design:type", Number)
    ], TaxOverrideModel.prototype, "taxAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TaxOverrideModel.prototype, "taxDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reason", String, true),
        __metadata("design:type", String)
    ], TaxOverrideModel.prototype, "reason", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAmountByTaxTypes", [TransactionLineTaxAmountByTaxTypeModel_1.TransactionLineTaxAmountByTaxTypeModel], true),
        __metadata("design:type", Array)
    ], TaxOverrideModel.prototype, "taxAmountByTaxTypes", void 0);
    TaxOverrideModel = __decorate([
        (0, json2typescript_1.JsonObject)("TaxOverrideModel")
    ], TaxOverrideModel);
    return TaxOverrideModel;
}());
