// avatax/lib/models/TaxRuleProductDetailModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxRuleProductDetailModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a tax rule product detail that changes the behavior of Avalara's tax engine for certain tax rules.
            
Avalara supports a two types of tax product detail.  For information about tax rule Product Types
HSCode and TaxCode
            
Because different types of tax rules have different behavior, some fields may change their behavior based on
the type of tax rule selected.  Please read the documentation for each field carefully and ensure that
the value you send is appropriate for the type of tax rule.
 * @export
 * @class TaxRuleProductDetailModel
 */
var TaxRuleProductDetailModel = exports.TaxRuleProductDetailModel = /** @class */ (function () {
    function TaxRuleProductDetailModel() {
        /**
         * @type {number}
         * @memberof TaxRuleProductDetailModel
         */
        this.taxRuleProductDetailId = undefined;
        /**
         * @type {number}
         * @memberof TaxRuleProductDetailModel
         */
        this.taxRuleId = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleProductDetailModel
         */
        this.productCode = undefined;
        /**
         * @type {Date}
         * @memberof TaxRuleProductDetailModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof TaxRuleProductDetailModel
         */
        this.endDate = undefined;
        /**
         * @type {number}
         * @memberof TaxRuleProductDetailModel
         */
        this.systemId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxRuleProductDetailId", Number, true),
        __metadata("design:type", Number)
    ], TaxRuleProductDetailModel.prototype, "taxRuleProductDetailId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxRuleId", Number, true),
        __metadata("design:type", Number)
    ], TaxRuleProductDetailModel.prototype, "taxRuleId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("productCode", String, true),
        __metadata("design:type", String)
    ], TaxRuleProductDetailModel.prototype, "productCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TaxRuleProductDetailModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TaxRuleProductDetailModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("systemId", Number, true),
        __metadata("design:type", Number)
    ], TaxRuleProductDetailModel.prototype, "systemId", void 0);
    TaxRuleProductDetailModel = __decorate([
        (0, json2typescript_1.JsonObject)("TaxRuleProductDetailModel")
    ], TaxRuleProductDetailModel);
    return TaxRuleProductDetailModel;
}());
