// avatax/lib/models/TaxRegionJurisdictionModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxRegionJurisdictionModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * The tax region jurisdiction model.
 * @export
 * @class TaxRegionJurisdictionModel
 */
var TaxRegionJurisdictionModel = exports.TaxRegionJurisdictionModel = /** @class */ (function () {
    function TaxRegionJurisdictionModel() {
        /**
         * @type {number}
         * @memberof TaxRegionJurisdictionModel
         */
        this.jurisdictionId = undefined;
        /**
         * @type {number}
         * @memberof TaxRegionJurisdictionModel
         */
        this.taxRegionId = undefined;
        /**
         * @type {number}
         * @memberof TaxRegionJurisdictionModel
         */
        this.jurisdictionLevelId = undefined;
        /**
         * @type {string}
         * @memberof TaxRegionJurisdictionModel
         */
        this.rockName = undefined;
        /**
         * @type {number}
         * @memberof TaxRegionJurisdictionModel
         */
        this.reportLevel = undefined;
        /**
         * @type {string}
         * @memberof TaxRegionJurisdictionModel
         */
        this.stateAssignedCode = undefined;
        /**
         * @type {number}
         * @memberof TaxRegionJurisdictionModel
         */
        this.taxAuthorityId = undefined;
        /**
         * @type {string}
         * @memberof TaxRegionJurisdictionModel
         */
        this.signatureCode = undefined;
        /**
         * @type {Date}
         * @memberof TaxRegionJurisdictionModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof TaxRegionJurisdictionModel
         */
        this.endDate = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionId", Number, true),
        __metadata("design:type", Number)
    ], TaxRegionJurisdictionModel.prototype, "jurisdictionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxRegionId", Number, true),
        __metadata("design:type", Number)
    ], TaxRegionJurisdictionModel.prototype, "taxRegionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionLevelId", Number, true),
        __metadata("design:type", Number)
    ], TaxRegionJurisdictionModel.prototype, "jurisdictionLevelId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rockName", String, true),
        __metadata("design:type", String)
    ], TaxRegionJurisdictionModel.prototype, "rockName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportLevel", Number, true),
        __metadata("design:type", Number)
    ], TaxRegionJurisdictionModel.prototype, "reportLevel", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("stateAssignedCode", String, true),
        __metadata("design:type", String)
    ], TaxRegionJurisdictionModel.prototype, "stateAssignedCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityId", Number, true),
        __metadata("design:type", Number)
    ], TaxRegionJurisdictionModel.prototype, "taxAuthorityId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("signatureCode", String, true),
        __metadata("design:type", String)
    ], TaxRegionJurisdictionModel.prototype, "signatureCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TaxRegionJurisdictionModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TaxRegionJurisdictionModel.prototype, "endDate", void 0);
    TaxRegionJurisdictionModel = __decorate([
        (0, json2typescript_1.JsonObject)("TaxRegionJurisdictionModel")
    ], TaxRegionJurisdictionModel);
    return TaxRegionJurisdictionModel;
}());
