// avatax/lib/models/TaxRegionModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxRegionModel = void 0;
var DenormalizedJurisModel_1 = require("./DenormalizedJurisModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * The tax region model.
 * @export
 * @class TaxRegionModel
 */
var TaxRegionModel = exports.TaxRegionModel = /** @class */ (function () {
    function TaxRegionModel() {
        /**
         * @type {number}
         * @memberof TaxRegionModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof TaxRegionModel
         */
        this.code = undefined;
        /**
         * @type {string}
         * @memberof TaxRegionModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof TaxRegionModel
         */
        this.county = undefined;
        /**
         * @type {string}
         * @memberof TaxRegionModel
         */
        this.city = undefined;
        /**
         * @type {string}
         * @memberof TaxRegionModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof TaxRegionModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof TaxRegionModel
         */
        this.serCode = undefined;
        /**
         * @type {string}
         * @memberof TaxRegionModel
         */
        this.signatureCode = undefined;
        /**
         * @type {Date}
         * @memberof TaxRegionModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof TaxRegionModel
         */
        this.endDate = undefined;
        /**
         * @type {boolean}
         * @memberof TaxRegionModel
         */
        this.isAcm = undefined;
        /**
         * @type {boolean}
         * @memberof TaxRegionModel
         */
        this.isSst = undefined;
        /**
         * @type {DenormalizedJurisModel[]}
         * @memberof TaxRegionModel
         */
        this.jurisdictions = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], TaxRegionModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("code", String, true),
        __metadata("design:type", String)
    ], TaxRegionModel.prototype, "code", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String, true),
        __metadata("design:type", String)
    ], TaxRegionModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("county", String, true),
        __metadata("design:type", String)
    ], TaxRegionModel.prototype, "county", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("city", String, true),
        __metadata("design:type", String)
    ], TaxRegionModel.prototype, "city", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], TaxRegionModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], TaxRegionModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("serCode", String, true),
        __metadata("design:type", String)
    ], TaxRegionModel.prototype, "serCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("signatureCode", String, true),
        __metadata("design:type", String)
    ], TaxRegionModel.prototype, "signatureCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TaxRegionModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TaxRegionModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isAcm", Boolean, true),
        __metadata("design:type", Boolean)
    ], TaxRegionModel.prototype, "isAcm", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSst", Boolean, true),
        __metadata("design:type", Boolean)
    ], TaxRegionModel.prototype, "isSst", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictions", [DenormalizedJurisModel_1.DenormalizedJurisModel], true),
        __metadata("design:type", Array)
    ], TaxRegionModel.prototype, "jurisdictions", void 0);
    TaxRegionModel = __decorate([
        (0, json2typescript_1.JsonObject)("TaxRegionModel")
    ], TaxRegionModel);
    return TaxRegionModel;
}());
