// avatax/lib/models/TaxRuleModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxRuleModel = void 0;
var Enums = require("../enums/index");
var TaxRuleProductDetailModel_1 = require("./TaxRuleProductDetailModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a tax rule that changes the behavior of Avalara's tax engine for certain products and/or entity use codes
in certain jurisdictions.
            
Avalara supports a few different types of tax rules.  For information about tax rule types, see
[TaxRuleTypeId](https://developer.avalara.com/api-reference/avatax/rest/v2/models/enums/TaxRuleTypeId/)
            
Because different types of tax rules have different behavior, some fields may change their behavior based on
the type of tax rule selected.  Please read the documentation for each field carefully and ensure that
the value you send is appropriate for the type of tax rule.
 * @export
 * @class TaxRuleModel
 */
var TaxRuleModel = exports.TaxRuleModel = /** @class */ (function () {
    function TaxRuleModel() {
        /**
         * @type {number}
         * @memberof TaxRuleModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof TaxRuleModel
         */
        this.companyId = undefined;
        /**
         * @type {number}
         * @memberof TaxRuleModel
         */
        this.taxCodeId = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.taxCode = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.stateFIPS = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.jurisName = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.jurisCode = undefined;
        /**
         * @type {Enums.JurisTypeId}
         * @memberof TaxRuleModel
         */
        this.jurisTypeId = undefined;
        /**
         * @type {Enums.JurisdictionType}
         * @memberof TaxRuleModel
         */
        this.jurisdictionTypeId = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.customerUsageType = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.entityUseCode = undefined;
        /**
         * @type {Enums.MatchingTaxType}
         * @memberof TaxRuleModel
         */
        this.taxTypeId = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.taxTypeCode = undefined;
        /**
         * @type {TaxRuleProductDetailModel[]}
         * @memberof TaxRuleModel
         */
        this.taxRuleProductDetail = undefined;
        /**
         * @type {Enums.RateType}
         * @memberof TaxRuleModel
         */
        this.rateTypeId = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.rateTypeCode = undefined;
        /**
         * @type {Enums.TaxRuleTypeId}
         * @memberof TaxRuleModel
         */
        this.taxRuleTypeId = undefined;
        /**
         * @type {boolean}
         * @memberof TaxRuleModel
         */
        this.isAllJuris = undefined;
        /**
         * @type {number}
         * @memberof TaxRuleModel
         */
        this.value = undefined;
        /**
         * @type {number}
         * @memberof TaxRuleModel
         */
        this.cap = undefined;
        /**
         * @type {number}
         * @memberof TaxRuleModel
         */
        this.threshold = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.options = undefined;
        /**
         * @type {Date}
         * @memberof TaxRuleModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof TaxRuleModel
         */
        this.endDate = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.description = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.countyFIPS = undefined;
        /**
         * @type {boolean}
         * @memberof TaxRuleModel
         */
        this.isSTPro = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.region = undefined;
        /**
         * @type {Enums.Sourcing}
         * @memberof TaxRuleModel
         */
        this.sourcing = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.taxTypeGroup = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.taxSubType = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.nonPassthroughExpression = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.currencyCode = undefined;
        /**
         * @type {number}
         * @memberof TaxRuleModel
         */
        this.preferredProgramId = undefined;
        /**
         * @type {number}
         * @memberof TaxRuleModel
         */
        this.uomId = undefined;
        /**
         * @type {Date}
         * @memberof TaxRuleModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof TaxRuleModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof TaxRuleModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof TaxRuleModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {string}
         * @memberof TaxRuleModel
         */
        this.unitOfBasis = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number),
        __metadata("design:type", Number)
    ], TaxRuleModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], TaxRuleModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCodeId", Number, true),
        __metadata("design:type", Number)
    ], TaxRuleModel.prototype, "taxCodeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCode", String, true),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "taxCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("stateFIPS", String, true),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "stateFIPS", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisName", String, true),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "jurisName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisCode", String),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "jurisCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisTypeId", Enums.JurisTypeIdConverter, true),
        __metadata("design:type", Number)
    ], TaxRuleModel.prototype, "jurisTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionTypeId", Enums.JurisdictionTypeConverter, true),
        __metadata("design:type", Number)
    ], TaxRuleModel.prototype, "jurisdictionTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerUsageType", String, true),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "customerUsageType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("entityUseCode", String, true),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "entityUseCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxTypeId", Enums.MatchingTaxTypeConverter, true),
        __metadata("design:type", Number)
    ], TaxRuleModel.prototype, "taxTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxTypeCode", String, true),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "taxTypeCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxRuleProductDetail", [TaxRuleProductDetailModel_1.TaxRuleProductDetailModel], true),
        __metadata("design:type", Array)
    ], TaxRuleModel.prototype, "taxRuleProductDetail", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rateTypeId", Enums.RateTypeConverter, true),
        __metadata("design:type", Number)
    ], TaxRuleModel.prototype, "rateTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rateTypeCode", String, true),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "rateTypeCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxRuleTypeId", Enums.TaxRuleTypeIdConverter),
        __metadata("design:type", Number)
    ], TaxRuleModel.prototype, "taxRuleTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isAllJuris", Boolean, true),
        __metadata("design:type", Boolean)
    ], TaxRuleModel.prototype, "isAllJuris", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("value", Number, true),
        __metadata("design:type", Number)
    ], TaxRuleModel.prototype, "value", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("cap", Number, true),
        __metadata("design:type", Number)
    ], TaxRuleModel.prototype, "cap", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("threshold", Number, true),
        __metadata("design:type", Number)
    ], TaxRuleModel.prototype, "threshold", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("options", String, true),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "options", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TaxRuleModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TaxRuleModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("countyFIPS", String, true),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "countyFIPS", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSTPro", Boolean, true),
        __metadata("design:type", Boolean)
    ], TaxRuleModel.prototype, "isSTPro", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("sourcing", Enums.SourcingConverter, true),
        __metadata("design:type", Number)
    ], TaxRuleModel.prototype, "sourcing", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxTypeGroup", String, true),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "taxTypeGroup", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxSubType", String, true),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "taxSubType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonPassthroughExpression", String, true),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "nonPassthroughExpression", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("currencyCode", String, true),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "currencyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("preferredProgramId", Number, true),
        __metadata("design:type", Number)
    ], TaxRuleModel.prototype, "preferredProgramId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("uomId", Number, true),
        __metadata("design:type", Number)
    ], TaxRuleModel.prototype, "uomId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TaxRuleModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], TaxRuleModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TaxRuleModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], TaxRuleModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("unitOfBasis", String, true),
        __metadata("design:type", String)
    ], TaxRuleModel.prototype, "unitOfBasis", void 0);
    TaxRuleModel = __decorate([
        (0, json2typescript_1.JsonObject)("TaxRuleModel")
    ], TaxRuleModel);
    return TaxRuleModel;
}());
