// avatax/lib/models/TaxTypeGroupModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxTypeGroupModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Represents a tax type group
 * @export
 * @class TaxTypeGroupModel
 */
var TaxTypeGroupModel = exports.TaxTypeGroupModel = /** @class */ (function () {
    function TaxTypeGroupModel() {
        /**
         * @type {number}
         * @memberof TaxTypeGroupModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof TaxTypeGroupModel
         */
        this.taxTypeGroup = undefined;
        /**
         * @type {string}
         * @memberof TaxTypeGroupModel
         */
        this.description = undefined;
        /**
         * @type {number}
         * @memberof TaxTypeGroupModel
         */
        this.subscriptionTypeId = undefined;
        /**
         * @type {string}
         * @memberof TaxTypeGroupModel
         */
        this.subscriptionDescription = undefined;
        /**
         * @type {string}
         * @memberof TaxTypeGroupModel
         */
        this.tabName = undefined;
        /**
         * @type {boolean}
         * @memberof TaxTypeGroupModel
         */
        this.showColumn = undefined;
        /**
         * @type {number}
         * @memberof TaxTypeGroupModel
         */
        this.displaySequence = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], TaxTypeGroupModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxTypeGroup", String, true),
        __metadata("design:type", String)
    ], TaxTypeGroupModel.prototype, "taxTypeGroup", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], TaxTypeGroupModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("subscriptionTypeId", Number, true),
        __metadata("design:type", Number)
    ], TaxTypeGroupModel.prototype, "subscriptionTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("subscriptionDescription", String, true),
        __metadata("design:type", String)
    ], TaxTypeGroupModel.prototype, "subscriptionDescription", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("tabName", String, true),
        __metadata("design:type", String)
    ], TaxTypeGroupModel.prototype, "tabName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("showColumn", Boolean, true),
        __metadata("design:type", Boolean)
    ], TaxTypeGroupModel.prototype, "showColumn", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("displaySequence", Number, true),
        __metadata("design:type", Number)
    ], TaxTypeGroupModel.prototype, "displaySequence", void 0);
    TaxTypeGroupModel = __decorate([
        (0, json2typescript_1.JsonObject)("TaxTypeGroupModel")
    ], TaxTypeGroupModel);
    return TaxTypeGroupModel;
}());
