// avatax/lib/models/TransactionModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionModel = void 0;
var Enums = require("../enums/index");
var TransactionLineModel_1 = require("./TransactionLineModel");
var TransactionAddressModel_1 = require("./TransactionAddressModel");
var TransactionLocationTypeModel_1 = require("./TransactionLocationTypeModel");
var TransactionSummary_1 = require("./TransactionSummary");
var TaxDetailsByTaxType_1 = require("./TaxDetailsByTaxType");
var TransactionParameterModel_1 = require("./TransactionParameterModel");
var TransactionUserDefinedFieldModel_1 = require("./TransactionUserDefinedFieldModel");
var AvaTaxMessage_1 = require("./AvaTaxMessage");
var InvoiceMessageModel_1 = require("./InvoiceMessageModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * This object represents a single transaction; for example, a sales invoice or purchase order.
 * @export
 * @class TransactionModel
 */
var TransactionModel = exports.TransactionModel = /** @class */ (function () {
    function TransactionModel() {
        /**
         * @type {number}
         * @memberof TransactionModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.code = undefined;
        /**
         * @type {number}
         * @memberof TransactionModel
         */
        this.companyId = undefined;
        /**
         * @type {Date}
         * @memberof TransactionModel
         */
        this.date = undefined;
        /**
         * @type {Date}
         * @memberof TransactionModel
         */
        this.paymentDate = undefined;
        /**
         * @type {Enums.DocumentStatus}
         * @memberof TransactionModel
         */
        this.status = undefined;
        /**
         * @type {Enums.DocumentType}
         * @memberof TransactionModel
         */
        this.type = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.batchCode = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.currencyCode = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.exchangeRateCurrencyCode = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.customerUsageType = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.entityUseCode = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.customerVendorCode = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.customerCode = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.exemptNo = undefined;
        /**
         * @type {boolean}
         * @memberof TransactionModel
         */
        this.reconciled = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.locationCode = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.reportingLocationCode = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.purchaseOrderNo = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.referenceCode = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.salespersonCode = undefined;
        /**
         * @type {Enums.TaxOverrideType}
         * @memberof TransactionModel
         */
        this.taxOverrideType = undefined;
        /**
         * @type {number}
         * @memberof TransactionModel
         */
        this.taxOverrideAmount = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.taxOverrideReason = undefined;
        /**
         * @type {number}
         * @memberof TransactionModel
         */
        this.totalAmount = undefined;
        /**
         * @type {number}
         * @memberof TransactionModel
         */
        this.totalExempt = undefined;
        /**
         * @type {number}
         * @memberof TransactionModel
         */
        this.totalDiscount = undefined;
        /**
         * @type {number}
         * @memberof TransactionModel
         */
        this.totalTax = undefined;
        /**
         * @type {number}
         * @memberof TransactionModel
         */
        this.totalTaxable = undefined;
        /**
         * @type {number}
         * @memberof TransactionModel
         */
        this.totalTaxCalculated = undefined;
        /**
         * @type {Enums.AdjustmentReason}
         * @memberof TransactionModel
         */
        this.adjustmentReason = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.adjustmentDescription = undefined;
        /**
         * @type {boolean}
         * @memberof TransactionModel
         */
        this.locked = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.country = undefined;
        /**
         * @type {number}
         * @memberof TransactionModel
         */
        this.version = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.softwareVersion = undefined;
        /**
         * @type {number}
         * @memberof TransactionModel
         */
        this.originAddressId = undefined;
        /**
         * @type {number}
         * @memberof TransactionModel
         */
        this.destinationAddressId = undefined;
        /**
         * @type {Date}
         * @memberof TransactionModel
         */
        this.exchangeRateEffectiveDate = undefined;
        /**
         * @type {number}
         * @memberof TransactionModel
         */
        this.exchangeRate = undefined;
        /**
         * @type {boolean}
         * @memberof TransactionModel
         */
        this.isSellerImporterOfRecord = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.description = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.email = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.businessIdentificationNo = undefined;
        /**
         * @type {Date}
         * @memberof TransactionModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof TransactionModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {Date}
         * @memberof TransactionModel
         */
        this.taxDate = undefined;
        /**
         * @type {TransactionLineModel[]}
         * @memberof TransactionModel
         */
        this.lines = undefined;
        /**
         * @type {TransactionAddressModel[]}
         * @memberof TransactionModel
         */
        this.addresses = undefined;
        /**
         * @type {TransactionLocationTypeModel[]}
         * @memberof TransactionModel
         */
        this.locationTypes = undefined;
        /**
         * @type {TransactionSummary[]}
         * @memberof TransactionModel
         */
        this.summary = undefined;
        /**
         * @type {TaxDetailsByTaxType[]}
         * @memberof TransactionModel
         */
        this.taxDetailsByTaxType = undefined;
        /**
         * @type {TransactionParameterModel[]}
         * @memberof TransactionModel
         */
        this.parameters = undefined;
        /**
         * @type {TransactionUserDefinedFieldModel[]}
         * @memberof TransactionModel
         */
        this.userDefinedFields = undefined;
        /**
         * @type {AvaTaxMessage[]}
         * @memberof TransactionModel
         */
        this.messages = undefined;
        /**
         * @type {InvoiceMessageModel[]}
         * @memberof TransactionModel
         */
        this.invoiceMessages = undefined;
        /**
         * @type {string}
         * @memberof TransactionModel
         */
        this.customerSupplierName = undefined;
        /**
         * @type {number}
         * @memberof TransactionModel
         */
        this.dataSourceId = undefined;
        /**
         * @type {Enums.DeliveryTerms}
         * @memberof TransactionModel
         */
        this.deliveryTerms = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("code", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "code", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("date", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TransactionModel.prototype, "date", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("paymentDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TransactionModel.prototype, "paymentDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("status", Enums.DocumentStatusConverter, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "status", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("type", Enums.DocumentTypeConverter, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "type", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("batchCode", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "batchCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("currencyCode", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "currencyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exchangeRateCurrencyCode", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "exchangeRateCurrencyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerUsageType", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "customerUsageType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("entityUseCode", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "entityUseCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerVendorCode", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "customerVendorCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerCode", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "customerCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptNo", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "exemptNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reconciled", Boolean, true),
        __metadata("design:type", Boolean)
    ], TransactionModel.prototype, "reconciled", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("locationCode", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "locationCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportingLocationCode", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "reportingLocationCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("purchaseOrderNo", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "purchaseOrderNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("referenceCode", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "referenceCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("salespersonCode", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "salespersonCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxOverrideType", Enums.TaxOverrideTypeConverter, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "taxOverrideType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxOverrideAmount", Number, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "taxOverrideAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxOverrideReason", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "taxOverrideReason", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalAmount", Number, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "totalAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalExempt", Number, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "totalExempt", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalDiscount", Number, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "totalDiscount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalTax", Number, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "totalTax", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalTaxable", Number, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "totalTaxable", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalTaxCalculated", Number, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "totalTaxCalculated", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("adjustmentReason", Enums.AdjustmentReasonConverter, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "adjustmentReason", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("adjustmentDescription", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "adjustmentDescription", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("locked", Boolean, true),
        __metadata("design:type", Boolean)
    ], TransactionModel.prototype, "locked", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("version", Number, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "version", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("softwareVersion", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "softwareVersion", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("originAddressId", Number, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "originAddressId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("destinationAddressId", Number, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "destinationAddressId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exchangeRateEffectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TransactionModel.prototype, "exchangeRateEffectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exchangeRate", Number, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "exchangeRate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSellerImporterOfRecord", Boolean, true),
        __metadata("design:type", Boolean)
    ], TransactionModel.prototype, "isSellerImporterOfRecord", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("email", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "email", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("businessIdentificationNo", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "businessIdentificationNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TransactionModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TransactionModel.prototype, "taxDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lines", [TransactionLineModel_1.TransactionLineModel], true),
        __metadata("design:type", Array)
    ], TransactionModel.prototype, "lines", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("addresses", [TransactionAddressModel_1.TransactionAddressModel], true),
        __metadata("design:type", Array)
    ], TransactionModel.prototype, "addresses", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("locationTypes", [TransactionLocationTypeModel_1.TransactionLocationTypeModel], true),
        __metadata("design:type", Array)
    ], TransactionModel.prototype, "locationTypes", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("summary", [TransactionSummary_1.TransactionSummary], true),
        __metadata("design:type", Array)
    ], TransactionModel.prototype, "summary", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxDetailsByTaxType", [TaxDetailsByTaxType_1.TaxDetailsByTaxType], true),
        __metadata("design:type", Array)
    ], TransactionModel.prototype, "taxDetailsByTaxType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("parameters", [TransactionParameterModel_1.TransactionParameterModel], true),
        __metadata("design:type", Array)
    ], TransactionModel.prototype, "parameters", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("userDefinedFields", [TransactionUserDefinedFieldModel_1.TransactionUserDefinedFieldModel], true),
        __metadata("design:type", Array)
    ], TransactionModel.prototype, "userDefinedFields", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("messages", [AvaTaxMessage_1.AvaTaxMessage], true),
        __metadata("design:type", Array)
    ], TransactionModel.prototype, "messages", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("invoiceMessages", [InvoiceMessageModel_1.InvoiceMessageModel], true),
        __metadata("design:type", Array)
    ], TransactionModel.prototype, "invoiceMessages", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerSupplierName", String, true),
        __metadata("design:type", String)
    ], TransactionModel.prototype, "customerSupplierName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dataSourceId", Number, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "dataSourceId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("deliveryTerms", Enums.DeliveryTermsConverter, true),
        __metadata("design:type", Number)
    ], TransactionModel.prototype, "deliveryTerms", void 0);
    TransactionModel = __decorate([
        (0, json2typescript_1.JsonObject)("TransactionModel")
    ], TransactionModel);
    return TransactionModel;
}());
