// avatax/lib/models/TransactionSummary.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionSummary = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
/**
 * Summary information about an overall transaction.
 * @export
 * @class TransactionSummary
 */
var TransactionSummary = exports.TransactionSummary = /** @class */ (function () {
    function TransactionSummary() {
        /**
         * @type {string}
         * @memberof TransactionSummary
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof TransactionSummary
         */
        this.region = undefined;
        /**
         * @type {Enums.JurisdictionType}
         * @memberof TransactionSummary
         */
        this.jurisType = undefined;
        /**
         * @type {string}
         * @memberof TransactionSummary
         */
        this.jurisCode = undefined;
        /**
         * @type {string}
         * @memberof TransactionSummary
         */
        this.jurisName = undefined;
        /**
         * @type {number}
         * @memberof TransactionSummary
         */
        this.taxAuthorityType = undefined;
        /**
         * @type {string}
         * @memberof TransactionSummary
         */
        this.stateAssignedNo = undefined;
        /**
         * @type {string}
         * @memberof TransactionSummary
         */
        this.taxType = undefined;
        /**
         * @type {string}
         * @memberof TransactionSummary
         */
        this.taxSubType = undefined;
        /**
         * @type {string}
         * @memberof TransactionSummary
         */
        this.taxName = undefined;
        /**
         * @type {string}
         * @memberof TransactionSummary
         */
        this.taxGroup = undefined;
        /**
         * @type {Enums.RateType}
         * @memberof TransactionSummary
         */
        this.rateType = undefined;
        /**
         * @type {string}
         * @memberof TransactionSummary
         */
        this.rateTypeCode = undefined;
        /**
         * @type {number}
         * @memberof TransactionSummary
         */
        this.taxable = undefined;
        /**
         * @type {number}
         * @memberof TransactionSummary
         */
        this.rate = undefined;
        /**
         * @type {number}
         * @memberof TransactionSummary
         */
        this.tax = undefined;
        /**
         * @type {number}
         * @memberof TransactionSummary
         */
        this.taxCalculated = undefined;
        /**
         * @type {number}
         * @memberof TransactionSummary
         */
        this.nonTaxable = undefined;
        /**
         * @type {number}
         * @memberof TransactionSummary
         */
        this.exemption = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], TransactionSummary.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], TransactionSummary.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisType", Enums.JurisdictionTypeConverter, true),
        __metadata("design:type", Number)
    ], TransactionSummary.prototype, "jurisType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisCode", String, true),
        __metadata("design:type", String)
    ], TransactionSummary.prototype, "jurisCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisName", String, true),
        __metadata("design:type", String)
    ], TransactionSummary.prototype, "jurisName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityType", Number, true),
        __metadata("design:type", Number)
    ], TransactionSummary.prototype, "taxAuthorityType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("stateAssignedNo", String, true),
        __metadata("design:type", String)
    ], TransactionSummary.prototype, "stateAssignedNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxType", String, true),
        __metadata("design:type", String)
    ], TransactionSummary.prototype, "taxType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxSubType", String, true),
        __metadata("design:type", String)
    ], TransactionSummary.prototype, "taxSubType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxName", String, true),
        __metadata("design:type", String)
    ], TransactionSummary.prototype, "taxName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxGroup", String, true),
        __metadata("design:type", String)
    ], TransactionSummary.prototype, "taxGroup", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rateType", Enums.RateTypeConverter, true),
        __metadata("design:type", Number)
    ], TransactionSummary.prototype, "rateType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rateTypeCode", String, true),
        __metadata("design:type", String)
    ], TransactionSummary.prototype, "rateTypeCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxable", Number, true),
        __metadata("design:type", Number)
    ], TransactionSummary.prototype, "taxable", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rate", Number, true),
        __metadata("design:type", Number)
    ], TransactionSummary.prototype, "rate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("tax", Number, true),
        __metadata("design:type", Number)
    ], TransactionSummary.prototype, "tax", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCalculated", Number, true),
        __metadata("design:type", Number)
    ], TransactionSummary.prototype, "taxCalculated", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonTaxable", Number, true),
        __metadata("design:type", Number)
    ], TransactionSummary.prototype, "nonTaxable", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemption", Number, true),
        __metadata("design:type", Number)
    ], TransactionSummary.prototype, "exemption", void 0);
    TransactionSummary = __decorate([
        (0, json2typescript_1.JsonObject)("TransactionSummary")
    ], TransactionSummary);
    return TransactionSummary;
}());
