// avatax/lib/models/TransactionAddressModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionAddressModel = void 0;
var Enums = require("../enums/index");
var JurisdictionModel_1 = require("./JurisdictionModel");
var json2typescript_1 = require("json2typescript");
/**
 * An address used within this transaction.
 * @export
 * @class TransactionAddressModel
 */
var TransactionAddressModel = exports.TransactionAddressModel = /** @class */ (function () {
    function TransactionAddressModel() {
        /**
         * @type {number}
         * @memberof TransactionAddressModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof TransactionAddressModel
         */
        this.transactionId = undefined;
        /**
         * @type {Enums.BoundaryLevel}
         * @memberof TransactionAddressModel
         */
        this.boundaryLevel = undefined;
        /**
         * @type {string}
         * @memberof TransactionAddressModel
         */
        this.line1 = undefined;
        /**
         * @type {string}
         * @memberof TransactionAddressModel
         */
        this.line2 = undefined;
        /**
         * @type {string}
         * @memberof TransactionAddressModel
         */
        this.line3 = undefined;
        /**
         * @type {string}
         * @memberof TransactionAddressModel
         */
        this.city = undefined;
        /**
         * @type {string}
         * @memberof TransactionAddressModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof TransactionAddressModel
         */
        this.postalCode = undefined;
        /**
         * @type {string}
         * @memberof TransactionAddressModel
         */
        this.country = undefined;
        /**
         * @type {number}
         * @memberof TransactionAddressModel
         */
        this.taxRegionId = undefined;
        /**
         * @type {string}
         * @memberof TransactionAddressModel
         */
        this.latitude = undefined;
        /**
         * @type {string}
         * @memberof TransactionAddressModel
         */
        this.longitude = undefined;
        /**
         * @type {JurisdictionModel[]}
         * @memberof TransactionAddressModel
         */
        this.jurisdictions = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], TransactionAddressModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("transactionId", Number, true),
        __metadata("design:type", Number)
    ], TransactionAddressModel.prototype, "transactionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("boundaryLevel", Enums.BoundaryLevelConverter, true),
        __metadata("design:type", Number)
    ], TransactionAddressModel.prototype, "boundaryLevel", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line1", String, true),
        __metadata("design:type", String)
    ], TransactionAddressModel.prototype, "line1", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line2", String, true),
        __metadata("design:type", String)
    ], TransactionAddressModel.prototype, "line2", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line3", String, true),
        __metadata("design:type", String)
    ], TransactionAddressModel.prototype, "line3", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("city", String, true),
        __metadata("design:type", String)
    ], TransactionAddressModel.prototype, "city", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], TransactionAddressModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("postalCode", String, true),
        __metadata("design:type", String)
    ], TransactionAddressModel.prototype, "postalCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], TransactionAddressModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxRegionId", Number, true),
        __metadata("design:type", Number)
    ], TransactionAddressModel.prototype, "taxRegionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("latitude", String, true),
        __metadata("design:type", String)
    ], TransactionAddressModel.prototype, "latitude", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("longitude", String, true),
        __metadata("design:type", String)
    ], TransactionAddressModel.prototype, "longitude", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictions", [JurisdictionModel_1.JurisdictionModel], true),
        __metadata("design:type", Array)
    ], TransactionAddressModel.prototype, "jurisdictions", void 0);
    TransactionAddressModel = __decorate([
        (0, json2typescript_1.JsonObject)("TransactionAddressModel")
    ], TransactionAddressModel);
    return TransactionAddressModel;
}());
