// avatax/lib/models/UomModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UomModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * The "Unit of Measurement" model captures information about a type of measurement.  Types of measurement refer to
different scales for the same dimension.  For example, measurements of type "Distance" may include units of measurement
such as meters, feet, inches, and miles.
 * @export
 * @class UomModel
 */
var UomModel = exports.UomModel = /** @class */ (function () {
    function UomModel() {
        /**
         * @type {number}
         * @memberof UomModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof UomModel
         */
        this.code = undefined;
        /**
         * @type {string}
         * @memberof UomModel
         */
        this.shortDesc = undefined;
        /**
         * @type {string}
         * @memberof UomModel
         */
        this.description = undefined;
        /**
         * @type {number}
         * @memberof UomModel
         */
        this.measurementTypeId = undefined;
        /**
         * @type {string}
         * @memberof UomModel
         */
        this.measurementTypeCode = undefined;
        /**
         * @type {string}
         * @memberof UomModel
         */
        this.siUOM = undefined;
        /**
         * @type {string}
         * @memberof UomModel
         */
        this.measurementTypeDescription = undefined;
        /**
         * @type {boolean}
         * @memberof UomModel
         */
        this.isSiUom = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], UomModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("code", String, true),
        __metadata("design:type", String)
    ], UomModel.prototype, "code", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("shortDesc", String, true),
        __metadata("design:type", String)
    ], UomModel.prototype, "shortDesc", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], UomModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("measurementTypeId", Number, true),
        __metadata("design:type", Number)
    ], UomModel.prototype, "measurementTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("measurementTypeCode", String, true),
        __metadata("design:type", String)
    ], UomModel.prototype, "measurementTypeCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("siUOM", String, true),
        __metadata("design:type", String)
    ], UomModel.prototype, "siUOM", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("measurementTypeDescription", String, true),
        __metadata("design:type", String)
    ], UomModel.prototype, "measurementTypeDescription", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSiUom", Boolean, true),
        __metadata("design:type", Boolean)
    ], UomModel.prototype, "isSiUom", void 0);
    UomModel = __decorate([
        (0, json2typescript_1.JsonObject)("UomModel")
    ], UomModel);
    return UomModel;
}());
