// avatax/lib/models/UserModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * An account user who is permitted to use AvaTax.
 * @export
 * @class UserModel
 */
var UserModel = exports.UserModel = /** @class */ (function () {
    function UserModel() {
        /**
         * @type {number}
         * @memberof UserModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof UserModel
         */
        this.accountId = undefined;
        /**
         * @type {number}
         * @memberof UserModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof UserModel
         */
        this.userName = undefined;
        /**
         * @type {string}
         * @memberof UserModel
         */
        this.firstName = undefined;
        /**
         * @type {string}
         * @memberof UserModel
         */
        this.lastName = undefined;
        /**
         * @type {string}
         * @memberof UserModel
         */
        this.email = undefined;
        /**
         * @type {string}
         * @memberof UserModel
         */
        this.postalCode = undefined;
        /**
         * @type {Enums.SecurityRoleId}
         * @memberof UserModel
         */
        this.securityRoleId = undefined;
        /**
         * @type {Enums.PasswordStatusId}
         * @memberof UserModel
         */
        this.passwordStatus = undefined;
        /**
         * @type {boolean}
         * @memberof UserModel
         */
        this.isActive = undefined;
        /**
         * @type {Date}
         * @memberof UserModel
         */
        this.createdDate = undefined;
        /**
         * @type {Date}
         * @memberof UserModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {string}
         * @memberof UserModel
         */
        this.subjectId = undefined;
        /**
         * @type {string}
         * @memberof UserModel
         */
        this.migratedDate = undefined;
        /**
         * @type {boolean}
         * @memberof UserModel
         */
        this.suppressNewUserEmail = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number),
        __metadata("design:type", Number)
    ], UserModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountId", Number),
        __metadata("design:type", Number)
    ], UserModel.prototype, "accountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], UserModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("userName", String),
        __metadata("design:type", String)
    ], UserModel.prototype, "userName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("firstName", String),
        __metadata("design:type", String)
    ], UserModel.prototype, "firstName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lastName", String),
        __metadata("design:type", String)
    ], UserModel.prototype, "lastName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("email", String),
        __metadata("design:type", String)
    ], UserModel.prototype, "email", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("postalCode", String, true),
        __metadata("design:type", String)
    ], UserModel.prototype, "postalCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("securityRoleId", Enums.SecurityRoleIdConverter),
        __metadata("design:type", Number)
    ], UserModel.prototype, "securityRoleId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("passwordStatus", Enums.PasswordStatusIdConverter, true),
        __metadata("design:type", Number)
    ], UserModel.prototype, "passwordStatus", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isActive", Boolean, true),
        __metadata("design:type", Boolean)
    ], UserModel.prototype, "isActive", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], UserModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], UserModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("subjectId", String, true),
        __metadata("design:type", String)
    ], UserModel.prototype, "subjectId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("migratedDate", String, true),
        __metadata("design:type", String)
    ], UserModel.prototype, "migratedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("suppressNewUserEmail", Boolean, true),
        __metadata("design:type", Boolean)
    ], UserModel.prototype, "suppressNewUserEmail", void 0);
    UserModel = __decorate([
        (0, json2typescript_1.JsonObject)("UserModel")
    ], UserModel);
    return UserModel;
}());
