// avatax/lib/models/VarianceResponseLine.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VarianceResponseLine = void 0;
var HsCode_1 = require("./HsCode");
var VarianceDetail_1 = require("./VarianceDetail");
var json2typescript_1 = require("json2typescript");
/**
 *
 * @export
 * @class VarianceResponseLine
 */
var VarianceResponseLine = exports.VarianceResponseLine = /** @class */ (function () {
    function VarianceResponseLine() {
        /**
         * @type {string}
         * @memberof VarianceResponseLine
         */
        this.lineNo = undefined;
        /**
         * @type {HsCode}
         * @memberof VarianceResponseLine
         */
        this.hsCodeVariance = undefined;
        /**
         * @type {string}
         * @memberof VarianceResponseLine
         */
        this.dutyRateVariance = undefined;
        /**
         * @type {number}
         * @memberof VarianceResponseLine
         */
        this.taxableVariance = undefined;
        /**
         * @type {number}
         * @memberof VarianceResponseLine
         */
        this.dutyVariance = undefined;
        /**
         * @type {number}
         * @memberof VarianceResponseLine
         */
        this.taxVariance = undefined;
        /**
         * @type {number}
         * @memberof VarianceResponseLine
         */
        this.totalTaxVariance = undefined;
        /**
         * @type {VarianceDetail[]}
         * @memberof VarianceResponseLine
         */
        this.unMappedDetails = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("lineNo", String, true),
        __metadata("design:type", String)
    ], VarianceResponseLine.prototype, "lineNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hsCodeVariance", HsCode_1.HsCode, true),
        __metadata("design:type", HsCode_1.HsCode)
    ], VarianceResponseLine.prototype, "hsCodeVariance", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dutyRateVariance", String, true),
        __metadata("design:type", String)
    ], VarianceResponseLine.prototype, "dutyRateVariance", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxableVariance", Number, true),
        __metadata("design:type", Number)
    ], VarianceResponseLine.prototype, "taxableVariance", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dutyVariance", Number, true),
        __metadata("design:type", Number)
    ], VarianceResponseLine.prototype, "dutyVariance", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxVariance", Number, true),
        __metadata("design:type", Number)
    ], VarianceResponseLine.prototype, "taxVariance", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalTaxVariance", Number, true),
        __metadata("design:type", Number)
    ], VarianceResponseLine.prototype, "totalTaxVariance", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("unMappedDetails", [VarianceDetail_1.VarianceDetail], true),
        __metadata("design:type", Array)
    ], VarianceResponseLine.prototype, "unMappedDetails", void 0);
    VarianceResponseLine = __decorate([
        (0, json2typescript_1.JsonObject)("VarianceResponseLine")
    ], VarianceResponseLine);
    return VarianceResponseLine;
}());
