// avatax/lib/models/VarianceResponseEntity.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VarianceResponseEntity = void 0;
var VarianceDetail_1 = require("./VarianceDetail");
var VarianceResponseLine_1 = require("./VarianceResponseLine");
var json2typescript_1 = require("json2typescript");
/**
 *
 * @export
 * @class VarianceResponseEntity
 */
var VarianceResponseEntity = exports.VarianceResponseEntity = /** @class */ (function () {
    function VarianceResponseEntity() {
        /**
         * @type {number}
         * @memberof VarianceResponseEntity
         */
        this.documentId = undefined;
        /**
         * @type {string}
         * @memberof VarianceResponseEntity
         */
        this.documentCode = undefined;
        /**
         * @type {number}
         * @memberof VarianceResponseEntity
         */
        this.customInvoiceId = undefined;
        /**
         * @type {number}
         * @memberof VarianceResponseEntity
         */
        this.varianceId = undefined;
        /**
         * @type {string}
         * @memberof VarianceResponseEntity
         */
        this.status = undefined;
        /**
         * @type {string}
         * @memberof VarianceResponseEntity
         */
        this.errorMessage = undefined;
        /**
         * @type {number}
         * @memberof VarianceResponseEntity
         */
        this.taxableVariance = undefined;
        /**
         * @type {number}
         * @memberof VarianceResponseEntity
         */
        this.dutyVariance = undefined;
        /**
         * @type {number}
         * @memberof VarianceResponseEntity
         */
        this.taxVariance = undefined;
        /**
         * @type {number}
         * @memberof VarianceResponseEntity
         */
        this.totalTaxVariance = undefined;
        /**
         * @type {VarianceDetail[]}
         * @memberof VarianceResponseEntity
         */
        this.unMappedDetails = undefined;
        /**
         * @type {VarianceResponseLine[]}
         * @memberof VarianceResponseEntity
         */
        this.varianceLines = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("documentId", Number, true),
        __metadata("design:type", Number)
    ], VarianceResponseEntity.prototype, "documentId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("documentCode", String, true),
        __metadata("design:type", String)
    ], VarianceResponseEntity.prototype, "documentCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customInvoiceId", Number, true),
        __metadata("design:type", Number)
    ], VarianceResponseEntity.prototype, "customInvoiceId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("varianceId", Number, true),
        __metadata("design:type", Number)
    ], VarianceResponseEntity.prototype, "varianceId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("status", String, true),
        __metadata("design:type", String)
    ], VarianceResponseEntity.prototype, "status", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("errorMessage", String, true),
        __metadata("design:type", String)
    ], VarianceResponseEntity.prototype, "errorMessage", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxableVariance", Number, true),
        __metadata("design:type", Number)
    ], VarianceResponseEntity.prototype, "taxableVariance", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dutyVariance", Number, true),
        __metadata("design:type", Number)
    ], VarianceResponseEntity.prototype, "dutyVariance", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxVariance", Number, true),
        __metadata("design:type", Number)
    ], VarianceResponseEntity.prototype, "taxVariance", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalTaxVariance", Number, true),
        __metadata("design:type", Number)
    ], VarianceResponseEntity.prototype, "totalTaxVariance", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("unMappedDetails", [VarianceDetail_1.VarianceDetail], true),
        __metadata("design:type", Array)
    ], VarianceResponseEntity.prototype, "unMappedDetails", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("varianceLines", [VarianceResponseLine_1.VarianceResponseLine], true),
        __metadata("design:type", Array)
    ], VarianceResponseEntity.prototype, "varianceLines", void 0);
    VarianceResponseEntity = __decorate([
        (0, json2typescript_1.JsonObject)("VarianceResponseEntity")
    ], VarianceResponseEntity);
    return VarianceResponseEntity;
}());
