// avatax/lib/models/WorksheetDocumentLine.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorksheetDocumentLine = void 0;
var Message_1 = require("./Message");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 *
 * @export
 * @class WorksheetDocumentLine
 */
var WorksheetDocumentLine = exports.WorksheetDocumentLine = /** @class */ (function () {
    function WorksheetDocumentLine() {
        /**
         * @type {Date}
         * @memberof WorksheetDocumentLine
         */
        this.reportingDate = undefined;
        /**
         * @type {string}
         * @memberof WorksheetDocumentLine
         */
        this.lineNo = undefined;
        /**
         * @type {number}
         * @memberof WorksheetDocumentLine
         */
        this.lineAmount = undefined;
        /**
         * @type {number}
         * @memberof WorksheetDocumentLine
         */
        this.exemptAmount = undefined;
        /**
         * @type {number}
         * @memberof WorksheetDocumentLine
         */
        this.taxableAmount = undefined;
        /**
         * @type {number}
         * @memberof WorksheetDocumentLine
         */
        this.taxAmount = undefined;
        /**
         * @type {Message[]}
         * @memberof WorksheetDocumentLine
         */
        this.messages = undefined;
        /**
         * @type {string}
         * @memberof WorksheetDocumentLine
         */
        this.resultCode = undefined;
        /**
         * @type {string}
         * @memberof WorksheetDocumentLine
         */
        this.transactionId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportingDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], WorksheetDocumentLine.prototype, "reportingDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lineNo", String, true),
        __metadata("design:type", String)
    ], WorksheetDocumentLine.prototype, "lineNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lineAmount", Number, true),
        __metadata("design:type", Number)
    ], WorksheetDocumentLine.prototype, "lineAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptAmount", Number, true),
        __metadata("design:type", Number)
    ], WorksheetDocumentLine.prototype, "exemptAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxableAmount", Number, true),
        __metadata("design:type", Number)
    ], WorksheetDocumentLine.prototype, "taxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAmount", Number, true),
        __metadata("design:type", Number)
    ], WorksheetDocumentLine.prototype, "taxAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("messages", [Message_1.Message], true),
        __metadata("design:type", Array)
    ], WorksheetDocumentLine.prototype, "messages", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("resultCode", String, true),
        __metadata("design:type", String)
    ], WorksheetDocumentLine.prototype, "resultCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("transactionId", String, true),
        __metadata("design:type", String)
    ], WorksheetDocumentLine.prototype, "transactionId", void 0);
    WorksheetDocumentLine = __decorate([
        (0, json2typescript_1.JsonObject)("WorksheetDocumentLine")
    ], WorksheetDocumentLine);
    return WorksheetDocumentLine;
}());
