// avatax/lib/enums/BatchStatus.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchStatusConverter = exports.BatchStatus = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var BatchStatus;
(function (BatchStatus) {
    BatchStatus[BatchStatus["Waiting"] = 0] = "Waiting";
    BatchStatus[BatchStatus["SystemErrors"] = 1] = "SystemErrors";
    BatchStatus[BatchStatus["Cancelled"] = 2] = "Cancelled";
    BatchStatus[BatchStatus["Completed"] = 3] = "Completed";
    BatchStatus[BatchStatus["Creating"] = 4] = "Creating";
    BatchStatus[BatchStatus["Deleted"] = 5] = "Deleted";
    BatchStatus[BatchStatus["Errors"] = 6] = "Errors";
    BatchStatus[BatchStatus["Paused"] = 7] = "Paused";
    BatchStatus[BatchStatus["Processing"] = 8] = "Processing";
    BatchStatus[BatchStatus["Cancelling"] = 9] = "Cancelling";
})(BatchStatus || (exports.BatchStatus = BatchStatus = {}));
var BatchStatusConverter = exports.BatchStatusConverter = /** @class */ (function () {
    function BatchStatusConverter() {
    }
    BatchStatusConverter.prototype.serialize = function (data) {
        return data;
    };
    BatchStatusConverter.prototype.deserialize = function (enumType) {
        return BatchStatus[enumType];
    };
    BatchStatusConverter = __decorate([
        json2typescript_1.JsonConverter
    ], BatchStatusConverter);
    return BatchStatusConverter;
}());
