// avatax/lib/enums/ChargedTo.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChargedToConverter = exports.ChargedTo = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var ChargedTo;
(function (ChargedTo) {
    ChargedTo[ChargedTo["Buyer"] = 0] = "Buyer";
    ChargedTo[ChargedTo["Seller"] = 1] = "Seller";
    ChargedTo[ChargedTo["ThirdParty"] = 2] = "ThirdParty";
})(ChargedTo || (exports.ChargedTo = ChargedTo = {}));
var ChargedToConverter = exports.ChargedToConverter = /** @class */ (function () {
    function ChargedToConverter() {
    }
    ChargedToConverter.prototype.serialize = function (data) {
        return data;
    };
    ChargedToConverter.prototype.deserialize = function (enumType) {
        return ChargedTo[enumType];
    };
    ChargedToConverter = __decorate([
        json2typescript_1.JsonConverter
    ], ChargedToConverter);
    return ChargedToConverter;
}());
