// avatax/lib/enums/DeliveryTerms.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeliveryTermsConverter = exports.DeliveryTerms = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var DeliveryTerms;
(function (DeliveryTerms) {
    DeliveryTerms[DeliveryTerms["DAP"] = 1] = "DAP";
    DeliveryTerms[DeliveryTerms["DDP"] = 2] = "DDP";
    DeliveryTerms[DeliveryTerms["FOB"] = 3] = "FOB";
    DeliveryTerms[DeliveryTerms["FCA"] = 4] = "FCA";
    DeliveryTerms[DeliveryTerms["FAS"] = 5] = "FAS";
    DeliveryTerms[DeliveryTerms["EXW"] = 6] = "EXW";
    DeliveryTerms[DeliveryTerms["DPU"] = 7] = "DPU";
    DeliveryTerms[DeliveryTerms["CPT"] = 8] = "CPT";
    DeliveryTerms[DeliveryTerms["CIP"] = 9] = "CIP";
    DeliveryTerms[DeliveryTerms["CIF"] = 10] = "CIF";
    DeliveryTerms[DeliveryTerms["CFR"] = 11] = "CFR";
})(DeliveryTerms || (exports.DeliveryTerms = DeliveryTerms = {}));
var DeliveryTermsConverter = exports.DeliveryTermsConverter = /** @class */ (function () {
    function DeliveryTermsConverter() {
    }
    DeliveryTermsConverter.prototype.serialize = function (data) {
        return data;
    };
    DeliveryTermsConverter.prototype.deserialize = function (enumType) {
        return DeliveryTerms[enumType];
    };
    DeliveryTermsConverter = __decorate([
        json2typescript_1.JsonConverter
    ], DeliveryTermsConverter);
    return DeliveryTermsConverter;
}());
