// avatax/lib/enums/DocumentStatus.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentStatusConverter = exports.DocumentStatus = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var DocumentStatus;
(function (DocumentStatus) {
    DocumentStatus[DocumentStatus["Temporary"] = 0] = "Temporary";
    DocumentStatus[DocumentStatus["Saved"] = 1] = "Saved";
    DocumentStatus[DocumentStatus["Posted"] = 2] = "Posted";
    DocumentStatus[DocumentStatus["Committed"] = 3] = "Committed";
    DocumentStatus[DocumentStatus["Cancelled"] = 4] = "Cancelled";
    DocumentStatus[DocumentStatus["Adjusted"] = 5] = "Adjusted";
    DocumentStatus[DocumentStatus["Queued"] = 6] = "Queued";
    DocumentStatus[DocumentStatus["PendingApproval"] = 7] = "PendingApproval";
    DocumentStatus[DocumentStatus["Any"] = -1] = "Any";
})(DocumentStatus || (exports.DocumentStatus = DocumentStatus = {}));
var DocumentStatusConverter = exports.DocumentStatusConverter = /** @class */ (function () {
    function DocumentStatusConverter() {
    }
    DocumentStatusConverter.prototype.serialize = function (data) {
        return data;
    };
    DocumentStatusConverter.prototype.deserialize = function (enumType) {
        return DocumentStatus[enumType];
    };
    DocumentStatusConverter = __decorate([
        json2typescript_1.JsonConverter
    ], DocumentStatusConverter);
    return DocumentStatusConverter;
}());
