// avatax/lib/enums/DocumentType.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentTypeConverter = exports.DocumentType = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var DocumentType;
(function (DocumentType) {
    DocumentType[DocumentType["SalesOrder"] = 0] = "SalesOrder";
    DocumentType[DocumentType["SalesInvoice"] = 1] = "SalesInvoice";
    DocumentType[DocumentType["PurchaseOrder"] = 2] = "PurchaseOrder";
    DocumentType[DocumentType["PurchaseInvoice"] = 3] = "PurchaseInvoice";
    DocumentType[DocumentType["ReturnOrder"] = 4] = "ReturnOrder";
    DocumentType[DocumentType["ReturnInvoice"] = 5] = "ReturnInvoice";
    DocumentType[DocumentType["InventoryTransferOrder"] = 6] = "InventoryTransferOrder";
    DocumentType[DocumentType["InventoryTransferInvoice"] = 7] = "InventoryTransferInvoice";
    DocumentType[DocumentType["ReverseChargeOrder"] = 8] = "ReverseChargeOrder";
    DocumentType[DocumentType["ReverseChargeInvoice"] = 9] = "ReverseChargeInvoice";
    DocumentType[DocumentType["CustomsInvoice"] = 10] = "CustomsInvoice";
    DocumentType[DocumentType["CustomsOrder"] = 11] = "CustomsOrder";
    DocumentType[DocumentType["Any"] = -1] = "Any";
})(DocumentType || (exports.DocumentType = DocumentType = {}));
var DocumentTypeConverter = exports.DocumentTypeConverter = /** @class */ (function () {
    function DocumentTypeConverter() {
    }
    DocumentTypeConverter.prototype.serialize = function (data) {
        return data;
    };
    DocumentTypeConverter.prototype.deserialize = function (enumType) {
        return DocumentType[enumType];
    };
    DocumentTypeConverter = __decorate([
        json2typescript_1.JsonConverter
    ], DocumentTypeConverter);
    return DocumentTypeConverter;
}());
