// avatax/lib/enums/ErrorCodeId.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ErrorCodeIdConverter = exports.ErrorCodeId = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var ErrorCodeId;
(function (ErrorCodeId) {
    ErrorCodeId[ErrorCodeId["ServerConfiguration"] = 1] = "ServerConfiguration";
    ErrorCodeId[ErrorCodeId["AccountInvalidException"] = 2] = "AccountInvalidException";
    ErrorCodeId[ErrorCodeId["CompanyInvalidException"] = 3] = "CompanyInvalidException";
    ErrorCodeId[ErrorCodeId["EntityNotFoundError"] = 4] = "EntityNotFoundError";
    ErrorCodeId[ErrorCodeId["ValueRequiredError"] = 5] = "ValueRequiredError";
    ErrorCodeId[ErrorCodeId["RangeError"] = 6] = "RangeError";
    ErrorCodeId[ErrorCodeId["RangeCompareError"] = 7] = "RangeCompareError";
    ErrorCodeId[ErrorCodeId["RangeSetError"] = 8] = "RangeSetError";
    ErrorCodeId[ErrorCodeId["TaxpayerNumberRequired"] = 9] = "TaxpayerNumberRequired";
    ErrorCodeId[ErrorCodeId["EntityLimitExceeded"] = 10] = "EntityLimitExceeded";
    ErrorCodeId[ErrorCodeId["CommonPassword"] = 11] = "CommonPassword";
    ErrorCodeId[ErrorCodeId["WeakPassword"] = 12] = "WeakPassword";
    ErrorCodeId[ErrorCodeId["StringLengthError"] = 13] = "StringLengthError";
    ErrorCodeId[ErrorCodeId["MaxStringLengthError"] = 14] = "MaxStringLengthError";
    ErrorCodeId[ErrorCodeId["EmailValidationError"] = 15] = "EmailValidationError";
    ErrorCodeId[ErrorCodeId["EmailMissingError"] = 16] = "EmailMissingError";
    ErrorCodeId[ErrorCodeId["InvalidQueryField"] = 17] = "InvalidQueryField";
    ErrorCodeId[ErrorCodeId["InvalidQueryValue"] = 18] = "InvalidQueryValue";
    ErrorCodeId[ErrorCodeId["SyntaxError"] = 19] = "SyntaxError";
    ErrorCodeId[ErrorCodeId["TooManyParametersError"] = 20] = "TooManyParametersError";
    ErrorCodeId[ErrorCodeId["UnterminatedValueError"] = 21] = "UnterminatedValueError";
    ErrorCodeId[ErrorCodeId["DeleteUserSelfError"] = 22] = "DeleteUserSelfError";
    ErrorCodeId[ErrorCodeId["OldPasswordInvalid"] = 23] = "OldPasswordInvalid";
    ErrorCodeId[ErrorCodeId["CannotChangePassword"] = 24] = "CannotChangePassword";
    ErrorCodeId[ErrorCodeId["ReadOnly"] = 25] = "ReadOnly";
    ErrorCodeId[ErrorCodeId["DateFormatError"] = 26] = "DateFormatError";
    ErrorCodeId[ErrorCodeId["NoDefaultCompany"] = 27] = "NoDefaultCompany";
    ErrorCodeId[ErrorCodeId["AccountTypeNotSupported"] = 28] = "AccountTypeNotSupported";
    ErrorCodeId[ErrorCodeId["InvalidAuthenticationType"] = 29] = "InvalidAuthenticationType";
    ErrorCodeId[ErrorCodeId["AuthenticationException"] = 30] = "AuthenticationException";
    ErrorCodeId[ErrorCodeId["AuthorizationException"] = 31] = "AuthorizationException";
    ErrorCodeId[ErrorCodeId["ValidationException"] = 32] = "ValidationException";
    ErrorCodeId[ErrorCodeId["InactiveUserError"] = 33] = "InactiveUserError";
    ErrorCodeId[ErrorCodeId["AuthenticationIncomplete"] = 34] = "AuthenticationIncomplete";
    ErrorCodeId[ErrorCodeId["BasicAuthIncorrect"] = 35] = "BasicAuthIncorrect";
    ErrorCodeId[ErrorCodeId["IdentityServerError"] = 36] = "IdentityServerError";
    ErrorCodeId[ErrorCodeId["BearerTokenInvalid"] = 37] = "BearerTokenInvalid";
    ErrorCodeId[ErrorCodeId["ModelRequiredException"] = 38] = "ModelRequiredException";
    ErrorCodeId[ErrorCodeId["AccountExpiredException"] = 39] = "AccountExpiredException";
    ErrorCodeId[ErrorCodeId["BearerTokenNotSupported"] = 41] = "BearerTokenNotSupported";
    ErrorCodeId[ErrorCodeId["InvalidSecurityRole"] = 42] = "InvalidSecurityRole";
    ErrorCodeId[ErrorCodeId["InvalidRegistrarAction"] = 43] = "InvalidRegistrarAction";
    ErrorCodeId[ErrorCodeId["RemoteServerError"] = 44] = "RemoteServerError";
    ErrorCodeId[ErrorCodeId["NoFilterCriteriaException"] = 45] = "NoFilterCriteriaException";
    ErrorCodeId[ErrorCodeId["OpenClauseException"] = 46] = "OpenClauseException";
    ErrorCodeId[ErrorCodeId["JsonFormatError"] = 47] = "JsonFormatError";
    ErrorCodeId[ErrorCodeId["InvalidDecimalValue"] = 48] = "InvalidDecimalValue";
    ErrorCodeId[ErrorCodeId["PermissionRequired"] = 49] = "PermissionRequired";
    ErrorCodeId[ErrorCodeId["UnhandledException"] = 50] = "UnhandledException";
    ErrorCodeId[ErrorCodeId["InactiveAccount"] = 51] = "InactiveAccount";
    ErrorCodeId[ErrorCodeId["LinkageNotAllowed"] = 52] = "LinkageNotAllowed";
    ErrorCodeId[ErrorCodeId["LinkageStatusUpdateNotSupported"] = 53] = "LinkageStatusUpdateNotSupported";
    ErrorCodeId[ErrorCodeId["ReportingCompanyMustHaveContactsError"] = 60] = "ReportingCompanyMustHaveContactsError";
    ErrorCodeId[ErrorCodeId["CompanyProfileNotSet"] = 61] = "CompanyProfileNotSet";
    ErrorCodeId[ErrorCodeId["CannotAssignUserToCompany"] = 62] = "CannotAssignUserToCompany";
    ErrorCodeId[ErrorCodeId["MustAssignUserToCompany"] = 63] = "MustAssignUserToCompany";
    ErrorCodeId[ErrorCodeId["InvalidTaxTypeMapping"] = 64] = "InvalidTaxTypeMapping";
    ErrorCodeId[ErrorCodeId["ModelStateInvalid"] = 70] = "ModelStateInvalid";
    ErrorCodeId[ErrorCodeId["DateRangeError"] = 80] = "DateRangeError";
    ErrorCodeId[ErrorCodeId["InvalidDateRangeError"] = 81] = "InvalidDateRangeError";
    ErrorCodeId[ErrorCodeId["RuleMustHaveTaxCode"] = 82] = "RuleMustHaveTaxCode";
    ErrorCodeId[ErrorCodeId["RuleTypeRestricted"] = 83] = "RuleTypeRestricted";
    ErrorCodeId[ErrorCodeId["AllJurisRuleLimits"] = 84] = "AllJurisRuleLimits";
    ErrorCodeId[ErrorCodeId["InvalidCompanyLocationSetting"] = 85] = "InvalidCompanyLocationSetting";
    ErrorCodeId[ErrorCodeId["InvalidAdjustmentType"] = 99] = "InvalidAdjustmentType";
    ErrorCodeId[ErrorCodeId["DeleteInformation"] = 100] = "DeleteInformation";
    ErrorCodeId[ErrorCodeId["DisableAuthenticationForSamlBasedAccounts"] = 101] = "DisableAuthenticationForSamlBasedAccounts";
    ErrorCodeId[ErrorCodeId["DisableResetPasswordForSamlBasedAccounts"] = 102] = "DisableResetPasswordForSamlBasedAccounts";
    ErrorCodeId[ErrorCodeId["OutOfRange"] = 118] = "OutOfRange";
    ErrorCodeId[ErrorCodeId["UnspecifiedTimeZone"] = 119] = "UnspecifiedTimeZone";
    ErrorCodeId[ErrorCodeId["CannotCreateDeletedObjects"] = 120] = "CannotCreateDeletedObjects";
    ErrorCodeId[ErrorCodeId["CannotModifyDeletedObjects"] = 121] = "CannotModifyDeletedObjects";
    ErrorCodeId[ErrorCodeId["ReturnNameNotFound"] = 122] = "ReturnNameNotFound";
    ErrorCodeId[ErrorCodeId["InvalidAddressTypeAndCategory"] = 123] = "InvalidAddressTypeAndCategory";
    ErrorCodeId[ErrorCodeId["DefaultCompanyLocation"] = 124] = "DefaultCompanyLocation";
    ErrorCodeId[ErrorCodeId["InvalidCountry"] = 125] = "InvalidCountry";
    ErrorCodeId[ErrorCodeId["InvalidCountryRegion"] = 126] = "InvalidCountryRegion";
    ErrorCodeId[ErrorCodeId["BrazilValidationError"] = 127] = "BrazilValidationError";
    ErrorCodeId[ErrorCodeId["BrazilExemptValidationError"] = 128] = "BrazilExemptValidationError";
    ErrorCodeId[ErrorCodeId["BrazilPisCofinsError"] = 129] = "BrazilPisCofinsError";
    ErrorCodeId[ErrorCodeId["JurisdictionNotFoundError"] = 130] = "JurisdictionNotFoundError";
    ErrorCodeId[ErrorCodeId["MedicalExciseError"] = 131] = "MedicalExciseError";
    ErrorCodeId[ErrorCodeId["RateDependsTaxabilityError"] = 132] = "RateDependsTaxabilityError";
    ErrorCodeId[ErrorCodeId["InvalidRateTypeCode"] = 134] = "InvalidRateTypeCode";
    ErrorCodeId[ErrorCodeId["RateTypeNotSupported"] = 135] = "RateTypeNotSupported";
    ErrorCodeId[ErrorCodeId["CannotUpdateNestedObjects"] = 136] = "CannotUpdateNestedObjects";
    ErrorCodeId[ErrorCodeId["UPCCodeInvalidChars"] = 137] = "UPCCodeInvalidChars";
    ErrorCodeId[ErrorCodeId["UPCCodeInvalidLength"] = 138] = "UPCCodeInvalidLength";
    ErrorCodeId[ErrorCodeId["IncorrectPathError"] = 139] = "IncorrectPathError";
    ErrorCodeId[ErrorCodeId["InvalidJurisdictionType"] = 140] = "InvalidJurisdictionType";
    ErrorCodeId[ErrorCodeId["MustConfirmResetLicenseKey"] = 141] = "MustConfirmResetLicenseKey";
    ErrorCodeId[ErrorCodeId["DuplicateCompanyCode"] = 142] = "DuplicateCompanyCode";
    ErrorCodeId[ErrorCodeId["TINFormatError"] = 143] = "TINFormatError";
    ErrorCodeId[ErrorCodeId["DuplicateNexusError"] = 144] = "DuplicateNexusError";
    ErrorCodeId[ErrorCodeId["UnknownNexusError"] = 145] = "UnknownNexusError";
    ErrorCodeId[ErrorCodeId["ParentNexusNotFound"] = 146] = "ParentNexusNotFound";
    ErrorCodeId[ErrorCodeId["InvalidTaxCodeType"] = 147] = "InvalidTaxCodeType";
    ErrorCodeId[ErrorCodeId["CannotActivateCompany"] = 148] = "CannotActivateCompany";
    ErrorCodeId[ErrorCodeId["DuplicateEntityProperty"] = 149] = "DuplicateEntityProperty";
    ErrorCodeId[ErrorCodeId["ReportingEntityError"] = 150] = "ReportingEntityError";
    ErrorCodeId[ErrorCodeId["InvalidReturnOperationError"] = 151] = "InvalidReturnOperationError";
    ErrorCodeId[ErrorCodeId["CannotDeleteCompany"] = 152] = "CannotDeleteCompany";
    ErrorCodeId[ErrorCodeId["CountryOverridesNotAvailable"] = 153] = "CountryOverridesNotAvailable";
    ErrorCodeId[ErrorCodeId["JurisdictionOverrideMismatch"] = 154] = "JurisdictionOverrideMismatch";
    ErrorCodeId[ErrorCodeId["DuplicateSystemTaxCode"] = 155] = "DuplicateSystemTaxCode";
    ErrorCodeId[ErrorCodeId["SSTOverridesNotAvailable"] = 156] = "SSTOverridesNotAvailable";
    ErrorCodeId[ErrorCodeId["NexusDateMismatch"] = 157] = "NexusDateMismatch";
    ErrorCodeId[ErrorCodeId["NexusParentDateMismatch"] = 159] = "NexusParentDateMismatch";
    ErrorCodeId[ErrorCodeId["BearerTokenParseUserIdError"] = 160] = "BearerTokenParseUserIdError";
    ErrorCodeId[ErrorCodeId["RetrieveUserError"] = 161] = "RetrieveUserError";
    ErrorCodeId[ErrorCodeId["InvalidConfigurationSetting"] = 162] = "InvalidConfigurationSetting";
    ErrorCodeId[ErrorCodeId["InvalidConfigurationValue"] = 163] = "InvalidConfigurationValue";
    ErrorCodeId[ErrorCodeId["InvalidEnumValue"] = 164] = "InvalidEnumValue";
    ErrorCodeId[ErrorCodeId["TaxCodeAssociatedTaxRule"] = 165] = "TaxCodeAssociatedTaxRule";
    ErrorCodeId[ErrorCodeId["CannotSwitchAccountId"] = 166] = "CannotSwitchAccountId";
    ErrorCodeId[ErrorCodeId["RequestIncomplete"] = 167] = "RequestIncomplete";
    ErrorCodeId[ErrorCodeId["AccountNotNew"] = 168] = "AccountNotNew";
    ErrorCodeId[ErrorCodeId["PasswordLengthInvalid"] = 169] = "PasswordLengthInvalid";
    ErrorCodeId[ErrorCodeId["InvalidPageKey"] = 170] = "InvalidPageKey";
    ErrorCodeId[ErrorCodeId["InvalidEcmsOverrideCode"] = 171] = "InvalidEcmsOverrideCode";
    ErrorCodeId[ErrorCodeId["AccountDoesNotExist"] = 172] = "AccountDoesNotExist";
    ErrorCodeId[ErrorCodeId["InvalidTaxType"] = 173] = "InvalidTaxType";
    ErrorCodeId[ErrorCodeId["IncorrectFieldValue"] = 174] = "IncorrectFieldValue";
    ErrorCodeId[ErrorCodeId["LeadingOrTrailingException"] = 175] = "LeadingOrTrailingException";
    ErrorCodeId[ErrorCodeId["NotEnoughAddressesInfo"] = 176] = "NotEnoughAddressesInfo";
    ErrorCodeId[ErrorCodeId["ReportNotInitiated"] = 177] = "ReportNotInitiated";
    ErrorCodeId[ErrorCodeId["FailedToBuildReport"] = 178] = "FailedToBuildReport";
    ErrorCodeId[ErrorCodeId["ReportNotFinished"] = 179] = "ReportNotFinished";
    ErrorCodeId[ErrorCodeId["FailedToDownloadReport"] = 181] = "FailedToDownloadReport";
    ErrorCodeId[ErrorCodeId["MalformedFilterException"] = 182] = "MalformedFilterException";
    ErrorCodeId[ErrorCodeId["ExpectedConjunctionError"] = 183] = "ExpectedConjunctionError";
    ErrorCodeId[ErrorCodeId["CriteriaNotSupportedError"] = 184] = "CriteriaNotSupportedError";
    ErrorCodeId[ErrorCodeId["CompanyAccountAndParentAccountMismatch"] = 185] = "CompanyAccountAndParentAccountMismatch";
    ErrorCodeId[ErrorCodeId["InvalidFileContentType"] = 186] = "InvalidFileContentType";
    ErrorCodeId[ErrorCodeId["RequestTooLarge"] = 187] = "RequestTooLarge";
    ErrorCodeId[ErrorCodeId["EcmsDisabled"] = 188] = "EcmsDisabled";
    ErrorCodeId[ErrorCodeId["UnknownConjunctionError"] = 189] = "UnknownConjunctionError";
    ErrorCodeId[ErrorCodeId["NoLinesDiscounted"] = 190] = "NoLinesDiscounted";
    ErrorCodeId[ErrorCodeId["FailedToDelete"] = 191] = "FailedToDelete";
    ErrorCodeId[ErrorCodeId["CircularCompanyHierarchies"] = 192] = "CircularCompanyHierarchies";
    ErrorCodeId[ErrorCodeId["DuplicateEntry"] = 193] = "DuplicateEntry";
    ErrorCodeId[ErrorCodeId["DuplicateFieldNameInOrderBy"] = 194] = "DuplicateFieldNameInOrderBy";
    ErrorCodeId[ErrorCodeId["CannotAdjustDocumentType"] = 195] = "CannotAdjustDocumentType";
    ErrorCodeId[ErrorCodeId["UserNoAccess"] = 196] = "UserNoAccess";
    ErrorCodeId[ErrorCodeId["InvalidEntry"] = 197] = "InvalidEntry";
    ErrorCodeId[ErrorCodeId["TransactionAlreadyCancelled"] = 198] = "TransactionAlreadyCancelled";
    ErrorCodeId[ErrorCodeId["QueryParameterOutOfRange"] = 199] = "QueryParameterOutOfRange";
    ErrorCodeId[ErrorCodeId["BatchSalesAuditMustBeZippedError"] = 200] = "BatchSalesAuditMustBeZippedError";
    ErrorCodeId[ErrorCodeId["BatchZipMustContainOneFileError"] = 201] = "BatchZipMustContainOneFileError";
    ErrorCodeId[ErrorCodeId["BatchInvalidFileTypeError"] = 202] = "BatchInvalidFileTypeError";
    ErrorCodeId[ErrorCodeId["BatchCannotSaveBatchFile"] = 203] = "BatchCannotSaveBatchFile";
    ErrorCodeId[ErrorCodeId["BatchCannotGetBatchFile"] = 204] = "BatchCannotGetBatchFile";
    ErrorCodeId[ErrorCodeId["BatchCannotDeleteBatchFile"] = 205] = "BatchCannotDeleteBatchFile";
    ErrorCodeId[ErrorCodeId["BatchMustContainOneFile"] = 206] = "BatchMustContainOneFile";
    ErrorCodeId[ErrorCodeId["MissingBatchFileContent"] = 207] = "MissingBatchFileContent";
    ErrorCodeId[ErrorCodeId["BatchCannotBeDeletedWhileProcessing"] = 208] = "BatchCannotBeDeletedWhileProcessing";
    ErrorCodeId[ErrorCodeId["InternalServerError"] = 209] = "InternalServerError";
    ErrorCodeId[ErrorCodeId["PointOfSaleFileSize"] = 250] = "PointOfSaleFileSize";
    ErrorCodeId[ErrorCodeId["PointOfSaleSetup"] = 251] = "PointOfSaleSetup";
    ErrorCodeId[ErrorCodeId["InvalidInputDate"] = 252] = "InvalidInputDate";
    ErrorCodeId[ErrorCodeId["GetTaxError"] = 300] = "GetTaxError";
    ErrorCodeId[ErrorCodeId["AddressConflictException"] = 301] = "AddressConflictException";
    ErrorCodeId[ErrorCodeId["DocumentCodeConflict"] = 303] = "DocumentCodeConflict";
    ErrorCodeId[ErrorCodeId["MissingAddress"] = 304] = "MissingAddress";
    ErrorCodeId[ErrorCodeId["InvalidParameterValue"] = 306] = "InvalidParameterValue";
    ErrorCodeId[ErrorCodeId["FetchLimit"] = 308] = "FetchLimit";
    ErrorCodeId[ErrorCodeId["InvalidAddress"] = 309] = "InvalidAddress";
    ErrorCodeId[ErrorCodeId["AddressLocationNotFound"] = 310] = "AddressLocationNotFound";
    ErrorCodeId[ErrorCodeId["MissingLine"] = 311] = "MissingLine";
    ErrorCodeId[ErrorCodeId["InvalidAddressTextCase"] = 312] = "InvalidAddressTextCase";
    ErrorCodeId[ErrorCodeId["DocumentNotCommitted"] = 313] = "DocumentNotCommitted";
    ErrorCodeId[ErrorCodeId["InvalidDocumentTypesToFetch"] = 315] = "InvalidDocumentTypesToFetch";
    ErrorCodeId[ErrorCodeId["TimeoutRequested"] = 316] = "TimeoutRequested";
    ErrorCodeId[ErrorCodeId["InvalidPostalCode"] = 317] = "InvalidPostalCode";
    ErrorCodeId[ErrorCodeId["InvalidSubscriptionDescription"] = 318] = "InvalidSubscriptionDescription";
    ErrorCodeId[ErrorCodeId["InvalidSubscriptionTypeId"] = 319] = "InvalidSubscriptionTypeId";
    ErrorCodeId[ErrorCodeId["CannotChangeFilingStatus"] = 401] = "CannotChangeFilingStatus";
    ErrorCodeId[ErrorCodeId["FEINFormatError"] = 402] = "FEINFormatError";
    ErrorCodeId[ErrorCodeId["ServerUnreachable"] = 500] = "ServerUnreachable";
    ErrorCodeId[ErrorCodeId["SubscriptionRequired"] = 600] = "SubscriptionRequired";
    ErrorCodeId[ErrorCodeId["AccountExists"] = 601] = "AccountExists";
    ErrorCodeId[ErrorCodeId["InvitationOnly"] = 602] = "InvitationOnly";
    ErrorCodeId[ErrorCodeId["AccountNotWhitelisted"] = 603] = "AccountNotWhitelisted";
    ErrorCodeId[ErrorCodeId["FreeTrialNotAvailable"] = 606] = "FreeTrialNotAvailable";
    ErrorCodeId[ErrorCodeId["AccountExistsDifferentEmail"] = 607] = "AccountExistsDifferentEmail";
    ErrorCodeId[ErrorCodeId["AvalaraIdentityApiError"] = 608] = "AvalaraIdentityApiError";
    ErrorCodeId[ErrorCodeId["InvalidIPAddress"] = 609] = "InvalidIPAddress";
    ErrorCodeId[ErrorCodeId["OfferCodeAlreadyApplied"] = 610] = "OfferCodeAlreadyApplied";
    ErrorCodeId[ErrorCodeId["AccountAlreadyExists"] = 611] = "AccountAlreadyExists";
    ErrorCodeId[ErrorCodeId["LicenseKeyNameAlreadyExistsForAccount"] = 612] = "LicenseKeyNameAlreadyExistsForAccount";
    ErrorCodeId[ErrorCodeId["UserAlreadyExist"] = 613] = "UserAlreadyExist";
    ErrorCodeId[ErrorCodeId["UserNotFound"] = 614] = "UserNotFound";
    ErrorCodeId[ErrorCodeId["UserManagementException"] = 615] = "UserManagementException";
    ErrorCodeId[ErrorCodeId["RefundTypeAndPercentageMismatch"] = 701] = "RefundTypeAndPercentageMismatch";
    ErrorCodeId[ErrorCodeId["InvalidDocumentTypeForRefund"] = 702] = "InvalidDocumentTypeForRefund";
    ErrorCodeId[ErrorCodeId["RefundTypeAndLineMismatch"] = 703] = "RefundTypeAndLineMismatch";
    ErrorCodeId[ErrorCodeId["RefundLinesRequired"] = 704] = "RefundLinesRequired";
    ErrorCodeId[ErrorCodeId["InvalidRefundType"] = 705] = "InvalidRefundType";
    ErrorCodeId[ErrorCodeId["RefundPercentageForTaxOnly"] = 706] = "RefundPercentageForTaxOnly";
    ErrorCodeId[ErrorCodeId["LineNoOutOfRange"] = 707] = "LineNoOutOfRange";
    ErrorCodeId[ErrorCodeId["RefundPercentageOutOfRange"] = 708] = "RefundPercentageOutOfRange";
    ErrorCodeId[ErrorCodeId["RefundPercentageMissing"] = 709] = "RefundPercentageMissing";
    ErrorCodeId[ErrorCodeId["MustUseCreateTransaction"] = 800] = "MustUseCreateTransaction";
    ErrorCodeId[ErrorCodeId["MustAcceptTermsAndConditions"] = 801] = "MustAcceptTermsAndConditions";
    ErrorCodeId[ErrorCodeId["FilingCalendarCannotBeDeleted"] = 900] = "FilingCalendarCannotBeDeleted";
    ErrorCodeId[ErrorCodeId["InvalidEffectiveDate"] = 901] = "InvalidEffectiveDate";
    ErrorCodeId[ErrorCodeId["NonOutletForm"] = 902] = "NonOutletForm";
    ErrorCodeId[ErrorCodeId["OverlappingFilingCalendar"] = 903] = "OverlappingFilingCalendar";
    ErrorCodeId[ErrorCodeId["FilingCalendarCannotBeEdited"] = 904] = "FilingCalendarCannotBeEdited";
    ErrorCodeId[ErrorCodeId["CannotModifyLockedTransaction"] = 1100] = "CannotModifyLockedTransaction";
    ErrorCodeId[ErrorCodeId["LineAlreadyExists"] = 1101] = "LineAlreadyExists";
    ErrorCodeId[ErrorCodeId["LineDoesNotExist"] = 1102] = "LineDoesNotExist";
    ErrorCodeId[ErrorCodeId["LinesNotSpecified"] = 1103] = "LinesNotSpecified";
    ErrorCodeId[ErrorCodeId["LineDetailsDoesNotExist"] = 1104] = "LineDetailsDoesNotExist";
    ErrorCodeId[ErrorCodeId["CannotCreateTransactionWithDeletedDataSource"] = 1105] = "CannotCreateTransactionWithDeletedDataSource";
    ErrorCodeId[ErrorCodeId["ShipToRegionRequiredWithDataSource"] = 1106] = "ShipToRegionRequiredWithDataSource";
    ErrorCodeId[ErrorCodeId["InvalidBusinessType"] = 1200] = "InvalidBusinessType";
    ErrorCodeId[ErrorCodeId["CannotModifyExemptCert"] = 1201] = "CannotModifyExemptCert";
    ErrorCodeId[ErrorCodeId["CertificatesError"] = 1203] = "CertificatesError";
    ErrorCodeId[ErrorCodeId["MissingRequiredFields"] = 1204] = "MissingRequiredFields";
    ErrorCodeId[ErrorCodeId["CertificatesNotSetup"] = 1205] = "CertificatesNotSetup";
    ErrorCodeId[ErrorCodeId["ConflictingExposureZone"] = 1208] = "ConflictingExposureZone";
    ErrorCodeId[ErrorCodeId["MissingFieldToCreateExposureZone"] = 1209] = "MissingFieldToCreateExposureZone";
    ErrorCodeId[ErrorCodeId["MissingExemptReason"] = 1210] = "MissingExemptReason";
    ErrorCodeId[ErrorCodeId["InvalidExemptReason"] = 1211] = "InvalidExemptReason";
    ErrorCodeId[ErrorCodeId["InvalidExemptionOperation"] = 1212] = "InvalidExemptionOperation";
    ErrorCodeId[ErrorCodeId["ConflictingFields"] = 1213] = "ConflictingFields";
    ErrorCodeId[ErrorCodeId["InvalidPdfOrImageFile"] = 1214] = "InvalidPdfOrImageFile";
    ErrorCodeId[ErrorCodeId["InvalidCoverLetterTitle"] = 1215] = "InvalidCoverLetterTitle";
    ErrorCodeId[ErrorCodeId["AccountNotProvisioned"] = 1216] = "AccountNotProvisioned";
    ErrorCodeId[ErrorCodeId["InvalidRequestContentType"] = 1217] = "InvalidRequestContentType";
    ErrorCodeId[ErrorCodeId["ExemptionPaginationLimits"] = 1218] = "ExemptionPaginationLimits";
    ErrorCodeId[ErrorCodeId["ExemptionSortLimits"] = 1219] = "ExemptionSortLimits";
    ErrorCodeId[ErrorCodeId["CustomerCantBeBothShipToAndBillTo"] = 1220] = "CustomerCantBeBothShipToAndBillTo";
    ErrorCodeId[ErrorCodeId["BillToCustomerExpected"] = 1221] = "BillToCustomerExpected";
    ErrorCodeId[ErrorCodeId["ShipToCustomerExpected"] = 1222] = "ShipToCustomerExpected";
    ErrorCodeId[ErrorCodeId["EcmsSstCertsRequired"] = 1223] = "EcmsSstCertsRequired";
    ErrorCodeId[ErrorCodeId["TransactionNotCancelled"] = 1300] = "TransactionNotCancelled";
    ErrorCodeId[ErrorCodeId["TooManyTransactions"] = 1301] = "TooManyTransactions";
    ErrorCodeId[ErrorCodeId["OnlyTaxDateOverrideIsAllowed"] = 1302] = "OnlyTaxDateOverrideIsAllowed";
    ErrorCodeId[ErrorCodeId["TransactionAlreadyExists"] = 1303] = "TransactionAlreadyExists";
    ErrorCodeId[ErrorCodeId["MultiDocumentTransactionAlreadyExists"] = 1304] = "MultiDocumentTransactionAlreadyExists";
    ErrorCodeId[ErrorCodeId["DateMismatch"] = 1305] = "DateMismatch";
    ErrorCodeId[ErrorCodeId["InvalidDocumentStatusForVerify"] = 1306] = "InvalidDocumentStatusForVerify";
    ErrorCodeId[ErrorCodeId["TotalAmountMismatch"] = 1307] = "TotalAmountMismatch";
    ErrorCodeId[ErrorCodeId["TotalTaxMismatch"] = 1308] = "TotalTaxMismatch";
    ErrorCodeId[ErrorCodeId["InvalidDocumentType"] = 1310] = "InvalidDocumentType";
    ErrorCodeId[ErrorCodeId["MultiDocumentPartiallyLocked"] = 1312] = "MultiDocumentPartiallyLocked";
    ErrorCodeId[ErrorCodeId["TransactionIsCommitted"] = 1313] = "TransactionIsCommitted";
    ErrorCodeId[ErrorCodeId["InvalidDocumentStatus"] = 1314] = "InvalidDocumentStatus";
    ErrorCodeId[ErrorCodeId["CommsConfigClientIdMissing"] = 1400] = "CommsConfigClientIdMissing";
    ErrorCodeId[ErrorCodeId["CommsConfigClientIdBadValue"] = 1401] = "CommsConfigClientIdBadValue";
    ErrorCodeId[ErrorCodeId["AccountInNewStatusException"] = 1404] = "AccountInNewStatusException";
    ErrorCodeId[ErrorCodeId["WorksheetException"] = 1405] = "WorksheetException";
    ErrorCodeId[ErrorCodeId["InvalidAccountOverride"] = 1406] = "InvalidAccountOverride";
    ErrorCodeId[ErrorCodeId["AccountOverrideNotAuthorized"] = 1407] = "AccountOverrideNotAuthorized";
    ErrorCodeId[ErrorCodeId["FieldNotQueryableError"] = 1408] = "FieldNotQueryableError";
    ErrorCodeId[ErrorCodeId["UsernameRequired"] = 1409] = "UsernameRequired";
    ErrorCodeId[ErrorCodeId["InvalidAuditMessage"] = 1410] = "InvalidAuditMessage";
    ErrorCodeId[ErrorCodeId["FieldNotOrderableError"] = 1411] = "FieldNotOrderableError";
    ErrorCodeId[ErrorCodeId["CannotDeleteParentBeforeChildNexus"] = 1500] = "CannotDeleteParentBeforeChildNexus";
    ErrorCodeId[ErrorCodeId["NexusChildDateMismatch"] = 1501] = "NexusChildDateMismatch";
    ErrorCodeId[ErrorCodeId["RemoteValidationError"] = 1502] = "RemoteValidationError";
    ErrorCodeId[ErrorCodeId["CannotModifySstNexus"] = 1503] = "CannotModifySstNexus";
    ErrorCodeId[ErrorCodeId["InvalidLocalNexusTypeId"] = 1504] = "InvalidLocalNexusTypeId";
    ErrorCodeId[ErrorCodeId["AdvancedRuleRequestRuleError"] = 1602] = "AdvancedRuleRequestRuleError";
    ErrorCodeId[ErrorCodeId["AdvancedRuleResponseRuleError"] = 1603] = "AdvancedRuleResponseRuleError";
    ErrorCodeId[ErrorCodeId["AdvancedRuleError"] = 1605] = "AdvancedRuleError";
    ErrorCodeId[ErrorCodeId["TaxRuleRequiresNexus"] = 1701] = "TaxRuleRequiresNexus";
    ErrorCodeId[ErrorCodeId["UPCCodeNotUnique"] = 1702] = "UPCCodeNotUnique";
    ErrorCodeId[ErrorCodeId["TaxCodeAssociatedWithItemCodeNotFound"] = 1704] = "TaxCodeAssociatedWithItemCodeNotFound";
    ErrorCodeId[ErrorCodeId["DuplicateSystemForItem"] = 1705] = "DuplicateSystemForItem";
    ErrorCodeId[ErrorCodeId["CannotDismissGlobalNotification"] = 1706] = "CannotDismissGlobalNotification";
    ErrorCodeId[ErrorCodeId["CannotUpdateAccountTypeId"] = 1713] = "CannotUpdateAccountTypeId";
    ErrorCodeId[ErrorCodeId["TaxpayerNumberIsRequired"] = 1714] = "TaxpayerNumberIsRequired";
    ErrorCodeId[ErrorCodeId["RequestLimitExceeded"] = 1715] = "RequestLimitExceeded";
    ErrorCodeId[ErrorCodeId["ConcurrentRequestLimitExceeded"] = 1716] = "ConcurrentRequestLimitExceeded";
    ErrorCodeId[ErrorCodeId["InvalidDocumentTypeForInspect"] = 1717] = "InvalidDocumentTypeForInspect";
    ErrorCodeId[ErrorCodeId["ServiceNotReady"] = 1718] = "ServiceNotReady";
    ErrorCodeId[ErrorCodeId["UpdateLocationRemittanceMismatchTypeAndCategory"] = 1719] = "UpdateLocationRemittanceMismatchTypeAndCategory";
    ErrorCodeId[ErrorCodeId["UpdateLocationRemittanceCheckExistingEffectiveDateError"] = 1720] = "UpdateLocationRemittanceCheckExistingEffectiveDateError";
    ErrorCodeId[ErrorCodeId["UpdateLocationRemittanceCheckExistingEndDateError"] = 1721] = "UpdateLocationRemittanceCheckExistingEndDateError";
    ErrorCodeId[ErrorCodeId["ErrorCountLimitExceededError"] = 1722] = "ErrorCountLimitExceededError";
    ErrorCodeId[ErrorCodeId["RateLimitExceededError"] = 1723] = "RateLimitExceededError";
    ErrorCodeId[ErrorCodeId["TaxCodeAndTaxCodeIdMissing"] = 1724] = "TaxCodeAndTaxCodeIdMissing";
    ErrorCodeId[ErrorCodeId["NexusAlreadyExists"] = 1725] = "NexusAlreadyExists";
    ErrorCodeId[ErrorCodeId["InvalidAddressTypeAndMarketPlaceOusideUsaFlag"] = 1726] = "InvalidAddressTypeAndMarketPlaceOusideUsaFlag";
    ErrorCodeId[ErrorCodeId["InvalidSettingSet"] = 1727] = "InvalidSettingSet";
    ErrorCodeId[ErrorCodeId["InvalidSettingName"] = 1728] = "InvalidSettingName";
    ErrorCodeId[ErrorCodeId["InvalidSettingValue"] = 1729] = "InvalidSettingValue";
    ErrorCodeId[ErrorCodeId["TooManyUserDefinedFields"] = 1730] = "TooManyUserDefinedFields";
    ErrorCodeId[ErrorCodeId["DuplicateUserDefinedFieldsFound"] = 1731] = "DuplicateUserDefinedFieldsFound";
    ErrorCodeId[ErrorCodeId["InvalidNameForUserDefinedField"] = 1732] = "InvalidNameForUserDefinedField";
    ErrorCodeId[ErrorCodeId["InvalidRestrictionType"] = 1733] = "InvalidRestrictionType";
    ErrorCodeId[ErrorCodeId["InvalidParameter"] = 1734] = "InvalidParameter";
    ErrorCodeId[ErrorCodeId["InvalidSystemCode"] = 1735] = "InvalidSystemCode";
    ErrorCodeId[ErrorCodeId["NoItemsForClassification"] = 1736] = "NoItemsForClassification";
    ErrorCodeId[ErrorCodeId["InvalidFileName"] = 1737] = "InvalidFileName";
    ErrorCodeId[ErrorCodeId["NoClassificationForSameHsCode"] = 1738] = "NoClassificationForSameHsCode";
    ErrorCodeId[ErrorCodeId["UnsupportedFileFormat"] = 1800] = "UnsupportedFileFormat";
    ErrorCodeId[ErrorCodeId["UnsupportedOutputFileType"] = 1801] = "UnsupportedOutputFileType";
    ErrorCodeId[ErrorCodeId["TaxProfileNotProvided"] = 1900] = "TaxProfileNotProvided";
    ErrorCodeId[ErrorCodeId["InvalidTaxProfile"] = 1901] = "InvalidTaxProfile";
    ErrorCodeId[ErrorCodeId["CompanyTaxProfileEntryRequired"] = 1902] = "CompanyTaxProfileEntryRequired";
    ErrorCodeId[ErrorCodeId["ErrorReadingTaxProfileEntry"] = 1903] = "ErrorReadingTaxProfileEntry";
    ErrorCodeId[ErrorCodeId["TraceDataNotAvailable"] = 2000] = "TraceDataNotAvailable";
    ErrorCodeId[ErrorCodeId["InvalidParameterUnitMeasurementType"] = 2100] = "InvalidParameterUnitMeasurementType";
    ErrorCodeId[ErrorCodeId["ParameterUnitRequired"] = 2101] = "ParameterUnitRequired";
    ErrorCodeId[ErrorCodeId["InvalidParameterValueDataType"] = 2102] = "InvalidParameterValueDataType";
    ErrorCodeId[ErrorCodeId["InvalidParameterAttributeType"] = 2103] = "InvalidParameterAttributeType";
    ErrorCodeId[ErrorCodeId["SubscriptionRequiredForParameter"] = 2104] = "SubscriptionRequiredForParameter";
    ErrorCodeId[ErrorCodeId["InvalidAccountType"] = 2105] = "InvalidAccountType";
    ErrorCodeId[ErrorCodeId["InvalidFirmSubscriptions"] = 2106] = "InvalidFirmSubscriptions";
    ErrorCodeId[ErrorCodeId["GenericTaxCodeForItem"] = 2200] = "GenericTaxCodeForItem";
    ErrorCodeId[ErrorCodeId["CannotCertifyCompany"] = 2201] = "CannotCertifyCompany";
    ErrorCodeId[ErrorCodeId["NoVoidedDocuments"] = 2202] = "NoVoidedDocuments";
    ErrorCodeId[ErrorCodeId["InadequateCommittedDocuments"] = 2203] = "InadequateCommittedDocuments";
    ErrorCodeId[ErrorCodeId["DocumentCodeIsGuid"] = 2204] = "DocumentCodeIsGuid";
    ErrorCodeId[ErrorCodeId["CustomerVendorCodeIsGuid"] = 2205] = "CustomerVendorCodeIsGuid";
    ErrorCodeId[ErrorCodeId["InadequateDocumentLineCount"] = 2206] = "InadequateDocumentLineCount";
    ErrorCodeId[ErrorCodeId["SameDocumentDescription"] = 2207] = "SameDocumentDescription";
    ErrorCodeId[ErrorCodeId["NoExemptionNoOrCustomerUsageType"] = 2208] = "NoExemptionNoOrCustomerUsageType";
    ErrorCodeId[ErrorCodeId["InadequateUniqueAddresses"] = 2209] = "InadequateUniqueAddresses";
    ErrorCodeId[ErrorCodeId["ItemCodesAreAllSame"] = 2210] = "ItemCodesAreAllSame";
    ErrorCodeId[ErrorCodeId["TaxCodesAreAllSame"] = 2211] = "TaxCodesAreAllSame";
    ErrorCodeId[ErrorCodeId["LocationCodeNotUsed"] = 2212] = "LocationCodeNotUsed";
    ErrorCodeId[ErrorCodeId["RepeatedLinesInDocument"] = 2213] = "RepeatedLinesInDocument";
    ErrorCodeId[ErrorCodeId["TaxDateOverrideAndNegativeLineAmount"] = 2214] = "TaxDateOverrideAndNegativeLineAmount";
    ErrorCodeId[ErrorCodeId["AllUSDCurrencyCodes"] = 2215] = "AllUSDCurrencyCodes";
    ErrorCodeId[ErrorCodeId["NoVATBuyerId"] = 2216] = "NoVATBuyerId";
    ErrorCodeId[ErrorCodeId["AllUSCountryCodes"] = 2217] = "AllUSCountryCodes";
    ErrorCodeId[ErrorCodeId["NoDocumentsToTest"] = 2218] = "NoDocumentsToTest";
    ErrorCodeId[ErrorCodeId["NoShippingCharge"] = 2219] = "NoShippingCharge";
    ErrorCodeId[ErrorCodeId["FailedToUpdateCompanyLocation"] = 2314] = "FailedToUpdateCompanyLocation";
    ErrorCodeId[ErrorCodeId["CompanyLocationDateRangeOverlap"] = 2315] = "CompanyLocationDateRangeOverlap";
    ErrorCodeId[ErrorCodeId["FieldLengthError"] = 2400] = "FieldLengthError";
    ErrorCodeId[ErrorCodeId["InputContainsBlacklistedCharacters"] = 2401] = "InputContainsBlacklistedCharacters";
    ErrorCodeId[ErrorCodeId["CannotCreateNestedObjects"] = 2402] = "CannotCreateNestedObjects";
    ErrorCodeId[ErrorCodeId["UserSubjectIdNotUpdated"] = 2500] = "UserSubjectIdNotUpdated";
    ErrorCodeId[ErrorCodeId["BatchTransactionTypeError"] = 2501] = "BatchTransactionTypeError";
    ErrorCodeId[ErrorCodeId["BatchTransactionLineLimitExceeded"] = 2502] = "BatchTransactionLineLimitExceeded";
    ErrorCodeId[ErrorCodeId["BatchCompanyIdAndCompanyCodeMismatch"] = 2503] = "BatchCompanyIdAndCompanyCodeMismatch";
    ErrorCodeId[ErrorCodeId["BatchCannotBeCancelledStatusError"] = 2504] = "BatchCannotBeCancelledStatusError";
    ErrorCodeId[ErrorCodeId["BatchCannotBeCancelledFormatError"] = 2505] = "BatchCannotBeCancelledFormatError";
    ErrorCodeId[ErrorCodeId["BatchTypeNotSupported"] = 2506] = "BatchTypeNotSupported";
    ErrorCodeId[ErrorCodeId["InvalidParameterDataType"] = 2600] = "InvalidParameterDataType";
    ErrorCodeId[ErrorCodeId["TagDoesNotExist"] = 2620] = "TagDoesNotExist";
    ErrorCodeId[ErrorCodeId["ObjectDeleted"] = 2660] = "ObjectDeleted";
    ErrorCodeId[ErrorCodeId["AssociatedObjectsDeleted"] = 2661] = "AssociatedObjectsDeleted";
    ErrorCodeId[ErrorCodeId["CannotDownloadReport"] = 2700] = "CannotDownloadReport";
    ErrorCodeId[ErrorCodeId["InvalidUnitOfBasis"] = 2800] = "InvalidUnitOfBasis";
    ErrorCodeId[ErrorCodeId["NotApplicableUnitOfBasis"] = 2801] = "NotApplicableUnitOfBasis";
    ErrorCodeId[ErrorCodeId["InvalidRateTypeTaxTypeMapping"] = 2802] = "InvalidRateTypeTaxTypeMapping";
    ErrorCodeId[ErrorCodeId["InvalidTaxTypeGroup"] = 2803] = "InvalidTaxTypeGroup";
    ErrorCodeId[ErrorCodeId["InvalidTaxSubType"] = 2804] = "InvalidTaxSubType";
    ErrorCodeId[ErrorCodeId["InvalidProductTypeId"] = 2805] = "InvalidProductTypeId";
    ErrorCodeId[ErrorCodeId["InvalidTaxRuleType"] = 2806] = "InvalidTaxRuleType";
    ErrorCodeId[ErrorCodeId["InvalidHsCode"] = 2807] = "InvalidHsCode";
    ErrorCodeId[ErrorCodeId["NotApplicableTaxType"] = 2808] = "NotApplicableTaxType";
    ErrorCodeId[ErrorCodeId["InvalidTaxTypeCode"] = 2809] = "InvalidTaxTypeCode";
    ErrorCodeId[ErrorCodeId["ContentAccessDenied"] = 2810] = "ContentAccessDenied";
    ErrorCodeId[ErrorCodeId["ContentNotFound"] = 2811] = "ContentNotFound";
    ErrorCodeId[ErrorCodeId["RegistrationNumberNotFound"] = 2812] = "RegistrationNumberNotFound";
})(ErrorCodeId || (exports.ErrorCodeId = ErrorCodeId = {}));
var ErrorCodeIdConverter = exports.ErrorCodeIdConverter = /** @class */ (function () {
    function ErrorCodeIdConverter() {
    }
    ErrorCodeIdConverter.prototype.serialize = function (data) {
        return data;
    };
    ErrorCodeIdConverter.prototype.deserialize = function (enumType) {
        return ErrorCodeId[enumType];
    };
    ErrorCodeIdConverter = __decorate([
        json2typescript_1.JsonConverter
    ], ErrorCodeIdConverter);
    return ErrorCodeIdConverter;
}());
