// avatax/lib/enums/FilingStatusId.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilingStatusIdConverter = exports.FilingStatusId = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var FilingStatusId;
(function (FilingStatusId) {
    FilingStatusId[FilingStatusId["PendingApproval"] = 0] = "PendingApproval";
    FilingStatusId[FilingStatusId["Dirty"] = 1] = "Dirty";
    FilingStatusId[FilingStatusId["ApprovedToFile"] = 2] = "ApprovedToFile";
    FilingStatusId[FilingStatusId["PendingFiling"] = 3] = "PendingFiling";
    FilingStatusId[FilingStatusId["PendingFilingOnBehalf"] = 4] = "PendingFilingOnBehalf";
    FilingStatusId[FilingStatusId["Filed"] = 5] = "Filed";
    FilingStatusId[FilingStatusId["FiledOnBehalf"] = 6] = "FiledOnBehalf";
    FilingStatusId[FilingStatusId["ReturnAccepted"] = 7] = "ReturnAccepted";
    FilingStatusId[FilingStatusId["ReturnAcceptedOnBehalf"] = 8] = "ReturnAcceptedOnBehalf";
    FilingStatusId[FilingStatusId["PaymentRemitted"] = 9] = "PaymentRemitted";
    FilingStatusId[FilingStatusId["Voided"] = 10] = "Voided";
    FilingStatusId[FilingStatusId["PendingReturn"] = 11] = "PendingReturn";
    FilingStatusId[FilingStatusId["PendingReturnOnBehalf"] = 12] = "PendingReturnOnBehalf";
    FilingStatusId[FilingStatusId["DoNotFile"] = 13] = "DoNotFile";
    FilingStatusId[FilingStatusId["ReturnRejected"] = 14] = "ReturnRejected";
    FilingStatusId[FilingStatusId["ReturnRejectedOnBehalf"] = 15] = "ReturnRejectedOnBehalf";
    FilingStatusId[FilingStatusId["ApprovedToFileOnBehalf"] = 16] = "ApprovedToFileOnBehalf";
    FilingStatusId[FilingStatusId["FiledByCustomer"] = 17] = "FiledByCustomer";
})(FilingStatusId || (exports.FilingStatusId = FilingStatusId = {}));
var FilingStatusIdConverter = exports.FilingStatusIdConverter = /** @class */ (function () {
    function FilingStatusIdConverter() {
    }
    FilingStatusIdConverter.prototype.serialize = function (data) {
        return data;
    };
    FilingStatusIdConverter.prototype.deserialize = function (enumType) {
        return FilingStatusId[enumType];
    };
    FilingStatusIdConverter = __decorate([
        json2typescript_1.JsonConverter
    ], FilingStatusIdConverter);
    return FilingStatusIdConverter;
}());
