// avatax/lib/enums/RateType.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RateTypeConverter = exports.RateType = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var RateType;
(function (RateType) {
    RateType[RateType["ReducedA"] = 65] = "ReducedA";
    RateType[RateType["ReducedB"] = 66] = "ReducedB";
    RateType[RateType["Food"] = 70] = "Food";
    RateType[RateType["General"] = 71] = "General";
    RateType[RateType["IncreasedStandard"] = 73] = "IncreasedStandard";
    RateType[RateType["LinenRental"] = 76] = "LinenRental";
    RateType[RateType["Medical"] = 77] = "Medical";
    RateType[RateType["Construction"] = 78] = "Construction";
    RateType[RateType["Parking"] = 80] = "Parking";
    RateType[RateType["SuperReduced"] = 81] = "SuperReduced";
    RateType[RateType["ReducedR"] = 82] = "ReducedR";
    RateType[RateType["Standard"] = 83] = "Standard";
    RateType[RateType["Leasing"] = 84] = "Leasing";
    RateType[RateType["LeasingLow"] = 85] = "LeasingLow";
    RateType[RateType["Services"] = 88] = "Services";
    RateType[RateType["Zero"] = 90] = "Zero";
})(RateType || (exports.RateType = RateType = {}));
var RateTypeConverter = exports.RateTypeConverter = /** @class */ (function () {
    function RateTypeConverter() {
    }
    RateTypeConverter.prototype.serialize = function (data) {
        return data;
    };
    RateTypeConverter.prototype.deserialize = function (enumType) {
        return RateType[enumType];
    };
    RateTypeConverter = __decorate([
        json2typescript_1.JsonConverter
    ], RateTypeConverter);
    return RateTypeConverter;
}());
