// avatax/lib/enums/ResolutionQuality.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResolutionQualityConverter = exports.ResolutionQuality = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var ResolutionQuality;
(function (ResolutionQuality) {
    ResolutionQuality[ResolutionQuality["NotCoded"] = 0] = "NotCoded";
    ResolutionQuality[ResolutionQuality["External"] = 1] = "External";
    ResolutionQuality[ResolutionQuality["CountryCentroid"] = 2] = "CountryCentroid";
    ResolutionQuality[ResolutionQuality["RegionCentroid"] = 3] = "RegionCentroid";
    ResolutionQuality[ResolutionQuality["PartialCentroid"] = 4] = "PartialCentroid";
    ResolutionQuality[ResolutionQuality["PostalCentroidGood"] = 5] = "PostalCentroidGood";
    ResolutionQuality[ResolutionQuality["PostalCentroidBetter"] = 6] = "PostalCentroidBetter";
    ResolutionQuality[ResolutionQuality["PostalCentroidBest"] = 7] = "PostalCentroidBest";
    ResolutionQuality[ResolutionQuality["Intersection"] = 8] = "Intersection";
    ResolutionQuality[ResolutionQuality["Interpolated"] = 9] = "Interpolated";
    ResolutionQuality[ResolutionQuality["Rooftop"] = 10] = "Rooftop";
    ResolutionQuality[ResolutionQuality["Constant"] = 11] = "Constant";
})(ResolutionQuality || (exports.ResolutionQuality = ResolutionQuality = {}));
var ResolutionQualityConverter = exports.ResolutionQualityConverter = /** @class */ (function () {
    function ResolutionQualityConverter() {
    }
    ResolutionQualityConverter.prototype.serialize = function (data) {
        return data;
    };
    ResolutionQualityConverter.prototype.deserialize = function (enumType) {
        return ResolutionQuality[enumType];
    };
    ResolutionQualityConverter = __decorate([
        json2typescript_1.JsonConverter
    ], ResolutionQualityConverter);
    return ResolutionQualityConverter;
}());
