// avatax/lib/enums/TaxRuleTypeId.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxRuleTypeIdConverter = exports.TaxRuleTypeId = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var TaxRuleTypeId;
(function (TaxRuleTypeId) {
    TaxRuleTypeId[TaxRuleTypeId["RateRule"] = 0] = "RateRule";
    TaxRuleTypeId[TaxRuleTypeId["RateOverrideRule"] = 1] = "RateOverrideRule";
    TaxRuleTypeId[TaxRuleTypeId["BaseRule"] = 2] = "BaseRule";
    TaxRuleTypeId[TaxRuleTypeId["ExemptEntityRule"] = 3] = "ExemptEntityRule";
    TaxRuleTypeId[TaxRuleTypeId["ProductTaxabilityRule"] = 4] = "ProductTaxabilityRule";
    TaxRuleTypeId[TaxRuleTypeId["NexusRule"] = 5] = "NexusRule";
    TaxRuleTypeId[TaxRuleTypeId["RateCapRule"] = 6] = "RateCapRule";
    TaxRuleTypeId[TaxRuleTypeId["TaxOverrideRule"] = 7] = "TaxOverrideRule";
    TaxRuleTypeId[TaxRuleTypeId["FeeRule"] = 8] = "FeeRule";
    TaxRuleTypeId[TaxRuleTypeId["OtherRule"] = 100] = "OtherRule";
})(TaxRuleTypeId || (exports.TaxRuleTypeId = TaxRuleTypeId = {}));
var TaxRuleTypeIdConverter = exports.TaxRuleTypeIdConverter = /** @class */ (function () {
    function TaxRuleTypeIdConverter() {
    }
    TaxRuleTypeIdConverter.prototype.serialize = function (data) {
        return data;
    };
    TaxRuleTypeIdConverter.prototype.deserialize = function (enumType) {
        return TaxRuleTypeId[enumType];
    };
    TaxRuleTypeIdConverter = __decorate([
        json2typescript_1.JsonConverter
    ], TaxRuleTypeIdConverter);
    return TaxRuleTypeIdConverter;
}());
