// avatax/lib/enums/BatchType.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchTypeConverter = exports.BatchType = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var BatchType;
(function (BatchType) {
    BatchType[BatchType["AvaCertUpdate"] = 0] = "AvaCertUpdate";
    BatchType[BatchType["AvaCertUpdateAll"] = 1] = "AvaCertUpdateAll";
    BatchType[BatchType["BatchMaintenance"] = 2] = "BatchMaintenance";
    BatchType[BatchType["CompanyLocationImport"] = 3] = "CompanyLocationImport";
    BatchType[BatchType["DocumentImport"] = 4] = "DocumentImport";
    BatchType[BatchType["ExemptCertImport"] = 5] = "ExemptCertImport";
    BatchType[BatchType["ItemImport"] = 6] = "ItemImport";
    BatchType[BatchType["SalesAuditExport"] = 7] = "SalesAuditExport";
    BatchType[BatchType["SstpTestDeckImport"] = 8] = "SstpTestDeckImport";
    BatchType[BatchType["TaxRuleImport"] = 9] = "TaxRuleImport";
    BatchType[BatchType["TransactionImport"] = 10] = "TransactionImport";
    BatchType[BatchType["UPCBulkImport"] = 11] = "UPCBulkImport";
    BatchType[BatchType["UPCValidationImport"] = 12] = "UPCValidationImport";
    BatchType[BatchType["CustomerSupplierImport"] = 13] = "CustomerSupplierImport";
    BatchType[BatchType["VarianceImport"] = 14] = "VarianceImport";
})(BatchType || (exports.BatchType = BatchType = {}));
var BatchTypeConverter = exports.BatchTypeConverter = /** @class */ (function () {
    function BatchTypeConverter() {
    }
    BatchTypeConverter.prototype.serialize = function (data) {
        return data;
    };
    BatchTypeConverter.prototype.deserialize = function (enumType) {
        return BatchType[enumType];
    };
    BatchTypeConverter = __decorate([
        json2typescript_1.JsonConverter
    ], BatchTypeConverter);
    return BatchTypeConverter;
}());
