// avatax/lib/enums/FailureCodes.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FailureCodesConverter = exports.FailureCodes = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var FailureCodes;
(function (FailureCodes) {
    FailureCodes[FailureCodes["BelowLegalDrinkingAge"] = 0] = "BelowLegalDrinkingAge";
    FailureCodes[FailureCodes["ShippingProhibitedToAddress"] = 1] = "ShippingProhibitedToAddress";
    FailureCodes[FailureCodes["MissingRequiredLicense"] = 2] = "MissingRequiredLicense";
    FailureCodes[FailureCodes["VolumeLimitExceeded"] = 3] = "VolumeLimitExceeded";
    FailureCodes[FailureCodes["InvalidFieldValue"] = 4] = "InvalidFieldValue";
    FailureCodes[FailureCodes["MissingRequiredField"] = 5] = "MissingRequiredField";
    FailureCodes[FailureCodes["InvalidFieldType"] = 6] = "InvalidFieldType";
    FailureCodes[FailureCodes["InvalidFormat"] = 7] = "InvalidFormat";
    FailureCodes[FailureCodes["InvalidDate"] = 8] = "InvalidDate";
    FailureCodes[FailureCodes["AlcoholContentLimitExceeded"] = 9] = "AlcoholContentLimitExceeded";
})(FailureCodes || (exports.FailureCodes = FailureCodes = {}));
var FailureCodesConverter = exports.FailureCodesConverter = /** @class */ (function () {
    function FailureCodesConverter() {
    }
    FailureCodesConverter.prototype.serialize = function (data) {
        return data;
    };
    FailureCodesConverter.prototype.deserialize = function (enumType) {
        return FailureCodes[enumType];
    };
    FailureCodesConverter = __decorate([
        json2typescript_1.JsonConverter
    ], FailureCodesConverter);
    return FailureCodesConverter;
}());
