// avatax/lib/enums/FilingFrequencyId.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilingFrequencyIdConverter = exports.FilingFrequencyId = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var FilingFrequencyId;
(function (FilingFrequencyId) {
    FilingFrequencyId[FilingFrequencyId["Monthly"] = 1] = "Monthly";
    FilingFrequencyId[FilingFrequencyId["Quarterly"] = 2] = "Quarterly";
    FilingFrequencyId[FilingFrequencyId["SemiAnnually"] = 3] = "SemiAnnually";
    FilingFrequencyId[FilingFrequencyId["Annually"] = 4] = "Annually";
    FilingFrequencyId[FilingFrequencyId["Bimonthly"] = 5] = "Bimonthly";
    FilingFrequencyId[FilingFrequencyId["Occasional"] = 6] = "Occasional";
    FilingFrequencyId[FilingFrequencyId["InverseQuarterly"] = 7] = "InverseQuarterly";
    FilingFrequencyId[FilingFrequencyId["Weekly"] = 8] = "Weekly";
})(FilingFrequencyId || (exports.FilingFrequencyId = FilingFrequencyId = {}));
var FilingFrequencyIdConverter = exports.FilingFrequencyIdConverter = /** @class */ (function () {
    function FilingFrequencyIdConverter() {
    }
    FilingFrequencyIdConverter.prototype.serialize = function (data) {
        return data;
    };
    FilingFrequencyIdConverter.prototype.deserialize = function (enumType) {
        return FilingFrequencyId[enumType];
    };
    FilingFrequencyIdConverter = __decorate([
        json2typescript_1.JsonConverter
    ], FilingFrequencyIdConverter);
    return FilingFrequencyIdConverter;
}());
