// avatax/lib/enums/FormTypeId.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormTypeIdConverter = exports.FormTypeId = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var FormTypeId;
(function (FormTypeId) {
    FormTypeId[FormTypeId["SalesAndUse"] = 0] = "SalesAndUse";
    FormTypeId[FormTypeId["Sales"] = 1] = "Sales";
    FormTypeId[FormTypeId["SellersUse"] = 2] = "SellersUse";
    FormTypeId[FormTypeId["Lodging"] = 3] = "Lodging";
    FormTypeId[FormTypeId["SalesAndLodging"] = 4] = "SalesAndLodging";
    FormTypeId[FormTypeId["ConsumerUse"] = 5] = "ConsumerUse";
    FormTypeId[FormTypeId["ResortAndRental"] = 6] = "ResortAndRental";
    FormTypeId[FormTypeId["TouristAndRental"] = 7] = "TouristAndRental";
    FormTypeId[FormTypeId["Prepayment"] = 8] = "Prepayment";
    FormTypeId[FormTypeId["PrepaymentAllowed"] = 9] = "PrepaymentAllowed";
})(FormTypeId || (exports.FormTypeId = FormTypeId = {}));
var FormTypeIdConverter = exports.FormTypeIdConverter = /** @class */ (function () {
    function FormTypeIdConverter() {
    }
    FormTypeIdConverter.prototype.serialize = function (data) {
        return data;
    };
    FormTypeIdConverter.prototype.deserialize = function (enumType) {
        return FormTypeId[enumType];
    };
    FormTypeIdConverter = __decorate([
        json2typescript_1.JsonConverter
    ], FormTypeIdConverter);
    return FormTypeIdConverter;
}());
