// avatax/lib/enums/NoticeCustomerType.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NoticeCustomerTypeConverter = exports.NoticeCustomerType = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var NoticeCustomerType;
(function (NoticeCustomerType) {
    NoticeCustomerType[NoticeCustomerType["AvaTaxReturns"] = 1] = "AvaTaxReturns";
    NoticeCustomerType[NoticeCustomerType["StandAlone"] = 2] = "StandAlone";
    NoticeCustomerType[NoticeCustomerType["Strategic"] = 3] = "Strategic";
    NoticeCustomerType[NoticeCustomerType["SST"] = 4] = "SST";
    NoticeCustomerType[NoticeCustomerType["TrustFile"] = 5] = "TrustFile";
    NoticeCustomerType[NoticeCustomerType["PWC"] = 6] = "PWC";
    NoticeCustomerType[NoticeCustomerType["HudsonGroup"] = 7] = "HudsonGroup";
    NoticeCustomerType[NoticeCustomerType["MRA"] = 8] = "MRA";
    NoticeCustomerType[NoticeCustomerType["None"] = 9] = "None";
})(NoticeCustomerType || (exports.NoticeCustomerType = NoticeCustomerType = {}));
var NoticeCustomerTypeConverter = exports.NoticeCustomerTypeConverter = /** @class */ (function () {
    function NoticeCustomerTypeConverter() {
    }
    NoticeCustomerTypeConverter.prototype.serialize = function (data) {
        return data;
    };
    NoticeCustomerTypeConverter.prototype.deserialize = function (enumType) {
        return NoticeCustomerType[enumType];
    };
    NoticeCustomerTypeConverter = __decorate([
        json2typescript_1.JsonConverter
    ], NoticeCustomerTypeConverter);
    return NoticeCustomerTypeConverter;
}());
