// avatax/lib/enums/SecurityRoleId.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityRoleIdConverter = exports.SecurityRoleId = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var SecurityRoleId;
(function (SecurityRoleId) {
    SecurityRoleId[SecurityRoleId["NoAccess"] = 0] = "NoAccess";
    SecurityRoleId[SecurityRoleId["SiteAdmin"] = 1] = "SiteAdmin";
    SecurityRoleId[SecurityRoleId["AccountOperator"] = 2] = "AccountOperator";
    SecurityRoleId[SecurityRoleId["AccountAdmin"] = 3] = "AccountAdmin";
    SecurityRoleId[SecurityRoleId["AccountUser"] = 4] = "AccountUser";
    SecurityRoleId[SecurityRoleId["SystemAdmin"] = 5] = "SystemAdmin";
    SecurityRoleId[SecurityRoleId["Registrar"] = 6] = "Registrar";
    SecurityRoleId[SecurityRoleId["CSPTester"] = 7] = "CSPTester";
    SecurityRoleId[SecurityRoleId["CSPAdmin"] = 8] = "CSPAdmin";
    SecurityRoleId[SecurityRoleId["SystemOperator"] = 9] = "SystemOperator";
    SecurityRoleId[SecurityRoleId["TechnicalSupportUser"] = 10] = "TechnicalSupportUser";
    SecurityRoleId[SecurityRoleId["TechnicalSupportAdmin"] = 11] = "TechnicalSupportAdmin";
    SecurityRoleId[SecurityRoleId["TreasuryUser"] = 12] = "TreasuryUser";
    SecurityRoleId[SecurityRoleId["TreasuryAdmin"] = 13] = "TreasuryAdmin";
    SecurityRoleId[SecurityRoleId["ComplianceUser"] = 14] = "ComplianceUser";
    SecurityRoleId[SecurityRoleId["ComplianceAdmin"] = 15] = "ComplianceAdmin";
    SecurityRoleId[SecurityRoleId["ProStoresOperator"] = 16] = "ProStoresOperator";
    SecurityRoleId[SecurityRoleId["CompanyUser"] = 17] = "CompanyUser";
    SecurityRoleId[SecurityRoleId["CompanyAdmin"] = 18] = "CompanyAdmin";
    SecurityRoleId[SecurityRoleId["ComplianceTempUser"] = 19] = "ComplianceTempUser";
    SecurityRoleId[SecurityRoleId["ComplianceRootUser"] = 20] = "ComplianceRootUser";
    SecurityRoleId[SecurityRoleId["ComplianceOperator"] = 21] = "ComplianceOperator";
    SecurityRoleId[SecurityRoleId["SSTAdmin"] = 22] = "SSTAdmin";
    SecurityRoleId[SecurityRoleId["FirmUser"] = 23] = "FirmUser";
    SecurityRoleId[SecurityRoleId["FirmAdmin"] = 24] = "FirmAdmin";
    SecurityRoleId[SecurityRoleId["BatchServiceAdmin"] = 25] = "BatchServiceAdmin";
    SecurityRoleId[SecurityRoleId["ECMAccountUser"] = 26] = "ECMAccountUser";
    SecurityRoleId[SecurityRoleId["ECMCompanyUser"] = 27] = "ECMCompanyUser";
})(SecurityRoleId || (exports.SecurityRoleId = SecurityRoleId = {}));
var SecurityRoleIdConverter = exports.SecurityRoleIdConverter = /** @class */ (function () {
    function SecurityRoleIdConverter() {
    }
    SecurityRoleIdConverter.prototype.serialize = function (data) {
        return data;
    };
    SecurityRoleIdConverter.prototype.deserialize = function (enumType) {
        return SecurityRoleId[enumType];
    };
    SecurityRoleIdConverter = __decorate([
        json2typescript_1.JsonConverter
    ], SecurityRoleIdConverter);
    return SecurityRoleIdConverter;
}());
