// avatax/lib/enums/SeverityLevel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SeverityLevelConverter = exports.SeverityLevel = void 0;
var json2typescript_1 = require("json2typescript");
/**
* @export
* @enum {string}
*/
var SeverityLevel;
(function (SeverityLevel) {
    SeverityLevel[SeverityLevel["Success"] = 0] = "Success";
    SeverityLevel[SeverityLevel["Warning"] = 1] = "Warning";
    SeverityLevel[SeverityLevel["Error"] = 2] = "Error";
    SeverityLevel[SeverityLevel["Exception"] = 3] = "Exception";
})(SeverityLevel || (exports.SeverityLevel = SeverityLevel = {}));
var SeverityLevelConverter = exports.SeverityLevelConverter = /** @class */ (function () {
    function SeverityLevelConverter() {
    }
    SeverityLevelConverter.prototype.serialize = function (data) {
        return data;
    };
    SeverityLevelConverter.prototype.deserialize = function (enumType) {
        return SeverityLevel[enumType];
    };
    SeverityLevelConverter = __decorate([
        json2typescript_1.JsonConverter
    ], SeverityLevelConverter);
    return SeverityLevelConverter;
}());
