// avatax/lib/models/AccountModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountModel = void 0;
var Enums = require("../enums/index");
var SubscriptionModel_1 = require("./SubscriptionModel");
var UserModel_1 = require("./UserModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * An AvaTax account.
 * @export
 * @class AccountModel
 */
var AccountModel = exports.AccountModel = /** @class */ (function () {
    function AccountModel() {
        /**
         * @type {number}
         * @memberof AccountModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof AccountModel
         */
        this.crmid = undefined;
        /**
         * @type {string}
         * @memberof AccountModel
         */
        this.name = undefined;
        /**
         * @type {Date}
         * @memberof AccountModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof AccountModel
         */
        this.endDate = undefined;
        /**
         * @type {Enums.AccountStatusId}
         * @memberof AccountModel
         */
        this.accountStatusId = undefined;
        /**
         * @type {Enums.AccountTypeId}
         * @memberof AccountModel
         */
        this.accountTypeId = undefined;
        /**
         * @type {Date}
         * @memberof AccountModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof AccountModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof AccountModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof AccountModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {SubscriptionModel[]}
         * @memberof AccountModel
         */
        this.subscriptions = undefined;
        /**
         * @type {UserModel[]}
         * @memberof AccountModel
         */
        this.users = undefined;
        /**
         * @type {boolean}
         * @memberof AccountModel
         */
        this.isSamlEnabled = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number),
        __metadata("design:type", Number)
    ], AccountModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("crmid", String, true),
        __metadata("design:type", String)
    ], AccountModel.prototype, "crmid", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String),
        __metadata("design:type", String)
    ], AccountModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AccountModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AccountModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountStatusId", Enums.AccountStatusIdConverter, true),
        __metadata("design:type", Number)
    ], AccountModel.prototype, "accountStatusId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountTypeId", Enums.AccountTypeIdConverter, true),
        __metadata("design:type", Number)
    ], AccountModel.prototype, "accountTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AccountModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], AccountModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AccountModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], AccountModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("subscriptions", [SubscriptionModel_1.SubscriptionModel], true),
        __metadata("design:type", Array)
    ], AccountModel.prototype, "subscriptions", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("users", [UserModel_1.UserModel], true),
        __metadata("design:type", Array)
    ], AccountModel.prototype, "users", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSamlEnabled", Boolean, true),
        __metadata("design:type", Boolean)
    ], AccountModel.prototype, "isSamlEnabled", void 0);
    AccountModel = __decorate([
        (0, json2typescript_1.JsonObject)("AccountModel")
    ], AccountModel);
    return AccountModel;
}());
