// avatax/lib/models/AccountMigrationStatusModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountMigrationStatusModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * An AvaTax account.
 * @export
 * @class AccountMigrationStatusModel
 */
var AccountMigrationStatusModel = exports.AccountMigrationStatusModel = /** @class */ (function () {
    function AccountMigrationStatusModel() {
        /**
         * @type {number}
         * @memberof AccountMigrationStatusModel
         */
        this.accountId = undefined;
        /**
         * @type {string}
         * @memberof AccountMigrationStatusModel
         */
        this.accountName = undefined;
        /**
         * @type {Date}
         * @memberof AccountMigrationStatusModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof AccountMigrationStatusModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {number}
         * @memberof AccountMigrationStatusModel
         */
        this.accountMigrationStatusId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountId", Number),
        __metadata("design:type", Number)
    ], AccountMigrationStatusModel.prototype, "accountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountName", String, true),
        __metadata("design:type", String)
    ], AccountMigrationStatusModel.prototype, "accountName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AccountMigrationStatusModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], AccountMigrationStatusModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountMigrationStatusId", Number, true),
        __metadata("design:type", Number)
    ], AccountMigrationStatusModel.prototype, "accountMigrationStatusId", void 0);
    AccountMigrationStatusModel = __decorate([
        (0, json2typescript_1.JsonObject)("AccountMigrationStatusModel")
    ], AccountMigrationStatusModel);
    return AccountMigrationStatusModel;
}());
