// avatax/lib/models/AccountConfigurationModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountConfigurationModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents one configuration setting for this account
 * @export
 * @class AccountConfigurationModel
 */
var AccountConfigurationModel = exports.AccountConfigurationModel = /** @class */ (function () {
    function AccountConfigurationModel() {
        /**
         * @type {number}
         * @memberof AccountConfigurationModel
         */
        this.accountId = undefined;
        /**
         * @type {string}
         * @memberof AccountConfigurationModel
         */
        this.category = undefined;
        /**
         * @type {string}
         * @memberof AccountConfigurationModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof AccountConfigurationModel
         */
        this.value = undefined;
        /**
         * @type {Date}
         * @memberof AccountConfigurationModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof AccountConfigurationModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof AccountConfigurationModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof AccountConfigurationModel
         */
        this.modifiedUserId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountId", Number, true),
        __metadata("design:type", Number)
    ], AccountConfigurationModel.prototype, "accountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("category", String),
        __metadata("design:type", String)
    ], AccountConfigurationModel.prototype, "category", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String),
        __metadata("design:type", String)
    ], AccountConfigurationModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("value", String, true),
        __metadata("design:type", String)
    ], AccountConfigurationModel.prototype, "value", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AccountConfigurationModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], AccountConfigurationModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AccountConfigurationModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], AccountConfigurationModel.prototype, "modifiedUserId", void 0);
    AccountConfigurationModel = __decorate([
        (0, json2typescript_1.JsonObject)("AccountConfigurationModel")
    ], AccountConfigurationModel);
    return AccountConfigurationModel;
}());
