// avatax/lib/models/ACHEntryDetailModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ACHEntryDetailModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * An edit to be made on a filing calendar.
 * @export
 * @class ACHEntryDetailModel
 */
var ACHEntryDetailModel = exports.ACHEntryDetailModel = /** @class */ (function () {
    function ACHEntryDetailModel() {
        /**
         * @type {number}
         * @memberof ACHEntryDetailModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof ACHEntryDetailModel
         */
        this.companyName = undefined;
        /**
         * @type {string}
         * @memberof ACHEntryDetailModel
         */
        this.state = undefined;
        /**
         * @type {string}
         * @memberof ACHEntryDetailModel
         */
        this.stateRegion = undefined;
        /**
         * @type {string}
         * @memberof ACHEntryDetailModel
         */
        this.individualId = undefined;
        /**
         * @type {string}
         * @memberof ACHEntryDetailModel
         */
        this.individualName = undefined;
        /**
         * @type {number}
         * @memberof ACHEntryDetailModel
         */
        this.amount = undefined;
        /**
         * @type {string}
         * @memberof ACHEntryDetailModel
         */
        this.traceNumber = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], ACHEntryDetailModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyName", String, true),
        __metadata("design:type", String)
    ], ACHEntryDetailModel.prototype, "companyName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("state", String, true),
        __metadata("design:type", String)
    ], ACHEntryDetailModel.prototype, "state", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("stateRegion", String, true),
        __metadata("design:type", String)
    ], ACHEntryDetailModel.prototype, "stateRegion", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("individualId", String, true),
        __metadata("design:type", String)
    ], ACHEntryDetailModel.prototype, "individualId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("individualName", String, true),
        __metadata("design:type", String)
    ], ACHEntryDetailModel.prototype, "individualName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("amount", Number, true),
        __metadata("design:type", Number)
    ], ACHEntryDetailModel.prototype, "amount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("traceNumber", String, true),
        __metadata("design:type", String)
    ], ACHEntryDetailModel.prototype, "traceNumber", void 0);
    ACHEntryDetailModel = __decorate([
        (0, json2typescript_1.JsonObject)("ACHEntryDetailModel")
    ], ACHEntryDetailModel);
    return ACHEntryDetailModel;
}());
