// avatax/lib/models/AddressResolutionModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddressResolutionModel = void 0;
var Enums = require("../enums/index");
var AddressInfo_1 = require("./AddressInfo");
var ValidatedAddressInfo_1 = require("./ValidatedAddressInfo");
var CoordinateInfo_1 = require("./CoordinateInfo");
var TaxAuthorityInfo_1 = require("./TaxAuthorityInfo");
var AvaTaxMessage_1 = require("./AvaTaxMessage");
var json2typescript_1 = require("json2typescript");
/**
 * Address Resolution Model
 * @export
 * @class AddressResolutionModel
 */
var AddressResolutionModel = exports.AddressResolutionModel = /** @class */ (function () {
    function AddressResolutionModel() {
        /**
         * @type {AddressInfo}
         * @memberof AddressResolutionModel
         */
        this.address = undefined;
        /**
         * @type {ValidatedAddressInfo[]}
         * @memberof AddressResolutionModel
         */
        this.validatedAddresses = undefined;
        /**
         * @type {CoordinateInfo}
         * @memberof AddressResolutionModel
         */
        this.coordinates = undefined;
        /**
         * @type {Enums.ResolutionQuality}
         * @memberof AddressResolutionModel
         */
        this.resolutionQuality = undefined;
        /**
         * @type {TaxAuthorityInfo[]}
         * @memberof AddressResolutionModel
         */
        this.taxAuthorities = undefined;
        /**
         * @type {AvaTaxMessage[]}
         * @memberof AddressResolutionModel
         */
        this.messages = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("address", AddressInfo_1.AddressInfo, true),
        __metadata("design:type", AddressInfo_1.AddressInfo)
    ], AddressResolutionModel.prototype, "address", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("validatedAddresses", [ValidatedAddressInfo_1.ValidatedAddressInfo], true),
        __metadata("design:type", Array)
    ], AddressResolutionModel.prototype, "validatedAddresses", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("coordinates", CoordinateInfo_1.CoordinateInfo, true),
        __metadata("design:type", CoordinateInfo_1.CoordinateInfo)
    ], AddressResolutionModel.prototype, "coordinates", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("resolutionQuality", Enums.ResolutionQualityConverter, true),
        __metadata("design:type", Number)
    ], AddressResolutionModel.prototype, "resolutionQuality", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorities", [TaxAuthorityInfo_1.TaxAuthorityInfo], true),
        __metadata("design:type", Array)
    ], AddressResolutionModel.prototype, "taxAuthorities", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("messages", [AvaTaxMessage_1.AvaTaxMessage], true),
        __metadata("design:type", Array)
    ], AddressResolutionModel.prototype, "messages", void 0);
    AddressResolutionModel = __decorate([
        (0, json2typescript_1.JsonObject)("AddressResolutionModel")
    ], AddressResolutionModel);
    return AddressResolutionModel;
}());
