// avatax/lib/models/AddTransactionLineModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddTransactionLineModel = void 0;
var Enums = require("../enums/index");
var LineItemModel_1 = require("./LineItemModel");
var json2typescript_1 = require("json2typescript");
/**
 * Model to add specific lines to exising transaction
 * @export
 * @class AddTransactionLineModel
 */
var AddTransactionLineModel = exports.AddTransactionLineModel = /** @class */ (function () {
    function AddTransactionLineModel() {
        /**
         * @type {string}
         * @memberof AddTransactionLineModel
         */
        this.companyCode = undefined;
        /**
         * @type {string}
         * @memberof AddTransactionLineModel
         */
        this.transactionCode = undefined;
        /**
         * @type {Enums.DocumentType}
         * @memberof AddTransactionLineModel
         */
        this.documentType = undefined;
        /**
         * @type {LineItemModel[]}
         * @memberof AddTransactionLineModel
         */
        this.lines = undefined;
        /**
         * @type {boolean}
         * @memberof AddTransactionLineModel
         */
        this.renumber = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyCode", String),
        __metadata("design:type", String)
    ], AddTransactionLineModel.prototype, "companyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("transactionCode", String),
        __metadata("design:type", String)
    ], AddTransactionLineModel.prototype, "transactionCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("documentType", Enums.DocumentTypeConverter, true),
        __metadata("design:type", Number)
    ], AddTransactionLineModel.prototype, "documentType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lines", [LineItemModel_1.LineItemModel]),
        __metadata("design:type", Array)
    ], AddTransactionLineModel.prototype, "lines", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("renumber", Boolean, true),
        __metadata("design:type", Boolean)
    ], AddTransactionLineModel.prototype, "renumber", void 0);
    AddTransactionLineModel = __decorate([
        (0, json2typescript_1.JsonObject)("AddTransactionLineModel")
    ], AddTransactionLineModel);
    return AddTransactionLineModel;
}());
