// avatax/lib/models/AdvancedRuleExecutionModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdvancedRuleExecutionModel = void 0;
var AdvancedRuleModel_1 = require("./AdvancedRuleModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Model representing an execution of an advanced rule for a company
 * @export
 * @class AdvancedRuleExecutionModel
 */
var AdvancedRuleExecutionModel = exports.AdvancedRuleExecutionModel = /** @class */ (function () {
    function AdvancedRuleExecutionModel() {
        /**
         * @type {string}
         * @memberof AdvancedRuleExecutionModel
         */
        this.ruleExecutionId = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleExecutionModel
         */
        this.name = undefined;
        /**
         * @type {Date}
         * @memberof AdvancedRuleExecutionModel
         */
        this.startDate = undefined;
        /**
         * @type {Date}
         * @memberof AdvancedRuleExecutionModel
         */
        this.endDate = undefined;
        /**
         * @type {boolean}
         * @memberof AdvancedRuleExecutionModel
         */
        this.enabled = undefined;
        /**
         * @type {boolean}
         * @memberof AdvancedRuleExecutionModel
         */
        this.continueOnError = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleExecutionModel
         */
        this.ruleId = undefined;
        /**
         * @type {AdvancedRuleModel}
         * @memberof AdvancedRuleExecutionModel
         */
        this.rule = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleExecutionModel
         */
        this.customerData = undefined;
        /**
         * @type {string}
         * @memberof AdvancedRuleExecutionModel
         */
        this.customerDataId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("ruleExecutionId", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleExecutionModel.prototype, "ruleExecutionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String),
        __metadata("design:type", String)
    ], AdvancedRuleExecutionModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("startDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AdvancedRuleExecutionModel.prototype, "startDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AdvancedRuleExecutionModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("enabled", Boolean, true),
        __metadata("design:type", Boolean)
    ], AdvancedRuleExecutionModel.prototype, "enabled", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("continueOnError", Boolean, true),
        __metadata("design:type", Boolean)
    ], AdvancedRuleExecutionModel.prototype, "continueOnError", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("ruleId", String),
        __metadata("design:type", String)
    ], AdvancedRuleExecutionModel.prototype, "ruleId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rule", AdvancedRuleModel_1.AdvancedRuleModel, true),
        __metadata("design:type", AdvancedRuleModel_1.AdvancedRuleModel)
    ], AdvancedRuleExecutionModel.prototype, "rule", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerData", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleExecutionModel.prototype, "customerData", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerDataId", String, true),
        __metadata("design:type", String)
    ], AdvancedRuleExecutionModel.prototype, "customerDataId", void 0);
    AdvancedRuleExecutionModel = __decorate([
        (0, json2typescript_1.JsonObject)("AdvancedRuleExecutionModel")
    ], AdvancedRuleExecutionModel);
    return AdvancedRuleExecutionModel;
}());
