// avatax/lib/models/AuditMultiDocumentModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuditMultiDocumentModel = void 0;
var Enums = require("../enums/index");
var ReconstructedMultiDocumentModel_1 = require("./ReconstructedMultiDocumentModel");
var OriginalApiRequestResponseModel_1 = require("./OriginalApiRequestResponseModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Information about a previously created MultiDocument transaction
 * @export
 * @class AuditMultiDocumentModel
 */
var AuditMultiDocumentModel = exports.AuditMultiDocumentModel = /** @class */ (function () {
    function AuditMultiDocumentModel() {
        /**
         * @type {ReconstructedMultiDocumentModel}
         * @memberof AuditMultiDocumentModel
         */
        this.reconstructed = undefined;
        /**
         * @type {string}
         * @memberof AuditMultiDocumentModel
         */
        this.code = undefined;
        /**
         * @type {Enums.DocumentType}
         * @memberof AuditMultiDocumentModel
         */
        this.type = undefined;
        /**
         * @type {Date}
         * @memberof AuditMultiDocumentModel
         */
        this.serverTimestamp = undefined;
        /**
         * @type {Date}
         * @memberof AuditMultiDocumentModel
         */
        this.serverDuration = undefined;
        /**
         * @type {Enums.ApiCallStatus}
         * @memberof AuditMultiDocumentModel
         */
        this.apiCallStatus = undefined;
        /**
         * @type {OriginalApiRequestResponseModel}
         * @memberof AuditMultiDocumentModel
         */
        this.original = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("reconstructed", ReconstructedMultiDocumentModel_1.ReconstructedMultiDocumentModel, true),
        __metadata("design:type", ReconstructedMultiDocumentModel_1.ReconstructedMultiDocumentModel)
    ], AuditMultiDocumentModel.prototype, "reconstructed", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("code", String, true),
        __metadata("design:type", String)
    ], AuditMultiDocumentModel.prototype, "code", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("type", Enums.DocumentTypeConverter, true),
        __metadata("design:type", Number)
    ], AuditMultiDocumentModel.prototype, "type", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("serverTimestamp", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AuditMultiDocumentModel.prototype, "serverTimestamp", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("serverDuration", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AuditMultiDocumentModel.prototype, "serverDuration", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("apiCallStatus", Enums.ApiCallStatusConverter, true),
        __metadata("design:type", Number)
    ], AuditMultiDocumentModel.prototype, "apiCallStatus", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("original", OriginalApiRequestResponseModel_1.OriginalApiRequestResponseModel, true),
        __metadata("design:type", OriginalApiRequestResponseModel_1.OriginalApiRequestResponseModel)
    ], AuditMultiDocumentModel.prototype, "original", void 0);
    AuditMultiDocumentModel = __decorate([
        (0, json2typescript_1.JsonObject)("AuditMultiDocumentModel")
    ], AuditMultiDocumentModel);
    return AuditMultiDocumentModel;
}());
