// avatax/lib/models/AuditTransactionModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuditTransactionModel = void 0;
var Enums = require("../enums/index");
var ReconstructedApiRequestResponseModel_1 = require("./ReconstructedApiRequestResponseModel");
var OriginalApiRequestResponseModel_1 = require("./OriginalApiRequestResponseModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Information about a previously created transaction
 * @export
 * @class AuditTransactionModel
 */
var AuditTransactionModel = exports.AuditTransactionModel = /** @class */ (function () {
    function AuditTransactionModel() {
        /**
         * @type {number}
         * @memberof AuditTransactionModel
         */
        this.companyId = undefined;
        /**
         * @type {ReconstructedApiRequestResponseModel}
         * @memberof AuditTransactionModel
         */
        this.reconstructed = undefined;
        /**
         * @type {Date}
         * @memberof AuditTransactionModel
         */
        this.serverTimestamp = undefined;
        /**
         * @type {Date}
         * @memberof AuditTransactionModel
         */
        this.serverDuration = undefined;
        /**
         * @type {Enums.ApiCallStatus}
         * @memberof AuditTransactionModel
         */
        this.apiCallStatus = undefined;
        /**
         * @type {OriginalApiRequestResponseModel}
         * @memberof AuditTransactionModel
         */
        this.original = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], AuditTransactionModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reconstructed", ReconstructedApiRequestResponseModel_1.ReconstructedApiRequestResponseModel, true),
        __metadata("design:type", ReconstructedApiRequestResponseModel_1.ReconstructedApiRequestResponseModel)
    ], AuditTransactionModel.prototype, "reconstructed", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("serverTimestamp", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AuditTransactionModel.prototype, "serverTimestamp", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("serverDuration", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AuditTransactionModel.prototype, "serverDuration", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("apiCallStatus", Enums.ApiCallStatusConverter, true),
        __metadata("design:type", Number)
    ], AuditTransactionModel.prototype, "apiCallStatus", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("original", OriginalApiRequestResponseModel_1.OriginalApiRequestResponseModel, true),
        __metadata("design:type", OriginalApiRequestResponseModel_1.OriginalApiRequestResponseModel)
    ], AuditTransactionModel.prototype, "original", void 0);
    AuditTransactionModel = __decorate([
        (0, json2typescript_1.JsonObject)("AuditTransactionModel")
    ], AuditTransactionModel);
    return AuditTransactionModel;
}());
