// avatax/lib/models/AvaFileFormModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AvaFileFormModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents information about a tax form known to Avalara
 * @export
 * @class AvaFileFormModel
 */
var AvaFileFormModel = exports.AvaFileFormModel = /** @class */ (function () {
    function AvaFileFormModel() {
        /**
         * @type {number}
         * @memberof AvaFileFormModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof AvaFileFormModel
         */
        this.returnName = undefined;
        /**
         * @type {string}
         * @memberof AvaFileFormModel
         */
        this.formName = undefined;
        /**
         * @type {string}
         * @memberof AvaFileFormModel
         */
        this.description = undefined;
        /**
         * @type {Date}
         * @memberof AvaFileFormModel
         */
        this.effDate = undefined;
        /**
         * @type {Date}
         * @memberof AvaFileFormModel
         */
        this.endDate = undefined;
        /**
         * @type {string}
         * @memberof AvaFileFormModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof AvaFileFormModel
         */
        this.country = undefined;
        /**
         * @type {Enums.FormTypeId}
         * @memberof AvaFileFormModel
         */
        this.formTypeId = undefined;
        /**
         * @type {Enums.FilingOptionTypeId}
         * @memberof AvaFileFormModel
         */
        this.filingOptionTypeId = undefined;
        /**
         * @type {Enums.DueDateTypeId}
         * @memberof AvaFileFormModel
         */
        this.dueDateTypeId = undefined;
        /**
         * @type {number}
         * @memberof AvaFileFormModel
         */
        this.dueDay = undefined;
        /**
         * @type {Enums.DueDateTypeId}
         * @memberof AvaFileFormModel
         */
        this.efileDueDateTypeId = undefined;
        /**
         * @type {number}
         * @memberof AvaFileFormModel
         */
        this.efileDueDay = undefined;
        /**
         * @type {Date}
         * @memberof AvaFileFormModel
         */
        this.efileDueTime = undefined;
        /**
         * @type {boolean}
         * @memberof AvaFileFormModel
         */
        this.hasVendorDiscount = undefined;
        /**
         * @type {Enums.RoundingTypeId}
         * @memberof AvaFileFormModel
         */
        this.roundingTypeId = undefined;
        /**
         * @type {Enums.OutletTypeId}
         * @memberof AvaFileFormModel
         */
        this.outletTypeId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], AvaFileFormModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("returnName", String, true),
        __metadata("design:type", String)
    ], AvaFileFormModel.prototype, "returnName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("formName", String, true),
        __metadata("design:type", String)
    ], AvaFileFormModel.prototype, "formName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], AvaFileFormModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AvaFileFormModel.prototype, "effDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AvaFileFormModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], AvaFileFormModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], AvaFileFormModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("formTypeId", Enums.FormTypeIdConverter, true),
        __metadata("design:type", Number)
    ], AvaFileFormModel.prototype, "formTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingOptionTypeId", Enums.FilingOptionTypeIdConverter, true),
        __metadata("design:type", Number)
    ], AvaFileFormModel.prototype, "filingOptionTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dueDateTypeId", Enums.DueDateTypeIdConverter, true),
        __metadata("design:type", Number)
    ], AvaFileFormModel.prototype, "dueDateTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dueDay", Number, true),
        __metadata("design:type", Number)
    ], AvaFileFormModel.prototype, "dueDay", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("efileDueDateTypeId", Enums.DueDateTypeIdConverter, true),
        __metadata("design:type", Number)
    ], AvaFileFormModel.prototype, "efileDueDateTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("efileDueDay", Number, true),
        __metadata("design:type", Number)
    ], AvaFileFormModel.prototype, "efileDueDay", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("efileDueTime", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], AvaFileFormModel.prototype, "efileDueTime", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hasVendorDiscount", Boolean, true),
        __metadata("design:type", Boolean)
    ], AvaFileFormModel.prototype, "hasVendorDiscount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("roundingTypeId", Enums.RoundingTypeIdConverter, true),
        __metadata("design:type", Number)
    ], AvaFileFormModel.prototype, "roundingTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("outletTypeId", Enums.OutletTypeIdConverter, true),
        __metadata("design:type", Number)
    ], AvaFileFormModel.prototype, "outletTypeId", void 0);
    AvaFileFormModel = __decorate([
        (0, json2typescript_1.JsonObject)("AvaFileFormModel")
    ], AvaFileFormModel);
    return AvaFileFormModel;
}());
