// avatax/lib/models/AvaTaxMessage.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AvaTaxMessage = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Informational or warning messages returned by AvaTax with a transaction
 * @export
 * @class AvaTaxMessage
 */
var AvaTaxMessage = exports.AvaTaxMessage = /** @class */ (function () {
    function AvaTaxMessage() {
        /**
         * @type {string}
         * @memberof AvaTaxMessage
         */
        this.summary = undefined;
        /**
         * @type {string}
         * @memberof AvaTaxMessage
         */
        this.details = undefined;
        /**
         * @type {string}
         * @memberof AvaTaxMessage
         */
        this.refersTo = undefined;
        /**
         * @type {string}
         * @memberof AvaTaxMessage
         */
        this.severity = undefined;
        /**
         * @type {string}
         * @memberof AvaTaxMessage
         */
        this.source = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("summary", String, true),
        __metadata("design:type", String)
    ], AvaTaxMessage.prototype, "summary", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("details", String, true),
        __metadata("design:type", String)
    ], AvaTaxMessage.prototype, "details", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("refersTo", String, true),
        __metadata("design:type", String)
    ], AvaTaxMessage.prototype, "refersTo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("severity", String, true),
        __metadata("design:type", String)
    ], AvaTaxMessage.prototype, "severity", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("source", String, true),
        __metadata("design:type", String)
    ], AvaTaxMessage.prototype, "source", void 0);
    AvaTaxMessage = __decorate([
        (0, json2typescript_1.JsonObject)("AvaTaxMessage")
    ], AvaTaxMessage);
    return AvaTaxMessage;
}());
