// avatax/lib/models/CertExpressInvitationModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CertExpressInvitationModel = void 0;
var Enums = require("../enums/index");
var CustomerModel_1 = require("./CustomerModel");
var CoverLetterModel_1 = require("./CoverLetterModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents an invitation for a customer to use CertExpress to self-report their own certificates.
This invitation is delivered by your choice of method, or you can present a hyperlink to the user
directly in your connector.  Your customer will be redirected to https://app.certexpress.com/ where
they can follow a step-by-step guide to enter information about their exemption certificates.  The
certificates entered will be recorded and automatically linked to their customer record.
 * @export
 * @class CertExpressInvitationModel
 */
var CertExpressInvitationModel = exports.CertExpressInvitationModel = /** @class */ (function () {
    function CertExpressInvitationModel() {
        /**
         * @type {number}
         * @memberof CertExpressInvitationModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof CertExpressInvitationModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof CertExpressInvitationModel
         */
        this.recipient = undefined;
        /**
         * @type {string}
         * @memberof CertExpressInvitationModel
         */
        this.customerCode = undefined;
        /**
         * @type {CustomerModel}
         * @memberof CertExpressInvitationModel
         */
        this.customer = undefined;
        /**
         * @type {CoverLetterModel}
         * @memberof CertExpressInvitationModel
         */
        this.coverLetter = undefined;
        /**
         * @type {string}
         * @memberof CertExpressInvitationModel
         */
        this.emailStatus = undefined;
        /**
         * @type {boolean}
         * @memberof CertExpressInvitationModel
         */
        this.coverLettersOnly = undefined;
        /**
         * @type {number[]}
         * @memberof CertExpressInvitationModel
         */
        this.exposureZones = undefined;
        /**
         * @type {number[]}
         * @memberof CertExpressInvitationModel
         */
        this.exemptReasons = undefined;
        /**
         * @type {Enums.CertificateRequestDeliveryMethod}
         * @memberof CertExpressInvitationModel
         */
        this.deliveryMethod = undefined;
        /**
         * @type {string}
         * @memberof CertExpressInvitationModel
         */
        this.message = undefined;
        /**
         * @type {Date}
         * @memberof CertExpressInvitationModel
         */
        this.date = undefined;
        /**
         * @type {string}
         * @memberof CertExpressInvitationModel
         */
        this.requestLink = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], CertExpressInvitationModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], CertExpressInvitationModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("recipient", String, true),
        __metadata("design:type", String)
    ], CertExpressInvitationModel.prototype, "recipient", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerCode", String, true),
        __metadata("design:type", String)
    ], CertExpressInvitationModel.prototype, "customerCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customer", CustomerModel_1.CustomerModel, true),
        __metadata("design:type", CustomerModel_1.CustomerModel)
    ], CertExpressInvitationModel.prototype, "customer", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("coverLetter", CoverLetterModel_1.CoverLetterModel, true),
        __metadata("design:type", CoverLetterModel_1.CoverLetterModel)
    ], CertExpressInvitationModel.prototype, "coverLetter", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("emailStatus", String, true),
        __metadata("design:type", String)
    ], CertExpressInvitationModel.prototype, "emailStatus", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("coverLettersOnly", Boolean, true),
        __metadata("design:type", Boolean)
    ], CertExpressInvitationModel.prototype, "coverLettersOnly", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exposureZones", [Number], true),
        __metadata("design:type", Array)
    ], CertExpressInvitationModel.prototype, "exposureZones", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptReasons", [Number], true),
        __metadata("design:type", Array)
    ], CertExpressInvitationModel.prototype, "exemptReasons", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("deliveryMethod", Enums.CertificateRequestDeliveryMethodConverter, true),
        __metadata("design:type", Number)
    ], CertExpressInvitationModel.prototype, "deliveryMethod", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("message", String, true),
        __metadata("design:type", String)
    ], CertExpressInvitationModel.prototype, "message", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("date", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CertExpressInvitationModel.prototype, "date", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("requestLink", String, true),
        __metadata("design:type", String)
    ], CertExpressInvitationModel.prototype, "requestLink", void 0);
    CertExpressInvitationModel = __decorate([
        (0, json2typescript_1.JsonObject)("CertExpressInvitationModel")
    ], CertExpressInvitationModel);
    return CertExpressInvitationModel;
}());
