// avatax/lib/models/ClassificationDetailsModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClassificationDetailsModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Represents classification details model
 * @export
 * @class ClassificationDetailsModel
 */
var ClassificationDetailsModel = exports.ClassificationDetailsModel = /** @class */ (function () {
    function ClassificationDetailsModel() {
        /**
         * @type {number}
         * @memberof ClassificationDetailsModel
         */
        this.classified = undefined;
        /**
         * @type {number}
         * @memberof ClassificationDetailsModel
         */
        this.inProgress = undefined;
        /**
         * @type {number}
         * @memberof ClassificationDetailsModel
         */
        this.notClassified = undefined;
        /**
         * @type {number}
         * @memberof ClassificationDetailsModel
         */
        this.failed = undefined;
        /**
         * @type {number}
         * @memberof ClassificationDetailsModel
         */
        this.notFound = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("classified", Number, true),
        __metadata("design:type", Number)
    ], ClassificationDetailsModel.prototype, "classified", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("inProgress", Number, true),
        __metadata("design:type", Number)
    ], ClassificationDetailsModel.prototype, "inProgress", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("notClassified", Number, true),
        __metadata("design:type", Number)
    ], ClassificationDetailsModel.prototype, "notClassified", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("failed", Number, true),
        __metadata("design:type", Number)
    ], ClassificationDetailsModel.prototype, "failed", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("notFound", Number, true),
        __metadata("design:type", Number)
    ], ClassificationDetailsModel.prototype, "notFound", void 0);
    ClassificationDetailsModel = __decorate([
        (0, json2typescript_1.JsonObject)("ClassificationDetailsModel")
    ], ClassificationDetailsModel);
    return ClassificationDetailsModel;
}());
