// avatax/lib/models/CommunicationsTSPairModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommunicationsTSPairModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Represents a transaction/service type pair for telecommunications tax
 * @export
 * @class CommunicationsTSPairModel
 */
var CommunicationsTSPairModel = exports.CommunicationsTSPairModel = /** @class */ (function () {
    function CommunicationsTSPairModel() {
        /**
         * @type {number}
         * @memberof CommunicationsTSPairModel
         */
        this.transactionTypeId = undefined;
        /**
         * @type {number}
         * @memberof CommunicationsTSPairModel
         */
        this.serviceTypeId = undefined;
        /**
         * @type {string}
         * @memberof CommunicationsTSPairModel
         */
        this.TransactionType = undefined;
        /**
         * @type {string}
         * @memberof CommunicationsTSPairModel
         */
        this.ServiceType = undefined;
        /**
         * @type {string}
         * @memberof CommunicationsTSPairModel
         */
        this.description = undefined;
        /**
         * @type {string[]}
         * @memberof CommunicationsTSPairModel
         */
        this.requiredParameters = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("transactionTypeId", Number),
        __metadata("design:type", Number)
    ], CommunicationsTSPairModel.prototype, "transactionTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("serviceTypeId", Number),
        __metadata("design:type", Number)
    ], CommunicationsTSPairModel.prototype, "serviceTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("TransactionType", String, true),
        __metadata("design:type", String)
    ], CommunicationsTSPairModel.prototype, "TransactionType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("ServiceType", String, true),
        __metadata("design:type", String)
    ], CommunicationsTSPairModel.prototype, "ServiceType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], CommunicationsTSPairModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("requiredParameters", [String], true),
        __metadata("design:type", Array)
    ], CommunicationsTSPairModel.prototype, "requiredParameters", void 0);
    CommunicationsTSPairModel = __decorate([
        (0, json2typescript_1.JsonObject)("CommunicationsTSPairModel")
    ], CommunicationsTSPairModel);
    return CommunicationsTSPairModel;
}());
