// avatax/lib/models/CompanyInitializationModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CompanyInitializationModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Company Initialization Model
 * @export
 * @class CompanyInitializationModel
 */
var CompanyInitializationModel = exports.CompanyInitializationModel = /** @class */ (function () {
    function CompanyInitializationModel() {
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.companyCode = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.vatRegistrationId = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.taxpayerIdNumber = undefined;
        /**
         * @type {boolean}
         * @memberof CompanyInitializationModel
         */
        this.isFein = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.line1 = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.line2 = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.line3 = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.city = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.postalCode = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.firstName = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.lastName = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.title = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.email = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.phoneNumber = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.mobileNumber = undefined;
        /**
         * @type {string}
         * @memberof CompanyInitializationModel
         */
        this.faxNumber = undefined;
        /**
         * @type {number}
         * @memberof CompanyInitializationModel
         */
        this.parentCompanyId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyCode", String, true),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "companyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("vatRegistrationId", String, true),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "vatRegistrationId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxpayerIdNumber", String, true),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "taxpayerIdNumber", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isFein", Boolean, true),
        __metadata("design:type", Boolean)
    ], CompanyInitializationModel.prototype, "isFein", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line1", String),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "line1", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line2", String, true),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "line2", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line3", String, true),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "line3", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("city", String),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "city", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("postalCode", String),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "postalCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("firstName", String),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "firstName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lastName", String),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "lastName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("title", String, true),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "title", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("email", String),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "email", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("phoneNumber", String),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "phoneNumber", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("mobileNumber", String, true),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "mobileNumber", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("faxNumber", String, true),
        __metadata("design:type", String)
    ], CompanyInitializationModel.prototype, "faxNumber", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("parentCompanyId", Number, true),
        __metadata("design:type", Number)
    ], CompanyInitializationModel.prototype, "parentCompanyId", void 0);
    CompanyInitializationModel = __decorate([
        (0, json2typescript_1.JsonObject)("CompanyInitializationModel")
    ], CompanyInitializationModel);
    return CompanyInitializationModel;
}());
