// avatax/lib/models/CompanyAddress.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CompanyAddress = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Company Address Information
 * @export
 * @class CompanyAddress
 */
var CompanyAddress = exports.CompanyAddress = /** @class */ (function () {
    function CompanyAddress() {
        /**
         * @type {string}
         * @memberof CompanyAddress
         */
        this.line = undefined;
        /**
         * @type {string}
         * @memberof CompanyAddress
         */
        this.city = undefined;
        /**
         * @type {string}
         * @memberof CompanyAddress
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof CompanyAddress
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof CompanyAddress
         */
        this.postalCode = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("line", String),
        __metadata("design:type", String)
    ], CompanyAddress.prototype, "line", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("city", String, true),
        __metadata("design:type", String)
    ], CompanyAddress.prototype, "city", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], CompanyAddress.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String),
        __metadata("design:type", String)
    ], CompanyAddress.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("postalCode", String),
        __metadata("design:type", String)
    ], CompanyAddress.prototype, "postalCode", void 0);
    CompanyAddress = __decorate([
        (0, json2typescript_1.JsonObject)("CompanyAddress")
    ], CompanyAddress);
    return CompanyAddress;
}());
