// avatax/lib/models/ComplianceTaxRateModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComplianceTaxRateModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * The tax rate model.
 * @export
 * @class ComplianceTaxRateModel
 */
var ComplianceTaxRateModel = exports.ComplianceTaxRateModel = /** @class */ (function () {
    function ComplianceTaxRateModel() {
        /**
         * @type {number}
         * @memberof ComplianceTaxRateModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof ComplianceTaxRateModel
         */
        this.rate = undefined;
        /**
         * @type {number}
         * @memberof ComplianceTaxRateModel
         */
        this.jurisdictionId = undefined;
        /**
         * @type {number}
         * @memberof ComplianceTaxRateModel
         */
        this.taxRegionId = undefined;
        /**
         * @type {Date}
         * @memberof ComplianceTaxRateModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof ComplianceTaxRateModel
         */
        this.endDate = undefined;
        /**
         * @type {string}
         * @memberof ComplianceTaxRateModel
         */
        this.rateTypeId = undefined;
        /**
         * @type {string}
         * @memberof ComplianceTaxRateModel
         */
        this.taxTypeId = undefined;
        /**
         * @type {string}
         * @memberof ComplianceTaxRateModel
         */
        this.taxName = undefined;
        /**
         * @type {number}
         * @memberof ComplianceTaxRateModel
         */
        this.unitOfBasisId = undefined;
        /**
         * @type {number}
         * @memberof ComplianceTaxRateModel
         */
        this.rateTypeTaxTypeMappingId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], ComplianceTaxRateModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rate", Number, true),
        __metadata("design:type", Number)
    ], ComplianceTaxRateModel.prototype, "rate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionId", Number, true),
        __metadata("design:type", Number)
    ], ComplianceTaxRateModel.prototype, "jurisdictionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxRegionId", Number, true),
        __metadata("design:type", Number)
    ], ComplianceTaxRateModel.prototype, "taxRegionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ComplianceTaxRateModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ComplianceTaxRateModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rateTypeId", String, true),
        __metadata("design:type", String)
    ], ComplianceTaxRateModel.prototype, "rateTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxTypeId", String, true),
        __metadata("design:type", String)
    ], ComplianceTaxRateModel.prototype, "taxTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxName", String, true),
        __metadata("design:type", String)
    ], ComplianceTaxRateModel.prototype, "taxName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("unitOfBasisId", Number, true),
        __metadata("design:type", Number)
    ], ComplianceTaxRateModel.prototype, "unitOfBasisId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rateTypeTaxTypeMappingId", Number, true),
        __metadata("design:type", Number)
    ], ComplianceTaxRateModel.prototype, "rateTypeTaxTypeMappingId", void 0);
    ComplianceTaxRateModel = __decorate([
        (0, json2typescript_1.JsonObject)("ComplianceTaxRateModel")
    ], ComplianceTaxRateModel);
    return ComplianceTaxRateModel;
}());
