// avatax/lib/models/CompanyDistanceThresholdModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CompanyDistanceThresholdModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * A company-distance-threshold model indicates the distance between a company
and the taxing borders of various countries.  Distance thresholds are necessary
to correctly calculate some value-added taxes.
            
Distance thresholds only apply to sales of goods in certain countries.  A distance threshold
is applied for each ship-from/ship-to combination of countries.  The threshold amount is defined by
the ship-to country.
            
Generally, if you have exceeded a distance threshold for taxes between a pair of countries, your tax calculation
will be determined to be the rate in the destination country.  If you have not exceeded the threshold,
your tax calculation will be determined to be the rate in the origin country.
            
The amount of a threshold is not tracked or managed in AvaTax, but the decision of your tax compliance department
as to whether you have exceeded this threshold is maintained in this object.
            
By default, you are considered to have exceeded tax thresholds. If you wish to change this default, you can create
a company-distance-threshold object to select the correct behavior for this origin/destination tax calculation process.
 * @export
 * @class CompanyDistanceThresholdModel
 */
var CompanyDistanceThresholdModel = exports.CompanyDistanceThresholdModel = /** @class */ (function () {
    function CompanyDistanceThresholdModel() {
        /**
         * @type {number}
         * @memberof CompanyDistanceThresholdModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof CompanyDistanceThresholdModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof CompanyDistanceThresholdModel
         */
        this.originCountry = undefined;
        /**
         * @type {string}
         * @memberof CompanyDistanceThresholdModel
         */
        this.destinationCountry = undefined;
        /**
         * @type {Date}
         * @memberof CompanyDistanceThresholdModel
         */
        this.effDate = undefined;
        /**
         * @type {Date}
         * @memberof CompanyDistanceThresholdModel
         */
        this.endDate = undefined;
        /**
         * @type {boolean}
         * @memberof CompanyDistanceThresholdModel
         */
        this.thresholdExceeded = undefined;
        /**
         * @type {string}
         * @memberof CompanyDistanceThresholdModel
         */
        this.type = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], CompanyDistanceThresholdModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], CompanyDistanceThresholdModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("originCountry", String),
        __metadata("design:type", String)
    ], CompanyDistanceThresholdModel.prototype, "originCountry", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("destinationCountry", String),
        __metadata("design:type", String)
    ], CompanyDistanceThresholdModel.prototype, "destinationCountry", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CompanyDistanceThresholdModel.prototype, "effDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CompanyDistanceThresholdModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("thresholdExceeded", Boolean, true),
        __metadata("design:type", Boolean)
    ], CompanyDistanceThresholdModel.prototype, "thresholdExceeded", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("type", String),
        __metadata("design:type", String)
    ], CompanyDistanceThresholdModel.prototype, "type", void 0);
    CompanyDistanceThresholdModel = __decorate([
        (0, json2typescript_1.JsonObject)("CompanyDistanceThresholdModel")
    ], CompanyDistanceThresholdModel);
    return CompanyDistanceThresholdModel;
}());
