// avatax/lib/models/CompanyUserDefinedFieldModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CompanyUserDefinedFieldModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 *
 * @export
 * @class CompanyUserDefinedFieldModel
 */
var CompanyUserDefinedFieldModel = exports.CompanyUserDefinedFieldModel = /** @class */ (function () {
    function CompanyUserDefinedFieldModel() {
        /**
         * @type {number}
         * @memberof CompanyUserDefinedFieldModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof CompanyUserDefinedFieldModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof CompanyUserDefinedFieldModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof CompanyUserDefinedFieldModel
         */
        this.friendlyName = undefined;
        /**
         * @type {Enums.UserDefinedFieldDataType}
         * @memberof CompanyUserDefinedFieldModel
         */
        this.dataType = undefined;
        /**
         * @type {Enums.UserDefinedFieldType}
         * @memberof CompanyUserDefinedFieldModel
         */
        this.userDefinedFieldType = undefined;
        /**
         * @type {number}
         * @memberof CompanyUserDefinedFieldModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof CompanyUserDefinedFieldModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof CompanyUserDefinedFieldModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {Date}
         * @memberof CompanyUserDefinedFieldModel
         */
        this.modifiedDate = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], CompanyUserDefinedFieldModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], CompanyUserDefinedFieldModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String),
        __metadata("design:type", String)
    ], CompanyUserDefinedFieldModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("friendlyName", String),
        __metadata("design:type", String)
    ], CompanyUserDefinedFieldModel.prototype, "friendlyName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dataType", Enums.UserDefinedFieldDataTypeConverter, true),
        __metadata("design:type", Number)
    ], CompanyUserDefinedFieldModel.prototype, "dataType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("userDefinedFieldType", Enums.UserDefinedFieldTypeConverter, true),
        __metadata("design:type", Number)
    ], CompanyUserDefinedFieldModel.prototype, "userDefinedFieldType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], CompanyUserDefinedFieldModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CompanyUserDefinedFieldModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], CompanyUserDefinedFieldModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CompanyUserDefinedFieldModel.prototype, "modifiedDate", void 0);
    CompanyUserDefinedFieldModel = __decorate([
        (0, json2typescript_1.JsonObject)("CompanyUserDefinedFieldModel")
    ], CompanyUserDefinedFieldModel);
    return CompanyUserDefinedFieldModel;
}());
