// avatax/lib/models/CountryCoefficientsEntity.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CountryCoefficientsEntity = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents the Country coefficients model, using which tax rules rates can be modified dynamically for CB transaciotns while applying tax rules
in order to reduce the variance for all the transactions at country level.
 * @export
 * @class CountryCoefficientsEntity
 */
var CountryCoefficientsEntity = exports.CountryCoefficientsEntity = /** @class */ (function () {
    function CountryCoefficientsEntity() {
        /**
         * @type {number}
         * @memberof CountryCoefficientsEntity
         */
        this.coefficientsId = undefined;
        /**
         * @type {number}
         * @memberof CountryCoefficientsEntity
         */
        this.accountId = undefined;
        /**
         * @type {number}
         * @memberof CountryCoefficientsEntity
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof CountryCoefficientsEntity
         */
        this.country = undefined;
        /**
         * @type {number}
         * @memberof CountryCoefficientsEntity
         */
        this.coefficient = undefined;
        /**
         * @type {string}
         * @memberof CountryCoefficientsEntity
         */
        this.taxSubTypeId = undefined;
        /**
         * @type {string}
         * @memberof CountryCoefficientsEntity
         */
        this.currencyCode = undefined;
        /**
         * @type {number}
         * @memberof CountryCoefficientsEntity
         */
        this.unitOfBasisId = undefined;
        /**
         * @type {boolean}
         * @memberof CountryCoefficientsEntity
         */
        this.isApplicable = undefined;
        /**
         * @type {Date}
         * @memberof CountryCoefficientsEntity
         */
        this.modifiedDate = undefined;
        /**
         * @type {Date}
         * @memberof CountryCoefficientsEntity
         */
        this.startDate = undefined;
        /**
         * @type {Date}
         * @memberof CountryCoefficientsEntity
         */
        this.endDate = undefined;
        /**
         * @type {number}
         * @memberof CountryCoefficientsEntity
         */
        this.modifiedUserId = undefined;
        /**
         * @type {Date}
         * @memberof CountryCoefficientsEntity
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof CountryCoefficientsEntity
         */
        this.createdUserId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("coefficientsId", Number, true),
        __metadata("design:type", Number)
    ], CountryCoefficientsEntity.prototype, "coefficientsId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountId", Number, true),
        __metadata("design:type", Number)
    ], CountryCoefficientsEntity.prototype, "accountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], CountryCoefficientsEntity.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String),
        __metadata("design:type", String)
    ], CountryCoefficientsEntity.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("coefficient", Number, true),
        __metadata("design:type", Number)
    ], CountryCoefficientsEntity.prototype, "coefficient", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxSubTypeId", String, true),
        __metadata("design:type", String)
    ], CountryCoefficientsEntity.prototype, "taxSubTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("currencyCode", String, true),
        __metadata("design:type", String)
    ], CountryCoefficientsEntity.prototype, "currencyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("unitOfBasisId", Number, true),
        __metadata("design:type", Number)
    ], CountryCoefficientsEntity.prototype, "unitOfBasisId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isApplicable", Boolean, true),
        __metadata("design:type", Boolean)
    ], CountryCoefficientsEntity.prototype, "isApplicable", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CountryCoefficientsEntity.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("startDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CountryCoefficientsEntity.prototype, "startDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CountryCoefficientsEntity.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], CountryCoefficientsEntity.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CountryCoefficientsEntity.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], CountryCoefficientsEntity.prototype, "createdUserId", void 0);
    CountryCoefficientsEntity = __decorate([
        (0, json2typescript_1.JsonObject)("CountryCoefficientsEntity")
    ], CountryCoefficientsEntity);
    return CountryCoefficientsEntity;
}());
