// avatax/lib/models/CreateCertExpressInvitationModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateCertExpressInvitationModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
/**
 * Represents an invitation for a customer to use CertExpress to self-report their own certificates.
This invitation is delivered by your choice of method, or you can present a hyperlink to the user
directly in your connector.  Your customer will be redirected to https://app.certexpress.com/ where
they can follow a step-by-step guide to enter information about their exemption certificates.  The
certificates entered will be recorded and automatically linked to their customer record.
 * @export
 * @class CreateCertExpressInvitationModel
 */
var CreateCertExpressInvitationModel = exports.CreateCertExpressInvitationModel = /** @class */ (function () {
    function CreateCertExpressInvitationModel() {
        /**
         * @type {string}
         * @memberof CreateCertExpressInvitationModel
         */
        this.recipient = undefined;
        /**
         * @type {string}
         * @memberof CreateCertExpressInvitationModel
         */
        this.coverLetterTitle = undefined;
        /**
         * @type {number[]}
         * @memberof CreateCertExpressInvitationModel
         */
        this.exposureZones = undefined;
        /**
         * @type {number[]}
         * @memberof CreateCertExpressInvitationModel
         */
        this.exemptReasons = undefined;
        /**
         * @type {Enums.CertificateRequestDeliveryMethod}
         * @memberof CreateCertExpressInvitationModel
         */
        this.deliveryMethod = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("recipient", String, true),
        __metadata("design:type", String)
    ], CreateCertExpressInvitationModel.prototype, "recipient", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("coverLetterTitle", String, true),
        __metadata("design:type", String)
    ], CreateCertExpressInvitationModel.prototype, "coverLetterTitle", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exposureZones", [Number], true),
        __metadata("design:type", Array)
    ], CreateCertExpressInvitationModel.prototype, "exposureZones", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptReasons", [Number], true),
        __metadata("design:type", Array)
    ], CreateCertExpressInvitationModel.prototype, "exemptReasons", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("deliveryMethod", Enums.CertificateRequestDeliveryMethodConverter, true),
        __metadata("design:type", Number)
    ], CreateCertExpressInvitationModel.prototype, "deliveryMethod", void 0);
    CreateCertExpressInvitationModel = __decorate([
        (0, json2typescript_1.JsonObject)("CreateCertExpressInvitationModel")
    ], CreateCertExpressInvitationModel);
    return CreateCertExpressInvitationModel;
}());
