// avatax/lib/models/CreateMultiDocumentModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateMultiDocumentModel = void 0;
var Enums = require("../enums/index");
var MultiDocumentLineItemModel_1 = require("./MultiDocumentLineItemModel");
var AddressesModel_1 = require("./AddressesModel");
var TransactionParameterModel_1 = require("./TransactionParameterModel");
var TransactionUserDefinedFieldModel_1 = require("./TransactionUserDefinedFieldModel");
var TaxOverrideModel_1 = require("./TaxOverrideModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * A MultiDocument transaction represents a sale or purchase that occurred between more than two companies.
            
A traditional transaction requires exactly two parties: a seller and a buyer.  MultiDocument transactions can
involve a marketplace of vendors, each of which contributes some portion of the final transaction.  Within
a MultiDocument transaction, each individual buyer and seller pair are matched up and converted to a separate
document.  This separation of documents allows each seller to file their taxes separately.
 * @export
 * @class CreateMultiDocumentModel
 */
var CreateMultiDocumentModel = exports.CreateMultiDocumentModel = /** @class */ (function () {
    function CreateMultiDocumentModel() {
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.code = undefined;
        /**
         * @type {MultiDocumentLineItemModel[]}
         * @memberof CreateMultiDocumentModel
         */
        this.lines = undefined;
        /**
         * @type {boolean}
         * @memberof CreateMultiDocumentModel
         */
        this.allowAdjust = undefined;
        /**
         * @type {Enums.DocumentType}
         * @memberof CreateMultiDocumentModel
         */
        this.type = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.companyCode = undefined;
        /**
         * @type {Date}
         * @memberof CreateMultiDocumentModel
         */
        this.date = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.salespersonCode = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.customerCode = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.customerUsageType = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.entityUseCode = undefined;
        /**
         * @type {number}
         * @memberof CreateMultiDocumentModel
         */
        this.discount = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.purchaseOrderNo = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.exemptionNo = undefined;
        /**
         * @type {AddressesModel}
         * @memberof CreateMultiDocumentModel
         */
        this.addresses = undefined;
        /**
         * @type {TransactionParameterModel[]}
         * @memberof CreateMultiDocumentModel
         */
        this.parameters = undefined;
        /**
         * @type {TransactionUserDefinedFieldModel[]}
         * @memberof CreateMultiDocumentModel
         */
        this.userDefinedFields = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.referenceCode = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.reportingLocationCode = undefined;
        /**
         * @type {boolean}
         * @memberof CreateMultiDocumentModel
         */
        this.commit = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.batchCode = undefined;
        /**
         * @type {TaxOverrideModel}
         * @memberof CreateMultiDocumentModel
         */
        this.taxOverride = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.currencyCode = undefined;
        /**
         * @type {Enums.ServiceMode}
         * @memberof CreateMultiDocumentModel
         */
        this.serviceMode = undefined;
        /**
         * @type {number}
         * @memberof CreateMultiDocumentModel
         */
        this.exchangeRate = undefined;
        /**
         * @type {Date}
         * @memberof CreateMultiDocumentModel
         */
        this.exchangeRateEffectiveDate = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.exchangeRateCurrencyCode = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.posLaneCode = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.businessIdentificationNo = undefined;
        /**
         * @type {boolean}
         * @memberof CreateMultiDocumentModel
         */
        this.isSellerImporterOfRecord = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.description = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.email = undefined;
        /**
         * @type {Enums.TaxDebugLevel}
         * @memberof CreateMultiDocumentModel
         */
        this.debugLevel = undefined;
        /**
         * @type {string}
         * @memberof CreateMultiDocumentModel
         */
        this.customerSupplierName = undefined;
        /**
         * @type {number}
         * @memberof CreateMultiDocumentModel
         */
        this.dataSourceId = undefined;
        /**
         * @type {Enums.DeliveryTerms}
         * @memberof CreateMultiDocumentModel
         */
        this.deliveryTerms = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("code", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "code", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lines", [MultiDocumentLineItemModel_1.MultiDocumentLineItemModel]),
        __metadata("design:type", Array)
    ], CreateMultiDocumentModel.prototype, "lines", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("allowAdjust", Boolean, true),
        __metadata("design:type", Boolean)
    ], CreateMultiDocumentModel.prototype, "allowAdjust", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("type", Enums.DocumentTypeConverter, true),
        __metadata("design:type", Number)
    ], CreateMultiDocumentModel.prototype, "type", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyCode", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "companyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("date", dateConverter_1.DateConverter),
        __metadata("design:type", Date)
    ], CreateMultiDocumentModel.prototype, "date", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("salespersonCode", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "salespersonCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerCode", String),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "customerCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerUsageType", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "customerUsageType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("entityUseCode", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "entityUseCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("discount", Number, true),
        __metadata("design:type", Number)
    ], CreateMultiDocumentModel.prototype, "discount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("purchaseOrderNo", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "purchaseOrderNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptionNo", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "exemptionNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("addresses", AddressesModel_1.AddressesModel, true),
        __metadata("design:type", AddressesModel_1.AddressesModel)
    ], CreateMultiDocumentModel.prototype, "addresses", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("parameters", [TransactionParameterModel_1.TransactionParameterModel], true),
        __metadata("design:type", Array)
    ], CreateMultiDocumentModel.prototype, "parameters", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("userDefinedFields", [TransactionUserDefinedFieldModel_1.TransactionUserDefinedFieldModel], true),
        __metadata("design:type", Array)
    ], CreateMultiDocumentModel.prototype, "userDefinedFields", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("referenceCode", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "referenceCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportingLocationCode", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "reportingLocationCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("commit", Boolean, true),
        __metadata("design:type", Boolean)
    ], CreateMultiDocumentModel.prototype, "commit", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("batchCode", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "batchCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxOverride", TaxOverrideModel_1.TaxOverrideModel, true),
        __metadata("design:type", TaxOverrideModel_1.TaxOverrideModel)
    ], CreateMultiDocumentModel.prototype, "taxOverride", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("currencyCode", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "currencyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("serviceMode", Enums.ServiceModeConverter, true),
        __metadata("design:type", Number)
    ], CreateMultiDocumentModel.prototype, "serviceMode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exchangeRate", Number, true),
        __metadata("design:type", Number)
    ], CreateMultiDocumentModel.prototype, "exchangeRate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exchangeRateEffectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CreateMultiDocumentModel.prototype, "exchangeRateEffectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exchangeRateCurrencyCode", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "exchangeRateCurrencyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("posLaneCode", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "posLaneCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("businessIdentificationNo", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "businessIdentificationNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSellerImporterOfRecord", Boolean, true),
        __metadata("design:type", Boolean)
    ], CreateMultiDocumentModel.prototype, "isSellerImporterOfRecord", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("email", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "email", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("debugLevel", Enums.TaxDebugLevelConverter, true),
        __metadata("design:type", Number)
    ], CreateMultiDocumentModel.prototype, "debugLevel", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerSupplierName", String, true),
        __metadata("design:type", String)
    ], CreateMultiDocumentModel.prototype, "customerSupplierName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dataSourceId", Number, true),
        __metadata("design:type", Number)
    ], CreateMultiDocumentModel.prototype, "dataSourceId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("deliveryTerms", Enums.DeliveryTermsConverter, true),
        __metadata("design:type", Number)
    ], CreateMultiDocumentModel.prototype, "deliveryTerms", void 0);
    CreateMultiDocumentModel = __decorate([
        (0, json2typescript_1.JsonObject)("CreateMultiDocumentModel")
    ], CreateMultiDocumentModel);
    return CreateMultiDocumentModel;
}());
