// avatax/lib/models/CustomerAttributeModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerAttributeModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * A Customer's linked attribute denoting what features applied to the customer. A customer can
be linked to multiple customer attributes and vice versa.
 * @export
 * @class CustomerAttributeModel
 */
var CustomerAttributeModel = exports.CustomerAttributeModel = /** @class */ (function () {
    function CustomerAttributeModel() {
        /**
         * @type {number}
         * @memberof CustomerAttributeModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof CustomerAttributeModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof CustomerAttributeModel
         */
        this.description = undefined;
        /**
         * @type {boolean}
         * @memberof CustomerAttributeModel
         */
        this.isSystemCode = undefined;
        /**
         * @type {boolean}
         * @memberof CustomerAttributeModel
         */
        this.isNonDeliver = undefined;
        /**
         * @type {boolean}
         * @memberof CustomerAttributeModel
         */
        this.isChangeable = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], CustomerAttributeModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String, true),
        __metadata("design:type", String)
    ], CustomerAttributeModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], CustomerAttributeModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSystemCode", Boolean, true),
        __metadata("design:type", Boolean)
    ], CustomerAttributeModel.prototype, "isSystemCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isNonDeliver", Boolean, true),
        __metadata("design:type", Boolean)
    ], CustomerAttributeModel.prototype, "isNonDeliver", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isChangeable", Boolean, true),
        __metadata("design:type", Boolean)
    ], CustomerAttributeModel.prototype, "isChangeable", void 0);
    CustomerAttributeModel = __decorate([
        (0, json2typescript_1.JsonObject)("CustomerAttributeModel")
    ], CustomerAttributeModel);
    return CustomerAttributeModel;
}());
