// avatax/lib/models/CycleSafeOptionResultModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CycleSafeOptionResultModel = void 0;
var FrequencyAvailableModel_1 = require("./FrequencyAvailableModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * CycleSafe Option Result
 * @export
 * @class CycleSafeOptionResultModel
 */
var CycleSafeOptionResultModel = exports.CycleSafeOptionResultModel = /** @class */ (function () {
    function CycleSafeOptionResultModel() {
        /**
         * @type {string}
         * @memberof CycleSafeOptionResultModel
         */
        this.taxFormCode = undefined;
        /**
         * @type {boolean}
         * @memberof CycleSafeOptionResultModel
         */
        this.mustCloneFilingCalendar = undefined;
        /**
         * @type {Date}
         * @memberof CycleSafeOptionResultModel
         */
        this.clonedCalendarEffDate = undefined;
        /**
         * @type {Date}
         * @memberof CycleSafeOptionResultModel
         */
        this.expiredCalendarEndDate = undefined;
        /**
         * @type {FrequencyAvailableModel[]}
         * @memberof CycleSafeOptionResultModel
         */
        this.frequenciesAvailable = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxFormCode", String, true),
        __metadata("design:type", String)
    ], CycleSafeOptionResultModel.prototype, "taxFormCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("mustCloneFilingCalendar", Boolean, true),
        __metadata("design:type", Boolean)
    ], CycleSafeOptionResultModel.prototype, "mustCloneFilingCalendar", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("clonedCalendarEffDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CycleSafeOptionResultModel.prototype, "clonedCalendarEffDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("expiredCalendarEndDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], CycleSafeOptionResultModel.prototype, "expiredCalendarEndDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("frequenciesAvailable", [FrequencyAvailableModel_1.FrequencyAvailableModel], true),
        __metadata("design:type", Array)
    ], CycleSafeOptionResultModel.prototype, "frequenciesAvailable", void 0);
    CycleSafeOptionResultModel = __decorate([
        (0, json2typescript_1.JsonObject)("CycleSafeOptionResultModel")
    ], CycleSafeOptionResultModel);
    return CycleSafeOptionResultModel;
}());
