// avatax/lib/models/DataSourceModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataSourceModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Data source object
 * @export
 * @class DataSourceModel
 */
var DataSourceModel = exports.DataSourceModel = /** @class */ (function () {
    function DataSourceModel() {
        /**
         * @type {number}
         * @memberof DataSourceModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof DataSourceModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof DataSourceModel
         */
        this.source = undefined;
        /**
         * @type {string}
         * @memberof DataSourceModel
         */
        this.instance = undefined;
        /**
         * @type {boolean}
         * @memberof DataSourceModel
         */
        this.isEnabled = undefined;
        /**
         * @type {boolean}
         * @memberof DataSourceModel
         */
        this.isSynced = undefined;
        /**
         * @type {boolean}
         * @memberof DataSourceModel
         */
        this.isAuthorized = undefined;
        /**
         * @type {Date}
         * @memberof DataSourceModel
         */
        this.lastSyncedDate = undefined;
        /**
         * @type {number}
         * @memberof DataSourceModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof DataSourceModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof DataSourceModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {Date}
         * @memberof DataSourceModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {Date}
         * @memberof DataSourceModel
         */
        this.deletedDate = undefined;
        /**
         * @type {boolean}
         * @memberof DataSourceModel
         */
        this.recalculate = undefined;
        /**
         * @type {string}
         * @memberof DataSourceModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof DataSourceModel
         */
        this.externalState = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], DataSourceModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number),
        __metadata("design:type", Number)
    ], DataSourceModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("source", String),
        __metadata("design:type", String)
    ], DataSourceModel.prototype, "source", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("instance", String, true),
        __metadata("design:type", String)
    ], DataSourceModel.prototype, "instance", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isEnabled", Boolean, true),
        __metadata("design:type", Boolean)
    ], DataSourceModel.prototype, "isEnabled", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSynced", Boolean, true),
        __metadata("design:type", Boolean)
    ], DataSourceModel.prototype, "isSynced", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isAuthorized", Boolean, true),
        __metadata("design:type", Boolean)
    ], DataSourceModel.prototype, "isAuthorized", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lastSyncedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], DataSourceModel.prototype, "lastSyncedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], DataSourceModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], DataSourceModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], DataSourceModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], DataSourceModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("deletedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], DataSourceModel.prototype, "deletedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("recalculate", Boolean, true),
        __metadata("design:type", Boolean)
    ], DataSourceModel.prototype, "recalculate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String, true),
        __metadata("design:type", String)
    ], DataSourceModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("externalState", String, true),
        __metadata("design:type", String)
    ], DataSourceModel.prototype, "externalState", void 0);
    DataSourceModel = __decorate([
        (0, json2typescript_1.JsonObject)("DataSourceModel")
    ], DataSourceModel);
    return DataSourceModel;
}());
