// avatax/lib/models/DenormalizedJurisModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DenormalizedJurisModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents information about a single legal taxing jurisdiction within a specific Avalara tax region.
 * @export
 * @class DenormalizedJurisModel
 */
var DenormalizedJurisModel = exports.DenormalizedJurisModel = /** @class */ (function () {
    function DenormalizedJurisModel() {
        /**
         * @type {Date}
         * @memberof DenormalizedJurisModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof DenormalizedJurisModel
         */
        this.endDate = undefined;
        /**
         * @type {string}
         * @memberof DenormalizedJurisModel
         */
        this.jurisCode = undefined;
        /**
         * @type {number}
         * @memberof DenormalizedJurisModel
         */
        this.jurisdictionId = undefined;
        /**
         * @type {Enums.JurisdictionType}
         * @memberof DenormalizedJurisModel
         */
        this.jurisType = undefined;
        /**
         * @type {string}
         * @memberof DenormalizedJurisModel
         */
        this.jurisName = undefined;
        /**
         * @type {string}
         * @memberof DenormalizedJurisModel
         */
        this.stateAssignedCode = undefined;
        /**
         * @type {number}
         * @memberof DenormalizedJurisModel
         */
        this.taxAuthorityId = undefined;
        /**
         * @type {string}
         * @memberof DenormalizedJurisModel
         */
        this.state = undefined;
        /**
         * @type {string}
         * @memberof DenormalizedJurisModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof DenormalizedJurisModel
         */
        this.county = undefined;
        /**
         * @type {string}
         * @memberof DenormalizedJurisModel
         */
        this.city = undefined;
        /**
         * @type {boolean}
         * @memberof DenormalizedJurisModel
         */
        this.isAcm = undefined;
        /**
         * @type {boolean}
         * @memberof DenormalizedJurisModel
         */
        this.isSst = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], DenormalizedJurisModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], DenormalizedJurisModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisCode", String, true),
        __metadata("design:type", String)
    ], DenormalizedJurisModel.prototype, "jurisCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionId", Number, true),
        __metadata("design:type", Number)
    ], DenormalizedJurisModel.prototype, "jurisdictionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisType", Enums.JurisdictionTypeConverter, true),
        __metadata("design:type", Number)
    ], DenormalizedJurisModel.prototype, "jurisType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisName", String, true),
        __metadata("design:type", String)
    ], DenormalizedJurisModel.prototype, "jurisName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("stateAssignedCode", String, true),
        __metadata("design:type", String)
    ], DenormalizedJurisModel.prototype, "stateAssignedCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityId", Number, true),
        __metadata("design:type", Number)
    ], DenormalizedJurisModel.prototype, "taxAuthorityId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("state", String, true),
        __metadata("design:type", String)
    ], DenormalizedJurisModel.prototype, "state", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], DenormalizedJurisModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("county", String, true),
        __metadata("design:type", String)
    ], DenormalizedJurisModel.prototype, "county", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("city", String, true),
        __metadata("design:type", String)
    ], DenormalizedJurisModel.prototype, "city", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isAcm", Boolean, true),
        __metadata("design:type", Boolean)
    ], DenormalizedJurisModel.prototype, "isAcm", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSst", Boolean, true),
        __metadata("design:type", Boolean)
    ], DenormalizedJurisModel.prototype, "isSst", void 0);
    DenormalizedJurisModel = __decorate([
        (0, json2typescript_1.JsonObject)("DenormalizedJurisModel")
    ], DenormalizedJurisModel);
    return DenormalizedJurisModel;
}());
