// avatax/lib/models/DeterminationFactorModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeterminationFactorModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * This object represents a single determination factor for a line that is being inspected through the InspectLine API.
 * @export
 * @class DeterminationFactorModel
 */
var DeterminationFactorModel = exports.DeterminationFactorModel = /** @class */ (function () {
    function DeterminationFactorModel() {
        /**
         * @type {string}
         * @memberof DeterminationFactorModel
         */
        this.code = undefined;
        /**
         * @type {string}
         * @memberof DeterminationFactorModel
         */
        this.description = undefined;
        /**
         * @type {string[]}
         * @memberof DeterminationFactorModel
         */
        this.ids = undefined;
        /**
         * @type {string[]}
         * @memberof DeterminationFactorModel
         */
        this.names = undefined;
        /**
         * @type {string}
         * @memberof DeterminationFactorModel
         */
        this.createdBy = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("code", String, true),
        __metadata("design:type", String)
    ], DeterminationFactorModel.prototype, "code", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], DeterminationFactorModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("ids", [String], true),
        __metadata("design:type", Array)
    ], DeterminationFactorModel.prototype, "ids", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("names", [String], true),
        __metadata("design:type", Array)
    ], DeterminationFactorModel.prototype, "names", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdBy", String, true),
        __metadata("design:type", String)
    ], DeterminationFactorModel.prototype, "createdBy", void 0);
    DeterminationFactorModel = __decorate([
        (0, json2typescript_1.JsonObject)("DeterminationFactorModel")
    ], DeterminationFactorModel);
    return DeterminationFactorModel;
}());
