// avatax/lib/models/EcmsModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EcmsModel = void 0;
var Enums = require("../enums/index");
var EcmsDetailModel_1 = require("./EcmsDetailModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Exempt certificate
 * @export
 * @class EcmsModel
 */
var EcmsModel = exports.EcmsModel = /** @class */ (function () {
    function EcmsModel() {
        /**
         * @type {number}
         * @memberof EcmsModel
         */
        this.exemptCertId = undefined;
        /**
         * @type {number}
         * @memberof EcmsModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof EcmsModel
         */
        this.customerCode = undefined;
        /**
         * @type {string}
         * @memberof EcmsModel
         */
        this.customerName = undefined;
        /**
         * @type {string}
         * @memberof EcmsModel
         */
        this.address1 = undefined;
        /**
         * @type {string}
         * @memberof EcmsModel
         */
        this.address2 = undefined;
        /**
         * @type {string}
         * @memberof EcmsModel
         */
        this.address3 = undefined;
        /**
         * @type {string}
         * @memberof EcmsModel
         */
        this.city = undefined;
        /**
         * @type {string}
         * @memberof EcmsModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof EcmsModel
         */
        this.postalCode = undefined;
        /**
         * @type {string}
         * @memberof EcmsModel
         */
        this.country = undefined;
        /**
         * @type {Enums.ExemptCertTypeId}
         * @memberof EcmsModel
         */
        this.exemptCertTypeId = undefined;
        /**
         * @type {string}
         * @memberof EcmsModel
         */
        this.documentRefNo = undefined;
        /**
         * @type {number}
         * @memberof EcmsModel
         */
        this.businessTypeId = undefined;
        /**
         * @type {string}
         * @memberof EcmsModel
         */
        this.businessTypeOtherDescription = undefined;
        /**
         * @type {string}
         * @memberof EcmsModel
         */
        this.exemptReasonId = undefined;
        /**
         * @type {string}
         * @memberof EcmsModel
         */
        this.exemptReasonOtherDescription = undefined;
        /**
         * @type {Date}
         * @memberof EcmsModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {string}
         * @memberof EcmsModel
         */
        this.regionsApplicable = undefined;
        /**
         * @type {Enums.ExemptCertStatusId}
         * @memberof EcmsModel
         */
        this.exemptCertStatusId = undefined;
        /**
         * @type {Date}
         * @memberof EcmsModel
         */
        this.createdDate = undefined;
        /**
         * @type {Date}
         * @memberof EcmsModel
         */
        this.lastTransactionDate = undefined;
        /**
         * @type {Date}
         * @memberof EcmsModel
         */
        this.expiryDate = undefined;
        /**
         * @type {number}
         * @memberof EcmsModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof EcmsModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof EcmsModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {string}
         * @memberof EcmsModel
         */
        this.countryIssued = undefined;
        /**
         * @type {string}
         * @memberof EcmsModel
         */
        this.avaCertId = undefined;
        /**
         * @type {Enums.ExemptCertReviewStatusId}
         * @memberof EcmsModel
         */
        this.exemptCertReviewStatusId = undefined;
        /**
         * @type {EcmsDetailModel[]}
         * @memberof EcmsModel
         */
        this.details = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptCertId", Number),
        __metadata("design:type", Number)
    ], EcmsModel.prototype, "exemptCertId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number),
        __metadata("design:type", Number)
    ], EcmsModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerCode", String),
        __metadata("design:type", String)
    ], EcmsModel.prototype, "customerCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerName", String, true),
        __metadata("design:type", String)
    ], EcmsModel.prototype, "customerName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("address1", String, true),
        __metadata("design:type", String)
    ], EcmsModel.prototype, "address1", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("address2", String, true),
        __metadata("design:type", String)
    ], EcmsModel.prototype, "address2", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("address3", String, true),
        __metadata("design:type", String)
    ], EcmsModel.prototype, "address3", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("city", String, true),
        __metadata("design:type", String)
    ], EcmsModel.prototype, "city", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String),
        __metadata("design:type", String)
    ], EcmsModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("postalCode", String, true),
        __metadata("design:type", String)
    ], EcmsModel.prototype, "postalCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String),
        __metadata("design:type", String)
    ], EcmsModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptCertTypeId", Enums.ExemptCertTypeIdConverter),
        __metadata("design:type", Number)
    ], EcmsModel.prototype, "exemptCertTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("documentRefNo", String, true),
        __metadata("design:type", String)
    ], EcmsModel.prototype, "documentRefNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("businessTypeId", Number),
        __metadata("design:type", Number)
    ], EcmsModel.prototype, "businessTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("businessTypeOtherDescription", String, true),
        __metadata("design:type", String)
    ], EcmsModel.prototype, "businessTypeOtherDescription", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptReasonId", String, true),
        __metadata("design:type", String)
    ], EcmsModel.prototype, "exemptReasonId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptReasonOtherDescription", String, true),
        __metadata("design:type", String)
    ], EcmsModel.prototype, "exemptReasonOtherDescription", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], EcmsModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("regionsApplicable", String),
        __metadata("design:type", String)
    ], EcmsModel.prototype, "regionsApplicable", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptCertStatusId", Enums.ExemptCertStatusIdConverter),
        __metadata("design:type", Number)
    ], EcmsModel.prototype, "exemptCertStatusId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], EcmsModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lastTransactionDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], EcmsModel.prototype, "lastTransactionDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("expiryDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], EcmsModel.prototype, "expiryDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], EcmsModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], EcmsModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], EcmsModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("countryIssued", String),
        __metadata("design:type", String)
    ], EcmsModel.prototype, "countryIssued", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("avaCertId", String, true),
        __metadata("design:type", String)
    ], EcmsModel.prototype, "avaCertId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptCertReviewStatusId", Enums.ExemptCertReviewStatusIdConverter, true),
        __metadata("design:type", Number)
    ], EcmsModel.prototype, "exemptCertReviewStatusId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("details", [EcmsDetailModel_1.EcmsDetailModel], true),
        __metadata("design:type", Array)
    ], EcmsModel.prototype, "details", void 0);
    EcmsModel = __decorate([
        (0, json2typescript_1.JsonObject)("EcmsModel")
    ], EcmsModel);
    return EcmsModel;
}());
