// avatax/lib/models/ErrorTransactionOutputModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ErrorTransactionOutputModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Error Transaction Model
 * @export
 * @class ErrorTransactionOutputModel
 */
var ErrorTransactionOutputModel = exports.ErrorTransactionOutputModel = /** @class */ (function () {
    function ErrorTransactionOutputModel() {
        /**
         * @type {string}
         * @memberof ErrorTransactionOutputModel
         */
        this.errorCode = undefined;
        /**
         * @type {string}
         * @memberof ErrorTransactionOutputModel
         */
        this.errorMessage = undefined;
        /**
         * @type {string}
         * @memberof ErrorTransactionOutputModel
         */
        this.avataxErrorJson = undefined;
        /**
         * @type {string}
         * @memberof ErrorTransactionOutputModel
         */
        this.avataxCreateTransactionJson = undefined;
        /**
         * @type {string}
         * @memberof ErrorTransactionOutputModel
         */
        this.datasource = undefined;
        /**
         * @type {Date}
         * @memberof ErrorTransactionOutputModel
         */
        this.documentDate = undefined;
        /**
         * @type {Date}
         * @memberof ErrorTransactionOutputModel
         */
        this.expiresAt = undefined;
        /**
         * @type {number}
         * @memberof ErrorTransactionOutputModel
         */
        this.amount = undefined;
        /**
         * @type {string}
         * @memberof ErrorTransactionOutputModel
         */
        this.datasourceSource = undefined;
        /**
         * @type {string}
         * @memberof ErrorTransactionOutputModel
         */
        this.shipToCountry = undefined;
        /**
         * @type {string}
         * @memberof ErrorTransactionOutputModel
         */
        this.shipToRegion = undefined;
        /**
         * @type {Enums.DocumentType}
         * @memberof ErrorTransactionOutputModel
         */
        this.documentType = undefined;
        /**
         * @type {string}
         * @memberof ErrorTransactionOutputModel
         */
        this.documentCode = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("errorCode", String, true),
        __metadata("design:type", String)
    ], ErrorTransactionOutputModel.prototype, "errorCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("errorMessage", String, true),
        __metadata("design:type", String)
    ], ErrorTransactionOutputModel.prototype, "errorMessage", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("avataxErrorJson", String, true),
        __metadata("design:type", String)
    ], ErrorTransactionOutputModel.prototype, "avataxErrorJson", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("avataxCreateTransactionJson", String, true),
        __metadata("design:type", String)
    ], ErrorTransactionOutputModel.prototype, "avataxCreateTransactionJson", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("datasource", String, true),
        __metadata("design:type", String)
    ], ErrorTransactionOutputModel.prototype, "datasource", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("documentDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ErrorTransactionOutputModel.prototype, "documentDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("expiresAt", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ErrorTransactionOutputModel.prototype, "expiresAt", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("amount", Number, true),
        __metadata("design:type", Number)
    ], ErrorTransactionOutputModel.prototype, "amount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("datasourceSource", String, true),
        __metadata("design:type", String)
    ], ErrorTransactionOutputModel.prototype, "datasourceSource", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("shipToCountry", String, true),
        __metadata("design:type", String)
    ], ErrorTransactionOutputModel.prototype, "shipToCountry", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("shipToRegion", String, true),
        __metadata("design:type", String)
    ], ErrorTransactionOutputModel.prototype, "shipToRegion", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("documentType", Enums.DocumentTypeConverter),
        __metadata("design:type", Number)
    ], ErrorTransactionOutputModel.prototype, "documentType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("documentCode", String),
        __metadata("design:type", String)
    ], ErrorTransactionOutputModel.prototype, "documentCode", void 0);
    ErrorTransactionOutputModel = __decorate([
        (0, json2typescript_1.JsonObject)("ErrorTransactionOutputModel")
    ], ErrorTransactionOutputModel);
    return ErrorTransactionOutputModel;
}());
