// avatax/lib/models/ErrorDetail.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ErrorDetail = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
/**
 *
 * @export
 * @class ErrorDetail
 */
var ErrorDetail = exports.ErrorDetail = /** @class */ (function () {
    function ErrorDetail() {
        /**
         * @type {Enums.ErrorCodeId}
         * @memberof ErrorDetail
         */
        this.code = undefined;
        /**
         * @type {number}
         * @memberof ErrorDetail
         */
        this.number = undefined;
        /**
         * @type {string}
         * @memberof ErrorDetail
         */
        this.message = undefined;
        /**
         * @type {string}
         * @memberof ErrorDetail
         */
        this.description = undefined;
        /**
         * @type {string}
         * @memberof ErrorDetail
         */
        this.faultCode = undefined;
        /**
         * @type {string}
         * @memberof ErrorDetail
         */
        this.faultSubCode = undefined;
        /**
         * @type {string}
         * @memberof ErrorDetail
         */
        this.helpLink = undefined;
        /**
         * @type {string}
         * @memberof ErrorDetail
         */
        this.refersTo = undefined;
        /**
         * @type {Enums.SeverityLevel}
         * @memberof ErrorDetail
         */
        this.severity = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("code", Enums.ErrorCodeIdConverter, true),
        __metadata("design:type", Number)
    ], ErrorDetail.prototype, "code", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("number", Number, true),
        __metadata("design:type", Number)
    ], ErrorDetail.prototype, "number", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("message", String, true),
        __metadata("design:type", String)
    ], ErrorDetail.prototype, "message", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], ErrorDetail.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("faultCode", String, true),
        __metadata("design:type", String)
    ], ErrorDetail.prototype, "faultCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("faultSubCode", String, true),
        __metadata("design:type", String)
    ], ErrorDetail.prototype, "faultSubCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("helpLink", String, true),
        __metadata("design:type", String)
    ], ErrorDetail.prototype, "helpLink", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("refersTo", String, true),
        __metadata("design:type", String)
    ], ErrorDetail.prototype, "refersTo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("severity", Enums.SeverityLevelConverter, true),
        __metadata("design:type", Number)
    ], ErrorDetail.prototype, "severity", void 0);
    ErrorDetail = __decorate([
        (0, json2typescript_1.JsonObject)("ErrorDetail")
    ], ErrorDetail);
    return ErrorDetail;
}());
