// avatax/lib/models/FilingsTaxDetailsModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilingsTaxDetailsModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Represents a listing of all tax calculation data for filings and for accruing to future filings.
 * @export
 * @class FilingsTaxDetailsModel
 */
var FilingsTaxDetailsModel = exports.FilingsTaxDetailsModel = /** @class */ (function () {
    function FilingsTaxDetailsModel() {
        /**
         * @type {string}
         * @memberof FilingsTaxDetailsModel
         */
        this.taxType = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxDetailsModel
         */
        this.salesAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxDetailsModel
         */
        this.nonTaxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxDetailsModel
         */
        this.taxAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingsTaxDetailsModel
         */
        this.numberOfNights = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxType", String, true),
        __metadata("design:type", String)
    ], FilingsTaxDetailsModel.prototype, "taxType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("salesAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxDetailsModel.prototype, "salesAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxDetailsModel.prototype, "nonTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxDetailsModel.prototype, "taxAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("numberOfNights", Number, true),
        __metadata("design:type", Number)
    ], FilingsTaxDetailsModel.prototype, "numberOfNights", void 0);
    FilingsTaxDetailsModel = __decorate([
        (0, json2typescript_1.JsonObject)("FilingsTaxDetailsModel")
    ], FilingsTaxDetailsModel);
    return FilingsTaxDetailsModel;
}());
