// avatax/lib/models/FilingCalendarModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilingCalendarModel = void 0;
var Enums = require("../enums/index");
var CompanyReturnSettingModel_1 = require("./CompanyReturnSettingModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a commitment to file a tax return on a recurring basis.
Only used if you subscribe to Avalara Returns.
 * @export
 * @class FilingCalendarModel
 */
var FilingCalendarModel = exports.FilingCalendarModel = /** @class */ (function () {
    function FilingCalendarModel() {
        /**
         * @type {number}
         * @memberof FilingCalendarModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof FilingCalendarModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.returnName = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.formCountry = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.formRegion = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.taxFormCode = undefined;
        /**
         * @type {number}
         * @memberof FilingCalendarModel
         */
        this.fiscalYearStartMonth = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.locationCode = undefined;
        /**
         * @type {Enums.OutletTypeId}
         * @memberof FilingCalendarModel
         */
        this.outletTypeId = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.paymentCurrency = undefined;
        /**
         * @type {Enums.FilingFrequencyId}
         * @memberof FilingCalendarModel
         */
        this.filingFrequencyId = undefined;
        /**
         * @type {number}
         * @memberof FilingCalendarModel
         */
        this.months = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.stateRegistrationId = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.localRegistrationId = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.employerIdentificationNumber = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.line1 = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.line2 = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.city = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.postalCode = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.mailingAddressLine1 = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.mailingAddressLine2 = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.mailingAddressCity = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.mailingAddressRegion = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.mailingAddressPostalCode = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.mailingAddressCountry = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.phone = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.customerFilingInstructions = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.legalEntityName = undefined;
        /**
         * @type {Date}
         * @memberof FilingCalendarModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof FilingCalendarModel
         */
        this.endDate = undefined;
        /**
         * @type {Enums.FilingTypeId}
         * @memberof FilingCalendarModel
         */
        this.filingTypeId = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.eFileUsername = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.eFilePassword = undefined;
        /**
         * @type {number}
         * @memberof FilingCalendarModel
         */
        this.prepayPercentage = undefined;
        /**
         * @type {boolean}
         * @memberof FilingCalendarModel
         */
        this.prePaymentRequired = undefined;
        /**
         * @type {number}
         * @memberof FilingCalendarModel
         */
        this.fixedPrepaymentAmount = undefined;
        /**
         * @type {Enums.MatchingTaxType}
         * @memberof FilingCalendarModel
         */
        this.taxTypeId = undefined;
        /**
         * @type {string[]}
         * @memberof FilingCalendarModel
         */
        this.taxTypes = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.internalNotes = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.alSignOn = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.alAccessCode = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.meBusinessCode = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.iaBen = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.ctReg = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.other1Name = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.other1Value = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.other2Name = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.other2Value = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.other3Name = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.other3Value = undefined;
        /**
         * @type {number}
         * @memberof FilingCalendarModel
         */
        this.taxAuthorityId = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.taxAuthorityName = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.taxAuthorityType = undefined;
        /**
         * @type {Date}
         * @memberof FilingCalendarModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof FilingCalendarModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof FilingCalendarModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof FilingCalendarModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.bulkAccountId = undefined;
        /**
         * @type {string}
         * @memberof FilingCalendarModel
         */
        this.siteCode = undefined;
        /**
         * @type {Enums.BulkAccountValidationStatus}
         * @memberof FilingCalendarModel
         */
        this.bulkAccountValidationStatus = undefined;
        /**
         * @type {CompanyReturnSettingModel[]}
         * @memberof FilingCalendarModel
         */
        this.settings = undefined;
        /**
         * @type {number}
         * @memberof FilingCalendarModel
         */
        this.autoLockOverrideDay = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], FilingCalendarModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number),
        __metadata("design:type", Number)
    ], FilingCalendarModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("returnName", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "returnName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("formCountry", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "formCountry", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("formRegion", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "formRegion", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxFormCode", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "taxFormCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("fiscalYearStartMonth", Number, true),
        __metadata("design:type", Number)
    ], FilingCalendarModel.prototype, "fiscalYearStartMonth", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("locationCode", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "locationCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("outletTypeId", Enums.OutletTypeIdConverter, true),
        __metadata("design:type", Number)
    ], FilingCalendarModel.prototype, "outletTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("paymentCurrency", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "paymentCurrency", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingFrequencyId", Enums.FilingFrequencyIdConverter),
        __metadata("design:type", Number)
    ], FilingCalendarModel.prototype, "filingFrequencyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("months", Number, true),
        __metadata("design:type", Number)
    ], FilingCalendarModel.prototype, "months", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("stateRegistrationId", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "stateRegistrationId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("localRegistrationId", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "localRegistrationId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("employerIdentificationNumber", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "employerIdentificationNumber", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line1", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "line1", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line2", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "line2", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("city", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "city", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("postalCode", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "postalCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("mailingAddressLine1", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "mailingAddressLine1", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("mailingAddressLine2", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "mailingAddressLine2", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("mailingAddressCity", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "mailingAddressCity", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("mailingAddressRegion", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "mailingAddressRegion", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("mailingAddressPostalCode", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "mailingAddressPostalCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("mailingAddressCountry", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "mailingAddressCountry", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("phone", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "phone", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerFilingInstructions", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "customerFilingInstructions", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("legalEntityName", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "legalEntityName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter),
        __metadata("design:type", Date)
    ], FilingCalendarModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingCalendarModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingTypeId", Enums.FilingTypeIdConverter, true),
        __metadata("design:type", Number)
    ], FilingCalendarModel.prototype, "filingTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("eFileUsername", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "eFileUsername", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("eFilePassword", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "eFilePassword", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("prepayPercentage", Number, true),
        __metadata("design:type", Number)
    ], FilingCalendarModel.prototype, "prepayPercentage", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("prePaymentRequired", Boolean, true),
        __metadata("design:type", Boolean)
    ], FilingCalendarModel.prototype, "prePaymentRequired", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("fixedPrepaymentAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingCalendarModel.prototype, "fixedPrepaymentAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxTypeId", Enums.MatchingTaxTypeConverter),
        __metadata("design:type", Number)
    ], FilingCalendarModel.prototype, "taxTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxTypes", [String], true),
        __metadata("design:type", Array)
    ], FilingCalendarModel.prototype, "taxTypes", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("internalNotes", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "internalNotes", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("alSignOn", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "alSignOn", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("alAccessCode", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "alAccessCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("meBusinessCode", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "meBusinessCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("iaBen", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "iaBen", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("ctReg", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "ctReg", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("other1Name", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "other1Name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("other1Value", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "other1Value", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("other2Name", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "other2Name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("other2Value", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "other2Value", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("other3Name", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "other3Name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("other3Value", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "other3Value", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityId", Number, true),
        __metadata("design:type", Number)
    ], FilingCalendarModel.prototype, "taxAuthorityId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityName", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "taxAuthorityName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityType", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "taxAuthorityType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingCalendarModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], FilingCalendarModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingCalendarModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], FilingCalendarModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("bulkAccountId", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "bulkAccountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("siteCode", String, true),
        __metadata("design:type", String)
    ], FilingCalendarModel.prototype, "siteCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("bulkAccountValidationStatus", Enums.BulkAccountValidationStatusConverter, true),
        __metadata("design:type", Number)
    ], FilingCalendarModel.prototype, "bulkAccountValidationStatus", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("settings", [CompanyReturnSettingModel_1.CompanyReturnSettingModel], true),
        __metadata("design:type", Array)
    ], FilingCalendarModel.prototype, "settings", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("autoLockOverrideDay", Number, true),
        __metadata("design:type", Number)
    ], FilingCalendarModel.prototype, "autoLockOverrideDay", void 0);
    FilingCalendarModel = __decorate([
        (0, json2typescript_1.JsonObject)("FilingCalendarModel")
    ], FilingCalendarModel);
    return FilingCalendarModel;
}());
