// avatax/lib/models/FilingModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilingModel = void 0;
var Enums = require("../enums/index");
var FilingRegionModel_1 = require("./FilingRegionModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a listing of all tax calculation data for filings and for accruing to future filings.
 * @export
 * @class FilingModel
 */
var FilingModel = exports.FilingModel = /** @class */ (function () {
    function FilingModel() {
        /**
         * @type {number}
         * @memberof FilingModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof FilingModel
         */
        this.companyId = undefined;
        /**
         * @type {number}
         * @memberof FilingModel
         */
        this.month = undefined;
        /**
         * @type {number}
         * @memberof FilingModel
         */
        this.year = undefined;
        /**
         * @type {Enums.WorksheetTypeId}
         * @memberof FilingModel
         */
        this.type = undefined;
        /**
         * @type {Date}
         * @memberof FilingModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof FilingModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof FilingModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof FilingModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {FilingRegionModel[]}
         * @memberof FilingModel
         */
        this.filingRegions = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], FilingModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], FilingModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("month", Number, true),
        __metadata("design:type", Number)
    ], FilingModel.prototype, "month", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("year", Number, true),
        __metadata("design:type", Number)
    ], FilingModel.prototype, "year", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("type", Enums.WorksheetTypeIdConverter, true),
        __metadata("design:type", Number)
    ], FilingModel.prototype, "type", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], FilingModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], FilingModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingRegions", [FilingRegionModel_1.FilingRegionModel], true),
        __metadata("design:type", Array)
    ], FilingModel.prototype, "filingRegions", void 0);
    FilingModel = __decorate([
        (0, json2typescript_1.JsonObject)("FilingModel")
    ], FilingModel);
    return FilingModel;
}());
