// avatax/lib/models/FilingRequestDataModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilingRequestDataModel = void 0;
var Enums = require("../enums/index");
var FilingAnswerModel_1 = require("./FilingAnswerModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a commitment to file a tax return on a recurring basis.
Only used if you subscribe to Avalara Returns.
 * @export
 * @class FilingRequestDataModel
 */
var FilingRequestDataModel = exports.FilingRequestDataModel = /** @class */ (function () {
    function FilingRequestDataModel() {
        /**
         * @type {number}
         * @memberof FilingRequestDataModel
         */
        this.companyReturnId = undefined;
        /**
         * @type {string}
         * @memberof FilingRequestDataModel
         */
        this.returnName = undefined;
        /**
         * @type {string}
         * @memberof FilingRequestDataModel
         */
        this.taxFormCode = undefined;
        /**
         * @type {Enums.FilingFrequencyId}
         * @memberof FilingRequestDataModel
         */
        this.filingFrequencyId = undefined;
        /**
         * @type {string}
         * @memberof FilingRequestDataModel
         */
        this.registrationId = undefined;
        /**
         * @type {number}
         * @memberof FilingRequestDataModel
         */
        this.months = undefined;
        /**
         * @type {number}
         * @memberof FilingRequestDataModel
         */
        this.fiscalYearStartMonth = undefined;
        /**
         * @type {Enums.MatchingTaxType}
         * @memberof FilingRequestDataModel
         */
        this.taxTypeId = undefined;
        /**
         * @type {string}
         * @memberof FilingRequestDataModel
         */
        this.locationCode = undefined;
        /**
         * @type {Date}
         * @memberof FilingRequestDataModel
         */
        this.effDate = undefined;
        /**
         * @type {Date}
         * @memberof FilingRequestDataModel
         */
        this.endDate = undefined;
        /**
         * @type {boolean}
         * @memberof FilingRequestDataModel
         */
        this.isClone = undefined;
        /**
         * @type {number}
         * @memberof FilingRequestDataModel
         */
        this.autoLockOverrideDay = undefined;
        /**
         * @type {string}
         * @memberof FilingRequestDataModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof FilingRequestDataModel
         */
        this.region = undefined;
        /**
         * @type {number}
         * @memberof FilingRequestDataModel
         */
        this.taxAuthorityId = undefined;
        /**
         * @type {string}
         * @memberof FilingRequestDataModel
         */
        this.taxAuthorityName = undefined;
        /**
         * @type {FilingAnswerModel[]}
         * @memberof FilingRequestDataModel
         */
        this.answers = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyReturnId", Number, true),
        __metadata("design:type", Number)
    ], FilingRequestDataModel.prototype, "companyReturnId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("returnName", String, true),
        __metadata("design:type", String)
    ], FilingRequestDataModel.prototype, "returnName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxFormCode", String, true),
        __metadata("design:type", String)
    ], FilingRequestDataModel.prototype, "taxFormCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingFrequencyId", Enums.FilingFrequencyIdConverter),
        __metadata("design:type", Number)
    ], FilingRequestDataModel.prototype, "filingFrequencyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("registrationId", String, true),
        __metadata("design:type", String)
    ], FilingRequestDataModel.prototype, "registrationId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("months", Number),
        __metadata("design:type", Number)
    ], FilingRequestDataModel.prototype, "months", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("fiscalYearStartMonth", Number, true),
        __metadata("design:type", Number)
    ], FilingRequestDataModel.prototype, "fiscalYearStartMonth", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxTypeId", Enums.MatchingTaxTypeConverter, true),
        __metadata("design:type", Number)
    ], FilingRequestDataModel.prototype, "taxTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("locationCode", String, true),
        __metadata("design:type", String)
    ], FilingRequestDataModel.prototype, "locationCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effDate", dateConverter_1.DateConverter),
        __metadata("design:type", Date)
    ], FilingRequestDataModel.prototype, "effDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingRequestDataModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isClone", Boolean, true),
        __metadata("design:type", Boolean)
    ], FilingRequestDataModel.prototype, "isClone", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("autoLockOverrideDay", Number, true),
        __metadata("design:type", Number)
    ], FilingRequestDataModel.prototype, "autoLockOverrideDay", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], FilingRequestDataModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], FilingRequestDataModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityId", Number, true),
        __metadata("design:type", Number)
    ], FilingRequestDataModel.prototype, "taxAuthorityId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityName", String, true),
        __metadata("design:type", String)
    ], FilingRequestDataModel.prototype, "taxAuthorityName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("answers", [FilingAnswerModel_1.FilingAnswerModel], true),
        __metadata("design:type", Array)
    ], FilingRequestDataModel.prototype, "answers", void 0);
    FilingRequestDataModel = __decorate([
        (0, json2typescript_1.JsonObject)("FilingRequestDataModel")
    ], FilingRequestDataModel);
    return FilingRequestDataModel;
}());
