// avatax/lib/models/FilingRegionModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilingRegionModel = void 0;
var Enums = require("../enums/index");
var FilingReturnModel_1 = require("./FilingReturnModel");
var FilingsCheckupSuggestedFormModel_1 = require("./FilingsCheckupSuggestedFormModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Regions
 * @export
 * @class FilingRegionModel
 */
var FilingRegionModel = exports.FilingRegionModel = /** @class */ (function () {
    function FilingRegionModel() {
        /**
         * @type {number}
         * @memberof FilingRegionModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof FilingRegionModel
         */
        this.filingId = undefined;
        /**
         * @type {string}
         * @memberof FilingRegionModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof FilingRegionModel
         */
        this.region = undefined;
        /**
         * @type {number}
         * @memberof FilingRegionModel
         */
        this.salesAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingRegionModel
         */
        this.taxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingRegionModel
         */
        this.taxAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingRegionModel
         */
        this.taxDueAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingRegionModel
         */
        this.collectAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingRegionModel
         */
        this.totalRemittanceAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingRegionModel
         */
        this.nonTaxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingRegionModel
         */
        this.consumerUseTaxAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingRegionModel
         */
        this.consumerUseNonTaxableAmount = undefined;
        /**
         * @type {number}
         * @memberof FilingRegionModel
         */
        this.consumerUseTaxableAmount = undefined;
        /**
         * @type {Date}
         * @memberof FilingRegionModel
         */
        this.approveDate = undefined;
        /**
         * @type {Date}
         * @memberof FilingRegionModel
         */
        this.startDate = undefined;
        /**
         * @type {Date}
         * @memberof FilingRegionModel
         */
        this.endDate = undefined;
        /**
         * @type {boolean}
         * @memberof FilingRegionModel
         */
        this.hasNexus = undefined;
        /**
         * @type {Enums.FilingStatusId}
         * @memberof FilingRegionModel
         */
        this.status = undefined;
        /**
         * @type {FilingReturnModel[]}
         * @memberof FilingRegionModel
         */
        this.returns = undefined;
        /**
         * @type {FilingsCheckupSuggestedFormModel[]}
         * @memberof FilingRegionModel
         */
        this.suggestReturns = undefined;
        /**
         * @type {Date}
         * @memberof FilingRegionModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof FilingRegionModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof FilingRegionModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof FilingRegionModel
         */
        this.modifiedUserId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], FilingRegionModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingId", Number, true),
        __metadata("design:type", Number)
    ], FilingRegionModel.prototype, "filingId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], FilingRegionModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], FilingRegionModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("salesAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingRegionModel.prototype, "salesAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingRegionModel.prototype, "taxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingRegionModel.prototype, "taxAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxDueAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingRegionModel.prototype, "taxDueAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("collectAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingRegionModel.prototype, "collectAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalRemittanceAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingRegionModel.prototype, "totalRemittanceAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingRegionModel.prototype, "nonTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("consumerUseTaxAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingRegionModel.prototype, "consumerUseTaxAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("consumerUseNonTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingRegionModel.prototype, "consumerUseNonTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("consumerUseTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], FilingRegionModel.prototype, "consumerUseTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("approveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingRegionModel.prototype, "approveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("startDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingRegionModel.prototype, "startDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingRegionModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hasNexus", Boolean, true),
        __metadata("design:type", Boolean)
    ], FilingRegionModel.prototype, "hasNexus", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("status", Enums.FilingStatusIdConverter, true),
        __metadata("design:type", Number)
    ], FilingRegionModel.prototype, "status", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("returns", [FilingReturnModel_1.FilingReturnModel], true),
        __metadata("design:type", Array)
    ], FilingRegionModel.prototype, "returns", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("suggestReturns", [FilingsCheckupSuggestedFormModel_1.FilingsCheckupSuggestedFormModel], true),
        __metadata("design:type", Array)
    ], FilingRegionModel.prototype, "suggestReturns", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingRegionModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], FilingRegionModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FilingRegionModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], FilingRegionModel.prototype, "modifiedUserId", void 0);
    FilingRegionModel = __decorate([
        (0, json2typescript_1.JsonObject)("FilingRegionModel")
    ], FilingRegionModel);
    return FilingRegionModel;
}());
