// avatax/lib/models/FilingsCheckupAuthorityModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilingsCheckupAuthorityModel = void 0;
var FilingsCheckupSuggestedFormModel_1 = require("./FilingsCheckupSuggestedFormModel");
var json2typescript_1 = require("json2typescript");
/**
 * Cycle Safe Expiration results.
 * @export
 * @class FilingsCheckupAuthorityModel
 */
var FilingsCheckupAuthorityModel = exports.FilingsCheckupAuthorityModel = /** @class */ (function () {
    function FilingsCheckupAuthorityModel() {
        /**
         * @type {number}
         * @memberof FilingsCheckupAuthorityModel
         */
        this.taxAuthorityId = undefined;
        /**
         * @type {string}
         * @memberof FilingsCheckupAuthorityModel
         */
        this.locationCode = undefined;
        /**
         * @type {string}
         * @memberof FilingsCheckupAuthorityModel
         */
        this.taxAuthorityName = undefined;
        /**
         * @type {number}
         * @memberof FilingsCheckupAuthorityModel
         */
        this.taxAuthorityTypeId = undefined;
        /**
         * @type {number}
         * @memberof FilingsCheckupAuthorityModel
         */
        this.jurisdictionId = undefined;
        /**
         * @type {number}
         * @memberof FilingsCheckupAuthorityModel
         */
        this.tax = undefined;
        /**
         * @type {string}
         * @memberof FilingsCheckupAuthorityModel
         */
        this.taxTypeId = undefined;
        /**
         * @type {FilingsCheckupSuggestedFormModel[]}
         * @memberof FilingsCheckupAuthorityModel
         */
        this.suggestedForms = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityId", Number, true),
        __metadata("design:type", Number)
    ], FilingsCheckupAuthorityModel.prototype, "taxAuthorityId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("locationCode", String, true),
        __metadata("design:type", String)
    ], FilingsCheckupAuthorityModel.prototype, "locationCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityName", String, true),
        __metadata("design:type", String)
    ], FilingsCheckupAuthorityModel.prototype, "taxAuthorityName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityTypeId", Number, true),
        __metadata("design:type", Number)
    ], FilingsCheckupAuthorityModel.prototype, "taxAuthorityTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionId", Number, true),
        __metadata("design:type", Number)
    ], FilingsCheckupAuthorityModel.prototype, "jurisdictionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("tax", Number, true),
        __metadata("design:type", Number)
    ], FilingsCheckupAuthorityModel.prototype, "tax", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxTypeId", String, true),
        __metadata("design:type", String)
    ], FilingsCheckupAuthorityModel.prototype, "taxTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("suggestedForms", [FilingsCheckupSuggestedFormModel_1.FilingsCheckupSuggestedFormModel], true),
        __metadata("design:type", Array)
    ], FilingsCheckupAuthorityModel.prototype, "suggestedForms", void 0);
    FilingsCheckupAuthorityModel = __decorate([
        (0, json2typescript_1.JsonObject)("FilingsCheckupAuthorityModel")
    ], FilingsCheckupAuthorityModel);
    return FilingsCheckupAuthorityModel;
}());
