// avatax/lib/models/FirmClientLinkageOutputModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FirmClientLinkageOutputModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Account Linkage output model
 * @export
 * @class FirmClientLinkageOutputModel
 */
var FirmClientLinkageOutputModel = exports.FirmClientLinkageOutputModel = /** @class */ (function () {
    function FirmClientLinkageOutputModel() {
        /**
         * @type {number}
         * @memberof FirmClientLinkageOutputModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof FirmClientLinkageOutputModel
         */
        this.firmAccountId = undefined;
        /**
         * @type {string}
         * @memberof FirmClientLinkageOutputModel
         */
        this.firmAccountName = undefined;
        /**
         * @type {number}
         * @memberof FirmClientLinkageOutputModel
         */
        this.clientAccountId = undefined;
        /**
         * @type {string}
         * @memberof FirmClientLinkageOutputModel
         */
        this.clientAccountName = undefined;
        /**
         * @type {Date}
         * @memberof FirmClientLinkageOutputModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof FirmClientLinkageOutputModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof FirmClientLinkageOutputModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof FirmClientLinkageOutputModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {Enums.FirmClientLinkageStatus}
         * @memberof FirmClientLinkageOutputModel
         */
        this.status = undefined;
        /**
         * @type {boolean}
         * @memberof FirmClientLinkageOutputModel
         */
        this.isDeleted = undefined;
        /**
         * @type {string}
         * @memberof FirmClientLinkageOutputModel
         */
        this.firmContactName = undefined;
        /**
         * @type {string}
         * @memberof FirmClientLinkageOutputModel
         */
        this.firmContactEmail = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], FirmClientLinkageOutputModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("firmAccountId", Number, true),
        __metadata("design:type", Number)
    ], FirmClientLinkageOutputModel.prototype, "firmAccountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("firmAccountName", String, true),
        __metadata("design:type", String)
    ], FirmClientLinkageOutputModel.prototype, "firmAccountName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("clientAccountId", Number, true),
        __metadata("design:type", Number)
    ], FirmClientLinkageOutputModel.prototype, "clientAccountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("clientAccountName", String, true),
        __metadata("design:type", String)
    ], FirmClientLinkageOutputModel.prototype, "clientAccountName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FirmClientLinkageOutputModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], FirmClientLinkageOutputModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FirmClientLinkageOutputModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], FirmClientLinkageOutputModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("status", Enums.FirmClientLinkageStatusConverter, true),
        __metadata("design:type", Number)
    ], FirmClientLinkageOutputModel.prototype, "status", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isDeleted", Boolean, true),
        __metadata("design:type", Boolean)
    ], FirmClientLinkageOutputModel.prototype, "isDeleted", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("firmContactName", String, true),
        __metadata("design:type", String)
    ], FirmClientLinkageOutputModel.prototype, "firmContactName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("firmContactEmail", String, true),
        __metadata("design:type", String)
    ], FirmClientLinkageOutputModel.prototype, "firmContactEmail", void 0);
    FirmClientLinkageOutputModel = __decorate([
        (0, json2typescript_1.JsonObject)("FirmClientLinkageOutputModel")
    ], FirmClientLinkageOutputModel);
    return FirmClientLinkageOutputModel;
}());
