// avatax/lib/models/FreeTrialRequestModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FreeTrialRequestModel = void 0;
var CompanyAddress_1 = require("./CompanyAddress");
var json2typescript_1 = require("json2typescript");
/**
 * Represents a request for a free trial account for AvaTax.
Free trial accounts are only available on the Sandbox environment.
 * @export
 * @class FreeTrialRequestModel
 */
var FreeTrialRequestModel = exports.FreeTrialRequestModel = /** @class */ (function () {
    function FreeTrialRequestModel() {
        /**
         * @type {string}
         * @memberof FreeTrialRequestModel
         */
        this.firstName = undefined;
        /**
         * @type {string}
         * @memberof FreeTrialRequestModel
         */
        this.lastName = undefined;
        /**
         * @type {string}
         * @memberof FreeTrialRequestModel
         */
        this.email = undefined;
        /**
         * @type {string}
         * @memberof FreeTrialRequestModel
         */
        this.company = undefined;
        /**
         * @type {string}
         * @memberof FreeTrialRequestModel
         */
        this.phone = undefined;
        /**
         * @type {string}
         * @memberof FreeTrialRequestModel
         */
        this.campaign = undefined;
        /**
         * @type {CompanyAddress}
         * @memberof FreeTrialRequestModel
         */
        this.companyAddress = undefined;
        /**
         * @type {string}
         * @memberof FreeTrialRequestModel
         */
        this.website = undefined;
        /**
         * @type {boolean}
         * @memberof FreeTrialRequestModel
         */
        this.haveReadAvalaraTermsAndConditions = undefined;
        /**
         * @type {boolean}
         * @memberof FreeTrialRequestModel
         */
        this.acceptAvalaraTermsAndConditions = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("firstName", String),
        __metadata("design:type", String)
    ], FreeTrialRequestModel.prototype, "firstName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lastName", String),
        __metadata("design:type", String)
    ], FreeTrialRequestModel.prototype, "lastName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("email", String),
        __metadata("design:type", String)
    ], FreeTrialRequestModel.prototype, "email", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("company", String),
        __metadata("design:type", String)
    ], FreeTrialRequestModel.prototype, "company", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("phone", String),
        __metadata("design:type", String)
    ], FreeTrialRequestModel.prototype, "phone", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("campaign", String, true),
        __metadata("design:type", String)
    ], FreeTrialRequestModel.prototype, "campaign", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyAddress", CompanyAddress_1.CompanyAddress),
        __metadata("design:type", CompanyAddress_1.CompanyAddress)
    ], FreeTrialRequestModel.prototype, "companyAddress", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("website", String, true),
        __metadata("design:type", String)
    ], FreeTrialRequestModel.prototype, "website", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("haveReadAvalaraTermsAndConditions", Boolean),
        __metadata("design:type", Boolean)
    ], FreeTrialRequestModel.prototype, "haveReadAvalaraTermsAndConditions", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("acceptAvalaraTermsAndConditions", Boolean),
        __metadata("design:type", Boolean)
    ], FreeTrialRequestModel.prototype, "acceptAvalaraTermsAndConditions", void 0);
    FreeTrialRequestModel = __decorate([
        (0, json2typescript_1.JsonObject)("FreeTrialRequestModel")
    ], FreeTrialRequestModel);
    return FreeTrialRequestModel;
}());
