// avatax/lib/models/FundingConfigurationModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FundingConfigurationModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Status of an Avalara Managed Returns funding configuration for a company
 * @export
 * @class FundingConfigurationModel
 */
var FundingConfigurationModel = exports.FundingConfigurationModel = /** @class */ (function () {
    function FundingConfigurationModel() {
        /**
         * @type {number}
         * @memberof FundingConfigurationModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof FundingConfigurationModel
         */
        this.systemType = undefined;
        /**
         * @type {string}
         * @memberof FundingConfigurationModel
         */
        this.currency = undefined;
        /**
         * @type {boolean}
         * @memberof FundingConfigurationModel
         */
        this.isFundingSetup = undefined;
        /**
         * @type {string}
         * @memberof FundingConfigurationModel
         */
        this.fundingMethod = undefined;
        /**
         * @type {Date}
         * @memberof FundingConfigurationModel
         */
        this.lastUpdated = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], FundingConfigurationModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("systemType", String, true),
        __metadata("design:type", String)
    ], FundingConfigurationModel.prototype, "systemType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("currency", String, true),
        __metadata("design:type", String)
    ], FundingConfigurationModel.prototype, "currency", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isFundingSetup", Boolean, true),
        __metadata("design:type", Boolean)
    ], FundingConfigurationModel.prototype, "isFundingSetup", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("fundingMethod", String, true),
        __metadata("design:type", String)
    ], FundingConfigurationModel.prototype, "fundingMethod", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lastUpdated", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FundingConfigurationModel.prototype, "lastUpdated", void 0);
    FundingConfigurationModel = __decorate([
        (0, json2typescript_1.JsonObject)("FundingConfigurationModel")
    ], FundingConfigurationModel);
    return FundingConfigurationModel;
}());
