// avatax/lib/models/FundingStatusModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FundingStatusModel = void 0;
var FundingESignMethodReturn_1 = require("./FundingESignMethodReturn");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Status of an Avalara Managed Returns funding configuration for a company
 * @export
 * @class FundingStatusModel
 */
var FundingStatusModel = exports.FundingStatusModel = /** @class */ (function () {
    function FundingStatusModel() {
        /**
         * @type {number}
         * @memberof FundingStatusModel
         */
        this.requestId = undefined;
        /**
         * @type {number}
         * @memberof FundingStatusModel
         */
        this.subledgerProfileID = undefined;
        /**
         * @type {string}
         * @memberof FundingStatusModel
         */
        this.companyID = undefined;
        /**
         * @type {string}
         * @memberof FundingStatusModel
         */
        this.domain = undefined;
        /**
         * @type {string}
         * @memberof FundingStatusModel
         */
        this.recipient = undefined;
        /**
         * @type {string}
         * @memberof FundingStatusModel
         */
        this.sender = undefined;
        /**
         * @type {string}
         * @memberof FundingStatusModel
         */
        this.documentKey = undefined;
        /**
         * @type {string}
         * @memberof FundingStatusModel
         */
        this.documentType = undefined;
        /**
         * @type {string}
         * @memberof FundingStatusModel
         */
        this.documentName = undefined;
        /**
         * @type {FundingESignMethodReturn}
         * @memberof FundingStatusModel
         */
        this.methodReturn = undefined;
        /**
         * @type {string}
         * @memberof FundingStatusModel
         */
        this.status = undefined;
        /**
         * @type {string}
         * @memberof FundingStatusModel
         */
        this.errorMessage = undefined;
        /**
         * @type {Date}
         * @memberof FundingStatusModel
         */
        this.lastPolled = undefined;
        /**
         * @type {Date}
         * @memberof FundingStatusModel
         */
        this.lastSigned = undefined;
        /**
         * @type {Date}
         * @memberof FundingStatusModel
         */
        this.lastActivated = undefined;
        /**
         * @type {number}
         * @memberof FundingStatusModel
         */
        this.templateRequestId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("requestId", Number, true),
        __metadata("design:type", Number)
    ], FundingStatusModel.prototype, "requestId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("subledgerProfileID", Number, true),
        __metadata("design:type", Number)
    ], FundingStatusModel.prototype, "subledgerProfileID", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyID", String, true),
        __metadata("design:type", String)
    ], FundingStatusModel.prototype, "companyID", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("domain", String, true),
        __metadata("design:type", String)
    ], FundingStatusModel.prototype, "domain", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("recipient", String, true),
        __metadata("design:type", String)
    ], FundingStatusModel.prototype, "recipient", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("sender", String, true),
        __metadata("design:type", String)
    ], FundingStatusModel.prototype, "sender", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("documentKey", String, true),
        __metadata("design:type", String)
    ], FundingStatusModel.prototype, "documentKey", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("documentType", String, true),
        __metadata("design:type", String)
    ], FundingStatusModel.prototype, "documentType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("documentName", String, true),
        __metadata("design:type", String)
    ], FundingStatusModel.prototype, "documentName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("methodReturn", FundingESignMethodReturn_1.FundingESignMethodReturn, true),
        __metadata("design:type", FundingESignMethodReturn_1.FundingESignMethodReturn)
    ], FundingStatusModel.prototype, "methodReturn", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("status", String, true),
        __metadata("design:type", String)
    ], FundingStatusModel.prototype, "status", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("errorMessage", String, true),
        __metadata("design:type", String)
    ], FundingStatusModel.prototype, "errorMessage", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lastPolled", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FundingStatusModel.prototype, "lastPolled", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lastSigned", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FundingStatusModel.prototype, "lastSigned", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lastActivated", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], FundingStatusModel.prototype, "lastActivated", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("templateRequestId", Number, true),
        __metadata("design:type", Number)
    ], FundingStatusModel.prototype, "templateRequestId", void 0);
    FundingStatusModel = __decorate([
        (0, json2typescript_1.JsonObject)("FundingStatusModel")
    ], FundingStatusModel);
    return FundingStatusModel;
}());
