// avatax/lib/models/IsoRegionModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IsoRegionModel = void 0;
var IsoLocalizedName_1 = require("./IsoLocalizedName");
var json2typescript_1 = require("json2typescript");
/**
 * Represents a region, province, or state within a country
 * @export
 * @class IsoRegionModel
 */
var IsoRegionModel = exports.IsoRegionModel = /** @class */ (function () {
    function IsoRegionModel() {
        /**
         * @type {string}
         * @memberof IsoRegionModel
         */
        this.countryCode = undefined;
        /**
         * @type {string}
         * @memberof IsoRegionModel
         */
        this.code = undefined;
        /**
         * @type {string}
         * @memberof IsoRegionModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof IsoRegionModel
         */
        this.classification = undefined;
        /**
         * @type {boolean}
         * @memberof IsoRegionModel
         */
        this.streamlinedSalesTax = undefined;
        /**
         * @type {IsoLocalizedName[]}
         * @memberof IsoRegionModel
         */
        this.localizedNames = undefined;
        /**
         * @type {boolean}
         * @memberof IsoRegionModel
         */
        this.isRegionTaxable = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("countryCode", String, true),
        __metadata("design:type", String)
    ], IsoRegionModel.prototype, "countryCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("code", String, true),
        __metadata("design:type", String)
    ], IsoRegionModel.prototype, "code", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String, true),
        __metadata("design:type", String)
    ], IsoRegionModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("classification", String, true),
        __metadata("design:type", String)
    ], IsoRegionModel.prototype, "classification", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("streamlinedSalesTax", Boolean, true),
        __metadata("design:type", Boolean)
    ], IsoRegionModel.prototype, "streamlinedSalesTax", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("localizedNames", [IsoLocalizedName_1.IsoLocalizedName], true),
        __metadata("design:type", Array)
    ], IsoRegionModel.prototype, "localizedNames", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isRegionTaxable", Boolean, true),
        __metadata("design:type", Boolean)
    ], IsoRegionModel.prototype, "isRegionTaxable", void 0);
    IsoRegionModel = __decorate([
        (0, json2typescript_1.JsonObject)("IsoRegionModel")
    ], IsoRegionModel);
    return IsoRegionModel;
}());
