// avatax/lib/models/ItemRestrictionOutputModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemRestrictionOutputModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a parameter associated with an item.
 * @export
 * @class ItemRestrictionOutputModel
 */
var ItemRestrictionOutputModel = exports.ItemRestrictionOutputModel = /** @class */ (function () {
    function ItemRestrictionOutputModel() {
        /**
         * @type {string}
         * @memberof ItemRestrictionOutputModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionOutputModel
         */
        this.countryOfImport = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionOutputModel
         */
        this.countryOfExport = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionOutputModel
         */
        this.countryOfManufacture = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionOutputModel
         */
        this.hsCode = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionOutputModel
         */
        this.itemCode = undefined;
        /**
         * @type {number}
         * @memberof ItemRestrictionOutputModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionOutputModel
         */
        this.restrictionType = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionOutputModel
         */
        this.regulation = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionOutputModel
         */
        this.governmentAgency = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionOutputModel
         */
        this.complianceMessage = undefined;
        /**
         * @type {Date}
         * @memberof ItemRestrictionOutputModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof ItemRestrictionOutputModel
         */
        this.createdUserId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", String, true),
        __metadata("design:type", String)
    ], ItemRestrictionOutputModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("countryOfImport", String, true),
        __metadata("design:type", String)
    ], ItemRestrictionOutputModel.prototype, "countryOfImport", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("countryOfExport", String, true),
        __metadata("design:type", String)
    ], ItemRestrictionOutputModel.prototype, "countryOfExport", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("countryOfManufacture", String, true),
        __metadata("design:type", String)
    ], ItemRestrictionOutputModel.prototype, "countryOfManufacture", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hsCode", String, true),
        __metadata("design:type", String)
    ], ItemRestrictionOutputModel.prototype, "hsCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("itemCode", String, true),
        __metadata("design:type", String)
    ], ItemRestrictionOutputModel.prototype, "itemCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], ItemRestrictionOutputModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("restrictionType", String, true),
        __metadata("design:type", String)
    ], ItemRestrictionOutputModel.prototype, "restrictionType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("regulation", String, true),
        __metadata("design:type", String)
    ], ItemRestrictionOutputModel.prototype, "regulation", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("governmentAgency", String, true),
        __metadata("design:type", String)
    ], ItemRestrictionOutputModel.prototype, "governmentAgency", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("complianceMessage", String, true),
        __metadata("design:type", String)
    ], ItemRestrictionOutputModel.prototype, "complianceMessage", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ItemRestrictionOutputModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], ItemRestrictionOutputModel.prototype, "createdUserId", void 0);
    ItemRestrictionOutputModel = __decorate([
        (0, json2typescript_1.JsonObject)("ItemRestrictionOutputModel")
    ], ItemRestrictionOutputModel);
    return ItemRestrictionOutputModel;
}());
