// avatax/lib/models/ItemParameterModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemParameterModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Represents a parameter associated with an item.
 * @export
 * @class ItemParameterModel
 */
var ItemParameterModel = exports.ItemParameterModel = /** @class */ (function () {
    function ItemParameterModel() {
        /**
         * @type {number}
         * @memberof ItemParameterModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof ItemParameterModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof ItemParameterModel
         */
        this.value = undefined;
        /**
         * @type {string}
         * @memberof ItemParameterModel
         */
        this.unit = undefined;
        /**
         * @type {number}
         * @memberof ItemParameterModel
         */
        this.itemId = undefined;
        /**
         * @type {boolean}
         * @memberof ItemParameterModel
         */
        this.isNeededForCalculation = undefined;
        /**
         * @type {boolean}
         * @memberof ItemParameterModel
         */
        this.isNeededForReturns = undefined;
        /**
         * @type {boolean}
         * @memberof ItemParameterModel
         */
        this.isNeededForClassification = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], ItemParameterModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String),
        __metadata("design:type", String)
    ], ItemParameterModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("value", String),
        __metadata("design:type", String)
    ], ItemParameterModel.prototype, "value", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("unit", String, true),
        __metadata("design:type", String)
    ], ItemParameterModel.prototype, "unit", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("itemId", Number, true),
        __metadata("design:type", Number)
    ], ItemParameterModel.prototype, "itemId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isNeededForCalculation", Boolean, true),
        __metadata("design:type", Boolean)
    ], ItemParameterModel.prototype, "isNeededForCalculation", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isNeededForReturns", Boolean, true),
        __metadata("design:type", Boolean)
    ], ItemParameterModel.prototype, "isNeededForReturns", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isNeededForClassification", Boolean, true),
        __metadata("design:type", Boolean)
    ], ItemParameterModel.prototype, "isNeededForClassification", void 0);
    ItemParameterModel = __decorate([
        (0, json2typescript_1.JsonObject)("ItemParameterModel")
    ], ItemParameterModel);
    return ItemParameterModel;
}());
