// avatax/lib/models/ItemRestrictionInputModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemRestrictionInputModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Represents a parameter associated with an item.
 * @export
 * @class ItemRestrictionInputModel
 */
var ItemRestrictionInputModel = exports.ItemRestrictionInputModel = /** @class */ (function () {
    function ItemRestrictionInputModel() {
        /**
         * @type {string}
         * @memberof ItemRestrictionInputModel
         */
        this.itemCode = undefined;
        /**
         * @type {number}
         * @memberof ItemRestrictionInputModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionInputModel
         */
        this.hsCode = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionInputModel
         */
        this.countryOfImport = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionInputModel
         */
        this.countryOfExport = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionInputModel
         */
        this.countryOfManufacture = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionInputModel
         */
        this.restrictionType = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionInputModel
         */
        this.regulation = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionInputModel
         */
        this.governmentAgency = undefined;
        /**
         * @type {string}
         * @memberof ItemRestrictionInputModel
         */
        this.complianceMessage = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("itemCode", String),
        __metadata("design:type", String)
    ], ItemRestrictionInputModel.prototype, "itemCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number),
        __metadata("design:type", Number)
    ], ItemRestrictionInputModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hsCode", String),
        __metadata("design:type", String)
    ], ItemRestrictionInputModel.prototype, "hsCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("countryOfImport", String),
        __metadata("design:type", String)
    ], ItemRestrictionInputModel.prototype, "countryOfImport", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("countryOfExport", String),
        __metadata("design:type", String)
    ], ItemRestrictionInputModel.prototype, "countryOfExport", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("countryOfManufacture", String),
        __metadata("design:type", String)
    ], ItemRestrictionInputModel.prototype, "countryOfManufacture", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("restrictionType", String),
        __metadata("design:type", String)
    ], ItemRestrictionInputModel.prototype, "restrictionType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("regulation", String),
        __metadata("design:type", String)
    ], ItemRestrictionInputModel.prototype, "regulation", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("governmentAgency", String, true),
        __metadata("design:type", String)
    ], ItemRestrictionInputModel.prototype, "governmentAgency", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("complianceMessage", String, true),
        __metadata("design:type", String)
    ], ItemRestrictionInputModel.prototype, "complianceMessage", void 0);
    ItemRestrictionInputModel = __decorate([
        (0, json2typescript_1.JsonObject)("ItemRestrictionInputModel")
    ], ItemRestrictionInputModel);
    return ItemRestrictionInputModel;
}());
