// avatax/lib/models/ItemTaxCodeClassificationRequestOutputModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemTaxCodeClassificationRequestOutputModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a tax code classification request output model
 * @export
 * @class ItemTaxCodeClassificationRequestOutputModel
 */
var ItemTaxCodeClassificationRequestOutputModel = exports.ItemTaxCodeClassificationRequestOutputModel = /** @class */ (function () {
    function ItemTaxCodeClassificationRequestOutputModel() {
        /**
         * @type {number}
         * @memberof ItemTaxCodeClassificationRequestOutputModel
         */
        this.requestId = undefined;
        /**
         * @type {number}
         * @memberof ItemTaxCodeClassificationRequestOutputModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof ItemTaxCodeClassificationRequestOutputModel
         */
        this.request = undefined;
        /**
         * @type {string}
         * @memberof ItemTaxCodeClassificationRequestOutputModel
         */
        this.status = undefined;
        /**
         * @type {string}
         * @memberof ItemTaxCodeClassificationRequestOutputModel
         */
        this.requestType = undefined;
        /**
         * @type {number}
         * @memberof ItemTaxCodeClassificationRequestOutputModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof ItemTaxCodeClassificationRequestOutputModel
         */
        this.createdDate = undefined;
        /**
         * @type {Date}
         * @memberof ItemTaxCodeClassificationRequestOutputModel
         */
        this.modifiedDate = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("requestId", Number, true),
        __metadata("design:type", Number)
    ], ItemTaxCodeClassificationRequestOutputModel.prototype, "requestId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], ItemTaxCodeClassificationRequestOutputModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("request", String, true),
        __metadata("design:type", String)
    ], ItemTaxCodeClassificationRequestOutputModel.prototype, "request", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("status", String, true),
        __metadata("design:type", String)
    ], ItemTaxCodeClassificationRequestOutputModel.prototype, "status", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("requestType", String, true),
        __metadata("design:type", String)
    ], ItemTaxCodeClassificationRequestOutputModel.prototype, "requestType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], ItemTaxCodeClassificationRequestOutputModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ItemTaxCodeClassificationRequestOutputModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], ItemTaxCodeClassificationRequestOutputModel.prototype, "modifiedDate", void 0);
    ItemTaxCodeClassificationRequestOutputModel = __decorate([
        (0, json2typescript_1.JsonObject)("ItemTaxCodeClassificationRequestOutputModel")
    ], ItemTaxCodeClassificationRequestOutputModel);
    return ItemTaxCodeClassificationRequestOutputModel;
}());
