// avatax/lib/models/JurisdictionModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.JurisdictionModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents information about a single legal taxing jurisdiction
 * @export
 * @class JurisdictionModel
 */
var JurisdictionModel = exports.JurisdictionModel = /** @class */ (function () {
    function JurisdictionModel() {
        /**
         * @type {string}
         * @memberof JurisdictionModel
         */
        this.code = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionModel
         */
        this.name = undefined;
        /**
         * @type {Enums.JurisdictionType}
         * @memberof JurisdictionModel
         */
        this.type = undefined;
        /**
         * @type {number}
         * @memberof JurisdictionModel
         */
        this.rate = undefined;
        /**
         * @type {number}
         * @memberof JurisdictionModel
         */
        this.salesRate = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionModel
         */
        this.signatureCode = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionModel
         */
        this.region = undefined;
        /**
         * @type {number}
         * @memberof JurisdictionModel
         */
        this.useRate = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionModel
         */
        this.city = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionModel
         */
        this.county = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionModel
         */
        this.shortName = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionModel
         */
        this.stateFips = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionModel
         */
        this.countyFips = undefined;
        /**
         * @type {string}
         * @memberof JurisdictionModel
         */
        this.placeFips = undefined;
        /**
         * @type {number}
         * @memberof JurisdictionModel
         */
        this.id = undefined;
        /**
         * @type {Date}
         * @memberof JurisdictionModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof JurisdictionModel
         */
        this.endDate = undefined;
        /**
         * @type {boolean}
         * @memberof JurisdictionModel
         */
        this.isAcm = undefined;
        /**
         * @type {boolean}
         * @memberof JurisdictionModel
         */
        this.isSst = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("code", String),
        __metadata("design:type", String)
    ], JurisdictionModel.prototype, "code", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String),
        __metadata("design:type", String)
    ], JurisdictionModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("type", Enums.JurisdictionTypeConverter),
        __metadata("design:type", Number)
    ], JurisdictionModel.prototype, "type", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rate", Number, true),
        __metadata("design:type", Number)
    ], JurisdictionModel.prototype, "rate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("salesRate", Number, true),
        __metadata("design:type", Number)
    ], JurisdictionModel.prototype, "salesRate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("signatureCode", String),
        __metadata("design:type", String)
    ], JurisdictionModel.prototype, "signatureCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], JurisdictionModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("useRate", Number, true),
        __metadata("design:type", Number)
    ], JurisdictionModel.prototype, "useRate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("city", String, true),
        __metadata("design:type", String)
    ], JurisdictionModel.prototype, "city", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("county", String, true),
        __metadata("design:type", String)
    ], JurisdictionModel.prototype, "county", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], JurisdictionModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("shortName", String, true),
        __metadata("design:type", String)
    ], JurisdictionModel.prototype, "shortName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("stateFips", String, true),
        __metadata("design:type", String)
    ], JurisdictionModel.prototype, "stateFips", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("countyFips", String, true),
        __metadata("design:type", String)
    ], JurisdictionModel.prototype, "countyFips", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("placeFips", String, true),
        __metadata("design:type", String)
    ], JurisdictionModel.prototype, "placeFips", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], JurisdictionModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], JurisdictionModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], JurisdictionModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isAcm", Boolean, true),
        __metadata("design:type", Boolean)
    ], JurisdictionModel.prototype, "isAcm", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSst", Boolean, true),
        __metadata("design:type", Boolean)
    ], JurisdictionModel.prototype, "isSst", void 0);
    JurisdictionModel = __decorate([
        (0, json2typescript_1.JsonObject)("JurisdictionModel")
    ], JurisdictionModel);
    return JurisdictionModel;
}());
