// avatax/lib/models/LocationQuestionModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocationQuestionModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
/**
 * Information about questions that the local jurisdictions require for each location
 * @export
 * @class LocationQuestionModel
 */
var LocationQuestionModel = exports.LocationQuestionModel = /** @class */ (function () {
    function LocationQuestionModel() {
        /**
         * @type {number}
         * @memberof LocationQuestionModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof LocationQuestionModel
         */
        this.question = undefined;
        /**
         * @type {string}
         * @memberof LocationQuestionModel
         */
        this.description = undefined;
        /**
         * @type {string}
         * @memberof LocationQuestionModel
         */
        this.regularExpression = undefined;
        /**
         * @type {string}
         * @memberof LocationQuestionModel
         */
        this.exampleValue = undefined;
        /**
         * @type {string}
         * @memberof LocationQuestionModel
         */
        this.jurisdictionName = undefined;
        /**
         * @type {Enums.JurisdictionType}
         * @memberof LocationQuestionModel
         */
        this.jurisdictionType = undefined;
        /**
         * @type {string}
         * @memberof LocationQuestionModel
         */
        this.jurisdictionCountry = undefined;
        /**
         * @type {string}
         * @memberof LocationQuestionModel
         */
        this.jurisdictionRegion = undefined;
        /**
         * @type {string}
         * @memberof LocationQuestionModel
         */
        this.helpText = undefined;
        /**
         * @type {number}
         * @memberof LocationQuestionModel
         */
        this.maxLength = undefined;
        /**
         * @type {boolean}
         * @memberof LocationQuestionModel
         */
        this.required = undefined;
        /**
         * @type {string}
         * @memberof LocationQuestionModel
         */
        this.dataType = undefined;
        /**
         * @type {string}
         * @memberof LocationQuestionModel
         */
        this.staticOptions = undefined;
        /**
         * @type {boolean}
         * @memberof LocationQuestionModel
         */
        this.unique = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number),
        __metadata("design:type", Number)
    ], LocationQuestionModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("question", String),
        __metadata("design:type", String)
    ], LocationQuestionModel.prototype, "question", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String),
        __metadata("design:type", String)
    ], LocationQuestionModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("regularExpression", String, true),
        __metadata("design:type", String)
    ], LocationQuestionModel.prototype, "regularExpression", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exampleValue", String, true),
        __metadata("design:type", String)
    ], LocationQuestionModel.prototype, "exampleValue", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionName", String, true),
        __metadata("design:type", String)
    ], LocationQuestionModel.prototype, "jurisdictionName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionType", Enums.JurisdictionTypeConverter, true),
        __metadata("design:type", Number)
    ], LocationQuestionModel.prototype, "jurisdictionType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionCountry", String, true),
        __metadata("design:type", String)
    ], LocationQuestionModel.prototype, "jurisdictionCountry", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionRegion", String, true),
        __metadata("design:type", String)
    ], LocationQuestionModel.prototype, "jurisdictionRegion", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("helpText", String, true),
        __metadata("design:type", String)
    ], LocationQuestionModel.prototype, "helpText", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("maxLength", Number, true),
        __metadata("design:type", Number)
    ], LocationQuestionModel.prototype, "maxLength", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("required", Boolean, true),
        __metadata("design:type", Boolean)
    ], LocationQuestionModel.prototype, "required", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dataType", String, true),
        __metadata("design:type", String)
    ], LocationQuestionModel.prototype, "dataType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("staticOptions", String, true),
        __metadata("design:type", String)
    ], LocationQuestionModel.prototype, "staticOptions", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("unique", Boolean, true),
        __metadata("design:type", Boolean)
    ], LocationQuestionModel.prototype, "unique", void 0);
    LocationQuestionModel = __decorate([
        (0, json2typescript_1.JsonObject)("LocationQuestionModel")
    ], LocationQuestionModel);
    return LocationQuestionModel;
}());
