// avatax/lib/models/LocationModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocationModel = void 0;
var Enums = require("../enums/index");
var LocationSettingModel_1 = require("./LocationSettingModel");
var LocationParameterModel_1 = require("./LocationParameterModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * A location where this company does business.
Some jurisdictions may require you to list all locations where your company does business.
 * @export
 * @class LocationModel
 */
var LocationModel = exports.LocationModel = /** @class */ (function () {
    function LocationModel() {
        /**
         * @type {number}
         * @memberof LocationModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof LocationModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof LocationModel
         */
        this.locationCode = undefined;
        /**
         * @type {string}
         * @memberof LocationModel
         */
        this.description = undefined;
        /**
         * @type {Enums.AddressTypeId}
         * @memberof LocationModel
         */
        this.addressTypeId = undefined;
        /**
         * @type {Enums.AddressCategoryId}
         * @memberof LocationModel
         */
        this.addressCategoryId = undefined;
        /**
         * @type {boolean}
         * @memberof LocationModel
         */
        this.isMarketplaceOutsideUsa = undefined;
        /**
         * @type {string}
         * @memberof LocationModel
         */
        this.line1 = undefined;
        /**
         * @type {string}
         * @memberof LocationModel
         */
        this.line2 = undefined;
        /**
         * @type {string}
         * @memberof LocationModel
         */
        this.line3 = undefined;
        /**
         * @type {string}
         * @memberof LocationModel
         */
        this.city = undefined;
        /**
         * @type {string}
         * @memberof LocationModel
         */
        this.county = undefined;
        /**
         * @type {string}
         * @memberof LocationModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof LocationModel
         */
        this.postalCode = undefined;
        /**
         * @type {string}
         * @memberof LocationModel
         */
        this.country = undefined;
        /**
         * @type {boolean}
         * @memberof LocationModel
         */
        this.isDefault = undefined;
        /**
         * @type {boolean}
         * @memberof LocationModel
         */
        this.isRegistered = undefined;
        /**
         * @type {string}
         * @memberof LocationModel
         */
        this.dbaName = undefined;
        /**
         * @type {string}
         * @memberof LocationModel
         */
        this.outletName = undefined;
        /**
         * @type {Date}
         * @memberof LocationModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof LocationModel
         */
        this.endDate = undefined;
        /**
         * @type {Date}
         * @memberof LocationModel
         */
        this.lastTransactionDate = undefined;
        /**
         * @type {Date}
         * @memberof LocationModel
         */
        this.registeredDate = undefined;
        /**
         * @type {Date}
         * @memberof LocationModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof LocationModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof LocationModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof LocationModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {LocationSettingModel[]}
         * @memberof LocationModel
         */
        this.settings = undefined;
        /**
         * @type {LocationParameterModel[]}
         * @memberof LocationModel
         */
        this.parameters = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number),
        __metadata("design:type", Number)
    ], LocationModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], LocationModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("locationCode", String),
        __metadata("design:type", String)
    ], LocationModel.prototype, "locationCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], LocationModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("addressTypeId", Enums.AddressTypeIdConverter),
        __metadata("design:type", Number)
    ], LocationModel.prototype, "addressTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("addressCategoryId", Enums.AddressCategoryIdConverter),
        __metadata("design:type", Number)
    ], LocationModel.prototype, "addressCategoryId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isMarketplaceOutsideUsa", Boolean, true),
        __metadata("design:type", Boolean)
    ], LocationModel.prototype, "isMarketplaceOutsideUsa", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line1", String),
        __metadata("design:type", String)
    ], LocationModel.prototype, "line1", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line2", String, true),
        __metadata("design:type", String)
    ], LocationModel.prototype, "line2", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("line3", String, true),
        __metadata("design:type", String)
    ], LocationModel.prototype, "line3", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("city", String, true),
        __metadata("design:type", String)
    ], LocationModel.prototype, "city", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("county", String, true),
        __metadata("design:type", String)
    ], LocationModel.prototype, "county", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], LocationModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("postalCode", String),
        __metadata("design:type", String)
    ], LocationModel.prototype, "postalCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String),
        __metadata("design:type", String)
    ], LocationModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isDefault", Boolean, true),
        __metadata("design:type", Boolean)
    ], LocationModel.prototype, "isDefault", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isRegistered", Boolean, true),
        __metadata("design:type", Boolean)
    ], LocationModel.prototype, "isRegistered", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dbaName", String, true),
        __metadata("design:type", String)
    ], LocationModel.prototype, "dbaName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("outletName", String, true),
        __metadata("design:type", String)
    ], LocationModel.prototype, "outletName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], LocationModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], LocationModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lastTransactionDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], LocationModel.prototype, "lastTransactionDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("registeredDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], LocationModel.prototype, "registeredDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], LocationModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], LocationModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], LocationModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], LocationModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("settings", [LocationSettingModel_1.LocationSettingModel], true),
        __metadata("design:type", Array)
    ], LocationModel.prototype, "settings", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("parameters", [LocationParameterModel_1.LocationParameterModel], true),
        __metadata("design:type", Array)
    ], LocationModel.prototype, "parameters", void 0);
    LocationModel = __decorate([
        (0, json2typescript_1.JsonObject)("LocationModel")
    ], LocationModel);
    return LocationModel;
}());
