// avatax/lib/models/LoginVerificationInputModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoginVerificationInputModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Represents a verification request using Skyscraper for a company
 * @export
 * @class LoginVerificationInputModel
 */
var LoginVerificationInputModel = exports.LoginVerificationInputModel = /** @class */ (function () {
    function LoginVerificationInputModel() {
        /**
         * @type {number}
         * @memberof LoginVerificationInputModel
         */
        this.companyId = undefined;
        /**
         * @type {number}
         * @memberof LoginVerificationInputModel
         */
        this.accountId = undefined;
        /**
         * @type {string}
         * @memberof LoginVerificationInputModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof LoginVerificationInputModel
         */
        this.taxFormCode = undefined;
        /**
         * @type {string}
         * @memberof LoginVerificationInputModel
         */
        this.username = undefined;
        /**
         * @type {string}
         * @memberof LoginVerificationInputModel
         */
        this.password = undefined;
        /**
         * @type {object}
         * @memberof LoginVerificationInputModel
         */
        this.additionalOptions = undefined;
        /**
         * @type {number}
         * @memberof LoginVerificationInputModel
         */
        this.bulkRequestId = undefined;
        /**
         * @type {number}
         * @memberof LoginVerificationInputModel
         */
        this.priority = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number),
        __metadata("design:type", Number)
    ], LoginVerificationInputModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountId", Number),
        __metadata("design:type", Number)
    ], LoginVerificationInputModel.prototype, "accountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String),
        __metadata("design:type", String)
    ], LoginVerificationInputModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxFormCode", String, true),
        __metadata("design:type", String)
    ], LoginVerificationInputModel.prototype, "taxFormCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("username", String),
        __metadata("design:type", String)
    ], LoginVerificationInputModel.prototype, "username", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("password", String),
        __metadata("design:type", String)
    ], LoginVerificationInputModel.prototype, "password", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("additionalOptions", Object, true),
        __metadata("design:type", Object)
    ], LoginVerificationInputModel.prototype, "additionalOptions", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("bulkRequestId", Number, true),
        __metadata("design:type", Number)
    ], LoginVerificationInputModel.prototype, "bulkRequestId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("priority", Number, true),
        __metadata("design:type", Number)
    ], LoginVerificationInputModel.prototype, "priority", void 0);
    LoginVerificationInputModel = __decorate([
        (0, json2typescript_1.JsonObject)("LoginVerificationInputModel")
    ], LoginVerificationInputModel);
    return LoginVerificationInputModel;
}());
