// avatax/lib/models/MarketplaceLocationModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MarketplaceLocationModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Marketplace Location Output model
 * @export
 * @class MarketplaceLocationModel
 */
var MarketplaceLocationModel = exports.MarketplaceLocationModel = /** @class */ (function () {
    function MarketplaceLocationModel() {
        /**
         * @type {string}
         * @memberof MarketplaceLocationModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof MarketplaceLocationModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof MarketplaceLocationModel
         */
        this.marketplaceId = undefined;
        /**
         * @type {string}
         * @memberof MarketplaceLocationModel
         */
        this.marketplace = undefined;
        /**
         * @type {Date}
         * @memberof MarketplaceLocationModel
         */
        this.marketplaceAdoptionDate = undefined;
        /**
         * @type {Date}
         * @memberof MarketplaceLocationModel
         */
        this.marketplaceEndDate = undefined;
        /**
         * @type {Date}
         * @memberof MarketplaceLocationModel
         */
        this.legislativeEffectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof MarketplaceLocationModel
         */
        this.enforcementDate = undefined;
        /**
         * @type {Date}
         * @memberof MarketplaceLocationModel
         */
        this.createdDate = undefined;
        /**
         * @type {Date}
         * @memberof MarketplaceLocationModel
         */
        this.modifiedDate = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], MarketplaceLocationModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], MarketplaceLocationModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("marketplaceId", String, true),
        __metadata("design:type", String)
    ], MarketplaceLocationModel.prototype, "marketplaceId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("marketplace", String, true),
        __metadata("design:type", String)
    ], MarketplaceLocationModel.prototype, "marketplace", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("marketplaceAdoptionDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], MarketplaceLocationModel.prototype, "marketplaceAdoptionDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("marketplaceEndDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], MarketplaceLocationModel.prototype, "marketplaceEndDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("legislativeEffectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], MarketplaceLocationModel.prototype, "legislativeEffectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("enforcementDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], MarketplaceLocationModel.prototype, "enforcementDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], MarketplaceLocationModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], MarketplaceLocationModel.prototype, "modifiedDate", void 0);
    MarketplaceLocationModel = __decorate([
        (0, json2typescript_1.JsonObject)("MarketplaceLocationModel")
    ], MarketplaceLocationModel);
    return MarketplaceLocationModel;
}());
