// avatax/lib/models/MultiTaxFilingReturnModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultiTaxFilingReturnModel = void 0;
var Enums = require("../enums/index");
var FilingsTaxSummaryModel_1 = require("./FilingsTaxSummaryModel");
var FilingsTaxDetailsModel_1 = require("./FilingsTaxDetailsModel");
var FilingReturnCreditModel_1 = require("./FilingReturnCreditModel");
var FilingAdjustmentModel_1 = require("./FilingAdjustmentModel");
var FilingAugmentationModel_1 = require("./FilingAugmentationModel");
var FilingPaymentModel_1 = require("./FilingPaymentModel");
var FilingAttachmentModel_1 = require("./FilingAttachmentModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Filing Returns Model
 * @export
 * @class MultiTaxFilingReturnModel
 */
var MultiTaxFilingReturnModel = exports.MultiTaxFilingReturnModel = /** @class */ (function () {
    function MultiTaxFilingReturnModel() {
        /**
         * @type {number}
         * @memberof MultiTaxFilingReturnModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof MultiTaxFilingReturnModel
         */
        this.filingCalendarId = undefined;
        /**
         * @type {string}
         * @memberof MultiTaxFilingReturnModel
         */
        this.registrationId = undefined;
        /**
         * @type {Enums.FilingStatusId}
         * @memberof MultiTaxFilingReturnModel
         */
        this.status = undefined;
        /**
         * @type {Enums.FilingFrequencyId}
         * @memberof MultiTaxFilingReturnModel
         */
        this.filingFrequency = undefined;
        /**
         * @type {Enums.FilingTypeId}
         * @memberof MultiTaxFilingReturnModel
         */
        this.filingType = undefined;
        /**
         * @type {string}
         * @memberof MultiTaxFilingReturnModel
         */
        this.formName = undefined;
        /**
         * @type {string}
         * @memberof MultiTaxFilingReturnModel
         */
        this.formCode = undefined;
        /**
         * @type {string}
         * @memberof MultiTaxFilingReturnModel
         */
        this.description = undefined;
        /**
         * @type {number}
         * @memberof MultiTaxFilingReturnModel
         */
        this.taxAuthorityId = undefined;
        /**
         * @type {Date}
         * @memberof MultiTaxFilingReturnModel
         */
        this.filedDate = undefined;
        /**
         * @type {Enums.AccrualType}
         * @memberof MultiTaxFilingReturnModel
         */
        this.accrualType = undefined;
        /**
         * @type {Date}
         * @memberof MultiTaxFilingReturnModel
         */
        this.startPeriod = undefined;
        /**
         * @type {Date}
         * @memberof MultiTaxFilingReturnModel
         */
        this.endPeriod = undefined;
        /**
         * @type {string}
         * @memberof MultiTaxFilingReturnModel
         */
        this.type = undefined;
        /**
         * @type {FilingsTaxSummaryModel}
         * @memberof MultiTaxFilingReturnModel
         */
        this.returnTaxSummary = undefined;
        /**
         * @type {FilingsTaxDetailsModel[]}
         * @memberof MultiTaxFilingReturnModel
         */
        this.returnTaxDetails = undefined;
        /**
         * @type {FilingReturnCreditModel}
         * @memberof MultiTaxFilingReturnModel
         */
        this.excludedCarryOverCredits = undefined;
        /**
         * @type {FilingReturnCreditModel}
         * @memberof MultiTaxFilingReturnModel
         */
        this.appliedCarryOverCredits = undefined;
        /**
         * @type {number}
         * @memberof MultiTaxFilingReturnModel
         */
        this.totalAdjustments = undefined;
        /**
         * @type {FilingAdjustmentModel[]}
         * @memberof MultiTaxFilingReturnModel
         */
        this.adjustments = undefined;
        /**
         * @type {number}
         * @memberof MultiTaxFilingReturnModel
         */
        this.totalAugmentations = undefined;
        /**
         * @type {FilingAugmentationModel[]}
         * @memberof MultiTaxFilingReturnModel
         */
        this.augmentations = undefined;
        /**
         * @type {number}
         * @memberof MultiTaxFilingReturnModel
         */
        this.totalPayments = undefined;
        /**
         * @type {FilingPaymentModel[]}
         * @memberof MultiTaxFilingReturnModel
         */
        this.payments = undefined;
        /**
         * @type {FilingAttachmentModel[]}
         * @memberof MultiTaxFilingReturnModel
         */
        this.attachments = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], MultiTaxFilingReturnModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingCalendarId", Number, true),
        __metadata("design:type", Number)
    ], MultiTaxFilingReturnModel.prototype, "filingCalendarId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("registrationId", String, true),
        __metadata("design:type", String)
    ], MultiTaxFilingReturnModel.prototype, "registrationId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("status", Enums.FilingStatusIdConverter, true),
        __metadata("design:type", Number)
    ], MultiTaxFilingReturnModel.prototype, "status", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingFrequency", Enums.FilingFrequencyIdConverter, true),
        __metadata("design:type", Number)
    ], MultiTaxFilingReturnModel.prototype, "filingFrequency", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingType", Enums.FilingTypeIdConverter, true),
        __metadata("design:type", Number)
    ], MultiTaxFilingReturnModel.prototype, "filingType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("formName", String, true),
        __metadata("design:type", String)
    ], MultiTaxFilingReturnModel.prototype, "formName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("formCode", String, true),
        __metadata("design:type", String)
    ], MultiTaxFilingReturnModel.prototype, "formCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], MultiTaxFilingReturnModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityId", Number, true),
        __metadata("design:type", Number)
    ], MultiTaxFilingReturnModel.prototype, "taxAuthorityId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], MultiTaxFilingReturnModel.prototype, "filedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accrualType", Enums.AccrualTypeConverter, true),
        __metadata("design:type", Number)
    ], MultiTaxFilingReturnModel.prototype, "accrualType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("startPeriod", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], MultiTaxFilingReturnModel.prototype, "startPeriod", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endPeriod", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], MultiTaxFilingReturnModel.prototype, "endPeriod", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("type", String, true),
        __metadata("design:type", String)
    ], MultiTaxFilingReturnModel.prototype, "type", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("returnTaxSummary", FilingsTaxSummaryModel_1.FilingsTaxSummaryModel, true),
        __metadata("design:type", FilingsTaxSummaryModel_1.FilingsTaxSummaryModel)
    ], MultiTaxFilingReturnModel.prototype, "returnTaxSummary", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("returnTaxDetails", [FilingsTaxDetailsModel_1.FilingsTaxDetailsModel], true),
        __metadata("design:type", Array)
    ], MultiTaxFilingReturnModel.prototype, "returnTaxDetails", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("excludedCarryOverCredits", FilingReturnCreditModel_1.FilingReturnCreditModel, true),
        __metadata("design:type", FilingReturnCreditModel_1.FilingReturnCreditModel)
    ], MultiTaxFilingReturnModel.prototype, "excludedCarryOverCredits", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("appliedCarryOverCredits", FilingReturnCreditModel_1.FilingReturnCreditModel, true),
        __metadata("design:type", FilingReturnCreditModel_1.FilingReturnCreditModel)
    ], MultiTaxFilingReturnModel.prototype, "appliedCarryOverCredits", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalAdjustments", Number, true),
        __metadata("design:type", Number)
    ], MultiTaxFilingReturnModel.prototype, "totalAdjustments", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("adjustments", [FilingAdjustmentModel_1.FilingAdjustmentModel], true),
        __metadata("design:type", Array)
    ], MultiTaxFilingReturnModel.prototype, "adjustments", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalAugmentations", Number, true),
        __metadata("design:type", Number)
    ], MultiTaxFilingReturnModel.prototype, "totalAugmentations", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("augmentations", [FilingAugmentationModel_1.FilingAugmentationModel], true),
        __metadata("design:type", Array)
    ], MultiTaxFilingReturnModel.prototype, "augmentations", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalPayments", Number, true),
        __metadata("design:type", Number)
    ], MultiTaxFilingReturnModel.prototype, "totalPayments", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("payments", [FilingPaymentModel_1.FilingPaymentModel], true),
        __metadata("design:type", Array)
    ], MultiTaxFilingReturnModel.prototype, "payments", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("attachments", [FilingAttachmentModel_1.FilingAttachmentModel], true),
        __metadata("design:type", Array)
    ], MultiTaxFilingReturnModel.prototype, "attachments", void 0);
    MultiTaxFilingReturnModel = __decorate([
        (0, json2typescript_1.JsonObject)("MultiTaxFilingReturnModel")
    ], MultiTaxFilingReturnModel);
    return MultiTaxFilingReturnModel;
}());
