// avatax/lib/models/MultiDocumentLineItemModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultiDocumentLineItemModel = void 0;
var Enums = require("../enums/index");
var AddressesModel_1 = require("./AddressesModel");
var TaxOverrideModel_1 = require("./TaxOverrideModel");
var TransactionLineParameterModel_1 = require("./TransactionLineParameterModel");
var TransactionLineUserDefinedFieldModel_1 = require("./TransactionLineUserDefinedFieldModel");
var json2typescript_1 = require("json2typescript");
/**
 * Represents one line item in a MultiDocument transaction
 * @export
 * @class MultiDocumentLineItemModel
 */
var MultiDocumentLineItemModel = exports.MultiDocumentLineItemModel = /** @class */ (function () {
    function MultiDocumentLineItemModel() {
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.companyCode = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.reportingLocationCode = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.number = undefined;
        /**
         * @type {number}
         * @memberof MultiDocumentLineItemModel
         */
        this.quantity = undefined;
        /**
         * @type {number}
         * @memberof MultiDocumentLineItemModel
         */
        this.amount = undefined;
        /**
         * @type {AddressesModel}
         * @memberof MultiDocumentLineItemModel
         */
        this.addresses = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.taxCode = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.customerUsageType = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.entityUseCode = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.itemCode = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.exemptionCode = undefined;
        /**
         * @type {boolean}
         * @memberof MultiDocumentLineItemModel
         */
        this.discounted = undefined;
        /**
         * @type {boolean}
         * @memberof MultiDocumentLineItemModel
         */
        this.taxIncluded = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.revenueAccount = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.ref1 = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.ref2 = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.description = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.businessIdentificationNo = undefined;
        /**
         * @type {TaxOverrideModel}
         * @memberof MultiDocumentLineItemModel
         */
        this.taxOverride = undefined;
        /**
         * @type {TransactionLineParameterModel[]}
         * @memberof MultiDocumentLineItemModel
         */
        this.parameters = undefined;
        /**
         * @type {TransactionLineUserDefinedFieldModel[]}
         * @memberof MultiDocumentLineItemModel
         */
        this.userDefinedFields = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.hsCode = undefined;
        /**
         * @type {number}
         * @memberof MultiDocumentLineItemModel
         */
        this.merchantSellerId = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.merchantSellerIdentifier = undefined;
        /**
         * @type {Enums.MarketplaceLiabilityType}
         * @memberof MultiDocumentLineItemModel
         */
        this.marketplaceLiabilityType = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.originationDocumentId = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.originationSite = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.category = undefined;
        /**
         * @type {string}
         * @memberof MultiDocumentLineItemModel
         */
        this.summary = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyCode", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "companyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportingLocationCode", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "reportingLocationCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("number", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "number", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("quantity", Number, true),
        __metadata("design:type", Number)
    ], MultiDocumentLineItemModel.prototype, "quantity", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("amount", Number),
        __metadata("design:type", Number)
    ], MultiDocumentLineItemModel.prototype, "amount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("addresses", AddressesModel_1.AddressesModel, true),
        __metadata("design:type", AddressesModel_1.AddressesModel)
    ], MultiDocumentLineItemModel.prototype, "addresses", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCode", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "taxCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerUsageType", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "customerUsageType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("entityUseCode", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "entityUseCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("itemCode", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "itemCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptionCode", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "exemptionCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("discounted", Boolean, true),
        __metadata("design:type", Boolean)
    ], MultiDocumentLineItemModel.prototype, "discounted", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxIncluded", Boolean, true),
        __metadata("design:type", Boolean)
    ], MultiDocumentLineItemModel.prototype, "taxIncluded", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("revenueAccount", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "revenueAccount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("ref1", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "ref1", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("ref2", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "ref2", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("businessIdentificationNo", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "businessIdentificationNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxOverride", TaxOverrideModel_1.TaxOverrideModel, true),
        __metadata("design:type", TaxOverrideModel_1.TaxOverrideModel)
    ], MultiDocumentLineItemModel.prototype, "taxOverride", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("parameters", [TransactionLineParameterModel_1.TransactionLineParameterModel], true),
        __metadata("design:type", Array)
    ], MultiDocumentLineItemModel.prototype, "parameters", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("userDefinedFields", [TransactionLineUserDefinedFieldModel_1.TransactionLineUserDefinedFieldModel], true),
        __metadata("design:type", Array)
    ], MultiDocumentLineItemModel.prototype, "userDefinedFields", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hsCode", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "hsCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("merchantSellerId", Number, true),
        __metadata("design:type", Number)
    ], MultiDocumentLineItemModel.prototype, "merchantSellerId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("merchantSellerIdentifier", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "merchantSellerIdentifier", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("marketplaceLiabilityType", Enums.MarketplaceLiabilityTypeConverter, true),
        __metadata("design:type", Number)
    ], MultiDocumentLineItemModel.prototype, "marketplaceLiabilityType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("originationDocumentId", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "originationDocumentId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("originationSite", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "originationSite", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("category", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "category", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("summary", String, true),
        __metadata("design:type", String)
    ], MultiDocumentLineItemModel.prototype, "summary", void 0);
    MultiDocumentLineItemModel = __decorate([
        (0, json2typescript_1.JsonObject)("MultiDocumentLineItemModel")
    ], MultiDocumentLineItemModel);
    return MultiDocumentLineItemModel;
}());
