// avatax/lib/models/NewAccountModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NewAccountModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents information about a newly created account
 * @export
 * @class NewAccountModel
 */
var NewAccountModel = exports.NewAccountModel = /** @class */ (function () {
    function NewAccountModel() {
        /**
         * @type {number}
         * @memberof NewAccountModel
         */
        this.accountId = undefined;
        /**
         * @type {string}
         * @memberof NewAccountModel
         */
        this.accountDetailsEmailedTo = undefined;
        /**
         * @type {Date}
         * @memberof NewAccountModel
         */
        this.createdDate = undefined;
        /**
         * @type {Date}
         * @memberof NewAccountModel
         */
        this.emailedDate = undefined;
        /**
         * @type {string}
         * @memberof NewAccountModel
         */
        this.limitations = undefined;
        /**
         * @type {string}
         * @memberof NewAccountModel
         */
        this.licenseKey = undefined;
        /**
         * @type {string}
         * @memberof NewAccountModel
         */
        this.paymentUrl = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountId", Number, true),
        __metadata("design:type", Number)
    ], NewAccountModel.prototype, "accountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountDetailsEmailedTo", String, true),
        __metadata("design:type", String)
    ], NewAccountModel.prototype, "accountDetailsEmailedTo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NewAccountModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("emailedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NewAccountModel.prototype, "emailedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("limitations", String, true),
        __metadata("design:type", String)
    ], NewAccountModel.prototype, "limitations", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("licenseKey", String, true),
        __metadata("design:type", String)
    ], NewAccountModel.prototype, "licenseKey", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("paymentUrl", String, true),
        __metadata("design:type", String)
    ], NewAccountModel.prototype, "paymentUrl", void 0);
    NewAccountModel = __decorate([
        (0, json2typescript_1.JsonObject)("NewAccountModel")
    ], NewAccountModel);
    return NewAccountModel;
}());
