// avatax/lib/models/NewFirmClientAccountRequestModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NewFirmClientAccountRequestModel = void 0;
var CompanyAddress_1 = require("./CompanyAddress");
var json2typescript_1 = require("json2typescript");
/**
 * Represents a request for a new account with Avalara for a new Firm client.
 * @export
 * @class NewFirmClientAccountRequestModel
 */
var NewFirmClientAccountRequestModel = exports.NewFirmClientAccountRequestModel = /** @class */ (function () {
    function NewFirmClientAccountRequestModel() {
        /**
         * @type {string}
         * @memberof NewFirmClientAccountRequestModel
         */
        this.accountName = undefined;
        /**
         * @type {string}
         * @memberof NewFirmClientAccountRequestModel
         */
        this.firstName = undefined;
        /**
         * @type {string}
         * @memberof NewFirmClientAccountRequestModel
         */
        this.lastName = undefined;
        /**
         * @type {string}
         * @memberof NewFirmClientAccountRequestModel
         */
        this.title = undefined;
        /**
         * @type {string}
         * @memberof NewFirmClientAccountRequestModel
         */
        this.phoneNumber = undefined;
        /**
         * @type {string}
         * @memberof NewFirmClientAccountRequestModel
         */
        this.email = undefined;
        /**
         * @type {string}
         * @memberof NewFirmClientAccountRequestModel
         */
        this.companyCode = undefined;
        /**
         * @type {CompanyAddress}
         * @memberof NewFirmClientAccountRequestModel
         */
        this.companyAddress = undefined;
        /**
         * @type {string}
         * @memberof NewFirmClientAccountRequestModel
         */
        this.taxPayerIdNumber = undefined;
        /**
         * @type {string[]}
         * @memberof NewFirmClientAccountRequestModel
         */
        this.properties = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountName", String),
        __metadata("design:type", String)
    ], NewFirmClientAccountRequestModel.prototype, "accountName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("firstName", String),
        __metadata("design:type", String)
    ], NewFirmClientAccountRequestModel.prototype, "firstName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lastName", String),
        __metadata("design:type", String)
    ], NewFirmClientAccountRequestModel.prototype, "lastName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("title", String, true),
        __metadata("design:type", String)
    ], NewFirmClientAccountRequestModel.prototype, "title", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("phoneNumber", String, true),
        __metadata("design:type", String)
    ], NewFirmClientAccountRequestModel.prototype, "phoneNumber", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("email", String),
        __metadata("design:type", String)
    ], NewFirmClientAccountRequestModel.prototype, "email", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyCode", String, true),
        __metadata("design:type", String)
    ], NewFirmClientAccountRequestModel.prototype, "companyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyAddress", CompanyAddress_1.CompanyAddress),
        __metadata("design:type", CompanyAddress_1.CompanyAddress)
    ], NewFirmClientAccountRequestModel.prototype, "companyAddress", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxPayerIdNumber", String, true),
        __metadata("design:type", String)
    ], NewFirmClientAccountRequestModel.prototype, "taxPayerIdNumber", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("properties", [String], true),
        __metadata("design:type", Array)
    ], NewFirmClientAccountRequestModel.prototype, "properties", void 0);
    NewFirmClientAccountRequestModel = __decorate([
        (0, json2typescript_1.JsonObject)("NewFirmClientAccountRequestModel")
    ], NewFirmClientAccountRequestModel);
    return NewFirmClientAccountRequestModel;
}());
