// avatax/lib/models/NexusModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NexusModel = void 0;
var Enums = require("../enums/index");
var NexusParameterDetailModel_1 = require("./NexusParameterDetailModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a declaration of nexus within a particular taxing jurisdiction.
            
To create a nexus declaration for your company, you must first call the Definitions API `ListNexus` to obtain a
list of Avalara-defined nexus.  Once you have determined which nexus you wish to declare, you should customize
only the user-selectable fields in this object.
            
The user selectable fields for the nexus object are `companyId`, `effectiveDate`, `endDate`, `localNexusTypeId`,
`taxId`, `nexusTypeId`, `hasPermanentEstablishment`, and `isSellerImporterOfRecord`.
            
When calling `CreateNexus` or `UpdateNexus`, all values in your nexus object except for the user-selectable fields
must match an Avalara-defined system nexus object.  You can retrieve a list of Avalara-defined system nexus objects
by calling `ListNexus`.  If any data does not match, AvaTax may not recognize your nexus declaration.
 * @export
 * @class NexusModel
 */
var NexusModel = exports.NexusModel = /** @class */ (function () {
    function NexusModel() {
        /**
         * @type {number}
         * @memberof NexusModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof NexusModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof NexusModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof NexusModel
         */
        this.region = undefined;
        /**
         * @type {Enums.JurisTypeId}
         * @memberof NexusModel
         */
        this.jurisTypeId = undefined;
        /**
         * @type {Enums.JurisdictionType}
         * @memberof NexusModel
         */
        this.jurisdictionTypeId = undefined;
        /**
         * @type {string}
         * @memberof NexusModel
         */
        this.jurisCode = undefined;
        /**
         * @type {string}
         * @memberof NexusModel
         */
        this.jurisName = undefined;
        /**
         * @type {Date}
         * @memberof NexusModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof NexusModel
         */
        this.endDate = undefined;
        /**
         * @type {string}
         * @memberof NexusModel
         */
        this.shortName = undefined;
        /**
         * @type {string}
         * @memberof NexusModel
         */
        this.signatureCode = undefined;
        /**
         * @type {string}
         * @memberof NexusModel
         */
        this.stateAssignedNo = undefined;
        /**
         * @type {Enums.NexusTypeId}
         * @memberof NexusModel
         */
        this.nexusTypeId = undefined;
        /**
         * @type {Enums.Sourcing}
         * @memberof NexusModel
         */
        this.sourcing = undefined;
        /**
         * @type {boolean}
         * @memberof NexusModel
         */
        this.hasLocalNexus = undefined;
        /**
         * @type {Enums.LocalNexusTypeId}
         * @memberof NexusModel
         */
        this.localNexusTypeId = undefined;
        /**
         * @type {boolean}
         * @memberof NexusModel
         */
        this.hasPermanentEstablishment = undefined;
        /**
         * @type {string}
         * @memberof NexusModel
         */
        this.taxId = undefined;
        /**
         * @type {boolean}
         * @memberof NexusModel
         */
        this.streamlinedSalesTax = undefined;
        /**
         * @type {boolean}
         * @memberof NexusModel
         */
        this.isSSTActive = undefined;
        /**
         * @type {Date}
         * @memberof NexusModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof NexusModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof NexusModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof NexusModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {string}
         * @memberof NexusModel
         */
        this.taxTypeGroup = undefined;
        /**
         * @type {string}
         * @memberof NexusModel
         */
        this.nexusTaxTypeGroup = undefined;
        /**
         * @type {number}
         * @memberof NexusModel
         */
        this.taxAuthorityId = undefined;
        /**
         * @type {boolean}
         * @memberof NexusModel
         */
        this.isSellerImporterOfRecord = undefined;
        /**
         * @type {string}
         * @memberof NexusModel
         */
        this.taxName = undefined;
        /**
         * @type {NexusParameterDetailModel[]}
         * @memberof NexusModel
         */
        this.parameters = undefined;
        /**
         * @type {boolean}
         * @memberof NexusModel
         */
        this.taxableNexus = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], NexusModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], NexusModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String),
        __metadata("design:type", String)
    ], NexusModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String),
        __metadata("design:type", String)
    ], NexusModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisTypeId", Enums.JurisTypeIdConverter, true),
        __metadata("design:type", Number)
    ], NexusModel.prototype, "jurisTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionTypeId", Enums.JurisdictionTypeConverter, true),
        __metadata("design:type", Number)
    ], NexusModel.prototype, "jurisdictionTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisCode", String),
        __metadata("design:type", String)
    ], NexusModel.prototype, "jurisCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisName", String),
        __metadata("design:type", String)
    ], NexusModel.prototype, "jurisName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NexusModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NexusModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("shortName", String, true),
        __metadata("design:type", String)
    ], NexusModel.prototype, "shortName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("signatureCode", String, true),
        __metadata("design:type", String)
    ], NexusModel.prototype, "signatureCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("stateAssignedNo", String, true),
        __metadata("design:type", String)
    ], NexusModel.prototype, "stateAssignedNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nexusTypeId", Enums.NexusTypeIdConverter, true),
        __metadata("design:type", Number)
    ], NexusModel.prototype, "nexusTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("sourcing", Enums.SourcingConverter, true),
        __metadata("design:type", Number)
    ], NexusModel.prototype, "sourcing", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hasLocalNexus", Boolean, true),
        __metadata("design:type", Boolean)
    ], NexusModel.prototype, "hasLocalNexus", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("localNexusTypeId", Enums.LocalNexusTypeIdConverter, true),
        __metadata("design:type", Number)
    ], NexusModel.prototype, "localNexusTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hasPermanentEstablishment", Boolean, true),
        __metadata("design:type", Boolean)
    ], NexusModel.prototype, "hasPermanentEstablishment", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxId", String, true),
        __metadata("design:type", String)
    ], NexusModel.prototype, "taxId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("streamlinedSalesTax", Boolean, true),
        __metadata("design:type", Boolean)
    ], NexusModel.prototype, "streamlinedSalesTax", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSSTActive", Boolean, true),
        __metadata("design:type", Boolean)
    ], NexusModel.prototype, "isSSTActive", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NexusModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], NexusModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NexusModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], NexusModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxTypeGroup", String, true),
        __metadata("design:type", String)
    ], NexusModel.prototype, "taxTypeGroup", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nexusTaxTypeGroup", String, true),
        __metadata("design:type", String)
    ], NexusModel.prototype, "nexusTaxTypeGroup", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityId", Number, true),
        __metadata("design:type", Number)
    ], NexusModel.prototype, "taxAuthorityId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSellerImporterOfRecord", Boolean, true),
        __metadata("design:type", Boolean)
    ], NexusModel.prototype, "isSellerImporterOfRecord", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxName", String, true),
        __metadata("design:type", String)
    ], NexusModel.prototype, "taxName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("parameters", [NexusParameterDetailModel_1.NexusParameterDetailModel], true),
        __metadata("design:type", Array)
    ], NexusModel.prototype, "parameters", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxableNexus", Boolean, true),
        __metadata("design:type", Boolean)
    ], NexusModel.prototype, "taxableNexus", void 0);
    NexusModel = __decorate([
        (0, json2typescript_1.JsonObject)("NexusModel")
    ], NexusModel);
    return NexusModel;
}());
