// avatax/lib/models/NexusTaxTypeGroupModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NexusTaxTypeGroupModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Represents a group of tax types
 * @export
 * @class NexusTaxTypeGroupModel
 */
var NexusTaxTypeGroupModel = exports.NexusTaxTypeGroupModel = /** @class */ (function () {
    function NexusTaxTypeGroupModel() {
        /**
         * @type {number}
         * @memberof NexusTaxTypeGroupModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof NexusTaxTypeGroupModel
         */
        this.nexusTaxTypeGroupId = undefined;
        /**
         * @type {string}
         * @memberof NexusTaxTypeGroupModel
         */
        this.description = undefined;
        /**
         * @type {number}
         * @memberof NexusTaxTypeGroupModel
         */
        this.subscriptionTypeId = undefined;
        /**
         * @type {string}
         * @memberof NexusTaxTypeGroupModel
         */
        this.subscriptionDescription = undefined;
        /**
         * @type {string}
         * @memberof NexusTaxTypeGroupModel
         */
        this.tabName = undefined;
        /**
         * @type {boolean}
         * @memberof NexusTaxTypeGroupModel
         */
        this.showColumn = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], NexusTaxTypeGroupModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nexusTaxTypeGroupId", String, true),
        __metadata("design:type", String)
    ], NexusTaxTypeGroupModel.prototype, "nexusTaxTypeGroupId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], NexusTaxTypeGroupModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("subscriptionTypeId", Number, true),
        __metadata("design:type", Number)
    ], NexusTaxTypeGroupModel.prototype, "subscriptionTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("subscriptionDescription", String, true),
        __metadata("design:type", String)
    ], NexusTaxTypeGroupModel.prototype, "subscriptionDescription", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("tabName", String, true),
        __metadata("design:type", String)
    ], NexusTaxTypeGroupModel.prototype, "tabName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("showColumn", Boolean, true),
        __metadata("design:type", Boolean)
    ], NexusTaxTypeGroupModel.prototype, "showColumn", void 0);
    NexusTaxTypeGroupModel = __decorate([
        (0, json2typescript_1.JsonObject)("NexusTaxTypeGroupModel")
    ], NexusTaxTypeGroupModel);
    return NexusTaxTypeGroupModel;
}());
