// avatax/lib/models/NoticeModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NoticeModel = void 0;
var Enums = require("../enums/index");
var NoticeCommentModel_1 = require("./NoticeCommentModel");
var NoticeFinanceModel_1 = require("./NoticeFinanceModel");
var NoticeResponsibilityDetailModel_1 = require("./NoticeResponsibilityDetailModel");
var NoticeRootCauseDetailModel_1 = require("./NoticeRootCauseDetailModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Represents a letter received from a tax authority regarding tax filing.
These letters often have the warning "Notice" printed at the top, which is why
they are called "Notices".
 * @export
 * @class NoticeModel
 */
var NoticeModel = exports.NoticeModel = /** @class */ (function () {
    function NoticeModel() {
        /**
         * @type {number}
         * @memberof NoticeModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof NoticeModel
         */
        this.companyId = undefined;
        /**
         * @type {number}
         * @memberof NoticeModel
         */
        this.statusId = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.status = undefined;
        /**
         * @type {Date}
         * @memberof NoticeModel
         */
        this.receivedDate = undefined;
        /**
         * @type {Date}
         * @memberof NoticeModel
         */
        this.closedDate = undefined;
        /**
         * @type {number}
         * @memberof NoticeModel
         */
        this.totalRemit = undefined;
        /**
         * @type {Enums.NoticeCustomerType}
         * @memberof NoticeModel
         */
        this.customerTypeId = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.region = undefined;
        /**
         * @type {number}
         * @memberof NoticeModel
         */
        this.taxAuthorityId = undefined;
        /**
         * @type {Enums.FilingFrequencyId}
         * @memberof NoticeModel
         */
        this.filingFrequency = undefined;
        /**
         * @type {Enums.TaxNoticeFilingTypeId}
         * @memberof NoticeModel
         */
        this.filingTypeId = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.ticketReferenceNo = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.ticketReferenceUrl = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.salesForceCase = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.salesForceCaseUrl = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.taxPeriod = undefined;
        /**
         * @type {number}
         * @memberof NoticeModel
         */
        this.reasonId = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.reason = undefined;
        /**
         * @type {number}
         * @memberof NoticeModel
         */
        this.typeId = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.type = undefined;
        /**
         * @type {Enums.FundingOption}
         * @memberof NoticeModel
         */
        this.customerFundingOptionId = undefined;
        /**
         * @type {Enums.NoticePriorityId}
         * @memberof NoticeModel
         */
        this.priorityId = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.customerComment = undefined;
        /**
         * @type {boolean}
         * @memberof NoticeModel
         */
        this.hideFromCustomer = undefined;
        /**
         * @type {Date}
         * @memberof NoticeModel
         */
        this.expectedResolutionDate = undefined;
        /**
         * @type {boolean}
         * @memberof NoticeModel
         */
        this.showResolutionDateToCustomer = undefined;
        /**
         * @type {number}
         * @memberof NoticeModel
         */
        this.closedByUserId = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.createdByUserName = undefined;
        /**
         * @type {number}
         * @memberof NoticeModel
         */
        this.ownedByUserId = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.description = undefined;
        /**
         * @type {number}
         * @memberof NoticeModel
         */
        this.avaFileFormId = undefined;
        /**
         * @type {number}
         * @memberof NoticeModel
         */
        this.revenueContactId = undefined;
        /**
         * @type {number}
         * @memberof NoticeModel
         */
        this.complianceContactId = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.taxFormCode = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.documentReference = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.jurisdictionName = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.jurisdictionType = undefined;
        /**
         * @type {NoticeCommentModel[]}
         * @memberof NoticeModel
         */
        this.comments = undefined;
        /**
         * @type {NoticeFinanceModel[]}
         * @memberof NoticeModel
         */
        this.finances = undefined;
        /**
         * @type {NoticeResponsibilityDetailModel[]}
         * @memberof NoticeModel
         */
        this.responsibility = undefined;
        /**
         * @type {NoticeRootCauseDetailModel[]}
         * @memberof NoticeModel
         */
        this.rootCause = undefined;
        /**
         * @type {Date}
         * @memberof NoticeModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof NoticeModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof NoticeModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof NoticeModel
         */
        this.modifiedUserId = undefined;
        /**
         * @type {string}
         * @memberof NoticeModel
         */
        this.registrationId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("statusId", Number),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "statusId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("status", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "status", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("receivedDate", dateConverter_1.DateConverter),
        __metadata("design:type", Date)
    ], NoticeModel.prototype, "receivedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("closedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NoticeModel.prototype, "closedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalRemit", Number, true),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "totalRemit", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerTypeId", Enums.NoticeCustomerTypeConverter),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "customerTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityId", Number, true),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "taxAuthorityId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingFrequency", Enums.FilingFrequencyIdConverter, true),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "filingFrequency", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("filingTypeId", Enums.TaxNoticeFilingTypeIdConverter, true),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "filingTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("ticketReferenceNo", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "ticketReferenceNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("ticketReferenceUrl", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "ticketReferenceUrl", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("salesForceCase", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "salesForceCase", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("salesForceCaseUrl", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "salesForceCaseUrl", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxPeriod", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "taxPeriod", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reasonId", Number),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "reasonId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reason", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "reason", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("typeId", Number, true),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "typeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("type", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "type", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerFundingOptionId", Enums.FundingOptionConverter, true),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "customerFundingOptionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("priorityId", Enums.NoticePriorityIdConverter),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "priorityId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerComment", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "customerComment", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hideFromCustomer", Boolean),
        __metadata("design:type", Boolean)
    ], NoticeModel.prototype, "hideFromCustomer", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("expectedResolutionDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NoticeModel.prototype, "expectedResolutionDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("showResolutionDateToCustomer", Boolean),
        __metadata("design:type", Boolean)
    ], NoticeModel.prototype, "showResolutionDateToCustomer", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("closedByUserId", Number, true),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "closedByUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdByUserName", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "createdByUserName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("ownedByUserId", Number, true),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "ownedByUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("avaFileFormId", Number, true),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "avaFileFormId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("revenueContactId", Number, true),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "revenueContactId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("complianceContactId", Number, true),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "complianceContactId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxFormCode", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "taxFormCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("documentReference", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "documentReference", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionName", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "jurisdictionName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionType", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "jurisdictionType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("comments", [NoticeCommentModel_1.NoticeCommentModel], true),
        __metadata("design:type", Array)
    ], NoticeModel.prototype, "comments", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("finances", [NoticeFinanceModel_1.NoticeFinanceModel], true),
        __metadata("design:type", Array)
    ], NoticeModel.prototype, "finances", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("responsibility", [NoticeResponsibilityDetailModel_1.NoticeResponsibilityDetailModel], true),
        __metadata("design:type", Array)
    ], NoticeModel.prototype, "responsibility", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rootCause", [NoticeRootCauseDetailModel_1.NoticeRootCauseDetailModel], true),
        __metadata("design:type", Array)
    ], NoticeModel.prototype, "rootCause", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NoticeModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], NoticeModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], NoticeModel.prototype, "modifiedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("registrationId", String, true),
        __metadata("design:type", String)
    ], NoticeModel.prototype, "registrationId", void 0);
    NoticeModel = __decorate([
        (0, json2typescript_1.JsonObject)("NoticeModel")
    ], NoticeModel);
    return NoticeModel;
}());
