// avatax/lib/models/ParameterUsageModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParameterUsageModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * usage of system defined parameters.
 * @export
 * @class ParameterUsageModel
 */
var ParameterUsageModel = exports.ParameterUsageModel = /** @class */ (function () {
    function ParameterUsageModel() {
        /**
         * @type {number}
         * @memberof ParameterUsageModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof ParameterUsageModel
         */
        this.parameterId = undefined;
        /**
         * @type {string}
         * @memberof ParameterUsageModel
         */
        this.productCode = undefined;
        /**
         * @type {string}
         * @memberof ParameterUsageModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof ParameterUsageModel
         */
        this.region = undefined;
        /**
         * @type {number}
         * @memberof ParameterUsageModel
         */
        this.systemId = undefined;
        /**
         * @type {string}
         * @memberof ParameterUsageModel
         */
        this.taxTypeId = undefined;
        /**
         * @type {string}
         * @memberof ParameterUsageModel
         */
        this.attributeType = undefined;
        /**
         * @type {string}
         * @memberof ParameterUsageModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof ParameterUsageModel
         */
        this.dataType = undefined;
        /**
         * @type {string}
         * @memberof ParameterUsageModel
         */
        this.helpText = undefined;
        /**
         * @type {string}
         * @memberof ParameterUsageModel
         */
        this.label = undefined;
        /**
         * @type {string}
         * @memberof ParameterUsageModel
         */
        this.helpUrl = undefined;
        /**
         * @type {string[]}
         * @memberof ParameterUsageModel
         */
        this.values = undefined;
        /**
         * @type {string[]}
         * @memberof ParameterUsageModel
         */
        this.valueDescriptions = undefined;
        /**
         * @type {string}
         * @memberof ParameterUsageModel
         */
        this.measurementType = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], ParameterUsageModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("parameterId", Number, true),
        __metadata("design:type", Number)
    ], ParameterUsageModel.prototype, "parameterId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("productCode", String, true),
        __metadata("design:type", String)
    ], ParameterUsageModel.prototype, "productCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], ParameterUsageModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], ParameterUsageModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("systemId", Number, true),
        __metadata("design:type", Number)
    ], ParameterUsageModel.prototype, "systemId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxTypeId", String, true),
        __metadata("design:type", String)
    ], ParameterUsageModel.prototype, "taxTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("attributeType", String, true),
        __metadata("design:type", String)
    ], ParameterUsageModel.prototype, "attributeType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String, true),
        __metadata("design:type", String)
    ], ParameterUsageModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dataType", String, true),
        __metadata("design:type", String)
    ], ParameterUsageModel.prototype, "dataType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("helpText", String, true),
        __metadata("design:type", String)
    ], ParameterUsageModel.prototype, "helpText", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("label", String, true),
        __metadata("design:type", String)
    ], ParameterUsageModel.prototype, "label", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("helpUrl", String, true),
        __metadata("design:type", String)
    ], ParameterUsageModel.prototype, "helpUrl", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("values", [String], true),
        __metadata("design:type", Array)
    ], ParameterUsageModel.prototype, "values", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("valueDescriptions", [String], true),
        __metadata("design:type", Array)
    ], ParameterUsageModel.prototype, "valueDescriptions", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("measurementType", String, true),
        __metadata("design:type", String)
    ], ParameterUsageModel.prototype, "measurementType", void 0);
    ParameterUsageModel = __decorate([
        (0, json2typescript_1.JsonObject)("ParameterUsageModel")
    ], ParameterUsageModel);
    return ParameterUsageModel;
}());
