// avatax/lib/models/PingResultModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PingResultModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
/**
 * Ping Result Model
 * @export
 * @class PingResultModel
 */
var PingResultModel = exports.PingResultModel = /** @class */ (function () {
    function PingResultModel() {
        /**
         * @type {string}
         * @memberof PingResultModel
         */
        this.version = undefined;
        /**
         * @type {boolean}
         * @memberof PingResultModel
         */
        this.authenticated = undefined;
        /**
         * @type {Enums.AuthenticationTypeId}
         * @memberof PingResultModel
         */
        this.authenticationType = undefined;
        /**
         * @type {string}
         * @memberof PingResultModel
         */
        this.authenticatedUserName = undefined;
        /**
         * @type {number}
         * @memberof PingResultModel
         */
        this.authenticatedUserId = undefined;
        /**
         * @type {number}
         * @memberof PingResultModel
         */
        this.authenticatedAccountId = undefined;
        /**
         * @type {number}
         * @memberof PingResultModel
         */
        this.authenticatedCompanyId = undefined;
        /**
         * @type {string}
         * @memberof PingResultModel
         */
        this.crmid = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("version", String, true),
        __metadata("design:type", String)
    ], PingResultModel.prototype, "version", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("authenticated", Boolean, true),
        __metadata("design:type", Boolean)
    ], PingResultModel.prototype, "authenticated", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("authenticationType", Enums.AuthenticationTypeIdConverter, true),
        __metadata("design:type", Number)
    ], PingResultModel.prototype, "authenticationType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("authenticatedUserName", String, true),
        __metadata("design:type", String)
    ], PingResultModel.prototype, "authenticatedUserName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("authenticatedUserId", Number, true),
        __metadata("design:type", Number)
    ], PingResultModel.prototype, "authenticatedUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("authenticatedAccountId", Number, true),
        __metadata("design:type", Number)
    ], PingResultModel.prototype, "authenticatedAccountId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("authenticatedCompanyId", Number, true),
        __metadata("design:type", Number)
    ], PingResultModel.prototype, "authenticatedCompanyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("crmid", String, true),
        __metadata("design:type", String)
    ], PingResultModel.prototype, "crmid", void 0);
    PingResultModel = __decorate([
        (0, json2typescript_1.JsonObject)("PingResultModel")
    ], PingResultModel);
    return PingResultModel;
}());
