// avatax/lib/models/PointOfSaleDataRequestModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PointOfSaleDataRequestModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Point-of-Sale Data Request Model
 * @export
 * @class PointOfSaleDataRequestModel
 */
var PointOfSaleDataRequestModel = exports.PointOfSaleDataRequestModel = /** @class */ (function () {
    function PointOfSaleDataRequestModel() {
        /**
         * @type {string}
         * @memberof PointOfSaleDataRequestModel
         */
        this.companyCode = undefined;
        /**
         * @type {Date}
         * @memberof PointOfSaleDataRequestModel
         */
        this.documentDate = undefined;
        /**
         * @type {Enums.PointOfSaleFileType}
         * @memberof PointOfSaleDataRequestModel
         */
        this.responseType = undefined;
        /**
         * @type {string[]}
         * @memberof PointOfSaleDataRequestModel
         */
        this.taxCodes = undefined;
        /**
         * @type {string[]}
         * @memberof PointOfSaleDataRequestModel
         */
        this.itemCodes = undefined;
        /**
         * @type {string[]}
         * @memberof PointOfSaleDataRequestModel
         */
        this.locationCodes = undefined;
        /**
         * @type {boolean}
         * @memberof PointOfSaleDataRequestModel
         */
        this.includeJurisCodes = undefined;
        /**
         * @type {Enums.PointOfSalePartnerId}
         * @memberof PointOfSaleDataRequestModel
         */
        this.partnerId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyCode", String),
        __metadata("design:type", String)
    ], PointOfSaleDataRequestModel.prototype, "companyCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("documentDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], PointOfSaleDataRequestModel.prototype, "documentDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("responseType", Enums.PointOfSaleFileTypeConverter, true),
        __metadata("design:type", Number)
    ], PointOfSaleDataRequestModel.prototype, "responseType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCodes", [String], true),
        __metadata("design:type", Array)
    ], PointOfSaleDataRequestModel.prototype, "taxCodes", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("itemCodes", [String], true),
        __metadata("design:type", Array)
    ], PointOfSaleDataRequestModel.prototype, "itemCodes", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("locationCodes", [String], true),
        __metadata("design:type", Array)
    ], PointOfSaleDataRequestModel.prototype, "locationCodes", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("includeJurisCodes", Boolean, true),
        __metadata("design:type", Boolean)
    ], PointOfSaleDataRequestModel.prototype, "includeJurisCodes", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("partnerId", Enums.PointOfSalePartnerIdConverter, true),
        __metadata("design:type", Number)
    ], PointOfSaleDataRequestModel.prototype, "partnerId", void 0);
    PointOfSaleDataRequestModel = __decorate([
        (0, json2typescript_1.JsonObject)("PointOfSaleDataRequestModel")
    ], PointOfSaleDataRequestModel);
    return PointOfSaleDataRequestModel;
}());
