// avatax/lib/models/PreferredProgramModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PreferredProgramModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * A preferred program is a customs and/or duty program that can be used to handle cross-border transactions.
Customers who sign up for a preferred program may obtain better terms for their customs and duty payments.
            
To indicate that your company has signed up for a preferred program, specify the `code` value from this
object as the value for the `AvaTax.LC.PreferredProgram` parameter in your transaction.
 * @export
 * @class PreferredProgramModel
 */
var PreferredProgramModel = exports.PreferredProgramModel = /** @class */ (function () {
    function PreferredProgramModel() {
        /**
         * @type {number}
         * @memberof PreferredProgramModel
         */
        this.id = undefined;
        /**
         * @type {string}
         * @memberof PreferredProgramModel
         */
        this.code = undefined;
        /**
         * @type {string}
         * @memberof PreferredProgramModel
         */
        this.originCountry = undefined;
        /**
         * @type {string}
         * @memberof PreferredProgramModel
         */
        this.destinationCountry = undefined;
        /**
         * @type {Date}
         * @memberof PreferredProgramModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof PreferredProgramModel
         */
        this.endDate = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], PreferredProgramModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("code", String, true),
        __metadata("design:type", String)
    ], PreferredProgramModel.prototype, "code", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("originCountry", String, true),
        __metadata("design:type", String)
    ], PreferredProgramModel.prototype, "originCountry", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("destinationCountry", String, true),
        __metadata("design:type", String)
    ], PreferredProgramModel.prototype, "destinationCountry", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], PreferredProgramModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], PreferredProgramModel.prototype, "endDate", void 0);
    PreferredProgramModel = __decorate([
        (0, json2typescript_1.JsonObject)("PreferredProgramModel")
    ], PreferredProgramModel);
    return PreferredProgramModel;
}());
