// avatax/lib/models/ProblemDetails.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProblemDetails = void 0;
var json2typescript_1 = require("json2typescript");
/**
 *
 * @export
 * @class ProblemDetails
 */
var ProblemDetails = exports.ProblemDetails = /** @class */ (function () {
    function ProblemDetails() {
        /**
         * @type {string}
         * @memberof ProblemDetails
         */
        this.type = undefined;
        /**
         * @type {string}
         * @memberof ProblemDetails
         */
        this.title = undefined;
        /**
         * @type {number}
         * @memberof ProblemDetails
         */
        this.status = undefined;
        /**
         * @type {string}
         * @memberof ProblemDetails
         */
        this.detail = undefined;
        /**
         * @type {string}
         * @memberof ProblemDetails
         */
        this.instance = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("type", String, true),
        __metadata("design:type", String)
    ], ProblemDetails.prototype, "type", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("title", String, true),
        __metadata("design:type", String)
    ], ProblemDetails.prototype, "title", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("status", Number, true),
        __metadata("design:type", Number)
    ], ProblemDetails.prototype, "status", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("detail", String, true),
        __metadata("design:type", String)
    ], ProblemDetails.prototype, "detail", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("instance", String, true),
        __metadata("design:type", String)
    ], ProblemDetails.prototype, "instance", void 0);
    ProblemDetails = __decorate([
        (0, json2typescript_1.JsonObject)("ProblemDetails")
    ], ProblemDetails);
    return ProblemDetails;
}());
