// avatax/lib/models/QueryRequestModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryRequestModel = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Represents a complex query request to parse using query filter guidelines from Microsoft REST standards
 * @export
 * @class QueryRequestModel
 */
var QueryRequestModel = exports.QueryRequestModel = /** @class */ (function () {
    function QueryRequestModel() {
        /**
         * @type {string}
         * @memberof QueryRequestModel
         */
        this.filter = undefined;
        /**
         * @type {string}
         * @memberof QueryRequestModel
         */
        this.include = undefined;
        /**
         * @type {number}
         * @memberof QueryRequestModel
         */
        this.maxResults = undefined;
        /**
         * @type {number}
         * @memberof QueryRequestModel
         */
        this.startIndex = undefined;
        /**
         * @type {string}
         * @memberof QueryRequestModel
         */
        this.sortBy = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("filter", String, true),
        __metadata("design:type", String)
    ], QueryRequestModel.prototype, "filter", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("include", String, true),
        __metadata("design:type", String)
    ], QueryRequestModel.prototype, "include", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("maxResults", Number, true),
        __metadata("design:type", Number)
    ], QueryRequestModel.prototype, "maxResults", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("startIndex", Number, true),
        __metadata("design:type", Number)
    ], QueryRequestModel.prototype, "startIndex", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("sortBy", String, true),
        __metadata("design:type", String)
    ], QueryRequestModel.prototype, "sortBy", void 0);
    QueryRequestModel = __decorate([
        (0, json2typescript_1.JsonObject)("QueryRequestModel")
    ], QueryRequestModel);
    return QueryRequestModel;
}());
