// avatax/lib/models/SettingModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * This object is used to keep track of custom information about a company.
            
The company settings system is a metadata system that you can use to store extra information
about a company.  Your integration or connector could use this data storage to keep track of
preference information, reminders, or any other storage that would need to persist even if
the customer uninstalls your application.
            
A setting can refer to any type of data you need to remember about this company object.
When creating this object, you may define your own `set`, `name`, and `value` parameters.
To define your own values, please choose a `set` name that begins with `X-` to indicate an extension.
 * @export
 * @class SettingModel
 */
var SettingModel = exports.SettingModel = /** @class */ (function () {
    function SettingModel() {
        /**
         * @type {number}
         * @memberof SettingModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof SettingModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof SettingModel
         */
        this.set = undefined;
        /**
         * @type {string}
         * @memberof SettingModel
         */
        this.name = undefined;
        /**
         * @type {string}
         * @memberof SettingModel
         */
        this.value = undefined;
        /**
         * @type {Date}
         * @memberof SettingModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof SettingModel
         */
        this.modifiedUserId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number),
        __metadata("design:type", Number)
    ], SettingModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], SettingModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("set", String),
        __metadata("design:type", String)
    ], SettingModel.prototype, "set", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String),
        __metadata("design:type", String)
    ], SettingModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("value", String, true),
        __metadata("design:type", String)
    ], SettingModel.prototype, "value", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], SettingModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], SettingModel.prototype, "modifiedUserId", void 0);
    SettingModel = __decorate([
        (0, json2typescript_1.JsonObject)("SettingModel")
    ], SettingModel);
    return SettingModel;
}());
