// avatax/lib/models/ShippingVerifyResult.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShippingVerifyResult = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
/**
 * The Response of the /shippingverify endpoint. Describes the result of checking all applicable shipping rules against each line in the transaction.
 * @export
 * @class ShippingVerifyResult
 */
var ShippingVerifyResult = exports.ShippingVerifyResult = /** @class */ (function () {
    function ShippingVerifyResult() {
        /**
         * @type {boolean}
         * @memberof ShippingVerifyResult
         */
        this.compliant = undefined;
        /**
         * @type {string}
         * @memberof ShippingVerifyResult
         */
        this.message = undefined;
        /**
         * @type {string}
         * @memberof ShippingVerifyResult
         */
        this.successMessages = undefined;
        /**
         * @type {string}
         * @memberof ShippingVerifyResult
         */
        this.failureMessages = undefined;
        /**
         * @type {Enums.FailureCodes[]}
         * @memberof ShippingVerifyResult
         */
        this.failureCodes = undefined;
        /**
         * @type {string[]}
         * @memberof ShippingVerifyResult
         */
        this.warningCodes = undefined;
        /**
         * @type {object[]}
         * @memberof ShippingVerifyResult
         */
        this.lines = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("compliant", Boolean, true),
        __metadata("design:type", Boolean)
    ], ShippingVerifyResult.prototype, "compliant", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("message", String, true),
        __metadata("design:type", String)
    ], ShippingVerifyResult.prototype, "message", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("successMessages", String, true),
        __metadata("design:type", String)
    ], ShippingVerifyResult.prototype, "successMessages", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("failureMessages", String, true),
        __metadata("design:type", String)
    ], ShippingVerifyResult.prototype, "failureMessages", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("failureCodes", [Enums.FailureCodes], true),
        __metadata("design:type", Array)
    ], ShippingVerifyResult.prototype, "failureCodes", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("warningCodes", [String], true),
        __metadata("design:type", Array)
    ], ShippingVerifyResult.prototype, "warningCodes", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lines", [Object], true),
        __metadata("design:type", Array)
    ], ShippingVerifyResult.prototype, "lines", void 0);
    ShippingVerifyResult = __decorate([
        (0, json2typescript_1.JsonObject)("ShippingVerifyResult")
    ], ShippingVerifyResult);
    return ShippingVerifyResult;
}());
