// avatax/lib/models/SkyscraperStatusModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkyscraperStatusModel = void 0;
var Enums = require("../enums/index");
var requiredFilingCalendarDataFieldModel_1 = require("./requiredFilingCalendarDataFieldModel");
var json2typescript_1 = require("json2typescript");
/**
 * Represents a list of statuses of returns available in skyscraper
 * @export
 * @class SkyscraperStatusModel
 */
var SkyscraperStatusModel = exports.SkyscraperStatusModel = /** @class */ (function () {
    function SkyscraperStatusModel() {
        /**
         * @type {string}
         * @memberof SkyscraperStatusModel
         */
        this.name = undefined;
        /**
         * @type {string[]}
         * @memberof SkyscraperStatusModel
         */
        this.taxFormCodes = undefined;
        /**
         * @type {string}
         * @memberof SkyscraperStatusModel
         */
        this.country = undefined;
        /**
         * @type {Enums.ScraperType}
         * @memberof SkyscraperStatusModel
         */
        this.scraperType = undefined;
        /**
         * @type {boolean}
         * @memberof SkyscraperStatusModel
         */
        this.isAvailable = undefined;
        /**
         * @type {string}
         * @memberof SkyscraperStatusModel
         */
        this.expectedResponseTime = undefined;
        /**
         * @type {string}
         * @memberof SkyscraperStatusModel
         */
        this.message = undefined;
        /**
         * @type {requiredFilingCalendarDataFieldModel[]}
         * @memberof SkyscraperStatusModel
         */
        this.requiredFilingCalendarDataFields = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("name", String, true),
        __metadata("design:type", String)
    ], SkyscraperStatusModel.prototype, "name", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxFormCodes", [String], true),
        __metadata("design:type", Array)
    ], SkyscraperStatusModel.prototype, "taxFormCodes", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], SkyscraperStatusModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("scraperType", Enums.ScraperTypeConverter, true),
        __metadata("design:type", Number)
    ], SkyscraperStatusModel.prototype, "scraperType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isAvailable", Boolean, true),
        __metadata("design:type", Boolean)
    ], SkyscraperStatusModel.prototype, "isAvailable", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("expectedResponseTime", String, true),
        __metadata("design:type", String)
    ], SkyscraperStatusModel.prototype, "expectedResponseTime", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("message", String, true),
        __metadata("design:type", String)
    ], SkyscraperStatusModel.prototype, "message", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("requiredFilingCalendarDataFields", [requiredFilingCalendarDataFieldModel_1.RequiredFilingCalendarDataFieldModel], true),
        __metadata("design:type", Array)
    ], SkyscraperStatusModel.prototype, "requiredFilingCalendarDataFields", void 0);
    SkyscraperStatusModel = __decorate([
        (0, json2typescript_1.JsonObject)("SkyscraperStatusModel")
    ], SkyscraperStatusModel);
    return SkyscraperStatusModel;
}());
