// avatax/lib/models/TaxDetailsByTaxType.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxDetailsByTaxType = void 0;
var TaxDetailsByTaxSubType_1 = require("./TaxDetailsByTaxSubType");
var json2typescript_1 = require("json2typescript");
/**
 * Tax Details by Tax Type
 * @export
 * @class TaxDetailsByTaxType
 */
var TaxDetailsByTaxType = exports.TaxDetailsByTaxType = /** @class */ (function () {
    function TaxDetailsByTaxType() {
        /**
         * @type {string}
         * @memberof TaxDetailsByTaxType
         */
        this.taxType = undefined;
        /**
         * @type {number}
         * @memberof TaxDetailsByTaxType
         */
        this.totalTaxable = undefined;
        /**
         * @type {number}
         * @memberof TaxDetailsByTaxType
         */
        this.totalExempt = undefined;
        /**
         * @type {number}
         * @memberof TaxDetailsByTaxType
         */
        this.totalNonTaxable = undefined;
        /**
         * @type {number}
         * @memberof TaxDetailsByTaxType
         */
        this.totalTax = undefined;
        /**
         * @type {TaxDetailsByTaxSubType[]}
         * @memberof TaxDetailsByTaxType
         */
        this.taxSubTypeDetails = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxType", String, true),
        __metadata("design:type", String)
    ], TaxDetailsByTaxType.prototype, "taxType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalTaxable", Number, true),
        __metadata("design:type", Number)
    ], TaxDetailsByTaxType.prototype, "totalTaxable", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalExempt", Number, true),
        __metadata("design:type", Number)
    ], TaxDetailsByTaxType.prototype, "totalExempt", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalNonTaxable", Number, true),
        __metadata("design:type", Number)
    ], TaxDetailsByTaxType.prototype, "totalNonTaxable", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalTax", Number, true),
        __metadata("design:type", Number)
    ], TaxDetailsByTaxType.prototype, "totalTax", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxSubTypeDetails", [TaxDetailsByTaxSubType_1.TaxDetailsByTaxSubType], true),
        __metadata("design:type", Array)
    ], TaxDetailsByTaxType.prototype, "taxSubTypeDetails", void 0);
    TaxDetailsByTaxType = __decorate([
        (0, json2typescript_1.JsonObject)("TaxDetailsByTaxType")
    ], TaxDetailsByTaxType);
    return TaxDetailsByTaxType;
}());
