// avatax/lib/models/TaxDetailsByTaxSubType.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxDetailsByTaxSubType = void 0;
var json2typescript_1 = require("json2typescript");
/**
 * Tax Details by Tax subtype
 * @export
 * @class TaxDetailsByTaxSubType
 */
var TaxDetailsByTaxSubType = exports.TaxDetailsByTaxSubType = /** @class */ (function () {
    function TaxDetailsByTaxSubType() {
        /**
         * @type {string}
         * @memberof TaxDetailsByTaxSubType
         */
        this.taxSubType = undefined;
        /**
         * @type {number}
         * @memberof TaxDetailsByTaxSubType
         */
        this.totalTaxable = undefined;
        /**
         * @type {number}
         * @memberof TaxDetailsByTaxSubType
         */
        this.totalExempt = undefined;
        /**
         * @type {number}
         * @memberof TaxDetailsByTaxSubType
         */
        this.totalNonTaxable = undefined;
        /**
         * @type {number}
         * @memberof TaxDetailsByTaxSubType
         */
        this.totalTax = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxSubType", String, true),
        __metadata("design:type", String)
    ], TaxDetailsByTaxSubType.prototype, "taxSubType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalTaxable", Number, true),
        __metadata("design:type", Number)
    ], TaxDetailsByTaxSubType.prototype, "totalTaxable", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalExempt", Number, true),
        __metadata("design:type", Number)
    ], TaxDetailsByTaxSubType.prototype, "totalExempt", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalNonTaxable", Number, true),
        __metadata("design:type", Number)
    ], TaxDetailsByTaxSubType.prototype, "totalNonTaxable", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalTax", Number, true),
        __metadata("design:type", Number)
    ], TaxDetailsByTaxSubType.prototype, "totalTax", void 0);
    TaxDetailsByTaxSubType = __decorate([
        (0, json2typescript_1.JsonObject)("TaxDetailsByTaxSubType")
    ], TaxDetailsByTaxSubType);
    return TaxDetailsByTaxSubType;
}());
