// avatax/lib/models/TransactionBatchItemModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionBatchItemModel = void 0;
var CreateTransactionModel_1 = require("./CreateTransactionModel");
var BatchAdjustTransactionModel_1 = require("./BatchAdjustTransactionModel");
var CreateOrAdjustTransactionModel_1 = require("./CreateOrAdjustTransactionModel");
var BatchVoidTransactionModel_1 = require("./BatchVoidTransactionModel");
var json2typescript_1 = require("json2typescript");
/**
 * Represents a transaction batch item.
Only one child transaction model should contain data.
 * @export
 * @class TransactionBatchItemModel
 */
var TransactionBatchItemModel = exports.TransactionBatchItemModel = /** @class */ (function () {
    function TransactionBatchItemModel() {
        /**
         * @type {string}
         * @memberof TransactionBatchItemModel
         */
        this.memo = undefined;
        /**
         * @type {CreateTransactionModel}
         * @memberof TransactionBatchItemModel
         */
        this.createTransactionModel = undefined;
        /**
         * @type {BatchAdjustTransactionModel}
         * @memberof TransactionBatchItemModel
         */
        this.adjustTransactionModel = undefined;
        /**
         * @type {CreateOrAdjustTransactionModel}
         * @memberof TransactionBatchItemModel
         */
        this.createOrAdjustTransactionModel = undefined;
        /**
         * @type {BatchVoidTransactionModel}
         * @memberof TransactionBatchItemModel
         */
        this.voidTransactionModel = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("memo", String, true),
        __metadata("design:type", String)
    ], TransactionBatchItemModel.prototype, "memo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createTransactionModel", CreateTransactionModel_1.CreateTransactionModel, true),
        __metadata("design:type", CreateTransactionModel_1.CreateTransactionModel)
    ], TransactionBatchItemModel.prototype, "createTransactionModel", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("adjustTransactionModel", BatchAdjustTransactionModel_1.BatchAdjustTransactionModel, true),
        __metadata("design:type", BatchAdjustTransactionModel_1.BatchAdjustTransactionModel)
    ], TransactionBatchItemModel.prototype, "adjustTransactionModel", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createOrAdjustTransactionModel", CreateOrAdjustTransactionModel_1.CreateOrAdjustTransactionModel, true),
        __metadata("design:type", CreateOrAdjustTransactionModel_1.CreateOrAdjustTransactionModel)
    ], TransactionBatchItemModel.prototype, "createOrAdjustTransactionModel", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("voidTransactionModel", BatchVoidTransactionModel_1.BatchVoidTransactionModel, true),
        __metadata("design:type", BatchVoidTransactionModel_1.BatchVoidTransactionModel)
    ], TransactionBatchItemModel.prototype, "voidTransactionModel", void 0);
    TransactionBatchItemModel = __decorate([
        (0, json2typescript_1.JsonObject)("TransactionBatchItemModel")
    ], TransactionBatchItemModel);
    return TransactionBatchItemModel;
}());
