// avatax/lib/models/TransactionLineModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionLineModel = void 0;
var Enums = require("../enums/index");
var TransactionLineDetailModel_1 = require("./TransactionLineDetailModel");
var TransactionLineLocationTypeModel_1 = require("./TransactionLineLocationTypeModel");
var TransactionLineParameterModel_1 = require("./TransactionLineParameterModel");
var TransactionLineUserDefinedFieldModel_1 = require("./TransactionLineUserDefinedFieldModel");
var TransactionLineTaxAmountByTaxTypeModel_1 = require("./TransactionLineTaxAmountByTaxTypeModel");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * One line item on this transaction.
 * @export
 * @class TransactionLineModel
 */
var TransactionLineModel = exports.TransactionLineModel = /** @class */ (function () {
    function TransactionLineModel() {
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.transactionId = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.lineNumber = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.boundaryOverrideId = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.customerUsageType = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.entityUseCode = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.description = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.destinationAddressId = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.originAddressId = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.discountAmount = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.discountTypeId = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.exemptAmount = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.exemptCertId = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.certificateId = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.exemptNo = undefined;
        /**
         * @type {boolean}
         * @memberof TransactionLineModel
         */
        this.isItemTaxable = undefined;
        /**
         * @type {boolean}
         * @memberof TransactionLineModel
         */
        this.isSSTP = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.itemCode = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.lineAmount = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.quantity = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.ref1 = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.ref2 = undefined;
        /**
         * @type {Date}
         * @memberof TransactionLineModel
         */
        this.reportingDate = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.revAccount = undefined;
        /**
         * @type {Enums.Sourcing}
         * @memberof TransactionLineModel
         */
        this.sourcing = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.tax = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.taxableAmount = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.taxCalculated = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.taxCode = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.taxCodeId = undefined;
        /**
         * @type {Date}
         * @memberof TransactionLineModel
         */
        this.taxDate = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.taxEngine = undefined;
        /**
         * @type {Enums.TaxOverrideType}
         * @memberof TransactionLineModel
         */
        this.taxOverrideType = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.businessIdentificationNo = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.taxOverrideAmount = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.taxOverrideReason = undefined;
        /**
         * @type {boolean}
         * @memberof TransactionLineModel
         */
        this.taxIncluded = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.merchantSellerId = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.merchantSellerIdentifier = undefined;
        /**
         * @type {Enums.MarketplaceLiabilityType}
         * @memberof TransactionLineModel
         */
        this.marketplaceLiabilityType = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.originationDocumentId = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.originationSite = undefined;
        /**
         * @type {TransactionLineDetailModel[]}
         * @memberof TransactionLineModel
         */
        this.details = undefined;
        /**
         * @type {TransactionLineDetailModel[]}
         * @memberof TransactionLineModel
         */
        this.accountPayableSalesTaxDetails = undefined;
        /**
         * @type {TransactionLineDetailModel[]}
         * @memberof TransactionLineModel
         */
        this.nonPassthroughDetails = undefined;
        /**
         * @type {TransactionLineLocationTypeModel[]}
         * @memberof TransactionLineModel
         */
        this.lineLocationTypes = undefined;
        /**
         * @type {TransactionLineParameterModel[]}
         * @memberof TransactionLineModel
         */
        this.parameters = undefined;
        /**
         * @type {TransactionLineUserDefinedFieldModel[]}
         * @memberof TransactionLineModel
         */
        this.userDefinedFields = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.hsCode = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.costInsuranceFreight = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.vatCode = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineModel
         */
        this.vatNumberTypeId = undefined;
        /**
         * @type {TransactionLineTaxAmountByTaxTypeModel[]}
         * @memberof TransactionLineModel
         */
        this.taxAmountByTaxTypes = undefined;
        /**
         * @type {Enums.DeemedSellerType}
         * @memberof TransactionLineModel
         */
        this.deemedSupplier = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.category = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineModel
         */
        this.summary = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("transactionId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "transactionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lineNumber", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "lineNumber", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("boundaryOverrideId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "boundaryOverrideId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("customerUsageType", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "customerUsageType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("entityUseCode", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "entityUseCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("destinationAddressId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "destinationAddressId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("originAddressId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "originAddressId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("discountAmount", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "discountAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("discountTypeId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "discountTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptAmount", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "exemptAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptCertId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "exemptCertId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("certificateId", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "certificateId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptNo", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "exemptNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isItemTaxable", Boolean, true),
        __metadata("design:type", Boolean)
    ], TransactionLineModel.prototype, "isItemTaxable", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSSTP", Boolean, true),
        __metadata("design:type", Boolean)
    ], TransactionLineModel.prototype, "isSSTP", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("itemCode", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "itemCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lineAmount", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "lineAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("quantity", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "quantity", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("ref1", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "ref1", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("ref2", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "ref2", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportingDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TransactionLineModel.prototype, "reportingDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("revAccount", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "revAccount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("sourcing", Enums.SourcingConverter, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "sourcing", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("tax", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "tax", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxableAmount", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "taxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCalculated", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "taxCalculated", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCode", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "taxCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCodeId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "taxCodeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], TransactionLineModel.prototype, "taxDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxEngine", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "taxEngine", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxOverrideType", Enums.TaxOverrideTypeConverter, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "taxOverrideType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("businessIdentificationNo", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "businessIdentificationNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxOverrideAmount", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "taxOverrideAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxOverrideReason", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "taxOverrideReason", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxIncluded", Boolean, true),
        __metadata("design:type", Boolean)
    ], TransactionLineModel.prototype, "taxIncluded", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("merchantSellerId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "merchantSellerId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("merchantSellerIdentifier", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "merchantSellerIdentifier", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("marketplaceLiabilityType", Enums.MarketplaceLiabilityTypeConverter, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "marketplaceLiabilityType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("originationDocumentId", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "originationDocumentId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("originationSite", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "originationSite", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("details", [TransactionLineDetailModel_1.TransactionLineDetailModel], true),
        __metadata("design:type", Array)
    ], TransactionLineModel.prototype, "details", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("accountPayableSalesTaxDetails", [TransactionLineDetailModel_1.TransactionLineDetailModel], true),
        __metadata("design:type", Array)
    ], TransactionLineModel.prototype, "accountPayableSalesTaxDetails", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonPassthroughDetails", [TransactionLineDetailModel_1.TransactionLineDetailModel], true),
        __metadata("design:type", Array)
    ], TransactionLineModel.prototype, "nonPassthroughDetails", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lineLocationTypes", [TransactionLineLocationTypeModel_1.TransactionLineLocationTypeModel], true),
        __metadata("design:type", Array)
    ], TransactionLineModel.prototype, "lineLocationTypes", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("parameters", [TransactionLineParameterModel_1.TransactionLineParameterModel], true),
        __metadata("design:type", Array)
    ], TransactionLineModel.prototype, "parameters", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("userDefinedFields", [TransactionLineUserDefinedFieldModel_1.TransactionLineUserDefinedFieldModel], true),
        __metadata("design:type", Array)
    ], TransactionLineModel.prototype, "userDefinedFields", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hsCode", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "hsCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("costInsuranceFreight", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "costInsuranceFreight", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("vatCode", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "vatCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("vatNumberTypeId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "vatNumberTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAmountByTaxTypes", [TransactionLineTaxAmountByTaxTypeModel_1.TransactionLineTaxAmountByTaxTypeModel], true),
        __metadata("design:type", Array)
    ], TransactionLineModel.prototype, "taxAmountByTaxTypes", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("deemedSupplier", Enums.DeemedSellerTypeConverter, true),
        __metadata("design:type", Number)
    ], TransactionLineModel.prototype, "deemedSupplier", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("category", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "category", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("summary", String, true),
        __metadata("design:type", String)
    ], TransactionLineModel.prototype, "summary", void 0);
    TransactionLineModel = __decorate([
        (0, json2typescript_1.JsonObject)("TransactionLineModel")
    ], TransactionLineModel);
    return TransactionLineModel;
}());
