// avatax/lib/models/TransactionLineDetailModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionLineDetailModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
/**
 * An individual tax detail element.  Represents the amount of tax calculated for a particular jurisdiction, for a particular line in an invoice.
 * @export
 * @class TransactionLineDetailModel
 */
var TransactionLineDetailModel = exports.TransactionLineDetailModel = /** @class */ (function () {
    function TransactionLineDetailModel() {
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.transactionLineId = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.transactionId = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.addressId = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineDetailModel
         */
        this.country = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineDetailModel
         */
        this.region = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineDetailModel
         */
        this.countyFIPS = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineDetailModel
         */
        this.stateFIPS = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.exemptAmount = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.exemptReasonId = undefined;
        /**
         * @type {boolean}
         * @memberof TransactionLineDetailModel
         */
        this.inState = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineDetailModel
         */
        this.jurisCode = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineDetailModel
         */
        this.jurisName = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.jurisdictionId = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineDetailModel
         */
        this.signatureCode = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineDetailModel
         */
        this.stateAssignedNo = undefined;
        /**
         * @type {Enums.JurisTypeId}
         * @memberof TransactionLineDetailModel
         */
        this.jurisType = undefined;
        /**
         * @type {Enums.JurisdictionType}
         * @memberof TransactionLineDetailModel
         */
        this.jurisdictionType = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.nonTaxableAmount = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.nonTaxableRuleId = undefined;
        /**
         * @type {Enums.TaxRuleTypeId}
         * @memberof TransactionLineDetailModel
         */
        this.nonTaxableType = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.rate = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.rateRuleId = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.rateSourceId = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineDetailModel
         */
        this.serCode = undefined;
        /**
         * @type {Enums.Sourcing}
         * @memberof TransactionLineDetailModel
         */
        this.sourcing = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.tax = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.taxableAmount = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineDetailModel
         */
        this.taxType = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineDetailModel
         */
        this.taxSubTypeId = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineDetailModel
         */
        this.taxTypeGroupId = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineDetailModel
         */
        this.taxName = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.taxAuthorityTypeId = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.taxRegionId = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.taxCalculated = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.taxOverride = undefined;
        /**
         * @type {Enums.RateType}
         * @memberof TransactionLineDetailModel
         */
        this.rateType = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineDetailModel
         */
        this.rateTypeCode = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.taxableUnits = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.nonTaxableUnits = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.exemptUnits = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineDetailModel
         */
        this.unitOfBasis = undefined;
        /**
         * @type {boolean}
         * @memberof TransactionLineDetailModel
         */
        this.isNonPassThru = undefined;
        /**
         * @type {boolean}
         * @memberof TransactionLineDetailModel
         */
        this.isFee = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.reportingTaxableUnits = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.reportingNonTaxableUnits = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.reportingExemptUnits = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.reportingTax = undefined;
        /**
         * @type {number}
         * @memberof TransactionLineDetailModel
         */
        this.reportingTaxCalculated = undefined;
        /**
         * @type {Enums.LiabilityType}
         * @memberof TransactionLineDetailModel
         */
        this.liabilityType = undefined;
        /**
         * @type {Enums.ChargedTo}
         * @memberof TransactionLineDetailModel
         */
        this.chargedTo = undefined;
        /**
         * @type {string}
         * @memberof TransactionLineDetailModel
         */
        this.avtUserBIN = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("transactionLineId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "transactionLineId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("transactionId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "transactionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("addressId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "addressId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("country", String, true),
        __metadata("design:type", String)
    ], TransactionLineDetailModel.prototype, "country", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("region", String, true),
        __metadata("design:type", String)
    ], TransactionLineDetailModel.prototype, "region", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("countyFIPS", String, true),
        __metadata("design:type", String)
    ], TransactionLineDetailModel.prototype, "countyFIPS", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("stateFIPS", String, true),
        __metadata("design:type", String)
    ], TransactionLineDetailModel.prototype, "stateFIPS", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptAmount", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "exemptAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptReasonId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "exemptReasonId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("inState", Boolean, true),
        __metadata("design:type", Boolean)
    ], TransactionLineDetailModel.prototype, "inState", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisCode", String, true),
        __metadata("design:type", String)
    ], TransactionLineDetailModel.prototype, "jurisCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisName", String, true),
        __metadata("design:type", String)
    ], TransactionLineDetailModel.prototype, "jurisName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "jurisdictionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("signatureCode", String, true),
        __metadata("design:type", String)
    ], TransactionLineDetailModel.prototype, "signatureCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("stateAssignedNo", String, true),
        __metadata("design:type", String)
    ], TransactionLineDetailModel.prototype, "stateAssignedNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisType", Enums.JurisTypeIdConverter, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "jurisType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("jurisdictionType", Enums.JurisdictionTypeConverter, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "jurisdictionType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonTaxableAmount", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "nonTaxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonTaxableRuleId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "nonTaxableRuleId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonTaxableType", Enums.TaxRuleTypeIdConverter, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "nonTaxableType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rate", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "rate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rateRuleId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "rateRuleId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rateSourceId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "rateSourceId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("serCode", String, true),
        __metadata("design:type", String)
    ], TransactionLineDetailModel.prototype, "serCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("sourcing", Enums.SourcingConverter, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "sourcing", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("tax", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "tax", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxableAmount", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "taxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxType", String, true),
        __metadata("design:type", String)
    ], TransactionLineDetailModel.prototype, "taxType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxSubTypeId", String, true),
        __metadata("design:type", String)
    ], TransactionLineDetailModel.prototype, "taxSubTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxTypeGroupId", String, true),
        __metadata("design:type", String)
    ], TransactionLineDetailModel.prototype, "taxTypeGroupId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxName", String, true),
        __metadata("design:type", String)
    ], TransactionLineDetailModel.prototype, "taxName", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxAuthorityTypeId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "taxAuthorityTypeId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxRegionId", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "taxRegionId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxCalculated", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "taxCalculated", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxOverride", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "taxOverride", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rateType", Enums.RateTypeConverter, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "rateType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("rateTypeCode", String, true),
        __metadata("design:type", String)
    ], TransactionLineDetailModel.prototype, "rateTypeCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxableUnits", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "taxableUnits", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("nonTaxableUnits", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "nonTaxableUnits", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("exemptUnits", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "exemptUnits", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("unitOfBasis", String, true),
        __metadata("design:type", String)
    ], TransactionLineDetailModel.prototype, "unitOfBasis", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isNonPassThru", Boolean, true),
        __metadata("design:type", Boolean)
    ], TransactionLineDetailModel.prototype, "isNonPassThru", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isFee", Boolean, true),
        __metadata("design:type", Boolean)
    ], TransactionLineDetailModel.prototype, "isFee", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportingTaxableUnits", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "reportingTaxableUnits", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportingNonTaxableUnits", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "reportingNonTaxableUnits", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportingExemptUnits", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "reportingExemptUnits", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportingTax", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "reportingTax", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("reportingTaxCalculated", Number, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "reportingTaxCalculated", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("liabilityType", Enums.LiabilityTypeConverter, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "liabilityType", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("chargedTo", Enums.ChargedToConverter, true),
        __metadata("design:type", Number)
    ], TransactionLineDetailModel.prototype, "chargedTo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("avtUserBIN", String, true),
        __metadata("design:type", String)
    ], TransactionLineDetailModel.prototype, "avtUserBIN", void 0);
    TransactionLineDetailModel = __decorate([
        (0, json2typescript_1.JsonObject)("TransactionLineDetailModel")
    ], TransactionLineDetailModel);
    return TransactionLineDetailModel;
}());
