// avatax/lib/models/UPCModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UPCModel = void 0;
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * One Universal Product Code object as defined for your company.
 * @export
 * @class UPCModel
 */
var UPCModel = exports.UPCModel = /** @class */ (function () {
    function UPCModel() {
        /**
         * @type {number}
         * @memberof UPCModel
         */
        this.id = undefined;
        /**
         * @type {number}
         * @memberof UPCModel
         */
        this.companyId = undefined;
        /**
         * @type {string}
         * @memberof UPCModel
         */
        this.upc = undefined;
        /**
         * @type {string}
         * @memberof UPCModel
         */
        this.legacyTaxCode = undefined;
        /**
         * @type {string}
         * @memberof UPCModel
         */
        this.description = undefined;
        /**
         * @type {Date}
         * @memberof UPCModel
         */
        this.effectiveDate = undefined;
        /**
         * @type {Date}
         * @memberof UPCModel
         */
        this.endDate = undefined;
        /**
         * @type {number}
         * @memberof UPCModel
         */
        this.usage = undefined;
        /**
         * @type {number}
         * @memberof UPCModel
         */
        this.isSystem = undefined;
        /**
         * @type {Date}
         * @memberof UPCModel
         */
        this.createdDate = undefined;
        /**
         * @type {number}
         * @memberof UPCModel
         */
        this.createdUserId = undefined;
        /**
         * @type {Date}
         * @memberof UPCModel
         */
        this.modifiedDate = undefined;
        /**
         * @type {number}
         * @memberof UPCModel
         */
        this.modifiedUserId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("id", Number, true),
        __metadata("design:type", Number)
    ], UPCModel.prototype, "id", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("companyId", Number, true),
        __metadata("design:type", Number)
    ], UPCModel.prototype, "companyId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("upc", String),
        __metadata("design:type", String)
    ], UPCModel.prototype, "upc", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("legacyTaxCode", String, true),
        __metadata("design:type", String)
    ], UPCModel.prototype, "legacyTaxCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("description", String),
        __metadata("design:type", String)
    ], UPCModel.prototype, "description", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("effectiveDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], UPCModel.prototype, "effectiveDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("endDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], UPCModel.prototype, "endDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("usage", Number, true),
        __metadata("design:type", Number)
    ], UPCModel.prototype, "usage", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("isSystem", Number, true),
        __metadata("design:type", Number)
    ], UPCModel.prototype, "isSystem", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], UPCModel.prototype, "createdDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("createdUserId", Number, true),
        __metadata("design:type", Number)
    ], UPCModel.prototype, "createdUserId", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], UPCModel.prototype, "modifiedDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("modifiedUserId", Number, true),
        __metadata("design:type", Number)
    ], UPCModel.prototype, "modifiedUserId", void 0);
    UPCModel = __decorate([
        (0, json2typescript_1.JsonObject)("UPCModel")
    ], UPCModel);
    return UPCModel;
}());
