// avatax/lib/models/VarianceLine.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VarianceLine = void 0;
var VarianceUnit_1 = require("./VarianceUnit");
var VarianceDetail_1 = require("./VarianceDetail");
var json2typescript_1 = require("json2typescript");
/**
 *
 * @export
 * @class VarianceLine
 */
var VarianceLine = exports.VarianceLine = /** @class */ (function () {
    function VarianceLine() {
        /**
         * @type {string}
         * @memberof VarianceLine
         */
        this.lineNo = undefined;
        /**
         * @type {string}
         * @memberof VarianceLine
         */
        this.hsCode = undefined;
        /**
         * @type {number}
         * @memberof VarianceLine
         */
        this.dutyRate = undefined;
        /**
         * @type {number}
         * @memberof VarianceLine
         */
        this.taxRate = undefined;
        /**
         * @type {VarianceUnit}
         * @memberof VarianceLine
         */
        this.amount = undefined;
        /**
         * @type {VarianceUnit}
         * @memberof VarianceLine
         */
        this.taxableAmount = undefined;
        /**
         * @type {VarianceUnit}
         * @memberof VarianceLine
         */
        this.dutyPaid = undefined;
        /**
         * @type {VarianceUnit}
         * @memberof VarianceLine
         */
        this.taxPaid = undefined;
        /**
         * @type {VarianceUnit}
         * @memberof VarianceLine
         */
        this.totalTaxPaid = undefined;
        /**
         * @type {VarianceDetail[]}
         * @memberof VarianceLine
         */
        this.details = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("lineNo", String, true),
        __metadata("design:type", String)
    ], VarianceLine.prototype, "lineNo", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("hsCode", String, true),
        __metadata("design:type", String)
    ], VarianceLine.prototype, "hsCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dutyRate", Number, true),
        __metadata("design:type", Number)
    ], VarianceLine.prototype, "dutyRate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxRate", Number, true),
        __metadata("design:type", Number)
    ], VarianceLine.prototype, "taxRate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("amount", VarianceUnit_1.VarianceUnit, true),
        __metadata("design:type", VarianceUnit_1.VarianceUnit)
    ], VarianceLine.prototype, "amount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxableAmount", VarianceUnit_1.VarianceUnit, true),
        __metadata("design:type", VarianceUnit_1.VarianceUnit)
    ], VarianceLine.prototype, "taxableAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("dutyPaid", VarianceUnit_1.VarianceUnit, true),
        __metadata("design:type", VarianceUnit_1.VarianceUnit)
    ], VarianceLine.prototype, "dutyPaid", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("taxPaid", VarianceUnit_1.VarianceUnit, true),
        __metadata("design:type", VarianceUnit_1.VarianceUnit)
    ], VarianceLine.prototype, "taxPaid", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalTaxPaid", VarianceUnit_1.VarianceUnit, true),
        __metadata("design:type", VarianceUnit_1.VarianceUnit)
    ], VarianceLine.prototype, "totalTaxPaid", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("details", [VarianceDetail_1.VarianceDetail], true),
        __metadata("design:type", Array)
    ], VarianceLine.prototype, "details", void 0);
    VarianceLine = __decorate([
        (0, json2typescript_1.JsonObject)("VarianceLine")
    ], VarianceLine);
    return VarianceLine;
}());
