// avatax/lib/models/VerifyMultiDocumentModel.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerifyMultiDocumentModel = void 0;
var Enums = require("../enums/index");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 * Verify that a MultiDocument object matches the information in your accounting system.
            
If all attributes of the MultiDocument object match the values in your request, the
MultiDocument object will be moved to the document status `Posted`.
            
For more information on document status, see [DocumentStatus](https://developer.avalara.com/api-reference/avatax/rest/v2/models/enums/DocumentStatus/).
 * @export
 * @class VerifyMultiDocumentModel
 */
var VerifyMultiDocumentModel = exports.VerifyMultiDocumentModel = /** @class */ (function () {
    function VerifyMultiDocumentModel() {
        /**
         * @type {string}
         * @memberof VerifyMultiDocumentModel
         */
        this.code = undefined;
        /**
         * @type {Enums.DocumentType}
         * @memberof VerifyMultiDocumentModel
         */
        this.type = undefined;
        /**
         * @type {Date}
         * @memberof VerifyMultiDocumentModel
         */
        this.verifyTransactionDate = undefined;
        /**
         * @type {number}
         * @memberof VerifyMultiDocumentModel
         */
        this.verifyTotalAmount = undefined;
        /**
         * @type {number}
         * @memberof VerifyMultiDocumentModel
         */
        this.verifyTotalTax = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("code", String),
        __metadata("design:type", String)
    ], VerifyMultiDocumentModel.prototype, "code", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("type", Enums.DocumentTypeConverter),
        __metadata("design:type", Number)
    ], VerifyMultiDocumentModel.prototype, "type", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("verifyTransactionDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], VerifyMultiDocumentModel.prototype, "verifyTransactionDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("verifyTotalAmount", Number, true),
        __metadata("design:type", Number)
    ], VerifyMultiDocumentModel.prototype, "verifyTotalAmount", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("verifyTotalTax", Number, true),
        __metadata("design:type", Number)
    ], VerifyMultiDocumentModel.prototype, "verifyTotalTax", void 0);
    VerifyMultiDocumentModel = __decorate([
        (0, json2typescript_1.JsonObject)("VerifyMultiDocumentModel")
    ], VerifyMultiDocumentModel);
    return VerifyMultiDocumentModel;
}());
