// avatax/lib/models/WorksheetDocument.js
"use strict";
/*
 * AvaTax Software Development Kit for JavaScript
 *
 * (c) 2004-2022 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @author     Jonathan Wenger <jonathan.wenger@avalara.com>
 * @author     Sachin Baijal <sachin.baijal@avalara.com>
 * @copyright  2004-2018 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @link       https://github.com/avadev/AvaTax-REST-V2-JS-SDK
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorksheetDocument = void 0;
var WorksheetDocumentLine_1 = require("./WorksheetDocumentLine");
var Message_1 = require("./Message");
var json2typescript_1 = require("json2typescript");
var dateConverter_1 = require("../utils/dateConverter");
/**
 *
 * @export
 * @class WorksheetDocument
 */
var WorksheetDocument = exports.WorksheetDocument = /** @class */ (function () {
    function WorksheetDocument() {
        /**
         * @type {string}
         * @memberof WorksheetDocument
         */
        this.docCode = undefined;
        /**
         * @type {Date}
         * @memberof WorksheetDocument
         */
        this.docDate = undefined;
        /**
         * @type {number}
         * @memberof WorksheetDocument
         */
        this.totalExempt = undefined;
        /**
         * @type {number}
         * @memberof WorksheetDocument
         */
        this.totalTaxable = undefined;
        /**
         * @type {number}
         * @memberof WorksheetDocument
         */
        this.totalTax = undefined;
        /**
         * @type {WorksheetDocumentLine[]}
         * @memberof WorksheetDocument
         */
        this.lines = undefined;
        /**
         * @type {Message[]}
         * @memberof WorksheetDocument
         */
        this.messages = undefined;
        /**
         * @type {string}
         * @memberof WorksheetDocument
         */
        this.resultCode = undefined;
        /**
         * @type {string}
         * @memberof WorksheetDocument
         */
        this.transactionId = undefined;
    }
    __decorate([
        (0, json2typescript_1.JsonProperty)("docCode", String, true),
        __metadata("design:type", String)
    ], WorksheetDocument.prototype, "docCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("docDate", dateConverter_1.DateConverter, true),
        __metadata("design:type", Date)
    ], WorksheetDocument.prototype, "docDate", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalExempt", Number, true),
        __metadata("design:type", Number)
    ], WorksheetDocument.prototype, "totalExempt", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalTaxable", Number, true),
        __metadata("design:type", Number)
    ], WorksheetDocument.prototype, "totalTaxable", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("totalTax", Number, true),
        __metadata("design:type", Number)
    ], WorksheetDocument.prototype, "totalTax", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("lines", [WorksheetDocumentLine_1.WorksheetDocumentLine], true),
        __metadata("design:type", Array)
    ], WorksheetDocument.prototype, "lines", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("messages", [Message_1.Message], true),
        __metadata("design:type", Array)
    ], WorksheetDocument.prototype, "messages", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("resultCode", String, true),
        __metadata("design:type", String)
    ], WorksheetDocument.prototype, "resultCode", void 0);
    __decorate([
        (0, json2typescript_1.JsonProperty)("transactionId", String, true),
        __metadata("design:type", String)
    ], WorksheetDocument.prototype, "transactionId", void 0);
    WorksheetDocument = __decorate([
        (0, json2typescript_1.JsonObject)("WorksheetDocument")
    ], WorksheetDocument);
    return WorksheetDocument;
}());
